
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Oncotrack FAIRification recipe</title>
    
  <link rel="stylesheet" href="../../../../_static/css/index.f658d18f9b420779cfdf24aa0a7e2d77.css">

    
  <link rel="stylesheet"
    href="../../../../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../../../../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../../../../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../../../../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../../../../_static/sphinx-book-theme.e7340bb3dbd8dde6db86f25597f54a1b.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/custom.css" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../../../../_static/js/index.d3f166471bb80abb5163.js">

    <script id="documentation_options" data-url_root="../../../../" src="../../../../_static/documentation_options.js"></script>
    <script src="../../../../_static/jquery.js"></script>
    <script src="../../../../_static/underscore.js"></script>
    <script src="../../../../_static/doctools.js"></script>
    <script src="../../../../_static/togglebutton.js"></script>
    <script src="../../../../_static/clipboard.min.js"></script>
    <script src="../../../../_static/copybutton.js"></script>
    <script src="../../../../_static/mermaid.js"></script>
    <script src="../../../../_static/mermaid.min.js"></script>
    <script src="../../../../_static/mermaid.core.js"></script>
    <script async="async" kind="hypothesis" src="https://hypothes.is/embed.js"></script>
    <script kind="utterances">

    var commentsRunWhenDOMLoaded = cb => {
    if (document.readyState != 'loading') {
        cb()
    } else if (document.addEventListener) {
        document.addEventListener('DOMContentLoaded', cb)
    } else {
        document.attachEvent('onreadystatechange', function() {
        if (document.readyState == 'complete') cb()
        })
    }
}

var addUtterances = () => {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src = "https://utteranc.es/client.js";
    script.async = "async";

    script.setAttribute("repo", "FAIRplus/the-fair-cookbook");
    script.setAttribute("issue-term", "pathname");
    script.setAttribute("theme", "github-light");
    script.setAttribute("label", "üí¨ comment");
    script.setAttribute("crossorigin", "anonymous");

    sections = document.querySelectorAll("div.section");
    if (sections !== null) {
        section = sections[sections.length-1];
        section.appendChild(script);
    }
}
commentsRunWhenDOMLoaded(addUtterances);
</script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../../../../_static/sphinx-book-theme.7d483ff0a819d6edff12ce0b1ead3928.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../../../../_static/sphinx-thebe.js"></script>
    <link rel="canonical" href="https://fairplus.github.io/the-fair-cookbook/content/recipes/applied-examples/oncotrack_raw/rawRecipe-oncotrack.html" />
    <link rel="shortcut icon" href="../../../../_static/favicon.ico"/>
    <link rel="index" title="Index" href="../../../../genindex.html" />
    <link rel="search" title="Search" href="../../../../search.html" />
    <link rel="next" title="RESOLUTE FAIRification Recipe" href="../resolute_raw/rawRecipe_resolute.html" />
    <link rel="prev" title="ND4BB dataset FAIRification recipe" href="../nd4bb_raw/FAIRification_CookBook_Recipe1_V02.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />


<!-- Opengraph tags -->
<meta property="og:url"         content="https://fairplus.github.io/the-fair-cookbook/content/recipes/applied-examples/oncotrack_raw/rawRecipe-oncotrack.html" />
<meta property="og:type"        content="article" />
<meta property="og:title"       content="Oncotrack FAIRification recipe" />
<meta property="og:description" content="Oncotrack FAIRification recipe  Ingredients  Metadata model  Oncotrack cohort metadata  Oncotrack drug sensitivity data  Oncotrack metadata template  Vocabulari" />
<meta property="og:image"       content="https://fairplus.github.io/the-fair-cookbook/_static/cookbook-logo-small.png" />

<meta name="twitter:card" content="summary" />


  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../../../../index.html">
  
  <img src="../../../../_static/cookbook-logo-small.png" class="logo" alt="logo">
  
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../../../../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../../../home.html">
   FAIR Cookbook
  </a>
 </li>
</ul>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../introduction.html">
   Introduction
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../introduction/the-turing-way.html">
     1. The Turing Way Book of Data Science
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../introduction/FAIR-cookbook-audience.html">
     2. Who should read the FAIR Cookbook?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../introduction/brief-FAIR-principles.html">
     3. What are the FAIR Principles?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../introduction/what-is-IMI.html">
     4. What is IMI?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../introduction/FAIR-in-IMI-context.html">
     5. FAIR in the context of IMI
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../introduction/IMI-Data-Access-Request-Process.html">
     6. Engaging with IMI Projects
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../introduction/FAIRplus-values.html">
     7. The Ethical Values behind FAIR and the FAIRPlus project
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../introduction/data-sharing-models.html">
     8. High level data sharing models
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../assessing-fairness.html">
   Assessing FAIR
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../assessing-fairness/cmmi/Data-Management-Maturity-Model.html">
     1. Data Management Maturity Model
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../assessing-fairness/cmmi/Datasets-Maturity-Model.html">
     2. FAIRplus Datasets Maturity Model
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../assessing-fairness/cmmi/Data-Usage-Areas.html">
     3. Data Usage Areas
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../../assessing-fairness/fair-indicators.html">
     4. FAIR Indicators
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../../assessing-fairness/cmmi/RDA-Indicators.html">
       4.4.1. RDA Indicators
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../../assessing-fairness/cmmi/FAIRPlus-Indicators.html">
       4.4.2. IMI FAIRPlus Indicators
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../../assessing-fairness/manual-fair-assessment.html">
     5. FAIR assessment by hand
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../../assessing-fairness/cmmi/evaluation-method.html">
       5.4.1. Evaluation Method
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../../assessing-fairness/cmmi/rda-evaluation.html">
       5.4.2. RDA Evaluation Methods
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../../assessing-fairness/automatic-fair-assessment.html">
     6. FAIR assessment by software
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../../assessing-fairness/fair-assessment-fairevaluator.html">
       6.4.1. Assessing FAIRness with FAIREvaluator
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../../assessing-fairness/fair-assessment-fairshake.html">
       6.4.2. FAIR assessment with FAIRShake
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../infrastructure.html">
   Infrastructure for FAIR
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../infrastructure/id-resolution.html">
     1. Identifier resolution services
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../infrastructure/data-catalog.html">
     2. Building a Dataset Catalogue
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../infrastructure/data-catalog-deployment.html">
     3. Deploying the IMI data catalogue
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../../infrastructure/vocabularies.html">
     4. Vocabularies and Terminology Infrastructure
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../../infrastructure/vocabularies/technical-and-architectural-selection-criteria-of-ontology-lookup-services.html">
       4.1. Technical &amp; architectural selection criteria of ontology lookup services
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../../infrastructure/vocabularies/Selecting_and_using_ontology_lookup_services.html">
       4.2. Selection and exploitation of Ontology Lookup Services
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../../infrastructure/vocabularies/UC3_R13_local_ontology_services.html">
       4.3. How to deploy a terminology service - EBI Ontology Lookup Service
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../findability.html">
   Improving Findability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../findability/identifiers.html">
     1. How to generate globally unique, resolvable and persistent identifiers
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../infrastructure/chemical-identities.html">
     2. Chemical Identity - InChi and SMILES
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../findability/identifiers-minids.html">
     3. Identifier Minting Service with Minid Client
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../findability/zenodo-deposition.html">
     4. Deposition to Zenodo Archive
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../../findability/seo.html">
     5. Findability: Search Engine Optimization
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../../findability/seo/bioschemas-data-page.html">
       5.8.1. Findability: Data Page Markup with Bioschemas
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../../findability/seo/bioschemas-dataset.html">
       5.8.2. Findability: Dataset Markup
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../../findability/seo/bioschemas-datacatalog.html">
       5.8.3. Findability: Data Catalog Markup
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../accessibility.html">
   Improving Accessibility
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../accessibility/anonymizing-patient-data.html">
     1. Anonymizing Patient Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../accessibility/how-to-use-sftp-for-transfer.html">
     2. How to use SFTP to transfer data files between institutions.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../accessibility/aspera.html">
     3. How to download files with Aspera
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../interoperability.html">
   Improving Interoperability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../interoperability/identifier-mapping.html">
     How to interlink data from different sources?
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../../interoperability/bridgedb-recipe.html">
     Identifier mapping with BridgeDB
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../../interoperability/bridgedb/bridgedb-example.html">
       Identifier mapping using BridgeDb WS
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../interoperability/selecting-ontologies.html">
     Which vocabulary to use?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../interoperability/ontology-new-term-request-recipe.html">
     Requesting terms addition to terminology artefacts
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../../interoperability/ontology-robot-recipe.html">
     Building an application ontology with Robot
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../../interoperability/ontology-robot/MSIO-robot-build-process.html">
       MSIO, an application ontology for Metabolomics
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../../interoperability/creating-minimal-metadata-profiles.html">
     Creating a Metadata Profile
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../../interoperability/transcriptomics-metadata.html">
       Establishing a Metadata Profile for Transcriptomics
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../../interoperability/covid19-sample-metadata-profile-shex-use-case.html">
       Metadata Profile Validation in RDF with Shape Expression: The Covid-19 sample metadata use case
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../interoperability/from-proprietary-to-open-standard-mzml-exemplar.html">
     From proprietary to open standard data format
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../interoperability/fastq-file-format-validators.html">
     File format validation, an example with FASTQ files
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../reusability.html">
   Improving Reusability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../reusability/ATI-licensing.html">
     1. Licensing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../reusability/ATI_licensing_software.html">
     2. Software Licenses
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../reusability/ATI_licensing_data.html">
     3. Data Licenses
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../reusability/expressing-data-use.html">
     4. Patterns and practice for expressing allowed data usage
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../reusability/provenance.html">
     5. Provenance Information
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../reusability/community-standards.html">
     6. Compliance with community annotation standards
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="../../applied-examples.html">
   Making IMI data FAIR
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../fair-data-matrix-recipe.html">
     Making a Data Matrix FAIR
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../fair-data-matrix/0-rose-metabolites-Python-data-handling.html">
       Bibliographic Reference:
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../fair-data-matrix/1-rose-metabolites-Python-analysis.html">
       Exploring and comparing Rose Scent profiles stored as tabular data packages with plotnine, a python port of R ggplot2
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../fair-data-matrix/2-rose-metabolites-Python-RDF-querying-analysis.html">
       Exploring a rose garden with SPARQL
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../fair-data-matrix/3-rose-metabolites-R-analysis.html">
       Exploring and comparing Rose Scent profiles stored as tabular data packages with ggplot2 R library
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../etox_raw/FAIRplus_Recipe_Ontology_mapping_the_eTox_dataset_scenario.html">
     Data normalization and integration through entity linking
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../nd4bb_raw/FAIRification_CookBook_Recipe1_V02.html">
     ND4BB dataset FAIRification recipe
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     Oncotrack FAIRification recipe
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../resolute_raw/rawRecipe_resolute.html">
     RESOLUTE FAIRification Recipe
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../../glossary-fair-terms.html">
   Glossary of FAIR things
  </a>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../help.html">
   Help
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../help/tips-tricks.html">
     1. Recipe Tips and Tricks
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../help/recipe-template.html">
     2. Recipe Template
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../help/how-to-create-recipe-with-gdoc.html">
     3. Using GoogleDocs to contribute a FAIR Cookbook Recipe
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../help/how-to-create-recipe-with-hackmd.html">
     4. How to use HackMD to write a FAIR Cookbook Recipe
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../help/how-to-create-recipe-with-git.html">
     5. How to create a FAIRPlus Coobook Recipe ?
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../boilerplate.html">
   Miscellaneous
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../boilerplate/license.html">
     1. License for this book
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../boilerplate/people.html">
     2. FAIR Cookbook Steering Council and Institutional Partners
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../boilerplate/governance.html">
     3. FAIR Cookbook Governance
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../boilerplate/code_of_conduct.html">
     4. FAIR Cookbook Code of Conduct
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../boilerplate/project_information.html">
     5. FAIR CookBook BoilerPlate
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../boilerplate/roadmap.html">
     6. FAIR cookbook development roadmap
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../boilerplate/contributing.html">
     7. Contributing to the FAIR cookbook
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../boilerplate/disclaimer.html">
     8. Disclaimer
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../../../../_sources/content/recipes/applied-examples/oncotrack_raw/rawRecipe-oncotrack.md"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.md</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/FAIRplus/the-fair-cookbook"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/FAIRplus/the-fair-cookbook/issues/new?title=Issue%20on%20page%20%2Fcontent/recipes/applied-examples/oncotrack_raw/rawRecipe-oncotrack.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        <a class="edit-button" href="https://github.com/FAIRplus/the-fair-cookbook/edit/migrating/content/recipes/applied-examples/oncotrack_raw/rawRecipe-oncotrack.md"><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Edit this page"><i class="fas fa-pencil-alt"></i>suggest edit</button></a>
    </div>
</div>


            <!-- Full screen (wrap in <a> to have style consistency -->
            <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                    data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                    title="Fullscreen mode"><i
                        class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i>
            Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#ingredients">
   Ingredients
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#objective">
   Objective
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#step-by-step-process">
   Step-by-Step Process
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#data-description">
     Data Description
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#metadata-etl-pipeline">
     Metadata ETL pipeline
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#data-curation">
     Data curation
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#drug-name-ontology">
       Drug name ontology
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#data-discrepancy">
       Data discrepancy
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#data-encoding">
       Data encoding
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#data-sharing">
     Data sharing
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#summary">
   Summary
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#future-plan">
   Future plan
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#supplementary-materials">
   Supplementary materials
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#authors">
   Authors
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#license">
   License
  </a>
 </li>
</ul>

        </nav>
        
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="oncotrack-fairification-recipe">
<span id="fcb-fairify-examples-oncotrack"></span><h1>Oncotrack FAIRification recipe<a class="headerlink" href="#oncotrack-fairification-recipe" title="Permalink to this headline">¬∂</a></h1>
<div class="section" id="ingredients">
<h2>Ingredients<a class="headerlink" href="#ingredients" title="Permalink to this headline">¬∂</a></h2>
<ul class="simple">
<li><p>Metadata model</p>
<ul>
<li><p><a class="reference external" href="https://drive.google.com/open?id=13xmazrUUWaCZVooL8d3HXZg7URfVlupM">Oncotrack cohort metadata</a></p></li>
<li><p><a class="reference external" href="https://drive.google.com/file/d/166IlLr-kZLBPoGLkxEyp7cLw8KFKtN1_/view?usp=sharing">Oncotrack drug sensitivity data</a></p></li>
<li><p><a class="reference external" href="https://docs.google.com/document/d/1OXSQsZbw2EEZOZrrXJIaJpsb60Y3-4N4W-wu9LX5oa8/edit?usp=sharing">Oncotrack metadata template</a></p></li>
</ul>
</li>
<li><p>Vocabularies and terminologies</p>
<ul>
<li><p>Pharmaceutical drug names follow the nomenclature of ChEBI and ChEMBL database. All drug ontologies are listed <a class="reference external" href="https://drive.google.com/open?id=1kYH76-3K3mkz6wfTydFoT_UPf8WP0fe_">here</a>.</p></li>
<li><p>All abbreviations and acronyms used in OncoTrack cohort metadata are listed in the <a class="reference external" href="https://drive.google.com/open?id=1voD6FGHVyHUgZEHFRfzh_HlTQa7CvW5-MtheGhydDAg">OncoTrack public metadata acronym table</a>.</p></li>
</ul>
</li>
<li><p>Data format</p>
<ul>
<li><p>Input data: Excel</p></li>
<li><p>Output data:</p>
<ul>
<li><p>tab-delimited text file</p></li>
<li><p>JSON file (JSON schema: <a class="reference external" href="https://drive.google.com/open?id=1DoyOZ1uMFv0aPpAQUPkf7mS8vAoS6DCY">BioSamples databases JSON schema</a>)</p></li>
</ul>
</li>
</ul>
</li>
<li><p>Tools and software</p>
<ul>
<li><p>Metadata extract and transform tool: R (Version 3.6.1)</p>
<ul>
<li><p>String parsing: R package stringi_1.4.3, stringr_1.4.0</p></li>
<li><p>Excel file Loading: readxl_1.3.1</p></li>
<li><p>Data structure transform:plyr_1.8.4</p></li>
<li><p>JSON format parsing: rjson_0.2.20</p></li>
</ul>
</li>
<li><p>JSON schema validator: <a class="reference external" href="https://github.com/elixir-europe/json-schema-validator">Elixir JSON schema validator</a></p></li>
<li><p>Ontology mapping: <a class="reference external" href="https://www.ebi.ac.uk/spot/zooma/">ZOOMA</a></p></li>
<li><p>File Integrity check: md5sum (GNU coreutils) 8.28</p></li>
</ul>
</li>
</ul>
</div>
<div class="section" id="objective">
<h2>Objective<a class="headerlink" href="#objective" title="Permalink to this headline">¬∂</a></h2>
<p>This FAIRification pipeline converts the OncoTrack sample metadata to a structured and consistent data format, improves the findability, interoperability, and reusability of the metadata. The FAIRified metadata provides an enriched context to other Oncotrack derived datasets.</p>
</div>
<div class="section" id="step-by-step-process">
<h2>Step-by-Step Process<a class="headerlink" href="#step-by-step-process" title="Permalink to this headline">¬∂</a></h2>
<p>The FAIRification of Oncotrack metadata includes four steps, accessing the data, metadata extraction, transform and load(ETL) pipeline, data curation and data sharing. (Figure 1)</p>
<!-- ![alt_text](image_0.png "image_tooltip")
_Figure 1: OncoTrack metadata FAIRification pipeline_ -->	
<div class="figure align-default" id="oncotrack-metadata-fairification-pipeline">
<a class="reference internal image-reference" href="../../../../_images/image_02.png"><img alt="OncoTrack metadata FAIRification pipeline." src="../../../../_images/image_02.png" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 32 </span><span class="caption-text">OncoTrack metadata FAIRification pipeline.</span><a class="headerlink" href="#oncotrack-metadata-fairification-pipeline" title="Permalink to this image">¬∂</a></p>
</div>
<div class="section" id="data-description">
<h3>Data Description<a class="headerlink" href="#data-description" title="Permalink to this headline">¬∂</a></h3>
<p>The OncoTrack colorectal carcinoma (CRC) biobank includes patient tumour samples, patient derived xenograft models (PDX) and patient-derived 3D  cell models (PDO) from 106 patients. Multi-omics experiments and various drug sensitivity studies have been performed on the CRC biobank.</p>
<p>The<a class="reference external" href="http://www.oncotrack.eu/home/index.html"> OncoTrack</a> sample metadata includes published sample metadata, drug sensitivity experiment results, and private cohort information. Here, we focus on the FAIRification of the public sample metadata. Drug sensitivity, as an important property of the patient-derived 3D cell culture models, is also included. The metadata is published on<a class="reference external" href="https://www.nature.com/articles/ncomms14262#Sec54"> Sch√ºtte, et al</a>. Annotations to the metadata are stored in<a class="reference external" href="https://owncloud.lcsb.uni.lu/apps/files/?dir=/ONCOTRACK&amp;fileid=10738223"> OwnCloud</a> with controlled access.</p>
<p>The original cohort metadata is stored in an Excel spreadsheet. Figure 2 is an example of the cohort metadata. Each sample is listed as a separate row in the spreadsheet. Sample attributes are listed in columns. The sample names follow the<a class="reference external" href="https://drive.google.com/file/d/1qTQ4cYsmD3AN9XYRxpwayOerc78LUzq6/view?usp=sharing"> OncoTrack identifier system guidelines</a>, which consists of patient ID, tumour type, and patient-derived cell culture model ID. For example,_ Sample ‚Äú150_MET1_XEN2‚Äù_ represents <em>‚ÄúThe second xenograft culture of the first metastatic tumour sample in Patient 150.‚Äù</em></p>
<!-- ![alt_text](image_1.png "image_tooltip")
_Figure 2: Example of OncoTrack public cohort metadata -->_
<div class="figure align-default" id="example-of-oncotrack-public-cohort-metadata">
<a class="reference internal image-reference" href="../../../../_images/image_11.png"><img alt="Example of OncoTrack public cohort metadata." src="../../../../_images/image_11.png" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 33 </span><span class="caption-text">Example of OncoTrack public cohort metadata.</span><a class="headerlink" href="#example-of-oncotrack-public-cohort-metadata" title="Permalink to this image">¬∂</a></p>
</div>
<p>The drug sensitivity data is also provided in Excel spreadsheets. Seventeen drugs are tested on patient-derived organoid(PDO) and patient-derived xenografts(PDX) models in five sets of experiments. Different sets have their unique measurement methods, drug response scales. For example, in <em>Sheet 1: PDO drug response</em> experiment, IC50 value is measured and defines the drug <em>response category</em> (Figure 3a). In <em>Sheet 2: PDX drug response test</em>, the <em>response evaluation criteria in solid tumours (<strong>RECIST</strong>)</em> scales defines the_ drug response category_ (Figure 3b). In both examples, the _drug response category _is marked by colour codes.</p>
<!-- ![alt_text](image_2.png "image_tooltip") -->
<div class="figure align-default" id="example-of-oncotrack">
<a class="reference internal image-reference" href="../../../../_images/image_2.png"><img alt="." src="../../../../_images/image_2.png" style="width: 800px;" /></a>
</div>
<!-- ![alt_text](image_3.png "image_tooltip")
_Figure 3: Example of Oncotrack drug response data_ -->
<div class="figure align-default" id="example-of-oncotrack-drug-response-data">
<a class="reference internal image-reference" href="../../../../_images/image_31.png"><img alt="Example of Oncotrack drug response data." src="../../../../_images/image_31.png" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 34 </span><span class="caption-text">Example of Oncotrack drug response data.</span><a class="headerlink" href="#example-of-oncotrack-drug-response-data" title="Permalink to this image">¬∂</a></p>
</div>
</div>
<div class="section" id="metadata-etl-pipeline">
<h3>Metadata ETL pipeline<a class="headerlink" href="#metadata-etl-pipeline" title="Permalink to this headline">¬∂</a></h3>
<p>The frequent use of acronyms and abbreviations in Oncotrack metadata and the inconsistent metadata structures make it difficult to interpret and reuse. To extract and annotate the metadata, the data owner and data curators first agreed on the structure and content of the FAIRified metadata, as well as the principles in metadata extraction.</p>
<p>Firstly, all missing values in the metadata shall be marked with _NAs _instead of being removed, to facilitate cross-sample comparisons. Secondly, all acronyms and abbreviations shall be expanded to their full forms to avoid ambiguity. Thirdly, the sample attribute values shall correspond to ontology terms if applicable. The final sample metadata template can be found<a class="reference external" href="https://docs.google.com/document/d/1OXSQsZbw2EEZOZrrXJIaJpsb60Y3-4N4W-wu9LX5oa8/edit"> here</a>.</p>
<p>The cohort metadata was converted to a<a class="reference external" href="https://docs.google.com/document/d/1pBjRe7rWO4xsUIVecIgxOm_SJaMWeWf8KBrsDTDVAmg/edit?usp=sharing"> tab-delimited table</a>, of which each row represents one sample, and each column represents one sample attribute. 144 samples from 106 patients were extracted, including 35 organoids and 59 xenografts. Tumour type and sample origin information was retrieved from the sample name. The disease name <em>‚Äúcolon and rectal cancer‚Äù</em> was replaced with ontology term<a class="reference external" href="http://www.ebi.ac.uk/efo/EFO_0005842"> ‚ÄúColorectal cancer‚Äù</a>. Figure 4 is an example of the extracted cohort metadata.</p>
<!-- ![alt_text](image_4.png "image_tooltip")
_Figure 4: Example of the extracted cohort metadata_ -->
<div class="figure align-default" id="example-of-the-extracted-cohort-metadata">
<a class="reference internal image-reference" href="../../../../_images/image_4.png"><img alt="Example of the extracted cohort metadata." src="../../../../_images/image_4.png" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 35 </span><span class="caption-text">Example of the extracted cohort metadata.</span><a class="headerlink" href="#example-of-the-extracted-cohort-metadata" title="Permalink to this image">¬∂</a></p>
</div>
<p>Drug sensitivity data were also extracted from the original spreadsheets. Each drug test per sample was listed as one entry. To coordinate different measurement approaches and response category scales, all drug sensitivity data were converted to a unified representation. All the measurement results were stored in Attribute <em>‚ÄúValue‚Äù</em>, the drug response categorizes criteria were stored in Attribute <em>‚ÄúUnit‚Äù,</em> and the _drug response category _was recalculated and stored in Attribute <em>‚ÄúResponse‚Äù.</em> Figure 5 is an example of the extracted drug sensitivity data. 1829 drug tests were extracted in total. The drug response summary is <a class="reference external" href="https://drive.google.com/file/d/1BNkuLtKUsqoAPJqDKqdhj2xYAhuxhQkf/view?usp=sharing">here</a>.</p>
<!-- ![alt_text](image_5.png "image_tooltip")
_Figure 5: Example of extracted drug sensitivity data_	 -->
<div class="figure align-default" id="example-of-extracted-drug-sensitivity-data">
<a class="reference internal image-reference" href="../../../../_images/image_51.png"><img alt="Example of extracted drug sensitivity data." src="../../../../_images/image_51.png" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 36 </span><span class="caption-text">Example of extracted drug sensitivity data.</span><a class="headerlink" href="#example-of-extracted-drug-sensitivity-data" title="Permalink to this image">¬∂</a></p>
</div>
<p>To associate drug sensitivity data with related sample cohort metadata and convert them to machine-readable formats, we combined cohort metadata and drug sensitivity data to JSON files. The<a class="reference external" href="https://drive.google.com/file/d/1DoyOZ1uMFv0aPpAQUPkf7mS8vAoS6DCY/view"> BioSamples JSON schema</a><span style="text-decoration:underline;"> </span>was used as the sample metadata template, allowing direct submissions to<a class="reference external" href="https://www.ebi.ac.uk/biosamples/"> BioSamples database</a> and can be easily converted to other formats, like XML, spreadsheets, etc.</p>
<p>The BioSamples JSON schema includes four blocks, submission metadata( e.g. submission domain, release date, update date), administrative metadata, (e.g. contacts, affiliations), general sample metadata (sample names, species, etc) and sample characteristics. All information is stored in key: value pairs. In sample characteristics, only data with flatten structure is accepted.</p>
<p>All cohort metadata and drug sensitivity data of each sample were combined into separate JSON files, named after the original sample identifier.</p>
<p>BioSamples JSON schema only supports strings as value types. Therefore, complex data, for example, drug sensitivity data, need additional parsing to fit into the schema. For each drug response experiments, four descriptive attributes were ‚Äú<em>Drug Name</em>‚Äù, ‚Äú<em>Value</em>‚Äù, ‚Äú<em>Unit</em>‚Äù and ‚Äú<em>Drug response level</em>‚Äù provided. To convert them to key: value pairs, the ‚Äú<em>drug name</em>‚Äù was kept as part of the attribute key. ‚ÄúValue‚Äù, ‚ÄúUnit‚Äù and ‚ÄúDrug response level‚Äù information were joint into one string. For example, the drug response of _regorafenib in Sample 302_MET_CELL_XEN _(Figure 6) was converted to <em>‚ÄúDrug response(regorafenib): Minor response, IC50(¬µM) = 29‚Äù.</em> For drugs that were tested more than once, a replicate ID was added to the attribute key.</p>
<p>Administrative information was also added to each JSON object, including institute, project name, and project website. Contact details were not provided, which needs further confirmation from the data owners.</p>
<p>All transformed metadata was validated against BioSamples schema with the<a class="reference external" href="https://github.com/elixir-europe/json-schema-validator"> Elixir JSON validator</a><span style="text-decoration:underline;">,</span> which is based on AJV with additional life science keyword validation.</p>
</div>
<div class="section" id="data-curation">
<h3>Data curation<a class="headerlink" href="#data-curation" title="Permalink to this headline">¬∂</a></h3>
<p>Data curation was implemented to increase the findability of the extracted metadata, including ontology mapping, encoding, and fixing data discrepancy.</p>
<div class="section" id="drug-name-ontology">
<h4>Drug name ontology<a class="headerlink" href="#drug-name-ontology" title="Permalink to this headline">¬∂</a></h4>
<p>Different types of drug names were used in the original drug sensitivity data, including generic names, trade names and their chemical names. Some in-development drugs were labelled by their company code. All drug names were mapped to a standardized nomenclature, together with ontology annotations, to increase the findability of drug response results.</p>
<p>The popular drug nomenclatures include International Nonproprietary Names (INNs), <a class="reference external" href="https://www.drugbank.ca/">DrugBank</a><span style="text-decoration:underline;"> </span>names, Chemical Entities of Biological Interest (<a class="reference external" href="https://www.ebi.ac.uk/chebi/">ChEBI</a>) database identifiers, and bioactivity database,<a class="reference external" href="https://www.ebi.ac.uk/chembl/"> ChEMBL</a><span style="text-decoration:underline;"> i</span>dentifiers. Among them, the ChEBI database cover most drugs in OncoTrack, is easy to access and has been curated as an ontology source in Open Biomedical Ontology (OBO). Hence, the drug names in ChEBI was selected as the primary drug names. For drugs or active compounds that were not included in ChEBI, their names in ChEMBL were selected.</p>
<p>The OncoTrack drug names were mapped to corresponding ontology terms in ChEBI and ChEMBL using ZOOMA. ZOOMA is an ontology annotation tool, which maps free texts to ontology terms based on a curated repository of annotation knowledge. In OncoTrack drug name mapping, only ChEBI and ChEMBL were selected as ontology sources, and all curation sources except GWAS were selected to improve the coverage and increase the mapping confidence. Figure 6 is an example of the ZOOMA mapping output. Only entries with high mapping confidences were selected. The rest were mapped manually to terms in ChEBI databases. Among all 17 drugs, Drug ‚ÄúBI 860585‚Äù didn‚Äôt map to any pharmaceutical terms, so the original name was kept. The complete mapping results are listed<a class="reference external" href="https://drive.google.com/open?id=1kYH76-3K3mkz6wfTydFoT_UPf8WP0fe_"> here</a>.</p>
<!-- 
![alt_text](image_6.jpg "image_tooltip")
_Figure 6: Example of  ZOOMA ontology mapping results_
 -->
<div class="figure align-default" id="example-of-zooma-ontology-mapping-results">
<a class="reference internal image-reference" href="../../../../_images/image_6.jpg"><img alt="Example of ZOOMA ontology mapping results." src="../../../../_images/image_6.jpg" style="width: 600px;" /></a>
<p class="caption"><span class="caption-number">Fig. 37 </span><span class="caption-text">Example of ZOOMA ontology mapping results.</span><a class="headerlink" href="#example-of-zooma-ontology-mapping-results" title="Permalink to this image">¬∂</a></p>
</div>
</div>
<div class="section" id="data-discrepancy">
<h4>Data discrepancy<a class="headerlink" href="#data-discrepancy" title="Permalink to this headline">¬∂</a></h4>
<p>Discrepancy and errors were found in the published metadata. In the cohort metadata, the tumour purity of sample ‚Äú150_MET1‚Äù, ‚Äú208_MET1‚Äù, ‚Äú209_MET2‚Äù and ‚Äú209_T2‚Äù were wrongly labelled. We compared these entries with the original records in the OncoTrack database and found discrepancies between the published metadata and Oncotrack database records. Those values were removed from extracted data until further confirmation from the data owner. The updated cohort metadata is <a class="reference external" href="https://drive.google.com/open?id=13xmazrUUWaCZVooL8d3HXZg7URfVlupM">here</a>.  In drug response metadata extraction, we recalculated the drug response level and found one error in <em>cetuximab</em> drug response level. The corrected drug response data is <a class="reference external" href="https://drive.google.com/open?id=166IlLr-kZLBPoGLkxEyp7cLw8KFKtN1_">here</a>.</p>
</div>
<div class="section" id="data-encoding">
<h4>Data encoding<a class="headerlink" href="#data-encoding" title="Permalink to this headline">¬∂</a></h4>
<p>Special letters were converted to ASCII encoding to avoid mojibake. All the ‚Äú¬µM‚Äùs in drug sensitivity data were converted to ‚ÄúmicroM‚Äù.</p>
</div>
</div>
<div class="section" id="data-sharing">
<h3>Data sharing<a class="headerlink" href="#data-sharing" title="Permalink to this headline">¬∂</a></h3>
<p>The curated metadata, both in summary spreadsheet format and JSON format, were uploaded to Owncloud and Google drive. The OncoTrack sample metadata can be submitted directly to the EMBL BioSamples database upon request. Supplementary figure 1 is an example of how the metadata will be displayed in the BioSamples database. Checksums were also provided to verify metadata integrity.</p>
</div>
</div>
<div class="section" id="summary">
<h2>Summary<a class="headerlink" href="#summary" title="Permalink to this headline">¬∂</a></h2>
<p>The public cohort metadata and drug sensitivity data were converted to tab-delimited summary files. For each sample, corresponding cohort metadata and drug data were combined to a separate JSON file, which can be later loaded to EBI BioSamples database. Metadata were translated to consistent terminologies and linked with ontology terms.</p>
</div>
<div class="section" id="future-plan">
<h2>Future plan<a class="headerlink" href="#future-plan" title="Permalink to this headline">¬∂</a></h2>
<p>The current ontology annotation is limited to drug names. Other ontologies can be added upon request. More detailed administrative metadata will be added once getting permission from the OncoTrack consortium. The license will be added according to the drug release policy.</p>
</div>
<div class="section" id="supplementary-materials">
<h2>Supplementary materials<a class="headerlink" href="#supplementary-materials" title="Permalink to this headline">¬∂</a></h2>
<p>Script 1: <a class="reference external" href="https://drive.google.com/file/d/1bbMMKZnygZ8kccdnSj2R_BRLDhd0vAS-/view?usp=sharing">Extract_cohort_metadata.R</a></p>
<p>Script 2: <a class="reference external" href="https://drive.google.com/open?id=1eNxdNFGN7GAw7Z0Cmk0k2BJiMMZ7wbel">Extract_drugResponse_metadata.R</a></p>
<p>Script 3: <a class="reference external" href="https://drive.google.com/file/d/18Ik3RryhWVFq9_2IIy6k8N-qpnP9Wgm3/view?usp=sharing">Transform to JSON.R</a></p>
<p>Supplementary figure 1: <a class="reference external" href="https://drive.google.com/open?id=1ALOwGwB2RpWksH4TuGeye8qEnUm0ULVH">Example of Sample 150-MET1-XEN2 in BioSamples database</a></p>
</div>
<hr class="docutils" />
<div class="section" id="authors">
<h2>Authors<a class="headerlink" href="#authors" title="Permalink to this headline">¬∂</a></h2>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="text-align:left head"><p>Name</p></th>
<th class="text-align:left head"><p>Affiliation</p></th>
<th class="text-align:left head"><p>orcid</p></th>
<th class="text-align:left head"><p>CrediT role</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-align:left"><p></p></td>
<td class="text-align:left"><p></p></td>
<td class="text-align:left"><p><a class="reference external" href="https://orcid.org/orcid.org/0000-0000-0000-0000">0000-0000-0000-0000</a></p></td>
<td class="text-align:left"><p>Writing - Original Draft</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p></p></td>
<td class="text-align:left"><p></p></td>
<td class="text-align:left"><p></p></td>
<td class="text-align:left"><p>Writing - Review &amp; Editing, Funding acquisition</p></td>
</tr>
</tbody>
</table>
</div>
<hr class="docutils" />
<div class="section" id="license">
<h2>License<a class="headerlink" href="#license" title="Permalink to this headline">¬∂</a></h2>
<p>This page is released under the Creative Commons 4.0 BY license.</p>
<p><a href="https://creativecommons.org/licenses/by/4.0/"><img src="https://mirrors.creativecommons.org/presskit/buttons/80x15/png/by.png" height="20"/></a></p>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./content/recipes/applied-examples/oncotrack_raw"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        
        <div class='prev-next-bottom'>
            
    <a class='left-prev' id="prev-link" href="../nd4bb_raw/FAIRification_CookBook_Recipe1_V02.html" title="previous page">ND4BB dataset FAIRification recipe</a>
    <a class='right-next' id="next-link" href="../resolute_raw/rawRecipe_resolute.html" title="next page">RESOLUTE FAIRification Recipe</a>

        </div>
        
        </div>
    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By The FAIRplus project members<br/>
        
            &copy; Copyright 2020 (CC-BY 4.0).<br/>
          <div class="extra_footer">
            <script>mermaid.init();</script> <div class="row_footer"> This project has received funding from the <a href="https://www.imi.europa.eu/">Innovative Medicines Initiative</a> 2 Joint Undertaking under grant agreement No 802750. This Joint Undertaking receives support from the European Union‚Äôs Horizon 2020 research and innovation programme and EFPIA Companies. This communication reflects the views of the authors and neither IMI nor the European Union, EFPIA or any Associated Partners are liable for any use that may be made of the information contained herein. </div>
<div class="row_footer"> This page was created by <a href="https://github.com/jupyter/jupyter-book/graphs/contributors">The Jupyter Book Community</a> </div>
<div class="row_footer"> <div class="column" style="center"><a href="https://europa.eu/"><img src="https://raw.githubusercontent.com/FAIRplus/the-fair-cookbook/migrating/_static/images/logo/eu-flag-recolored-fairplus.png" height="20" alt="EU flag" style="float:center""></a></div> <div class="column"><a href="https://www.imi.europa.eu/"><img src="https://raw.githubusercontent.com/FAIRplus/the-fair-cookbook/migrating/_static/images/logo/imi-logo-fairplus-v1.png" height="20" alt="IMI logo" style="float:center""></a></div> <div class="column"><a href="https://efpia.eu"><img src="https://raw.githubusercontent.com/FAIRplus/the-fair-cookbook/migrating/_static/images/logo/efpia-logo-fairplus2.png" height="20" alt="EFPIA logo" style="float:center""></a></div> <br/> </div>
          </div>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../../../../_static/js/index.d3f166471bb80abb5163.js"></script>


    
    <!-- Google Analytics -->
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-171697285-1', 'auto');
      ga('set', 'anonymizeIp', true);
      ga('send', 'pageview');
    </script>
    <script async src='https://www.google-analytics.com/analytics.js'></script>
    <!-- End Google Analytics -->
    
  </body>
</html>