
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Creating a community annotation compliant ISA descriptor: Dataset Maturity Level 2</title>

  <link rel="stylesheet" href="../../../_static/css/index.f658d18f9b420779cfdf24aa0a7e2d77.css">


  <link rel="stylesheet"
    href="../../../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">



  <link rel="stylesheet"
    href="../../../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../../../_static/vendor/lato_latin-ext/1.44.1/index.css">


    <link rel="stylesheet" href="../../../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../../../_static/sphinx-book-theme.e7340bb3dbd8dde6db86f25597f54a1b.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/materialdesignicons.min.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/search.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/animate.min.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/vuetify.min.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/search-wizard.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/custom.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />

  <link rel="preload" as="script" href="../../../_static/js/index.d3f166471bb80abb5163.js">

    <script id="documentation_options" data-url_root="../../../" src="../../../_static/documentation_options.js"></script>
    <script src="../../../_static/jquery.js"></script>
    <script src="../../../_static/underscore.js"></script>
    <script src="../../../_static/doctools.js"></script>
    <script src="../../../_static/togglebutton.js"></script>
    <script src="../../../_static/clipboard.min.js"></script>
    <script src="../../../_static/copybutton.js"></script>
    <script src="../../../_static/js/mermaid.js"></script>
    <script src="../../../_static/js/vue.min.js"></script>
    <script src="../../../_static/js/mermaid.min.js"></script>
    <script src="../../../_static/js/searchNodal.js"></script>
    <script src="../../../_static/js/mermaid.core.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../../../_static/sphinx-book-theme.7d483ff0a819d6edff12ce0b1ead3928.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../../../_static/sphinx-thebe.js"></script>
    <link rel="canonical" href="https://fairplus.github.io/the-fair-cookbook/content/recipes/maturity/isa-api-programmatic-rebuild-of-BII-S-3.html" />
    <link rel="shortcut icon" href="../../../_static/favicon.ico"/>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" />
    <link rel="next" title="From ISA JSON to ISA JSON-LD (RDF): Dataset Maturity Level 3" href="isa-json-conversion-to-rdf-linked-data.html" />
    <link rel="prev" title="Creating a basic ISA descriptor: Dataset Maturity Level 1" href="create-a-simple-ISA-descriptor.html" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<meta name="docsearch:language" content="en" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/vuetify/2.6.9/vuetify.js"
        integrity="sha512-cgrSlRKhr/ovJePDoGIdcfZ3wiGcc0fT4Q7hAo0x3P1lyLwUmv2NgJtCo2gzFbxqFjKvNnCfVUjZ1XADm0/ZdQ=="
        crossorigin="anonymous" referrerpolicy="no-referrer">
</script>

<!-- Opengraph tags -->
<meta property="og:url"         content="https://fairplus.github.io/the-fair-cookbook/content/recipes/maturity/isa-api-programmatic-rebuild-of-BII-S-3.html" />
<meta property="og:type"        content="article" />
<meta property="og:title"       content="Creating a community annotation compliant ISA descriptor: Dataset Maturity Level 2" />
<meta property="og:description" content="An online, open and live resource for the Life Sciences with recipes that help you to make and keep data Findable, Accessible, Interoperable and Reusable; in one word FAIR." />
 <meta property="og:image"       content="/_static/images/homepage.png" /> 
<meta name="twitter:card" content="summary" />
  </head>
<body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">



<v-app id="app" class="pa-0">
    <v-container fluid class="pa-0">
        <v-row class="pa-0" no-gutters>
            <v-container fluid class="pa-0">
                <v-row>
                    <div class="topbar container-xl fixed-top pl-0 primary lighten-1 py-2" style="height:65px !important;">
    <div class="topbar-contents row" style="margin: initial !important;">
        <div class="col topbar-main d-flex align-center">
            
                <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                        data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                        aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                        title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                    <i class="fas fa-bars"></i>
                    <i class="fas fa-arrow-left"></i>
                    <i class="fas fa-arrow-up"></i>
                </button>
            

            
                <a href="../../../index.html" style="width: 300px">
                    <img
                        src="../../../_static/cookbook-logo-small.png"
                        class="logo"
                        alt="The FAIR Cookbook"
                        style="width: 100%"
                    >
                </a>
            

            <v-spacer> </v-spacer>
            <div class="mr-4 extraButtons">
                <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger d-flex justify-center align-center" style="height:100%">

    <v-menu offset-y content-class="elevation-0" open-on-hover>
        <template v-slot:activator="{ on, attrs }">
            <v-btn
              id="dropdown-buttons-trigger"
              color="white"
              outlined
              v-bind="attrs"
              v-on="on"
              flat
              solo
              rounded
              :fab="$vuetify.breakpoint.mdAndDown"
              :small="$vuetify.breakpoint.mdAndDown"
              :class="$vuetify.breakpoint.mdAndDown ? 'pa-0' : ''"
            >
                <v-icon :class="$vuetify.breakpoint.lgAndUp ? 'mr-3' : 'mx-1'" left>fab fa-github</v-icon>
                <span v-if="$vuetify.breakpoint.lgAndUp"> GitHub </span>
            </v-btn>
        </template>
        <div class="homepageMenu d-flex flex-column">
            
            <v-btn
              color="grey darken-1"
              dark
              flat
              solo
              rounded
              href="https://github.com/FAIRplus/the-fair-cookbook"
              class="mt-1"
              target="_blank"
            >
                <v-icon class="mr-3" left>fab fa-github</v-icon>
                repository
            </v-btn>
            

            
            <v-btn
              color="grey darken-1"
              dark
              flat
              solo
              rounded
              href="https://github.com/FAIRplus/the-fair-cookbook/issues/new?title=Issue%20on%20page%20%2Fcontent/recipes/maturity/isa-api-programmatic-rebuild-of-BII-S-3.html&body=Your%20issue%20content%20here."
              class="mt-1"
              target="_blank"
            >
                <v-icon class="mr-3" left>fas fa-lightbulb</v-icon>
                open issue
            </v-btn>
            

            
            <v-btn
              color="grey darken-1"
              dark
              flat
              solo
              rounded
              href="https://github.com/FAIRplus/the-fair-cookbook/edit/main/content/recipes/maturity/isa-api-programmatic-rebuild-of-BII-S-3.ipynb"
              class="mt-1"
              target="_blank"
            >
                <v-icon class="mx-3" left>fas fa-pencil-alt</v-icon>
                suggest edit
            </v-btn>
            
        </div>
    </v-menu>
</div>

                <!-- Launch buttons -->

<div class="dropdown-buttons-trigger mr-3 d-flex justify-center align-center" style="height:100%">
    <v-menu offset-y content-class="elevation-0" open-on-hover>
        <template v-slot:activator="{ on, attrs }">
            <v-btn
              id="dropdown-buttons-trigger"
              color="white"
              outlined
              v-bind="attrs"
              v-on="on"
              flat
              solo
              rounded
              :fab="$vuetify.breakpoint.mdAndDown"
              :small="$vuetify.breakpoint.mdAndDown"
              :class="$vuetify.breakpoint.mdAndDown ? 'pa-0' : ''"
            >
                <v-icon :class="$vuetify.breakpoint.lgAndUp ? 'mr-3' : 'mx-0'" left>far fa-play-circle</v-icon>
                <span v-if="$vuetify.breakpoint.lgAndUp"> Execute notebook </span>
            </v-btn>
        </template>
        <div class="homepageMenu d-flex flex-column">
            
            <v-btn
              color="grey darken-1"
              dark
              flat
              solo
              rounded
              href="https://mybinder.org/v2/gh/FAIRplus/the-fair-cookbook/main?urlpath=lab/tree/content/recipes/maturity/isa-api-programmatic-rebuild-of-BII-S-3.ipynb"
              class="mt-1 d-flex justify-start"
              target="_blank"
            >
                <img style="width:20px; height:auto" src="../../../_static/images/logo_binder.svg" alt="Interact on binder" class="mr-3">
                Binder
            </v-btn>
            

            

            
            <v-btn
              color="grey darken-1"
              dark
              flat
              solo
              rounded
              href="https://colab.research.google.com/github/FAIRplus/the-fair-cookbook/blob/main/content/recipes/maturity/isa-api-programmatic-rebuild-of-BII-S-3.ipynb"
              class="mt-1 d-flex justify-start"
              target="_blank"
            >
                <img style="width:20px; height:auto" src="../../../_static/images/logo_colab.png" alt="Interact on Colab" class="mr-3">
                Colab
            </v-btn>
            

            
        </div>
    </v-menu>
</div>

            </div>
                
                    <script>
    Vue.component("Searchfield", {
      data () { return {
        query: null
      } },
      methods: {
        search() {
            if (this.query) window.location.href = `../../search-wizard.html?q=` + this.query
            else window.location.href = `../../search-wizard.html`
        }
      },
      template: `
      <div style="width:250px">
            <v-text-field
              v-model="query"
              append-icon="fa-search"
              label="Search Wizard..."
              dense
              rounded
              single-line
              solo
              hide-details
              class="white primary--text"
              @click:append="search"
              @keyup.enter="search"
            />
      </div>
      `
    })
</script>

<Searchfield> </Searchfield>






                
                
                
        </div>
    </div>
</div>
                </v-row>
                
                    <v-row class="d-flex">
                    
                        <div
                            class=" bd-sidebar site-navigation show"
                            id="site-navigation"
                            style="width:350px"
                        >
                            
                                <div class="nav-drawer d-flex flex-column grey lighten-3" style="height:100%;padding-top: 40px;">

    <div class="testTOC">
        <script>
    Vue.component("Navigation", {
        data() { return { recipes: {} }},
        methods: {
            async getData(){
                const recipes = await fetch('/_static/recipes.json')
                this.recipes = await recipes.json();
                this.parseDom()
            },
            parseDom() {
                const links = Array.prototype.slice.call(document.getElementsByClassName("reference internal"))
                const icon = '<i class="fas fa-play-circle grey lighten-3 secondary--text ml-2"></i>'
                links.forEach(link => {
                    const recipeSplit = link.innerText.split(". ")
                    const recipeName = recipeSplit[recipeSplit.length - 1].trim()
                    recipeSplit.pop()
                    recipeSplit.forEach(pref => { pref = pref.trim() })
                    const prefix = recipeSplit.length > 0 ? recipeSplit.join(".") + "." : null
                    const content = prefix ? `${prefix} ${recipeName}` : recipeName
                    if (Object.keys(this.recipes).includes(recipeName)) {
                        if (this.recipes[recipeName].executable === 'Yes') link.innerHTML = content + icon
                    }
                })
            }
        },
        mounted() { this.getData() }
    })
</script>


<Navigation> </Navigation>

    </div>

    <div class="nav-tree grey px-5 lighten-3 pb-5" style="flex-grow: 1; border-bottom: 1px solid #ccc !important">
        
        
            <nav class="bd-links pt-10" id="bd-docs-nav" aria-label="Main navigation">
    <p class="caption collapsible-parent">
 <span class="caption-text">
  FOREWORD
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/FAIR-cookbook-audience.html">
   Introduction
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/brief-FAIR-principles.html">
   Introducing the FAIR Principles
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/FAIRplus-values.html">
   Reflecting on the ethical values of FAIR
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/fairification-process.html">
   Introducing our FAIRification framework
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/priorization.html">
   Prioritizing projects for FAIRification
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/metadata-fair.html">
   Framing FAIR and the notion of metadata
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/FAIR-and-knowledge-graphs.html">
   Understanding the relation between FAIR and Knowledge Graphs
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/training-datasets.html">
   Training for FAIRification with open or synthetic biomedical datasets
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/public-kg-resource-integration.html">
   Raising Awareness in Public Knowledge Graphs for Life Sciences
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/fair-cro.html">
   Reflecting on Practical Considerations for CROs to play FAIR
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/dpia.html">
   Data Protection Impact Assessment and Data Privacy
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/glossary-fair-terms.html">
   Glossary
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Recipes at a Glance
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../../search-wizard.html">
   All Recipes In a Table
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  FAIR RECIPES
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../findability.html">
   Findability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/identifiers.html">
     1. Introducing unique, persistent identifiers
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/chemical-identities.html">
     2. Creating InChI &amp; SMILES identifiers for chemical structures
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/iupac-names.html">
     3. Creating InChIKeys for IUPAC names
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/identifiers-minids.html">
     4. Minting identifiers with Globus Minid client
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/zenodo-deposition.html">
     5. Depositing to generic repositories - Zenodo use case
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/registeringDatasets.html">
     6. Registering datasets with Wikidata
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/checksum-create.html">
     7. Creating file checksums
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/checksum-validate.html">
     8. Validating checksums to verify file integrity
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../findability/seo.html">
     9. Introducing Search Engine Optimization (SEO)
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../findability/seo/bioschemas-data-page.html">
       9.9.1. Marking up Data pages with Schema.org &amp; Bioschemas for SEO
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../findability/seo/bioschemas-dataset.html">
       9.9.2. Marking up Dataset page with Schema.org &amp; Bioschemas for SEO
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../findability/seo/bioschemas-datacatalog.html">
       9.9.3. Marking up Data Catalogue page with Schema.org &amp; Bioschemas for SEO
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../accessibility.html">
   Accessibility
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../accessibility/sftp.html">
     1. Transferring data with SFTP protocol
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../accessibility/aspera.html">
     2. Downloading data with Aspera protocol
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../interoperability.html">
   Interoperability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../interoperability/registering-swisslipids-identifiers.html">
     1. Registering SwissLipids identifiers in Wikidata
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../interoperability/identifier-mapping.html">
     2. Interlinking data from different sources
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../interoperability/bridgedb-recipe.html">
     3. Mapping identifiers with BridgeDb
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../interoperability/bridgedb/bridgedb-example.html">
       3.11.1. Mapping identifiers using BridgeDb web services
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../interoperability/introduction-terminologies-ontologies.html">
     4. Introducing terminologies and ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../interoperability/selecting-ontologies.html">
     5. Selecting terminologies and ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../interoperability/ontology-new-term-request-recipe.html">
     6. Requesting new terms from terminologies and ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../interoperability/ontology-operations-tools.html">
     7. Introducing ontology-related tools and services
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../interoperability/ontology-robot-recipe.html">
     8. Building an application ontology with ROBOT
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../interoperability/ontology-robot-recipe/MSIO-robot-build-process.html">
       8.11.1. Building an application ontology for metabolomics - MSIO
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../interoperability/ontology-robot-recipe/competency-questions.html">
       8.11.2. Defining competency questions for the Ontology ROBOT use case
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../interoperability/ontology-alignment-oxo.html">
     9. Mapping Ontologies with OxO, EBI Ontology Xref Service
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../interoperability/creating-data-dictionary.html">
     10. Creating a data/variable dictionary
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../interoperability/creating-minimal-metadata-profiles.html">
     11. Creating a metadata profile
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../interoperability/transcriptomics-metadata.html">
       11.5.1. Outlining a metadata profile for transcriptomics
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../interoperability/covid19-sample-metadata-profile-shex-use-case.html">
       11.5.2. Building a community compliant metadata profile - The Covid19 sample profile use case
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../interoperability/bioactivity-profile.html">
       11.5.3. Outlining a metadata profile for Bioactivity data
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../interoperability/c4c-clinical-trials.html">
       11.5.4. Creating a metadata profile for clinical trial protocols
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../interoperability/from-proprietary-to-open-standard-mzml-exemplar.html">
     12. Converting from proprietary to open format
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../interoperability/fastq-file-format-validators.html">
     13. Validating file format - FASTQ example
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../interoperability/rdf-conversion.html">
     14. Inventorying tools for converting data to RDF
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../interoperability/ETL-tools.html">
     15. Surveying extraction, transformation, load (ETL) tools
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../interoperability/VCF-to-FHIR-conversion.html">
     16. Expressing Clinical Genetic Information as FHIR JSON
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../interoperability/vcf2fhir/FAIRcookbook-vcf2fhir.html">
       16.9.1. Converting VCF file to FHIR JSON
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../interoperability/creating-knowledge-graph-from-unstructured-text.html">
     17. Creating knowledge graphs from unstructured text
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../interoperability/nlp2kg/creating-knowledge-graph-from-text.html">
       17.12.1. unstructured text to graph as executable notebook
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../reusability.html">
   Reusability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/ATI-licensing.html">
     1. Licensing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/ATI_licensing_software.html">
     2. Licensing Software
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/ATI_licensing_data.html">
     3. Licensing Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/expressing-data-use.html">
     4. Declaring data permitted uses
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/provenance.html">
     5. Introducing Provenance Information
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../infrastructure.html">
   Infrastructure
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/id-resolution.html">
     1. Introducing identifier resolution services
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/gupri.html">
     2. Creating resolvable identifiers
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/data-catalog.html">
     3. Building a catalogue of datasets
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/dats-model.html">
     4. Introducing the DATS model
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/data-catalog-deployment.html">
     5. Deploying a data catalogue - The IMI data catalogue example
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../infrastructure/vocabularies.html">
     6. Vocabulary management
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../infrastructure/vocabularies/technical-and-architectural-selection-criteria-of-ontology-lookup-services.html">
       6.1. Introducing vocabulary portals and lookup services
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../infrastructure/vocabularies/Selecting_and_using_ontology_lookup_services.html">
       6.2. Selecting an ontology lookup service
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../infrastructure/vocabularies/UC3_R13_local_ontology_services.html">
       6.3. Deploying EBI Ontology Loopkup Service
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/openrefine-nvt.html">
     7. Using OpenRefine and Karma for FAIRification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/fair-api.html">
     8. Developing FAIR API for the Web
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../assessing-fairness.html">
   Assessment
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../assessing-fairness/fair-assessment-fairevaluator.html">
     1. Assessing with FAIR Evaluator
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../assessing-fairness/fair-assessment-fairshake.html">
     2. Assessing with FAIRshake
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  FAIR Maturity
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="../maturity.html">
   Changing culture with the Dataset Maturity Model
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/miappe.html">
     Improving dataset maturity - MIAPPE-compliant submission to EMBL-EBI databases
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/plant-pheno-data-publication.html">
     Publishing plant phenotypic data
    </a>
   </li>
   <li class="toctree-l2 current active collapsible-parent">
    <a class="reference internal" href="isa-maturity-progression.html">
     Moving through maturity levels with ISA
    </a>
    <ul class="current collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="create-a-simple-ISA-descriptor.html">
       Minimal Data Maturity with ISA - ISA-Tab and free text
      </a>
     </li>
     <li class="toctree-l3 current active">
      <a class="current reference internal" href="#">
       Enhanced Data Maturity with ISA - ISA-JSON and ontology markup
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="isa-json-conversion-to-rdf-linked-data.html">
       Moving to a semantically typed version - ISA-JSON-LD
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="isa-as-RO.html">
       Dissemination - Packaging ISA as a Research Object (RO)
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../applied-examples/fair-data-matrix-recipe.html">
     Making an omics data matrix FAIR
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/0-rose-metabolites-Python-data-handling.html">
       FAIRifying Data Matrices - Step1 - Starting material
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/1-rose-metabolites-Python-analysis.html">
       FAIRifying Data Matrices - Step2 - Structuring data
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/2-rose-metabolites-Python-RDF-querying-analysis.html">
       FAIRifying Data Matrices - Step3 - Exploring data with SPARQL
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/3-rose-metabolites-R-analysis.html">
       FAIRifying Data Matrices - Step4 - Integrating data
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/fair-workflows.html">
     Making Computational Workflows FAIR
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  FAIRified Datasets
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../applied-examples.html">
   Applied examples
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/etox.html">
     1. IMI eTox - toxicity datasets
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/nd4bb.html">
     2. IMI nd4bb - chemical activities datasets
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/oncotrack_raw/rawRecipe-oncotrack.html">
     3. Readying IMI Oncotrack - clinical cohort datasets for deposition to EBI Biosamples
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/resolute_raw/resolute.html">
     4. Depositing IMI ReSOLUTE transcriptomics datasets to EBI repositories
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/ehden-ohdsi.html">
     5. Enhancing discoverability of EHDEN OHDSI data with Schema.org markup
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/eubopen-hcs-bioimage.html">
     6. Depositing IMI EUBOPEN High-Content Screening data to EBI BioImage Archive
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/approach-cdisc.html">
     7. Mapping IMI APPROACH datasets to CDISC-SDTM standard
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  AFTERWORD
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../afterword/fair-reflections.html">
   Lessons learned from the FAIR journey and project outlook
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  GLOSSARY
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/glossary-fair-terms.html">
   Glossary of terms and abbreviations
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  JOIN US
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../community.html">
   Community
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../afterword/people.html">
     Boards and contributors
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../afterword/code_of_conduct.html">
     Code of conduct
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../afterword.html">
     Platform
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../afterword/the-turing-way.html">
       Leveraging the Turing Way Book
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../help.html">
   Contribute
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../afterword/contributing.html">
     1. How to contribute
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/how-to-create-recipe-with-gdoc.html">
     2. Add via GoogleDoc
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/how-to-create-recipe-with-hackmd.html">
     3. Add via HackMD
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/how-to-create-recipe-with-git.html">
     4. Add via Git
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/recipe-template.html">
     5. Git recipe template
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/tips-tricks.html">
     6. Tips and tricks
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/myst.html">
     7. Markdown cheatsheet
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>

</nav>
        
        
             <!-- To handle the deprecated key -->

        
    </div>

    <div class="text-center px-5 lighten-3 grey py-2">
        Powered by <a href="https://jupyterbook.org" class="secondary--text">Jupyter Book</a>
    </div>
</div>
                        </div>
                        
                            <main class="bd-content" role="main" style="width:1%; flex-grow:1;">
                                <v-container fluid id="main-content" class="pa-0">
                                    <v-row class="d-flex">
                                        
                                            <v-col class="col-xl-10 col-lg-10 col-md-9 col-12 px-10 pt-10 mt-3">
                                                
  <div class="section" id="creating-a-community-annotation-compliant-isa-descriptor-dataset-maturity-level-2">
<h1>Creating a community annotation compliant ISA descriptor: Dataset Maturity Level 2<a class="headerlink" href="#creating-a-community-annotation-compliant-isa-descriptor-dataset-maturity-level-2" title="Permalink to this headline">¶</a></h1>
<br/>
<br/>
<script type="application/ld+json"> {"@context": {"bs": "https://bioschema.org/", "dc": "http://purl.org/dc/terms/", "owl": "http://www.w3.org/2002/07/owl#", "sdo": "https://schema.org/"}, "@type": ["sdo:HowTo", "sdo:LearningResource"], "@id": "https://w3id.org/faircookbook/FCB064", "sdo:name": "Enhanced Data Maturity with ISA - ISA-JSON and ontology markup", "sdo:description": "Enhanced Data Maturity with ISA - ISA-JSON and ontology markup", "sdo:keywords": [], "dc:conformsTo": {"@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE", "@type": "CreativeWork"}, "sdo:inLanguage": ["en-GB"], "sdo:audience": ["Principal Investigator|T4FS_0000441", "Data Manager|T4FS_0000520", "Data Scientist|T4FS_0000566"], "sdo:isPartOf": [{"@type": "CreativeWork", "@id": "https://fairplus.github.io/the-fair-cookbook/", "sdo:url": "https://fairplus.github.io/the-fair-cookbook/", "sdo:name": "The FAIR Cookbook"}], "sdo:learningResourceType": "Hands-on", "sdo:identifier": {"@type": "PropertyValue", "@id": "", "sdo:name": "FCB064", "dso:url": "https://w3id.org/faircookbook/FCB064", "owl:sameAs": "https://identifiers.org/fcb:FCB064"}, "sdo:license": {"@type": "CreativeWork", "@id": "https://creativecommons.org/licenses/by/4.0/", "sdo:url": "https://creativecommons.org/licenses/by/4.0/", "sdo:name": "CC-BY-4.0"}, "sdo:timeRequired": "PT0H15M0S", "sdo:totalTime": "PT0H15M0S", "sdo:step": [{"@type": "sdo:HowToSection", "sdo:position": "1", "sdo:name": "Overview", "sdo:url": "https://w3id.org/faircookbook/FCB064#main-objectives", "sdo:itemListElement": [{"@type": "sdo:HowToStep", "sdo:position": "1", "sdo:name": "", "sdo:itemListElement": [{"@type": "sdo:HowToDirection", "sdo:position": "1", "sdo:name": "step-by-step section", "sdo:text": "Follow the step-by-step section.", "sdo:url": "https://w3id.org/faircookbook/FCB064#main-objectives"}, {"@type": "sdo:HowToTip", "sdo:position": "2", "sdo:name": "Reporting an issue", "sdo:text": "Use our GitHub issue tracker to report any error. Thanks!", "sdo:url": "https://github.com/FAIRplus/the-fair-cookbook/issues/new/choose"}]}]}, {"@type": "sdo:HowToSection", "sdo:position": "2", "sdo:name": "Conclusion", "sdo:url": "https://w3id.org/faircookbook/FCB064#conclusion", "sdo:itemListElement": [{"@type": "sdo:HowToStep", "sdo:position": "1", "sdo:name": "", "sdo:itemListElement": [{"@type": "sdo:HowToDirection", "sdo:position": "1", "sdo:name": "conclusion section", "sdo:text": "What to read next?", "sdo:url": "https://w3id.org/faircookbook/FCB064#conclusion"}, {"@type": "sdo:HowToTip", "sdo:position": "2", "sdo:name": "Reporting an issue", "sdo:text": "Use our GitHub issue tracker to report any error. Thanks!", "sdo:url": "https://github.com/FAIRplus/the-fair-cookbook/issues/new/choose"}]}]}]} </script>
<div class="card my-10 recipe d-flex">
<div class="card-header purple-dark text--white">
<div class="overview">
<div class="overviewLabel">Recipe Overview</div>
<div class="overviewContent">
<div class="item"><div class="itemContainer">
<i class="itemIcon fa fa-clock fa-2x text--white"></i>
<div class="itemContent">
<div class="label"> Reading Time </div>
15 minutes
</div>
</div></div>
<div class="item"><div class="itemContainer">
<i class="itemIcon fa fa-play-circle fa-2x text--white"></i>
<div class="itemContent">
<div class="label"> Executable Code </div>
Yes
</div>
</div></div>
<div class="item" style="margin-bottom:10px"><div class="itemContainer">
<i class="itemIcon fa fa-fire fa-2x text--white" style="margin-left:4px;margin-right: 5px;"></i>
<div class="itemContent">
<div class="label" style="position:relative; left:-6px">Difficulty</div>
<i class="fa fa-fire"></i>
<i class="fa fa-fire"></i>
<i class="fa fa-fire" style="color:grey"></i>
<i class="fa fa-fire" style="color:grey"></i>
<i class="fa fa-fire" style="color:grey"></i>
</div>
</div></div>
</div></div>
</div>
<div class="card-body text--purple-dark">
<div class="title">
<div style="flex-grow:1; margin-right:5px">Enhanced Data Maturity with ISA - ISA-JSON and ontology markup</div>
<img src="https://fairplus.github.io/the-fair-cookbook/_static/images/fairplus-mini.png" alt="FAIRPlus logo"> </img>
</div>
<div class="section">
<i class="sectionIcon fa fa-laptop fa-2x"></i><div class="sectionContent">
<div class="label">Recipe Type</div>
<div class="sectionValue">Hands-on</div>
</div>
</div>
<div class="section" style="flex-grow: 1;">
<i class="sectionIcon fa fa-users fa-2x"></i><div class="sectionContent">
<div class="label">Audience</div>
<div class="sectionValue"><a href="http://purl.obolibrary.org/obo/T4FS_0000441">Principal Investigator</a>, <a href="http://purl.obolibrary.org/obo/T4FS_0000520">Data Manager</a>, <a href="http://purl.obolibrary.org/obo/T4FS_0000566">Data Scientist</a></div>
</div>
</div>
<div class="section" style="flex-grow: 1;">
<svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="signal" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 600" class="bars level-2">
<path class="bar5" d="M216 288h-48c-8.84 0-16 7.16-16 16v192c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V304c0-8.84-7.16-16-16-16zM88 384H40c-8.84 0-16 7.16-16 16v96c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16v-96c0-8.84-7.16-16-16-16zm256-192h-48c-8.84 0-16 7.16-16 16v288c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V208c0-8.84-7.16-16-16-16zm128-96h-48c-8.84 0-16 7.16-16 16v384c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V112c0-8.84-7.16-16-16-16zM600 0h-48c-8.84 0-16 7.16-16 16v480c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V16c0-8.84-7.16-16-16-16z"/>
<path class="bar4" d="M216 288h-48c-8.84 0-16 7.16-16 16v192c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V304c0-8.84-7.16-16-16-16zM88 384H40c-8.84 0-16 7.16-16 16v96c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16v-96c0-8.84-7.16-16-16-16zm256-192h-48c-8.84 0-16 7.16-16 16v288c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V208c0-8.84-7.16-16-16-16zm128-96h-48c-8.84 0-16 7.16-16 16v384c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V112c0-8.84-7.16-16-16-16zM600" />
<path class="bar3" d="M216 288h-48c-8.84 0-16 7.16-16 16v192c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V304c0-8.84-7.16-16-16-16zM88 384H40c-8.84 0-16 7.16-16 16v96c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16v-96c0-8.84-7.16-16-16-16zm256-192h-48c-8.84 0-16 7.16-16 16v288c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V208c0-8.84-7.16-16-16-16zm128-96h-48c-8.84" />
<path class="bar1" d="M216 288h-48c-8.84 0-16 7.16-16 16v192c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V304c0-8.84-7.16-16-16-16zM88 384H40c-8.84 0-16 7.16-16 16v96c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16v-96c0-8.84-7.16-16-16-16zm256-192h-48c-8.84"/>
<path class="bar2" d="M216 288h-48c-8.84 0-16 7.16-16 16v192c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V304c0-8.84-7.16-16-16-16zM88 384H40c-8.84"/>
</svg>
<div class="sectionContent">
<div class="label">Maturity Level & Indicator</div>
<div class="sectionValue"><v-chip-group><v-chip href="https://w3id.org/Data-Maturity/DSM-1-C1" class="secondary white--text" target="_blank" label>DSM-1-C1</v-chip></v-chip-group></div>
<div class="tooltip">hover me
<span class="tooltiptext">Tooltip text</span></div>
</div>
</div><div class="card-footer text--orange sphinx-bs.badge.badge-primary"> Cite me with 
<a href="https://w3id.org/faircookbook/FCB064" class="text--purple-dark">FCB064
</a></div>
</div>
</div>
<div class="section" id="abstract">
<h2>Abstract:<a class="headerlink" href="#abstract" title="Permalink to this headline">¶</a></h2>
<p>With the recipe, we will continue using the ISA model to capture study metadata and dataset description, but we will be
<strong>moving up</strong> on the scale of <strong>data set maturity</strong> by showing how to implement four important aspects:</p>
<ul class="simple">
<li><p>community agreed experimental design descriptors</p></li>
<li><p>community agreed annotation levels for description of biological materials</p></li>
<li><p>extensive semantic annotation and reduction of free text annotation</p></li>
<li><p>machine-readable and more interoperable format: serialization to ISA-JSON</p></li>
</ul>
</div>
<div class="section" id="let-s-get-started-loading-the-isa-api">
<h2>Let’s get started! Loading the ISA-API<a class="headerlink" href="#let-s-get-started-loading-the-isa-api" title="Permalink to this headline">¶</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">isatools.model</span> <span class="kn">import</span> <span class="p">(</span>
    <span class="n">Comment</span><span class="p">,</span>
    <span class="n">Investigation</span><span class="p">,</span>
    <span class="n">Study</span><span class="p">,</span>
    <span class="n">StudyFactor</span><span class="p">,</span>
    <span class="n">FactorValue</span><span class="p">,</span>
    <span class="n">OntologyAnnotation</span><span class="p">,</span>
    <span class="n">Characteristic</span><span class="p">,</span>
    <span class="n">OntologySource</span><span class="p">,</span>
    <span class="n">Material</span><span class="p">,</span>
    <span class="n">Sample</span><span class="p">,</span>
    <span class="n">Source</span><span class="p">,</span>
    <span class="n">Protocol</span><span class="p">,</span>
    <span class="n">ProtocolParameter</span><span class="p">,</span>
    <span class="n">ParameterValue</span><span class="p">,</span>
    <span class="n">Process</span><span class="p">,</span>
    <span class="n">Publication</span><span class="p">,</span>
    <span class="n">Person</span><span class="p">,</span>
    <span class="n">Assay</span><span class="p">,</span>
    <span class="n">DataFile</span><span class="p">,</span>
    <span class="n">plink</span>
<span class="p">)</span>

<span class="kn">import</span> <span class="nn">os</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="creating-an-isa-investigation">
<h2>Creating an ISA investigation<a class="headerlink" href="#creating-an-isa-investigation" title="Permalink to this headline">¶</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">investigation</span> <span class="o">=</span> <span class="n">Investigation</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="declaring-semantic-resources-used-to-annotated-isa-objects">
<h2>Declaring <code class="docutils literal notranslate"><span class="pre">Semantic</span> <span class="pre">Resources</span></code> used to annotated ISA objects<a class="headerlink" href="#declaring-semantic-resources-used-to-annotated-isa-objects" title="Permalink to this headline">¶</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">ncbitaxon</span> <span class="o">=</span> <span class="n">OntologySource</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;NCIBTaxon&quot;</span><span class="p">,</span>
                           <span class="n">description</span><span class="o">=</span><span class="s2">&quot;NCBI Taxonomy&quot;</span><span class="p">)</span>
<span class="n">efo</span>   <span class="o">=</span> <span class="n">OntologySource</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;EFO&quot;</span><span class="p">,</span>
                       <span class="n">description</span><span class="o">=</span><span class="s2">&quot;Experimental Factor Ontology&quot;</span><span class="p">)</span>
<span class="n">obi</span>   <span class="o">=</span> <span class="n">OntologySource</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s1">&#39;OBI&#39;</span><span class="p">,</span>
                       <span class="n">description</span><span class="o">=</span><span class="s2">&quot;Ontology for Biomedical Investigations&quot;</span><span class="p">)</span>
<span class="n">chebi</span> <span class="o">=</span> <span class="n">OntologySource</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;CHEBI&quot;</span><span class="p">,</span>
                       <span class="n">description</span><span class="o">=</span><span class="s2">&quot;Chemical Entity of Biological Interest&quot;</span><span class="p">)</span>
<span class="n">pato</span>  <span class="o">=</span> <span class="n">OntologySource</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s1">&#39;PATO&#39;</span><span class="p">,</span>
                       <span class="n">description</span><span class="o">=</span><span class="s2">&quot;Phenotype and Trait Ontology&quot;</span><span class="p">)</span>

<span class="n">investigation</span><span class="o">.</span><span class="n">ontology_source_references</span> <span class="o">=</span> <span class="p">[</span><span class="n">chebi</span><span class="p">,</span><span class="n">efo</span><span class="p">,</span><span class="n">obi</span><span class="p">,</span><span class="n">pato</span><span class="p">,</span><span class="n">ncbitaxon</span><span class="p">]</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="building-and-annotation-isa-objects-to-a-community-compliant-level">
<h2>Building and Annotation ISA objects to a <code class="docutils literal notranslate"><span class="pre">Community</span> <span class="pre">Compliant</span></code> level<a class="headerlink" href="#building-and-annotation-isa-objects-to-a-community-compliant-level" title="Permalink to this headline">¶</a></h2>
<p>The annotation requirements we are dealing with are:</p>
<ul class="simple">
<li><p>those of <a class="reference internal" href="../maturity.html"><span class="doc std std-doc">ENA Short Read Archive Repository</span></a> on the one hand</p></li>
<li><p>those of a Minimal Information checklist: MINSEQE minimum information for a sequencing experiment.</p></li>
</ul>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">study</span> <span class="o">=</span> <span class="n">Study</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="s2">&quot;s_BII-S-3-synthesis.txt&quot;</span><span class="p">)</span>
<span class="n">study</span><span class="o">.</span><span class="n">identifier</span> <span class="o">=</span> <span class="s2">&quot;BII-S-3-synth&quot;</span>
<span class="n">study</span><span class="o">.</span><span class="n">title</span> <span class="o">=</span> <span class="s2">&quot;Metagenomes and Metatranscriptomes of phytoplankton blooms from an ocean acidification mesocosm experiment&quot;</span>
<span class="n">study</span><span class="o">.</span><span class="n">description</span> <span class="o">=</span> <span class="s2">&quot;Sequencing the metatranscriptome can provide information about the response of organisms to </span><span class="se">\</span>
<span class="s2">varying environmental conditions. We present a methodology for obtaining random whole-community mRNA from a complex </span><span class="se">\</span>
<span class="s2">microbial assemblage using Pyrosequencing. The metatranscriptome had, with minimum contamination by ribosomal RNA,</span><span class="se">\</span>
<span class="s2">significant coverage of abundant transcripts, and included significantly more potentially novel proteins than in the</span><span class="se">\</span>
<span class="s2">metagenome. This experiment is part of a much larger experiment. We have produced 4 454 metatranscriptomic datasets </span><span class="se">\</span>
<span class="s2">and 6 454 metagenomic datasets. These were derived from 4 samples.&quot;</span>
<span class="n">study</span><span class="o">.</span><span class="n">submission_date</span> <span class="o">=</span> <span class="s2">&quot;2008-08-15&quot;</span>
<span class="n">study</span><span class="o">.</span><span class="n">public_release_date</span> <span class="o">=</span> <span class="s2">&quot;2008-08-15&quot;</span>
</pre></div>
</div>
</div>
</div>
<p>These NCBI SRA related ISA Comments fields are required and must be present for the <code class="docutils literal notranslate"><span class="pre">ISA</span> <span class="pre">SRAconverter</span></code> to be invoked later</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">src_comment_sra1</span> <span class="o">=</span> <span class="n">Comment</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;SRA Broker Name&quot;</span><span class="p">,</span> <span class="n">value</span><span class="o">=</span><span class="s2">&quot;OXFORD&quot;</span><span class="p">)</span>
<span class="n">src_comment_sra2</span> <span class="o">=</span> <span class="n">Comment</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;SRA Center Name&quot;</span><span class="p">,</span> <span class="n">value</span><span class="o">=</span><span class="s2">&quot;OXFORD&quot;</span><span class="p">)</span>
<span class="n">src_comment_sra3</span> <span class="o">=</span> <span class="n">Comment</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;SRA Center Project Name&quot;</span><span class="p">,</span> <span class="n">value</span><span class="o">=</span><span class="s2">&quot;OXFORD&quot;</span><span class="p">)</span>
<span class="n">src_comment_sra4</span> <span class="o">=</span> <span class="n">Comment</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;SRA Lab Name&quot;</span><span class="p">,</span> <span class="n">value</span><span class="o">=</span><span class="s2">&quot;Oxford e-Research Centre&quot;</span><span class="p">)</span>
<span class="n">src_comment_sra5</span> <span class="o">=</span> <span class="n">Comment</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;SRA Submission Action&quot;</span><span class="p">,</span> <span class="n">value</span><span class="o">=</span><span class="s2">&quot;ADD&quot;</span><span class="p">)</span>
<span class="n">study</span><span class="o">.</span><span class="n">comments</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">src_comment_sra1</span><span class="p">)</span>
<span class="n">study</span><span class="o">.</span><span class="n">comments</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">src_comment_sra2</span><span class="p">)</span>
<span class="n">study</span><span class="o">.</span><span class="n">comments</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">src_comment_sra3</span><span class="p">)</span>
<span class="n">study</span><span class="o">.</span><span class="n">comments</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">src_comment_sra4</span><span class="p">)</span>
<span class="n">study</span><span class="o">.</span><span class="n">comments</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">src_comment_sra5</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>These <code class="docutils literal notranslate"><span class="pre">ISA</span> <span class="pre">Comments</span></code> are optional and may be used to report <strong>funding information</strong></p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">src_comment_st1</span> <span class="o">=</span> <span class="n">Comment</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;Study Funding Agency&quot;</span><span class="p">,</span> <span class="n">value</span><span class="o">=</span><span class="s2">&quot;&quot;</span><span class="p">)</span>
<span class="n">src_comment_st2</span> <span class="o">=</span> <span class="n">Comment</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;Study Grant Number&quot;</span><span class="p">,</span> <span class="n">value</span><span class="o">=</span><span class="s2">&quot;&quot;</span><span class="p">)</span>
<span class="n">study</span><span class="o">.</span><span class="n">comments</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">src_comment_st1</span><span class="p">)</span>
<span class="n">study</span><span class="o">.</span><span class="n">comments</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">src_comment_st2</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="declaring-all-the-protocols-used-in-the-isa-study">
<h3>Declaring all the protocols used in the ISA study.<a class="headerlink" href="#declaring-all-the-protocols-used-in-the-isa-study" title="Permalink to this headline">¶</a></h3>
<div class="admonition-tip admonition">
<p class="admonition-title">tip</p>
<p>Note also the declaration of <code class="docutils literal notranslate"><span class="pre">ISA</span> <span class="pre">Protocol</span> <span class="pre">Parameters</span></code> when needed.</p>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">study</span><span class="o">.</span><span class="n">protocols</span> <span class="o">=</span> <span class="p">[</span> 
    <span class="n">Protocol</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;environmental material collection - standard procedure 1&quot;</span><span class="p">,</span>
             <span class="n">description</span><span class="o">=</span><span class="s2">&quot;Waters samples were prefiltered through a 1.6 um GF/A glass fibre filter to reduce Eukaryotic </span><span class="se">\</span>
<span class="s2">             contamination. Filtrate was then collected on a 0.2 um Sterivex (millipore) filter which was frozen in </span><span class="se">\</span>
<span class="s2">             liquid nitrogen until nucelic acid extraction. CO2 bubbled through 11000 L mesocosm to simulate ocean </span><span class="se">\</span>
<span class="s2">             acidification predicted conditions. Then phosphate and nitrate were added to induce a phytoplankton bloom.&quot;</span><span class="p">,</span>
             <span class="n">protocol_type</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;sample collection&quot;</span><span class="p">),</span>
             <span class="n">parameters</span><span class="o">=</span><span class="p">[</span>
              <span class="n">ProtocolParameter</span><span class="p">(</span><span class="n">parameter_name</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;filter pore size&quot;</span><span class="p">))</span>
             <span class="p">]</span>
            <span class="p">),</span>
    <span class="n">Protocol</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;aliquoting-0&quot;</span><span class="p">,</span>
             <span class="n">description</span><span class="o">=</span><span class="s2">&quot;aliquoting&quot;</span><span class="p">,</span>
             <span class="n">protocol_type</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;sample collection&quot;</span><span class="p">)),</span>
    <span class="n">Protocol</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;nucleic acid extraction&quot;</span><span class="p">,</span>
             <span class="n">description</span><span class="o">=</span><span class="s2">&quot;Total nucleic acid extraction was done as quickly as possible using the method of </span><span class="se">\</span>
<span class="s2">             Neufeld et al, 2007.&quot;</span><span class="p">,</span>
             <span class="n">protocol_type</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;nucleic acid extraction&quot;</span><span class="p">)</span>
    <span class="p">),</span>
    <span class="n">Protocol</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;mRNA extraction - standard procedure 3&quot;</span><span class="p">,</span>
             <span class="n">description</span><span class="o">=</span><span class="s2">&quot;RNA MinElute + substrative Hybridization + MEGAclear For transcriptomics, total RNA</span><span class="se">\</span>
<span class="s2">             was separated from the columns using the RNA MinElute clean-up kit (Qiagen) and checked for integrity of</span><span class="se">\</span>
<span class="s2">             rRNA using an Agilent bioanalyser (RNA nano6000 chip). High integrity rRNA is essential for subtractive</span><span class="se">\</span>
<span class="s2">             hybridization. Samples were treated with Turbo DNA-free enzyme (Ambion) to remove contaminating DNA.</span><span class="se">\</span>
<span class="s2">             The rRNA was removed from mRNA by subtractive hybridization (Microbe Express Kit, Ambion), and absence</span><span class="se">\</span>
<span class="s2">             of rRNA and DNA contamination was confirmed using the Agilent bioanalyser. The mRNA was further</span><span class="se">\</span>
<span class="s2">             purified with the MEGAclearTM kit (Ambion). Reverse transcription of mRNA was performed using</span><span class="se">\</span>
<span class="s2">             the SuperScript III enzyme (Invitrogen) with random hexamer primers (Promega).</span><span class="se">\</span>
<span class="s2">             The cDNA was treated with RiboShredderTM RNase Blend (Epicentre) to remove trace RNA contaminants. </span><span class="se">\</span>
<span class="s2">             To improve the yield of cDNA, samples were subjected to random amplification using the </span><span class="se">\</span>
<span class="s2">             GenomiPhi V2 method (GE Healthcare). GenomiPhi technology produces branched DNA molecules </span><span class="se">\</span>
<span class="s2">             that are recalcitrant to the pyrosequencing methodology. Therefore amplified samples were treated with</span><span class="se">\</span>
<span class="s2">             S1 nuclease using the method of Zhang et al.2006.&quot;</span><span class="p">,</span>
             <span class="n">protocol_type</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;labeling&quot;</span><span class="p">)</span> <span class="c1">#nucleic acid extraction</span>
    <span class="p">),</span>
    <span class="n">Protocol</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;genomic DNA extraction - standard procedure 4&quot;</span><span class="p">,</span>
             <span class="n">description</span><span class="o">=</span><span class="s2">&quot;superscript+random hexamer primer&quot;</span><span class="p">,</span>
             <span class="n">protocol_type</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;nucleic acid extraction&quot;</span><span class="p">)</span>
    <span class="p">),</span>
    <span class="n">Protocol</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;reverse transcription - standard procedure 5&quot;</span><span class="p">,</span>
             <span class="n">description</span><span class="o">=</span><span class="s2">&quot;&quot;</span><span class="p">,</span>
             <span class="n">protocol_type</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;reverse transcription&quot;</span><span class="p">),</span>
    <span class="p">),</span>
     <span class="n">Protocol</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;library construction&quot;</span><span class="p">,</span>
              <span class="n">description</span><span class="o">=</span><span class="s2">&quot;&quot;</span><span class="p">,</span>
              <span class="n">protocol_type</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;library construction&quot;</span><span class="p">),</span>
              <span class="n">parameters</span><span class="o">=</span><span class="p">[</span><span class="n">ProtocolParameter</span><span class="p">(</span><span class="n">parameter_name</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;library strategy&quot;</span><span class="p">)),</span>
                          <span class="n">ProtocolParameter</span><span class="p">(</span><span class="n">parameter_name</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;library layout&quot;</span><span class="p">)),</span>
                          <span class="n">ProtocolParameter</span><span class="p">(</span><span class="n">parameter_name</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;library selection&quot;</span><span class="p">))</span>
            <span class="p">]</span>
    <span class="p">),</span>   

    <span class="n">Protocol</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;nucleic acid sequencing&quot;</span><span class="p">,</span> <span class="c1">#pyrosequencing - standard procedure 6&quot;,</span>
             <span class="n">description</span><span class="o">=</span><span class="s2">&quot;1. Sample Input and Fragmentation: The Genome Sequencer FLX System supports the sequencing of </span><span class="se">\</span>
<span class="s2">             samples from a wide variety of starting materials including genomic DNA, PCR products, BACs, and cDNA. </span><span class="se">\</span>
<span class="s2">             Samples such as genomic DNA and BACs are fractionated into small, 300- to 800-base pair fragments. </span><span class="se">\</span>
<span class="s2">             For smaller samples, such as small non-coding RNA or PCR amplicons, fragmentation is not required. </span><span class="se">\</span>
<span class="s2">             Instead, short PCR products amplified using Genome Sequencer fusion primers can be used for immobilization</span><span class="se">\</span>
<span class="s2">             onto DNA capture beads as shown below.&quot;</span><span class="p">,</span>
             <span class="n">protocol_type</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;nucleic acid sequencing&quot;</span><span class="p">),</span>
             <span class="n">parameters</span><span class="o">=</span><span class="p">[</span><span class="n">ProtocolParameter</span><span class="p">(</span><span class="n">parameter_name</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;sequencing instrument&quot;</span><span class="p">))</span>
            <span class="p">]</span>
    <span class="p">),</span>
    <span class="n">Protocol</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;sequence analysis - standard procedure 7&quot;</span><span class="p">,</span>
             <span class="n">description</span><span class="o">=</span><span class="s2">&quot;&quot;</span><span class="p">,</span>
             <span class="n">protocol_type</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;data transformation&quot;</span><span class="p">)</span>
    <span class="p">)</span>
<span class="p">]</span>
</pre></div>
</div>
</div>
</div>
<p>Adding a Study Design descriptor to the ISA Study object</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">intervention_design</span> <span class="o">=</span> <span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term_source</span><span class="o">=</span><span class="n">obi</span><span class="p">)</span>
<span class="n">intervention_design</span><span class="o">.</span><span class="n">term</span> <span class="o">=</span> <span class="s2">&quot;intervention design&quot;</span>
<span class="n">intervention_design</span><span class="o">.</span><span class="n">term_accession</span> <span class="o">=</span> <span class="s2">&quot;http://purl.obolibrary.org/obo/OBI_0000115&quot;</span>

<span class="n">study</span><span class="o">.</span><span class="n">design_descriptors</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">intervention_design</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div>
<div class="section" id="declaring-the-isa-study-factors-i-e-the-independent-variables-of-the-experiment">
<h2>Declaring the <code class="docutils literal notranslate"><span class="pre">ISA</span> <span class="pre">Study</span> <span class="pre">Factors</span></code>, i.e. The Independent Variables of the Experiment<a class="headerlink" href="#declaring-the-isa-study-factors-i-e-the-independent-variables-of-the-experiment" title="Permalink to this headline">¶</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">study</span><span class="o">.</span><span class="n">factors</span> <span class="o">=</span> <span class="p">[</span>
    <span class="n">StudyFactor</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;compound&quot;</span><span class="p">,</span>
                <span class="n">factor_type</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;chemical substance&quot;</span><span class="p">,</span>
                                               <span class="n">term_accession</span><span class="o">=</span><span class="s2">&quot;http://purl.obolibrary.org/obo/CHEBI_59999&quot;</span><span class="p">,</span>
                                               <span class="n">term_source</span><span class="o">=</span><span class="n">chebi</span><span class="p">)),</span>
    <span class="n">StudyFactor</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;dose&quot;</span><span class="p">,</span>
                <span class="n">factor_type</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;dose&quot;</span><span class="p">,</span>
                                               <span class="n">term_accession</span><span class="o">=</span><span class="s2">&quot;http://www.ebi.ac.uk/efo/EFO_0000428&quot;</span><span class="p">,</span>
                                               <span class="n">term_source</span><span class="o">=</span><span class="n">efo</span><span class="p">)),</span>
    <span class="n">StudyFactor</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;collection time&quot;</span><span class="p">,</span>
                <span class="n">factor_type</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;time&quot;</span><span class="p">,</span>
                                               <span class="n">term_accession</span><span class="o">=</span><span class="s2">&quot;http://purl.obolibrary.org/obo/PATO_0000165&quot;</span><span class="p">,</span>
                                               <span class="n">term_source</span><span class="o">=</span><span class="n">pato</span><span class="p">))</span>
<span class="p">]</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="declaring-the-isa-factorvalues-i-e-the-factor-levels-to-each-of-the-indepedent-variables-or-isa-study-factor">
<h2>Declaring the <code class="docutils literal notranslate"><span class="pre">ISA</span> <span class="pre">FactorValues</span></code> i.e. The factor levels to each of the Indepedent Variables or <code class="docutils literal notranslate"><span class="pre">ISA</span> <span class="pre">Study</span> <span class="pre">Factor</span></code>.<a class="headerlink" href="#declaring-the-isa-factorvalues-i-e-the-factor-levels-to-each-of-the-indepedent-variables-or-isa-study-factor" title="Permalink to this headline">¶</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fv1</span> <span class="o">=</span> <span class="n">FactorValue</span><span class="p">(</span><span class="n">factor_name</span><span class="o">=</span><span class="n">study</span><span class="o">.</span><span class="n">factors</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">value</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;carbon dioxide&quot;</span><span class="p">))</span>
<span class="n">fv2</span> <span class="o">=</span> <span class="n">FactorValue</span><span class="p">(</span><span class="n">factor_name</span><span class="o">=</span><span class="n">study</span><span class="o">.</span><span class="n">factors</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">value</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;high&quot;</span><span class="p">))</span>
<span class="n">fv3</span> <span class="o">=</span> <span class="n">FactorValue</span><span class="p">(</span><span class="n">factor_name</span><span class="o">=</span><span class="n">study</span><span class="o">.</span><span class="n">factors</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">value</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;normal&quot;</span><span class="p">))</span>
<span class="n">fv4</span> <span class="o">=</span> <span class="n">FactorValue</span><span class="p">(</span><span class="n">factor_name</span><span class="o">=</span><span class="n">study</span><span class="o">.</span><span class="n">factors</span><span class="p">[</span><span class="mi">2</span><span class="p">],</span> <span class="n">value</span><span class="o">=</span><span class="s2">&quot;may 13th, 2006&quot;</span><span class="p">)</span>
<span class="n">fv5</span> <span class="o">=</span> <span class="n">FactorValue</span><span class="p">(</span><span class="n">factor_name</span><span class="o">=</span><span class="n">study</span><span class="o">.</span><span class="n">factors</span><span class="p">[</span><span class="mi">2</span><span class="p">],</span> <span class="n">value</span><span class="o">=</span><span class="s2">&quot;may 19th, 2006&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="adding-the-publications-associated-to-the-study">
<h2>Adding the publications associated to the study<a class="headerlink" href="#adding-the-publications-associated-to-the-study" title="Permalink to this headline">¶</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">study</span><span class="o">.</span><span class="n">publications</span> <span class="o">=</span> <span class="p">[</span>
    <span class="n">Publication</span><span class="p">(</span><span class="n">doi</span><span class="o">=</span><span class="s2">&quot;10.1371/journal.pone.0003042&quot;</span><span class="p">,</span>
                <span class="n">pubmed_id</span><span class="o">=</span><span class="s2">&quot;18725995&quot;</span><span class="p">,</span>
                <span class="n">title</span><span class="o">=</span><span class="s2">&quot;Detection of large numbers of novel sequences in the metatranscriptomes </span><span class="se">\</span>
<span class="s2">                      of complex marine microbial communities.&quot;</span><span class="p">,</span>
                <span class="n">status</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;indexed in PubMed&quot;</span><span class="p">),</span>
                <span class="n">author_list</span><span class="o">=</span><span class="s2">&quot;Gilbert JA, Field D, Huang Y, Edwards R, Li W, Gilna P, Joint I.&quot;</span><span class="p">),</span>
    <span class="n">Publication</span><span class="p">(</span><span class="n">doi</span><span class="o">=</span><span class="s2">&quot;10.1111/j.1462-2920.2008.01745.x&quot;</span><span class="p">,</span>
                <span class="n">title</span><span class="o">=</span><span class="s2">&quot;Potential for phosphonoacetate utilization by marine bacteria in temperate coastal waters&quot;</span><span class="p">,</span>
                <span class="n">pubmed_id</span><span class="o">=</span><span class="s2">&quot;18783384&quot;</span><span class="p">,</span>
                <span class="n">status</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;indexed in PubMed&quot;</span><span class="p">),</span>
                <span class="n">author_list</span><span class="o">=</span><span class="s2">&quot;Gilbert JA, Thomas S, Cooley NA, Kulakova A, Field D, Booth T, McGrath JW, Quinn JP, Joint I.&quot;</span><span class="p">)</span>
<span class="p">]</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="adding-the-authors-of-the-study">
<h2>Adding the authors of the study<a class="headerlink" href="#adding-the-authors-of-the-study" title="Permalink to this headline">¶</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">study</span><span class="o">.</span><span class="n">contacts</span> <span class="o">=</span> <span class="p">[</span>
    <span class="n">Person</span><span class="p">(</span><span class="n">first_name</span><span class="o">=</span><span class="s2">&quot;Jack&quot;</span><span class="p">,</span>
           <span class="n">last_name</span><span class="o">=</span><span class="s2">&quot;Gilbert&quot;</span><span class="p">,</span>
           <span class="n">affiliation</span><span class="o">=</span><span class="s2">&quot;Plymouth Marine Laboratory&quot;</span><span class="p">,</span>
           <span class="n">email</span><span class="o">=</span><span class="s2">&quot;jagi@pml.ac.uk&quot;</span><span class="p">,</span>
           <span class="n">address</span><span class="o">=</span><span class="s2">&quot;Prospect Place, Plymouth, United Kingdom&quot;</span><span class="p">,</span>
           <span class="n">comments</span><span class="o">=</span><span class="p">[</span><span class="n">Comment</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;Study Person REF&quot;</span><span class="p">,</span>
                             <span class="n">value</span><span class="o">=</span><span class="s2">&quot;&quot;</span><span class="p">)],</span>
           <span class="n">roles</span><span class="o">=</span><span class="p">[</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;principal investigator role&quot;</span><span class="p">),</span>
                   <span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;SRA Inform On Status&quot;</span><span class="p">),</span>
                   <span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;SRA Inform On Error&quot;</span><span class="p">)]</span>
    <span class="p">),</span>
    <span class="n">Person</span><span class="p">(</span><span class="n">first_name</span><span class="o">=</span><span class="s2">&quot;Dawn&quot;</span><span class="p">,</span>
           <span class="n">last_name</span><span class="o">=</span><span class="s2">&quot;Field&quot;</span><span class="p">,</span>
           <span class="n">affiliation</span><span class="o">=</span><span class="s2">&quot;NERC Centre for Ecology and Hydrology&quot;</span><span class="p">,</span>
           <span class="n">address</span><span class="o">=</span><span class="s2">&quot;CEH Oxford, Oxford, United Kingdom&quot;</span><span class="p">,</span>
           <span class="n">comments</span><span class="o">=</span><span class="p">[</span><span class="n">Comment</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;Study Person REF&quot;</span><span class="p">,</span>
                             <span class="n">value</span><span class="o">=</span><span class="s2">&quot;&quot;</span><span class="p">)],</span>
           <span class="n">roles</span><span class="o">=</span><span class="p">[</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;principal investigator role&quot;</span><span class="p">)]</span>
    <span class="p">)</span>
<span class="p">]</span>
</pre></div>
</div>
</div>
</div>
<p>Now, creating ISA Source and Sample objects and building what is the BII-S-3 Study table</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">study</span><span class="o">.</span><span class="n">sources</span> <span class="o">=</span> <span class="p">[</span><span class="n">Source</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;GSM255770&quot;</span><span class="p">),</span> <span class="n">Source</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;GSM255771&quot;</span><span class="p">),</span><span class="n">Source</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;GSM255772&quot;</span><span class="p">),</span><span class="n">Source</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;GSM255773&quot;</span><span class="p">)]</span>
<span class="n">study</span><span class="o">.</span><span class="n">samples</span> <span class="o">=</span> <span class="p">[</span><span class="n">Sample</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;GSM255770&quot;</span><span class="p">),</span> <span class="n">Sample</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;GSM255771&quot;</span><span class="p">),</span> <span class="n">Sample</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;GSM255772&quot;</span><span class="p">),</span> <span class="n">Sample</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;GSM255773&quot;</span><span class="p">)]</span>

<span class="c1"># Note how the treatment groups are defined as sets of factor values attached to the ISA.Sample object</span>
<span class="n">study</span><span class="o">.</span><span class="n">samples</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">factor_values</span><span class="o">=</span><span class="p">[</span><span class="n">fv1</span><span class="p">,</span><span class="n">fv2</span><span class="p">,</span><span class="n">fv4</span><span class="p">]</span>
<span class="n">study</span><span class="o">.</span><span class="n">samples</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">factor_values</span><span class="o">=</span><span class="p">[</span><span class="n">fv1</span><span class="p">,</span><span class="n">fv3</span><span class="p">,</span><span class="n">fv4</span><span class="p">]</span>
<span class="n">study</span><span class="o">.</span><span class="n">samples</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span><span class="o">.</span><span class="n">factor_values</span><span class="o">=</span><span class="p">[</span><span class="n">fv1</span><span class="p">,</span><span class="n">fv2</span><span class="p">,</span><span class="n">fv5</span><span class="p">]</span>
<span class="n">study</span><span class="o">.</span><span class="n">samples</span><span class="p">[</span><span class="mi">3</span><span class="p">]</span><span class="o">.</span><span class="n">factor_values</span><span class="o">=</span><span class="p">[</span><span class="n">fv1</span><span class="p">,</span><span class="n">fv3</span><span class="p">,</span><span class="n">fv5</span><span class="p">]</span>

<span class="n">characteristic_organism</span> <span class="o">=</span> <span class="n">Characteristic</span><span class="p">(</span><span class="n">category</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;Organism&quot;</span><span class="p">),</span>
                                         <span class="n">value</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;marine metagenome&quot;</span><span class="p">,</span>
                                                                  <span class="n">term_source</span><span class="o">=</span><span class="n">ncbitaxon</span><span class="p">,</span>
                                                                  <span class="n">term_accession</span><span class="o">=</span><span class="s2">&quot;http://purl.obolibrary.org/obo/NCBITaxon_408172&quot;</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<p>Now creating a Process showing a <code class="docutils literal notranslate"><span class="pre">Protocol</span> <span class="pre">Application</span></code> using Source as input and producing Sample as output.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">study</span><span class="o">.</span><span class="n">sources</span><span class="p">)):</span>
    <span class="n">study</span><span class="o">.</span><span class="n">sources</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">characteristics</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">characteristic_organism</span><span class="p">)</span>
    <span class="n">study</span><span class="o">.</span><span class="n">process_sequence</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Process</span><span class="p">(</span><span class="n">executes_protocol</span><span class="o">=</span><span class="n">study</span><span class="o">.</span><span class="n">protocols</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span>
                                         <span class="n">inputs</span><span class="o">=</span><span class="p">[</span><span class="n">study</span><span class="o">.</span><span class="n">sources</span><span class="p">[</span><span class="n">i</span><span class="p">]],</span>
                                         <span class="n">outputs</span><span class="o">=</span><span class="p">[</span><span class="n">study</span><span class="o">.</span><span class="n">samples</span><span class="p">[</span><span class="n">i</span><span class="p">]])</span>
                                 <span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>Now appending the ISA Study object to the ISA Investigation object</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">investigation</span><span class="o">.</span><span class="n">studies</span> <span class="o">=</span> <span class="p">[</span><span class="n">study</span><span class="p">]</span>
</pre></div>
</div>
</div>
</div>
<p>Now, creating the ISA objects to represent assays and acquired raw data
Starting by declaring the 2 types of assays used in BII-S-3:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">assay</span> <span class="o">=</span> <span class="n">Assay</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="s2">&quot;a_gilbert-assay-Gx.txt&quot;</span><span class="p">)</span>

<span class="n">assay</span><span class="o">.</span><span class="n">measurement_type</span> <span class="o">=</span> <span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;metagenome sequencing&quot;</span><span class="p">,</span>
                                            <span class="n">term_accession</span><span class="o">=</span><span class="s2">&quot;http://purl.obolibrary.org/obo/OBI_0002623&quot;</span><span class="p">,</span>
                                            <span class="n">term_source</span><span class="o">=</span><span class="n">obi</span><span class="p">)</span>

<span class="n">assay</span><span class="o">.</span><span class="n">technology_type</span> <span class="o">=</span> <span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;nucleotide sequencing&quot;</span><span class="p">,</span>
                                           <span class="n">term_accession</span><span class="o">=</span><span class="s2">&quot;http://purl.obolibrary.org/obo/OBI_0000626&quot;</span><span class="p">,</span>
                                           <span class="n">term_source</span><span class="o">=</span><span class="n">obi</span><span class="p">)</span>


<span class="n">assay_tx</span> <span class="o">=</span> <span class="n">Assay</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="s2">&quot;a_gilbert-assay-Tx.txt&quot;</span><span class="p">)</span>

<span class="n">assay_tx</span><span class="o">.</span><span class="n">measurement_type</span> <span class="o">=</span> <span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;transcription profiling&quot;</span><span class="p">,</span>
                                               <span class="n">term_accession</span><span class="o">=</span><span class="s2">&quot;http://purl.obolibrary.org/obo/OBI_0000424&quot;</span><span class="p">,</span>
                                               <span class="n">term_source</span><span class="o">=</span><span class="n">obi</span><span class="p">)</span>
<span class="n">assay_tx</span><span class="o">.</span><span class="n">technology_type</span> <span class="o">=</span> <span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;nucleotide sequencing&quot;</span><span class="p">,</span>
                                              <span class="n">term_accession</span><span class="o">=</span><span class="s2">&quot;http://purl.obolibrary.org/obo/OBI_0000626&quot;</span><span class="p">,</span>
                                              <span class="n">term_source</span><span class="o">=</span><span class="n">obi</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>make sure the <code class="docutils literal notranslate"><span class="pre">ISA</span> <span class="pre">plink</span> <span class="pre">function</span></code> connects the first and last protocols found in a chain of protocols.</p>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">sample</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">study</span><span class="o">.</span><span class="n">samples</span><span class="p">):</span>
    
<span class="c1">#     # create an aliquoting process which creates new samples from existing samples created in a study.processSequence</span>
<span class="c1">#     aliquoting_process = Process(executes_protocol=study.protocols[1])</span>
<span class="c1">#     aliquot = Sample(name=&quot;aliquot-{}&quot;.format(i), derives_from=sample)</span>
<span class="c1">#     aliquoting_process.inputs.append(sample)</span>
<span class="c1">#     aliquoting_process.outputs.append(aliquot)</span>

    <span class="c1"># create an extraction process that executes the extraction protocol</span>

    <span class="n">extraction_process</span> <span class="o">=</span> <span class="n">Process</span><span class="p">(</span><span class="n">executes_protocol</span><span class="o">=</span><span class="n">study</span><span class="o">.</span><span class="n">protocols</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>

    <span class="c1"># extraction process takes as input a sample, and produces an extract material as output</span>
    
    <span class="n">char_ext</span> <span class="o">=</span> <span class="n">Characteristic</span><span class="p">(</span><span class="n">category</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;Material Type&quot;</span><span class="p">),</span>
                              <span class="n">value</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;pellet&quot;</span><span class="p">))</span>
    
    <span class="n">char_ext1</span> <span class="o">=</span> <span class="n">Characteristic</span><span class="p">(</span><span class="n">category</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;quantity&quot;</span><span class="p">),</span>
                               <span class="n">value</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;loads&quot;</span><span class="p">))</span>

    <span class="c1"># extraction_process.inputs.append(aliquot)</span>
    <span class="n">extraction_process</span><span class="o">.</span><span class="n">inputs</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">sample</span><span class="p">)</span>
    <span class="n">material</span> <span class="o">=</span> <span class="n">Material</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;extract-</span><span class="si">{}</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">i</span><span class="p">))</span>
    <span class="n">material</span><span class="o">.</span><span class="n">type</span> <span class="o">=</span> <span class="s2">&quot;Extract Name&quot;</span>
    <span class="n">material</span><span class="o">.</span><span class="n">characteristics</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">char_ext</span><span class="p">)</span>
    <span class="n">material</span><span class="o">.</span><span class="n">characteristics</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">char_ext1</span><span class="p">)</span>
    <span class="n">extraction_process</span><span class="o">.</span><span class="n">outputs</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">material</span><span class="p">)</span>

    <span class="c1"># create a sequencing process that executes the sequencing protocol</span>

    <span class="n">sequencing_process</span> <span class="o">=</span> <span class="n">Process</span><span class="p">(</span><span class="n">executes_protocol</span><span class="o">=</span><span class="n">study</span><span class="o">.</span><span class="n">protocols</span><span class="p">[</span><span class="mi">7</span><span class="p">])</span>
    <span class="n">sequencing_process</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="s2">&quot;assay-name-</span><span class="si">{}</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">i</span><span class="p">)</span>
    <span class="n">sequencing_process</span><span class="o">.</span><span class="n">inputs</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">extraction_process</span><span class="o">.</span><span class="n">outputs</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
    <span class="c1"># sequencing_process.inputs.append(material)</span>

    <span class="c1"># Sequencing process usually has an output data file</span>

    <span class="n">datafile</span> <span class="o">=</span> <span class="n">DataFile</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="s2">&quot;sequenced-data-</span><span class="si">{}</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">i</span><span class="p">),</span> <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Raw Data File&quot;</span><span class="p">)</span>
    <span class="n">data_comment</span> <span class="o">=</span> <span class="n">Comment</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;data_comment&quot;</span><span class="p">,</span><span class="n">value</span><span class="o">=</span><span class="s2">&quot;data_value&quot;</span><span class="p">)</span>
    <span class="n">datafile</span><span class="o">.</span><span class="n">comments</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">data_comment</span><span class="p">)</span>
    <span class="n">sequencing_process</span><span class="o">.</span><span class="n">outputs</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">datafile</span><span class="p">)</span>

    <span class="c1"># Ensure Processes are linked forward and backward. plink(from_process, to_process) is a function to set</span>
    <span class="c1"># these links for you. It is found in the isatools.model package</span>

    <span class="c1"># plink(aliquoting_process, sequencing_process)</span>
    <span class="n">plink</span><span class="p">(</span><span class="n">extraction_process</span><span class="p">,</span> <span class="n">sequencing_process</span><span class="p">)</span>
    <span class="c1"># make sure the extract, data file, and the processes are attached to the assay</span>


    <span class="n">assay</span><span class="o">.</span><span class="n">samples</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">sample</span><span class="p">)</span>
    <span class="c1"># assay.samples.append(aliquot)</span>
    <span class="n">assay</span><span class="o">.</span><span class="n">other_material</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">material</span><span class="p">)</span>
    <span class="n">assay</span><span class="o">.</span><span class="n">data_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">datafile</span><span class="p">)</span>
    
    <span class="n">assay</span><span class="o">.</span><span class="n">process_sequence</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">extraction_process</span><span class="p">)</span>
    <span class="n">assay</span><span class="o">.</span><span class="n">process_sequence</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">sequencing_process</span><span class="p">)</span>

    <span class="c1"># create an extraction process that executes the extraction protocol</span>

    <span class="n">extraction_process_tx</span> <span class="o">=</span> <span class="n">Process</span><span class="p">(</span><span class="n">executes_protocol</span><span class="o">=</span><span class="n">study</span><span class="o">.</span><span class="n">protocols</span><span class="p">[</span><span class="mi">2</span><span class="p">])</span>

    <span class="c1"># extraction process takes as input a sample, and produces an extract material as output</span>

    <span class="n">extraction_process_tx</span><span class="o">.</span><span class="n">inputs</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">sample</span><span class="p">)</span>
    <span class="c1"># extraction_process_tx.outputs.append(sample)</span>
    <span class="c1"># extraction_process_tx.outputs=[]</span>
    <span class="c1"># material_tx = Material(name=&quot;extract-{}&quot;.format(i))</span>
    <span class="c1"># material_tx.type = &quot;Extract Name&quot;</span>
    <span class="c1"># extraction_process_tx.outputs.append(material_tx)</span>

    <span class="c1"># labeling process takes as input an extract, and produces a labeled extract material as output</span>
    <span class="n">labeling_process_tx</span> <span class="o">=</span> <span class="n">Process</span><span class="p">(</span><span class="n">executes_protocol</span><span class="o">=</span><span class="n">study</span><span class="o">.</span><span class="n">protocols</span><span class="p">[</span><span class="mi">3</span><span class="p">])</span>
    <span class="c1"># labeling_process_tx.inputs.append(sample)</span>
    <span class="c1"># labeling_process_tx.inputs=[]</span>
    <span class="n">material_tx</span> <span class="o">=</span> <span class="n">Material</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;le-</span><span class="si">{}</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">i</span><span class="p">))</span>
    <span class="n">material_tx</span><span class="o">.</span><span class="n">type</span> <span class="o">=</span> <span class="s2">&quot;Labeled Extract Name&quot;</span>
    <span class="n">labeling_process_tx</span><span class="o">.</span><span class="n">outputs</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">material_tx</span><span class="p">)</span>
    
    
    <span class="c1"># create a sequencing process that executes the sequencing protocol</span>
    <span class="c1"># seq_pv1 = ParameterValue(category=&quot;library layout&quot;, value=&quot;SINGLE&quot;)</span>
    <span class="n">seq_pv2</span> <span class="o">=</span> <span class="n">ParameterValue</span><span class="p">(</span><span class="n">category</span><span class="o">=</span><span class="n">ProtocolParameter</span><span class="p">(</span><span class="n">parameter_name</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;library layout&quot;</span><span class="p">)),</span>
                             <span class="n">value</span><span class="o">=</span><span class="n">OntologyAnnotation</span><span class="p">(</span><span class="n">term</span><span class="o">=</span><span class="s2">&quot;SINGLE&quot;</span><span class="p">))</span>
    <span class="n">sequencing_process_tx</span> <span class="o">=</span> <span class="n">Process</span><span class="p">(</span><span class="n">executes_protocol</span><span class="o">=</span><span class="n">study</span><span class="o">.</span><span class="n">protocols</span><span class="p">[</span><span class="mi">6</span><span class="p">],</span> <span class="n">parameter_values</span><span class="o">=</span><span class="p">[</span><span class="n">seq_pv2</span><span class="p">])</span>
    <span class="n">sequencing_process_tx</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="s2">&quot;assay-name-tx-</span><span class="si">{}</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">i</span><span class="p">)</span>
    <span class="n">sequencing_process_tx</span><span class="o">.</span><span class="n">inputs</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">labeling_process_tx</span><span class="o">.</span><span class="n">outputs</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>

    <span class="c1"># Sequencing process usually has an output data file</span>
    
    <span class="n">datafile_tx</span> <span class="o">=</span> <span class="n">DataFile</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="s2">&quot;sequenced-data-tx-</span><span class="si">{}</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">i</span><span class="p">),</span> <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Raw Data File&quot;</span><span class="p">)</span>
    <span class="n">sequencing_process_tx</span><span class="o">.</span><span class="n">outputs</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">datafile_tx</span><span class="p">)</span>

    <span class="c1"># Ensure Processes are linked forward and backward. plink(from_process, to_process) is a function to set</span>
    <span class="c1"># these links for you. It is found in the isatools.model package</span>

   
    <span class="c1"># plink(extraction_process_tx, labeling_process_tx)</span>
    <span class="c1"># plink(labeling_process_tx, sequencing_process_tx)</span>
    <span class="n">plink</span><span class="p">(</span><span class="n">extraction_process_tx</span><span class="p">,</span> <span class="n">sequencing_process_tx</span><span class="p">)</span>
    <span class="c1"># make sure the extract, data file, and the processes are attached to the assay</span>


    <span class="n">assay_tx</span><span class="o">.</span><span class="n">samples</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">sample</span><span class="p">)</span>
    <span class="n">assay_tx</span><span class="o">.</span><span class="n">other_material</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">material_tx</span><span class="p">)</span>
    <span class="n">assay_tx</span><span class="o">.</span><span class="n">data_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">datafile_tx</span><span class="p">)</span>
    
    <span class="n">assay_tx</span><span class="o">.</span><span class="n">process_sequence</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">extraction_process_tx</span><span class="p">)</span>
    <span class="n">assay_tx</span><span class="o">.</span><span class="n">process_sequence</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">labeling_process_tx</span><span class="p">)</span> 
    <span class="n">assay_tx</span><span class="o">.</span><span class="n">process_sequence</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">sequencing_process_tx</span><span class="p">)</span>    
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="adding-both-assays-to-the-isa-study-object">
<h2>Adding both assays to the ISA Study object<a class="headerlink" href="#adding-both-assays-to-the-isa-study-object" title="Permalink to this headline">¶</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">study</span><span class="o">.</span><span class="n">assays</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">assay</span><span class="p">)</span>
<span class="n">study</span><span class="o">.</span><span class="n">assays</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">assay_tx</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># assay_g = investigation.studies[0].assays[1]</span>
<span class="c1"># assay_t = investigation.studies[0].assays[0]</span>

<span class="c1"># assay_t.samples=investigation.studies[0].samples</span>


<span class="c1"># extract1 = Material(name=&quot;GSM255770.e1&quot;)</span>
<span class="c1"># extract1.type = &quot;Extract Name&quot;</span>
<span class="c1"># extract2 = Material(name=&quot;GSM255771.e1&quot;)</span>
<span class="c1"># extract2.type = &quot;Extract Name&quot;</span>
<span class="c1"># extract3 = Material(name=&quot;GSM255772.e1&quot;)</span>
<span class="c1"># extract3.type = &quot;Extract Name&quot;</span>
<span class="c1"># extract4 = Material(name=&quot;GSM255773.e1&quot;)</span>
<span class="c1"># extract4.type = &quot;Extract Name&quot;</span>
<span class="c1"># extract5 = Material(name=&quot;GSM255774.e1&quot;)</span>
<span class="c1"># extract5.type = &quot;Extract Name&quot;</span>

<span class="c1"># assay_t.other_material.append(extract1)</span>
<span class="c1"># assay_t.other_material.append(extract2)</span>
<span class="c1"># assay_t.other_material.append(extract3)</span>
<span class="c1"># assay_t.other_material.append(extract4)</span>
<span class="c1"># assay_t.other_material.append(extract5)</span>


<span class="c1"># for i in range(len(study.samples)):</span>

<span class="c1">#     assay_t.process_sequence.append(Process(</span>
<span class="c1">#             executes_protocol=study.protocols[1],</span>
<span class="c1">#                 inputs=study.samples[i],</span>
<span class="c1">#                 outputs=assay_t.other_material[i]</span>
<span class="c1">#         ))</span>
    
<span class="c1">#     data=DataFile(filename=&quot;sequenced-data-{}&quot;.format(i), label=&quot;Raw Data File&quot;)  </span>
    
<span class="c1">#     assay_t.process_sequence.append(Process(</span>
<span class="c1">#         executes_protocol=study.protocols[3],</span>
<span class="c1">#                 inputs=assay_t.other_material[i]</span>
<span class="c1">#     ))</span>
    
<span class="c1">#     plink(assay_t.process_sequence[0], assay_t.process_sequence[1])</span>
<span class="c1">#     assay_t.process_sequence[-1].outputs.append(data)</span>
<span class="c1">#     assay_t.data_files.append(data)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="writing-the-isa-object-representation-to-file-with-the-isa-api-dump-function">
<h2>Writing the ISA object representation to file with the ISA-API <code class="docutils literal notranslate"><span class="pre">dump</span></code> function<a class="headerlink" href="#writing-the-isa-object-representation-to-file-with-the-isa-api-dump-function" title="Permalink to this headline">¶</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">isatools.isatab</span> <span class="kn">import</span> <span class="n">dump</span>

<span class="c1"># note the use of the flag for explicit serialization on factor values on assay tables</span>
<span class="n">dump</span><span class="p">(</span><span class="n">investigation</span><span class="p">,</span> <span class="s2">&quot;./output/BII-S-3-synth/&quot;</span><span class="p">,</span> <span class="n">write_factor_values_in_assay_table</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="reading-the-isa-document-from-disk-back-in-loading-it-into-memory-and-writing-to-disk-again-to-check-that-the-isa-api-load-function-works-nominally">
<h2>Reading the ISA document from disk back in, loading it into memory and writing to disk again to check that the ISA-API load function works nominally<a class="headerlink" href="#reading-the-isa-document-from-disk-back-in-loading-it-into-memory-and-writing-to-disk-again-to-check-that-the-isa-api-load-function-works-nominally" title="Permalink to this headline">¶</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">isatools.isatab</span> <span class="kn">import</span> <span class="n">load</span>
<span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="s2">&quot;./output/BII-S-3-synth/&quot;</span><span class="p">,</span><span class="s2">&quot;i_investigation.txt&quot;</span><span class="p">))</span> <span class="k">as</span> <span class="n">bii3stest</span><span class="p">:</span>
    <span class="n">roundtrip</span> <span class="o">=</span> <span class="n">load</span><span class="p">(</span><span class="n">bii3stest</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">isatools.convert</span> <span class="kn">import</span> <span class="n">isatab2json</span>
<span class="kn">from</span> <span class="nn">isatools</span> <span class="kn">import</span> <span class="n">isajson</span>
<span class="kn">import</span> <span class="nn">json</span>
<span class="n">isa_json</span> <span class="o">=</span> <span class="n">isatab2json</span><span class="o">.</span><span class="n">convert</span><span class="p">(</span><span class="s1">&#39;./output/BII-S-3-synth/&#39;</span><span class="p">,</span> <span class="n">validate_first</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">use_new_parser</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="n">isa_j</span> <span class="o">=</span> <span class="n">json</span><span class="o">.</span><span class="n">dumps</span><span class="p">(</span>
            <span class="n">isa_json</span><span class="p">,</span> <span class="bp">cls</span><span class="o">=</span><span class="n">isajson</span><span class="o">.</span><span class="n">ISAJSONEncoder</span><span class="p">,</span> <span class="n">sort_keys</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">indent</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> <span class="n">separators</span><span class="o">=</span><span class="p">(</span><span class="s1">&#39;,&#39;</span><span class="p">,</span> <span class="s1">&#39;: &#39;</span><span class="p">)</span>
        <span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>et Voilà!</p>
</div>
<div class="section" id="authors">
<h2>Authors<a class="headerlink" href="#authors" title="Permalink to this headline">¶</a></h2>
<details class="sphinx-bs dropdown card mb-3">
<summary class="summary-title card-header">
<strong>Authors</strong><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p class="card-text">Name</p></th>
<th class="head"><p class="card-text">ORCID</p></th>
<th class="head"><p class="card-text">Affiliation</p></th>
<th class="head"><p class="card-text">Type</p></th>
<th class="head"><p class="card-text">ELIXIR Node</p></th>
<th class="head"><p class="card-text">Contribution</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p class="card-text"><div class="firstCol"><a target="_blank" href="https://github.com/{github_handle}"><img class="avatar-style" src="https://avatars.githubusercontent.com/proccaserra"></img><div class="d-block">Philippe Rocca-Serra</div></a></div></p></td>
<td><p class="card-text"><a target="_blank" href="https://orcid.org/0000-0001-9853-5668"><i class="fab fa-orcid fa-2x text--orange"></i></a></p></td>
<td><p class="card-text">University of Oxford</p></td>
<td><p class="card-text"><i class="fas fa-graduation-cap fa-1x text--orange" alt="Academic"></i></p></td>
<td><p class="card-text"><img class="elixir-style" src="../../../_static/images/logo/Elixir/ELIXIR-UK.svg" ></img></p></td>
<td><p class="card-text">Writing - Original Draft</p></td>
</tr>
</tbody>
</table>
</div>
</details></div>
<div class="section" id="license">
<h2>License<a class="headerlink" href="#license" title="Permalink to this headline">¶</a></h2>
<details class="sphinx-bs dropdown card mb-3">
<summary class="summary-title card-header">
<strong>License</strong><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<a href="https://creativecommons.org/licenses/by/4.0/">
    <img src="../../../_static/images/logo/cc.svg"  height="25"/>
    <img src="../../../_static/images/logo/by.svg"  height="25"/></a>
    <a href="https://creativecommons.org/licenses/by/4.0/"> The Creative Commons 4.0 BY license</a></br>
    <br/>
</div>
</details></div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "isa-api-py39"
        },
        kernelOptions: {
            kernelName: "isa-api-py39",
            path: "./content/recipes/maturity"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'isa-api-py39'</script>

                                            </v-col>
                                            <div class="d-none d-md-block col-md-2 bd-toc show collapse custom-bdtoc pt-5">
                                                <div style="position: fixed">
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i>
            Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#abstract">
   Abstract:
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#let-s-get-started-loading-the-isa-api">
   Let’s get started! Loading the ISA-API
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#creating-an-isa-investigation">
   Creating an ISA investigation
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#declaring-semantic-resources-used-to-annotated-isa-objects">
   Declaring
   <code class="docutils literal notranslate">
    <span class="pre">
     Semantic
    </span>
    <span class="pre">
     Resources
    </span>
   </code>
   used to annotated ISA objects
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#building-and-annotation-isa-objects-to-a-community-compliant-level">
   Building and Annotation ISA objects to a
   <code class="docutils literal notranslate">
    <span class="pre">
     Community
    </span>
    <span class="pre">
     Compliant
    </span>
   </code>
   level
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#declaring-all-the-protocols-used-in-the-isa-study">
     Declaring all the protocols used in the ISA study.
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#declaring-the-isa-study-factors-i-e-the-independent-variables-of-the-experiment">
   Declaring the
   <code class="docutils literal notranslate">
    <span class="pre">
     ISA
    </span>
    <span class="pre">
     Study
    </span>
    <span class="pre">
     Factors
    </span>
   </code>
   , i.e. The Independent Variables of the Experiment
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#declaring-the-isa-factorvalues-i-e-the-factor-levels-to-each-of-the-indepedent-variables-or-isa-study-factor">
   Declaring the
   <code class="docutils literal notranslate">
    <span class="pre">
     ISA
    </span>
    <span class="pre">
     FactorValues
    </span>
   </code>
   i.e. The factor levels to each of the Indepedent Variables or
   <code class="docutils literal notranslate">
    <span class="pre">
     ISA
    </span>
    <span class="pre">
     Study
    </span>
    <span class="pre">
     Factor
    </span>
   </code>
   .
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#adding-the-publications-associated-to-the-study">
   Adding the publications associated to the study
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#adding-the-authors-of-the-study">
   Adding the authors of the study
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#adding-both-assays-to-the-isa-study-object">
   Adding both assays to the ISA Study object
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#writing-the-isa-object-representation-to-file-with-the-isa-api-dump-function">
   Writing the ISA object representation to file with the ISA-API
   <code class="docutils literal notranslate">
    <span class="pre">
     dump
    </span>
   </code>
   function
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#reading-the-isa-document-from-disk-back-in-loading-it-into-memory-and-writing-to-disk-again-to-check-that-the-isa-api-load-function-works-nominally">
   Reading the ISA document from disk back in, loading it into memory and writing to disk again to check that the ISA-API load function works nominally
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#authors">
   Authors
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#license">
   License
  </a>
 </li>
</ul>

        </nav>
        </div>
                                            </div>
                                        
                                    </v-row>

                                    
                                        <v-row class="mb-5">
                                            <div class='pl-10 prev-next-bottom mr-10' style="width:82%">
                                                
<a class='left-prev' id="prev-link" href="create-a-simple-ISA-descriptor.html" title="previous page">Creating a basic ISA descriptor: Dataset Maturity Level 1</a>
<a class='right-next' id="next-link" href="isa-json-conversion-to-rdf-linked-data.html" title="next page">From ISA JSON to ISA JSON-LD (RDF): Dataset Maturity Level 3</a>

                                            </div>
                                        </v-row>
                                    
                                </v-container>
                            </main>
                        
                    
                </v-row>
                
            </v-container>
        </v-row>
    </v-container>
</v-app>


  <script src="../../../_static/js/index.d3f166471bb80abb5163.js"></script>



<!-- Google Analytics -->
<script>
    window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
    ga('create', 'UA-171697285-1', 'auto');
    ga('set', 'anonymizeIp', true);
    ga('send', 'pageview');
</script>
<script async src='https://www.google-analytics.com/analytics.js'></script>
<!-- End Google Analytics -->

<v-container fluid>
    <v-row no-gutter>
        <v-col cols="12"><footer class="footer" style=" margin-top: 12px;  ">
    <div class="container-md mt-4">
      <div class="row" style="display:flex; justify-content: center">
          <div class="col-12 col-sm-12 col-md-4 col-lg-4 col-xl-3">
            <div class="footerImages">
              <a href="https://europa.eu/"><img src="/_static/images/logo/eu-flag.png" height="40" alt="EU flag"></a>
              <a href="https://www.imi.europa.eu/"><img src="/_static/images/logo/imi-logo.png" height="40" alt="IMI logo"></a>
              <a href="https://efpia.eu"><img src="/_static/images/logo/efpia-logo.png" height="40" alt="EFPIA logo"></a>
            </div>
            <div class="mb-3"> Grant agreement 802750 </div>
          </div>
          <div class="col-5 col-sm-2 col-xs-5 col-md-2 col-lg-2 col-xl-2">
              <ul>
                <li><a href="mailto:faircookbook-ed@elixir-europe.org"> Contact us </a></li>
                <li><a href="https://fairplus-project.eu" target="_blank"> FAIRplus </a> </li>
              </ul>
          </div>
          <div class="col-5 col-sm-2 col-xs-5 col-md-2 col-lg-2 col-xl-2">
              <ul>
                <li> <a href="https://faircookbook.elixir-europe.org/content/recipes/afterword/people.html  "> Contributors </a> </li> </li>
                <li> <a href="https://faircookbook.elixir-europe.org/content/recipes/help.html"> How to contribute </a> </li>
              </ul>
          </div>
          <div class="col-12 col-sm-4 col-xs-12 col-md-3 col-lg-3 col-xl-3 pb-5">
            <a class="mr-3" href="https://elixir-europe.org/" target="_blank"><img src="/_static/images/logo/Elixir/ELIXIR.png" height="50" alt="Elixir logo"></a>
            <a href="https://fairplus-project.eu/"><img src="https://fairplus-project.eu/images/fairplus-logo.png" alt="FAIRplus logo" style="background:white;padding:5px;height:33px;"></a>
          </div>
          <div class="col-12 pb-4">
              Recommended in the
              <a
                href="https://www.imi.europa.eu/sites/default/files/uploads/documents/resources-for-projects/IMI2_OpenAcesGuidelines_Updated2021.pdf"
                target="_blank"
                class="primary--text"
              >
                 IMI/IHI Project Guidelines
              </a> and the
              <a
                href="link to https://ec.europa.eu/info/funding-tenders/opportunities/docs/2021-2027/horizon/wp-call/2023-2024/wp-4-health_horizon-2023-2024_en.pdf"
                target="_blank"
                class="primary--text"
              >
                  Horizon Europe Work Programme for Health
              </a> as the resource for FAIR Data Management guidelines and good practices for the Life Sciences.
          </div>
      </div>
    </div>

    <div class="container-fluid" style="padding:0">
        <div class="row" style="padding:0; margin:0">
            <div class="col-12 purple-dark">
                The FAIR Cookbook is licensed under the
                <a href="https://creativecommons.org/licenses/by/4.0/" target="_blank">
                    Creative Commons Attribution 4.0 International
                </a> (CC BY 4.0) license.
            </div>
        </div>
        <div class="row extra_footer d-none">
            <script> mermaid.init() </script>
        </div>
    </div>

</footer> </v-col>
    </v-row>
</v-container>
<script type="application/javascript">
    const vuetify = new Vuetify({
        theme: {
            themes: {
                light: {
                    primary: '#421C6E',
                    secondary: '#FC7A4A',
                    accent: '#8038D1',
                    error: '#FF5252',
                    info: '#2196F3',
                    success: '#7E0038',
                    warning: '#FFC107'
                }
            }
        },
        icons: { iconfont: 'fa4' }
    })
    const my_vue = new Vue({ el: '#app', vuetify: vuetify })

</script>
  </body>
</html>