
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>7. bioimage-deposition</title>

  <link rel="stylesheet" href="../../../_static/css/index.f658d18f9b420779cfdf24aa0a7e2d77.css">


  <link rel="stylesheet"
    href="../../../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">



  <link rel="stylesheet"
    href="../../../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../../../_static/vendor/lato_latin-ext/1.44.1/index.css">


    <link rel="stylesheet" href="../../../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../../../_static/sphinx-book-theme.e7340bb3dbd8dde6db86f25597f54a1b.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/vuetify.min.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/search.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/search-wizard.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/materialdesignicons.min.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/custom.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/animate.min.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />

  <link rel="preload" as="script" href="../../../_static/js/index.d3f166471bb80abb5163.js">

    <script id="documentation_options" data-url_root="../../../" src="../../../_static/documentation_options.js"></script>
    <script src="../../../_static/jquery.js"></script>
    <script src="../../../_static/underscore.js"></script>
    <script src="../../../_static/doctools.js"></script>
    <script src="../../../_static/togglebutton.js"></script>
    <script src="../../../_static/clipboard.min.js"></script>
    <script src="../../../_static/copybutton.js"></script>
    <script src="../../../_static/js/searchNodal.js"></script>
    <script src="../../../_static/js/vue.min.js"></script>
    <script src="../../../_static/js/mermaid.min.js"></script>
    <script src="../../../_static/js/mermaid.js"></script>
    <script src="../../../_static/js/mermaid.core.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../../../_static/sphinx-book-theme.7d483ff0a819d6edff12ce0b1ead3928.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../../../_static/sphinx-thebe.js"></script>
    <link rel="canonical" href="https://fairplus.github.io/the-fair-cookbook/content/recipes/applied-examples/BIA_submission.html" />
    <link rel="shortcut icon" href="../../../_static/favicon.ico"/>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" />
    <link rel="next" title="8. Mapping of clinical trial data to CDISC-SDTM: a practical example based on APPROACH and ABIRISK" href="approach-cdisc.html" />
    <link rel="prev" title="6. IMI EUBOPEN FAIR High-Content Screening data deposition" href="eubopen-hcs-bioimage.html" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<meta name="docsearch:language" content="en" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/vuetify/2.6.9/vuetify.js"
        integrity="sha512-cgrSlRKhr/ovJePDoGIdcfZ3wiGcc0fT4Q7hAo0x3P1lyLwUmv2NgJtCo2gzFbxqFjKvNnCfVUjZ1XADm0/ZdQ=="
        crossorigin="anonymous" referrerpolicy="no-referrer">
</script>

<!-- Opengraph tags -->
<meta property="og:url"         content="https://fairplus.github.io/the-fair-cookbook/content/recipes/applied-examples/BIA_submission.html" />
<meta property="og:type"        content="article" />
<meta property="og:title"       content="bioimage-deposition" />
<meta property="og:description" content="An online, open and live resource for the Life Sciences with recipes that help you to make and keep data Findable, Accessible, Interoperable and Reusable; in one word FAIR." />
 <meta property="og:image"       content="/_static/images/homepage.png" /> 
<meta name="twitter:card" content="summary" />
  </head>
<body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">



<v-app id="app" class="pa-0">
    <v-container fluid class="pa-0">
        <v-row class="pa-0" no-gutters>
            <v-container fluid class="pa-0">
                <v-row>
                    <div class="topbar container-xl fixed-top pl-0 primary lighten-1 py-2" style="height:65px !important;">
    <div class="topbar-contents row" style="margin: initial !important;">
        <div class="col topbar-main d-flex align-center">
            
                <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                        data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                        aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                        title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                    <i class="fas fa-bars"></i>
                    <i class="fas fa-arrow-left"></i>
                    <i class="fas fa-arrow-up"></i>
                </button>
            

            
                <a href="../../../index.html" style="width: 300px">
                    <img
                        src="../../../_static/cookbook-logo-small.png"
                        class="logo"
                        alt="The FAIR Cookbook"
                        style="width: 100%"
                    >
                </a>
            

            <v-spacer> </v-spacer>
            <div class="mr-4 extraButtons">
                <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger d-flex justify-center align-center" style="height:100%">

    <v-menu offset-y content-class="elevation-0" open-on-hover>
        <template v-slot:activator="{ on, attrs }">
            <v-btn
              id="dropdown-buttons-trigger"
              color="white"
              outlined
              v-bind="attrs"
              v-on="on"
              flat
              solo
              rounded
              :fab="$vuetify.breakpoint.mdAndDown"
              :small="$vuetify.breakpoint.mdAndDown"
              :class="$vuetify.breakpoint.mdAndDown ? 'pa-0' : ''"
            >
                <v-icon :class="$vuetify.breakpoint.lgAndUp ? 'mr-3' : 'mx-1'" left>fab fa-github</v-icon>
                <span v-if="$vuetify.breakpoint.lgAndUp"> GitHub </span>
            </v-btn>
        </template>
        <div class="homepageMenu d-flex flex-column">
            
            <v-btn
              color="grey darken-1"
              dark
              flat
              solo
              rounded
              href="https://github.com/FAIRplus/the-fair-cookbook"
              class="mt-1"
              target="_blank"
            >
                <v-icon class="mr-3" left>fab fa-github</v-icon>
                repository
            </v-btn>
            

            
            <v-btn
              color="grey darken-1"
              dark
              flat
              solo
              rounded
              href="https://github.com/FAIRplus/the-fair-cookbook/issues/new?title=Issue%20on%20page%20%2Fcontent/recipes/applied-examples/BIA_submission.html&body=Your%20issue%20content%20here."
              class="mt-1"
              target="_blank"
            >
                <v-icon class="mr-3" left>fas fa-lightbulb</v-icon>
                open issue
            </v-btn>
            

            
            <v-btn
              color="grey darken-1"
              dark
              flat
              solo
              rounded
              href="https://github.com/FAIRplus/the-fair-cookbook/edit/main/content/recipes/applied-examples/BIA_submission.md"
              class="mt-1"
              target="_blank"
            >
                <v-icon class="mx-3" left>fas fa-pencil-alt</v-icon>
                suggest edit
            </v-btn>
            
        </div>
    </v-menu>
</div>

                <!-- Launch buttons -->

            </div>
                
                    <script>
    Vue.component("Searchfield", {
      data () { return {
        query: null
      } },
      methods: {
        search() {
            if (this.query) window.location.href = `../../search-wizard.html?q=` + this.query
            else window.location.href = `../../search-wizard.html`
        }
      },
      template: `
      <div style="width:250px">
            <v-text-field
              v-model="query"
              append-icon="fa-search"
              label="Search Wizard..."
              dense
              rounded
              single-line
              solo
              hide-details
              class="white primary--text"
              @click:append="search"
              @keyup.enter="search"
            />
      </div>
      `
    })
</script>

<Searchfield> </Searchfield>






                
                
                
        </div>
    </div>
</div>
                </v-row>
                
                    <v-row class="d-flex">
                    
                        <div
                            class=" bd-sidebar site-navigation show"
                            id="site-navigation"
                            style="width:350px"
                        >
                            
                                <div class="nav-drawer d-flex flex-column grey lighten-3" style="height:100%;padding-top: 40px;">

    <div class="testTOC">
        <script>
    Vue.component("Navigation", {
        data() { return { recipes: {} }},
        methods: {
            async getData(){
                const recipes = await fetch('/_static/recipes.json')
                this.recipes = await recipes.json();
                this.parseDom()
            },
            parseDom() {
                const links = Array.prototype.slice.call(document.getElementsByClassName("reference internal"))
                const icon = '<i class="fas fa-play-circle grey lighten-3 secondary--text ml-2"></i>'
                links.forEach(link => {
                    const recipeSplit = link.innerText.split(". ")
                    const recipeName = recipeSplit[recipeSplit.length - 1].trim()
                    recipeSplit.pop()
                    recipeSplit.forEach(pref => { pref = pref.trim() })
                    const prefix = recipeSplit.length > 0 ? recipeSplit.join(".") + "." : null
                    const content = prefix ? `${prefix} ${recipeName}` : recipeName
                    if (Object.keys(this.recipes).includes(recipeName)) {
                        if (this.recipes[recipeName].executable === 'Yes') link.innerHTML = content + icon
                    }
                })
            }
        },
        mounted() { this.getData() }
    })
</script>


<Navigation> </Navigation>

    </div>

    <div class="nav-tree grey px-5 lighten-3 pb-5" style="flex-grow: 1; border-bottom: 1px solid #ccc !important">
        
        
            <nav class="bd-links pt-10" id="bd-docs-nav" aria-label="Main navigation">
    <p class="caption collapsible-parent">
 <span class="caption-text">
  FOREWORD
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/FAIR-cookbook-audience.html">
   Introduction
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/brief-FAIR-principles.html">
   Introducing the FAIR Principles
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/FAIRplus-values.html">
   Reflecting on the ethical values of FAIR
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/fairification-process.html">
   Introducing our FAIRification framework
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/priorization.html">
   Prioritizing projects for FAIRification
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/metadata-fair.html">
   Framing FAIR and the notion of metadata
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/FAIR-and-knowledge-graphs.html">
   Understanding the relation between FAIR and Knowledge Graphs
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/training-datasets.html">
   Training for FAIRification with open or synthetic biomedical datasets
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/public-kg-resource-integration.html">
   Raising Awareness in Public Knowledge Graphs for Life Sciences
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/fair-cro.html">
   Reflecting on Practical Considerations for CROs to play FAIR
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/dpia.html">
   Data Protection Impact Assessment and Data Privacy
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/glossary-fair-terms.html">
   Glossary
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Recipes at a Glance
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../../search-wizard.html">
   All Recipes In a Table
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  FAIR RECIPES
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../findability.html">
   Findability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/identifiers.html">
     1. Introducing unique, persistent identifiers
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/chemical-identities.html">
     2. Creating InChI &amp; SMILES identifiers for chemical structures
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/iupac-names.html">
     3. Creating InChIKeys for IUPAC names
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/identifiers-minids.html">
     4. Minting identifiers with Globus Minid client
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/zenodo-deposition.html">
     5. Depositing to generic repositories - Zenodo use case
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/registeringDatasets.html">
     6. Registering datasets with Wikidata
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/checksum-create.html">
     7. Creating file checksums
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/checksum-validate.html">
     8. Validating checksums to verify file integrity
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../findability/seo.html">
     9. Introducing Search Engine Optimization (SEO)
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../findability/seo/bioschemas-data-page.html">
       9.9.1. Marking up Data pages with Schema.org &amp; Bioschemas for SEO
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../findability/seo/bioschemas-dataset.html">
       9.9.2. Marking up Dataset page with Schema.org &amp; Bioschemas for SEO
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../findability/seo/bioschemas-datacatalog.html">
       9.9.3. Marking up Data Catalogue page with Schema.org &amp; Bioschemas for SEO
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../accessibility.html">
   Accessibility
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../accessibility/sftp.html">
     1. Transferring data with SFTP protocol
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../accessibility/aspera.html">
     2. Downloading data with Aspera protocol
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../interoperability.html">
   Interoperability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../interoperability/registering-swisslipids-identifiers.html">
     1. Registering SwissLipids identifiers in Wikidata
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../interoperability/identifier-mapping.html">
     2. Interlinking data from different sources
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../interoperability/bridgedb-recipe.html">
     3. Mapping identifiers with BridgeDb
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../interoperability/bridgedb/bridgedb-example.html">
       3.11.1. Mapping identifiers using BridgeDb web services
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../interoperability/introduction-terminologies-ontologies.html">
     4. Introducing terminologies and ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../interoperability/selecting-ontologies.html">
     5. Selecting terminologies and ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../interoperability/ontology-new-term-request-recipe.html">
     6. Requesting new terms from terminologies and ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../interoperability/ontology-operations-tools.html">
     7. Introducing ontology-related tools and services
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../interoperability/ontology-robot-recipe.html">
     8. Building an application ontology with ROBOT
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../interoperability/ontology-robot-recipe/MSIO-robot-build-process.html">
       8.11.1. Building an application ontology for metabolomics - MSIO
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../interoperability/ontology-robot-recipe/competency-questions.html">
       8.11.2. Defining competency questions for the Ontology ROBOT use case
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../interoperability/ontology-alignment-oxo.html">
     9. Mapping Ontologies with OxO, EBI Ontology Xref Service
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../interoperability/creating-data-dictionary.html">
     10. Creating a data/variable dictionary
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../interoperability/creating-minimal-metadata-profiles.html">
     11. Creating a metadata profile
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../interoperability/transcriptomics-metadata.html">
       11.5.1. Outlining a metadata profile for transcriptomics
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../interoperability/covid19-sample-metadata-profile-shex-use-case.html">
       11.5.2. Building a community compliant metadata profile - The Covid19 sample profile use case
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../interoperability/bioactivity-profile.html">
       11.5.3. Outlining a metadata profile for Bioactivity data
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../interoperability/c4c-clinical-trials.html">
       11.5.4. Creating a metadata profile for clinical trial protocols
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../interoperability/from-proprietary-to-open-standard-mzml-exemplar.html">
     12. Converting from proprietary to open format
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../interoperability/fastq-file-format-validators.html">
     13. Validating file format - FASTQ example
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../interoperability/rdf-conversion.html">
     14. Inventorying tools for converting data to RDF
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../interoperability/ETL-tools.html">
     15. Surveying extraction, transformation, load (ETL) tools
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../interoperability/VCF-to-FHIR-conversion.html">
     16. Expressing Clinical Genetic Information as FHIR JSON
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../interoperability/vcf2fhir/FAIRcookbook-vcf2fhir.html">
       16.9.1. Converting VCF file to FHIR JSON
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../interoperability/creating-knowledge-graph-from-unstructured-text.html">
     17. Creating knowledge graphs from unstructured text
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../interoperability/nlp2kg/creating-knowledge-graph-from-text.html">
       17.12.1. unstructured text to graph as executable notebook
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../reusability.html">
   Reusability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/ATI-licensing.html">
     1. Licensing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/ATI_licensing_software.html">
     2. Licensing Software
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/ATI_licensing_data.html">
     3. Licensing Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/expressing-data-use.html">
     4. Declaring data permitted uses
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/provenance.html">
     5. Introducing Provenance Information
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../infrastructure.html">
   Infrastructure
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/id-resolution.html">
     1. Introducing identifier resolution services
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/gupri.html">
     2. Creating resolvable identifiers
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/data-catalog.html">
     3. Building a catalogue of datasets
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/dats-model.html">
     4. Introducing the DATS model
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/data-catalog-deployment.html">
     5. Deploying a data catalogue - The IMI data catalogue example
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../infrastructure/vocabularies.html">
     6. Vocabulary management
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../infrastructure/vocabularies/technical-and-architectural-selection-criteria-of-ontology-lookup-services.html">
       6.1. Introducing vocabulary portals and lookup services
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../infrastructure/vocabularies/Selecting_and_using_ontology_lookup_services.html">
       6.2. Selecting an ontology lookup service
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../infrastructure/vocabularies/UC3_R13_local_ontology_services.html">
       6.3. Deploying EBI Ontology Loopkup Service
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/openrefine-nvt.html">
     7. Using OpenRefine and Karma for FAIRification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/fair-api.html">
     8. Developing FAIR API for the Web
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../assessing-fairness.html">
   Assessment
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../assessing-fairness/fair-assessment-fairevaluator.html">
     1. Assessing with FAIR Evaluator
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../assessing-fairness/fair-assessment-fairshake.html">
     2. Assessing with FAIRshake
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  FAIR Maturity
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../maturity.html">
   Changing culture with the Dataset Maturity Model
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/miappe.html">
     Improving dataset maturity - MIAPPE-compliant submission to EMBL-EBI databases
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/plant-pheno-data-publication.html">
     Publishing plant phenotypic data
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../maturity/isa-maturity-progression.html">
     Moving through maturity levels with ISA
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../maturity/create-a-simple-ISA-descriptor.html">
       Minimal Data Maturity with ISA - ISA-Tab and free text
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../maturity/isa-api-programmatic-rebuild-of-BII-S-3.html">
       Enhanced Data Maturity with ISA - ISA-JSON and ontology markup
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../maturity/isa-json-conversion-to-rdf-linked-data.html">
       Moving to a semantically typed version - ISA-JSON-LD
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../maturity/isa-as-RO.html">
       Dissemination - Packaging ISA as a Research Object (RO)
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="fair-data-matrix-recipe.html">
     Making an omics data matrix FAIR
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="fair-data-matrix/0-rose-metabolites-Python-data-handling.html">
       FAIRifying Data Matrices - Step1 - Starting material
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="fair-data-matrix/1-rose-metabolites-Python-analysis.html">
       FAIRifying Data Matrices - Step2 - Structuring data
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="fair-data-matrix/2-rose-metabolites-Python-RDF-querying-analysis.html">
       FAIRifying Data Matrices - Step3 - Exploring data with SPARQL
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="fair-data-matrix/3-rose-metabolites-R-analysis.html">
       FAIRifying Data Matrices - Step4 - Integrating data
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="fair-workflows.html">
     Making Computational Workflows FAIR
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  FAIRified Datasets
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="../applied-examples.html">
   Applied examples
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="etox.html">
     1. IMI eTox - toxicity datasets
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="nd4bb.html">
     2. IMI nd4bb - chemical activities datasets
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="oncotrack_raw/rawRecipe-oncotrack.html">
     3. Readying IMI Oncotrack - clinical cohort datasets for deposition to EBI Biosamples
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="resolute_raw/resolute.html">
     4. Depositing IMI ReSOLUTE transcriptomics datasets to EBI repositories
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ehden-ohdsi.html">
     5. Enhancing discoverability of EHDEN OHDSI data with Schema.org markup
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="eubopen-hcs-bioimage.html">
     6. Depositing IMI EUBOPEN High-Content Screening data to EBI BioImage Archive
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     7. Depositing epifluorescence and confocal microscopy data to EBI BioImage Archive
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="approach-cdisc.html">
     8. Mapping IMI APPROACH datasets to CDISC-SDTM standard
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  AFTERWORD
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../afterword/fair-reflections.html">
   Lessons learned from the FAIR journey and project outlook
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  GLOSSARY
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/glossary-fair-terms.html">
   Glossary of terms and abbreviations
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  JOIN US
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../community.html">
   Community
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../afterword/people.html">
     Boards and contributors
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../afterword/code_of_conduct.html">
     Code of conduct
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../afterword.html">
     Platform
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../afterword/the-turing-way.html">
       Leveraging the Turing Way Book
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../help.html">
   Contribute
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../afterword/contributing.html">
     1. How to contribute
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/how-to-create-recipe-with-gdoc.html">
     2. Add via GoogleDoc
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/how-to-create-recipe-with-hackmd.html">
     3. Add via HackMD
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/how-to-create-recipe-with-git.html">
     4. Add via Git
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/recipe-template.html">
     5. Git recipe template
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/tips-tricks.html">
     6. Tips and tricks
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/myst.html">
     7. Markdown cheatsheet
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>

</nav>
        
        
             <!-- To handle the deprecated key -->

        
    </div>

    <div class="text-center px-5 lighten-3 grey py-2">
        Powered by <a href="https://jupyterbook.org" class="secondary--text">Jupyter Book</a>
    </div>
</div>
                        </div>
                        
                            <main class="bd-content" role="main" style="width:1%; flex-grow:1;">
                                <v-container fluid id="main-content" class="pa-0">
                                    <v-row class="d-flex">
                                        
                                            <v-col class="col-xl-10 col-lg-10 col-md-9 col-12 px-10 pt-10 mt-3">
                                                
  <div class="section" id="bioimage-deposition">
<span id="fcb-bioimage"></span><h1><span class="section-number">7. </span>bioimage-deposition<a class="headerlink" href="#bioimage-deposition" title="Permalink to this headline">¶</a></h1>
<script type="application/ld+json"> {"@context": {"bs": "https://bioschema.org/", "dc": "http://purl.org/dc/terms/", "owl": "http://www.w3.org/2002/07/owl#", "sdo": "https://schema.org/"}, "@type": ["sdo:HowTo", "sdo:LearningResource"], "@id": "https://w3id.org/faircookbook/FCB086", "sdo:name": "Depositing Covid-19 image data to BioImage Archive", "sdo:description": "Depositing Covid-19 image data to BioImage Archive", "sdo:keywords": [], "dc:conformsTo": {"@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE", "@type": "CreativeWork"}, "sdo:inLanguage": ["en-GB"], "sdo:audience": ["Data Producer|NA", "Data Manager|T4FS_0000520"], "sdo:isPartOf": [{"@type": "CreativeWork", "@id": "https://fairplus.github.io/the-fair-cookbook/", "sdo:url": "https://fairplus.github.io/the-fair-cookbook/", "sdo:name": "The FAIR Cookbook"}], "sdo:learningResourceType": "Experience Report / Applied Example", "sdo:identifier": {"@type": "PropertyValue", "@id": "", "sdo:name": "FCB086", "dso:url": "https://w3id.org/faircookbook/FCB086", "owl:sameAs": "https://identifiers.org/fcb:FCB086"}, "sdo:license": {"@type": "CreativeWork", "@id": "https://creativecommons.org/licenses/by/4.0/", "sdo:url": "https://creativecommons.org/licenses/by/4.0/", "sdo:name": "CC-BY-4.0"}, "sdo:timeRequired": "PT0H20M0S", "sdo:totalTime": "PT0H20M0S", "sdo:step": [{"@type": "sdo:HowToSection", "sdo:position": "1", "sdo:name": "Overview", "sdo:url": "https://w3id.org/faircookbook/FCB086#main-objectives", "sdo:itemListElement": [{"@type": "sdo:HowToStep", "sdo:position": "1", "sdo:name": "", "sdo:itemListElement": [{"@type": "sdo:HowToDirection", "sdo:position": "1", "sdo:name": "step-by-step section", "sdo:text": "Follow the step-by-step section.", "sdo:url": "https://w3id.org/faircookbook/FCB086#main-objectives"}, {"@type": "sdo:HowToTip", "sdo:position": "2", "sdo:name": "Reporting an issue", "sdo:text": "Use our GitHub issue tracker to report any error. Thanks!", "sdo:url": "https://github.com/FAIRplus/the-fair-cookbook/issues/new/choose"}]}]}, {"@type": "sdo:HowToSection", "sdo:position": "2", "sdo:name": "Conclusion", "sdo:url": "https://w3id.org/faircookbook/FCB086#conclusion", "sdo:itemListElement": [{"@type": "sdo:HowToStep", "sdo:position": "1", "sdo:name": "", "sdo:itemListElement": [{"@type": "sdo:HowToDirection", "sdo:position": "1", "sdo:name": "conclusion section", "sdo:text": "What to read next?", "sdo:url": "https://w3id.org/faircookbook/FCB086#conclusion"}, {"@type": "sdo:HowToTip", "sdo:position": "2", "sdo:name": "Reporting an issue", "sdo:text": "Use our GitHub issue tracker to report any error. Thanks!", "sdo:url": "https://github.com/FAIRplus/the-fair-cookbook/issues/new/choose"}]}]}]} </script>
<div class="card my-10 recipe d-flex">
<div class="card-header purple-dark text--white">
<div class="overview">
<div class="overviewLabel">Recipe Overview</div>
<div class="overviewContent">
<div class="item"><div class="itemContainer">
<i class="itemIcon fa fa-clock fa-2x text--white"></i>
<div class="itemContent">
<div class="label"> Reading Time </div>
20 minutes
</div>
</div></div>
<div class="item"><div class="itemContainer">
<i class="itemIcon fa fa-play-circle fa-2x text--white"></i>
<div class="itemContent">
<div class="label"> Executable Code </div>
No
</div>
</div></div>
<div class="item" style="margin-bottom:10px"><div class="itemContainer">
<i class="itemIcon fa fa-fire fa-2x text--white" style="margin-left:4px;margin-right: 5px;"></i>
<div class="itemContent">
<div class="label" style="position:relative; left:-6px">Difficulty</div>
<i class="fa fa-fire"></i>
<i class="fa fa-fire"></i>
<i class="fa fa-fire"></i>
<i class="fa fa-fire" style="color:grey"></i>
<i class="fa fa-fire" style="color:grey"></i>
</div>
</div></div>
</div></div>
</div>
<div class="card-body text--purple-dark">
<div class="title">
<div style="flex-grow:1; margin-right:5px">Depositing Covid-19 image data to BioImage Archive</div>
<img src="https://fairplus.github.io/the-fair-cookbook/_static/images/fairplus-mini.png" alt="FAIRPlus logo"> </img>
</div>
<div class="section">
<i class="sectionIcon fa fa-laptop fa-2x"></i><div class="sectionContent">
<div class="label">Recipe Type</div>
<div class="sectionValue">Experience Report / Applied Example</div>
</div>
</div>
<div class="section" style="flex-grow: 1;">
<i class="sectionIcon fa fa-users fa-2x"></i><div class="sectionContent">
<div class="label">Audience</div>
<div class="sectionValue">Data Producer, <a href="http://purl.obolibrary.org/obo/T4FS_0000520">Data Manager</a></div>
</div>
</div>
<div class="section" style="flex-grow: 1;">
<svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="signal" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 600" class="bars level-3">
<path class="bar5" d="M216 288h-48c-8.84 0-16 7.16-16 16v192c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V304c0-8.84-7.16-16-16-16zM88 384H40c-8.84 0-16 7.16-16 16v96c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16v-96c0-8.84-7.16-16-16-16zm256-192h-48c-8.84 0-16 7.16-16 16v288c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V208c0-8.84-7.16-16-16-16zm128-96h-48c-8.84 0-16 7.16-16 16v384c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V112c0-8.84-7.16-16-16-16zM600 0h-48c-8.84 0-16 7.16-16 16v480c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V16c0-8.84-7.16-16-16-16z"/>
<path class="bar4" d="M216 288h-48c-8.84 0-16 7.16-16 16v192c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V304c0-8.84-7.16-16-16-16zM88 384H40c-8.84 0-16 7.16-16 16v96c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16v-96c0-8.84-7.16-16-16-16zm256-192h-48c-8.84 0-16 7.16-16 16v288c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V208c0-8.84-7.16-16-16-16zm128-96h-48c-8.84 0-16 7.16-16 16v384c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V112c0-8.84-7.16-16-16-16zM600" />
<path class="bar3" d="M216 288h-48c-8.84 0-16 7.16-16 16v192c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V304c0-8.84-7.16-16-16-16zM88 384H40c-8.84 0-16 7.16-16 16v96c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16v-96c0-8.84-7.16-16-16-16zm256-192h-48c-8.84 0-16 7.16-16 16v288c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V208c0-8.84-7.16-16-16-16zm128-96h-48c-8.84" />
<path class="bar1" d="M216 288h-48c-8.84 0-16 7.16-16 16v192c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V304c0-8.84-7.16-16-16-16zM88 384H40c-8.84 0-16 7.16-16 16v96c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16v-96c0-8.84-7.16-16-16-16zm256-192h-48c-8.84"/>
<path class="bar2" d="M216 288h-48c-8.84 0-16 7.16-16 16v192c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V304c0-8.84-7.16-16-16-16zM88 384H40c-8.84"/>
</svg>
<div class="sectionContent">
<div class="label">Maturity Level & Indicator</div>
<div class="sectionValue"><v-chip-group><v-chip href="https://w3id.org/Data-Maturity/DSM-1-C0" class="secondary white--text" target="_blank" label>DSM-1-C0</v-chip><v-chip href="https://w3id.org/Data-Maturity/DSM-1-C1" class="secondary white--text" target="_blank" label>DSM-1-C1</v-chip></v-chip-group></div>
<div class="tooltip">hover me
<span class="tooltiptext">Tooltip text</span></div>
</div>
</div><div class="card-footer text--orange sphinx-bs.badge.badge-primary"> Cite me with 
<a href="https://w3id.org/faircookbook/FCB086" class="text--purple-dark">FCB086
</a></div>
</div>
</div>
<div class="section" id="overview">
<h2><span class="section-number">7.1. </span>Overview<a class="headerlink" href="#overview" title="Permalink to this headline">¶</a></h2>
<p>In addition to making sure that data and metadata are well-structured, the FAIR data principles also involve sharing
research outputs in ways that go beyond paper publications. However, even the best and most thorough deposition
ecosystem would be worthless if no data were deposited. Therefore, this <strong>datatype-specific recipe for bio-imaging data</strong> provides:</p>
<ul class="simple">
<li><p>information on available data and metadata standards and repositories.</p></li>
<li><p>a step-by-step recipe on data deposition in the BioImage Archive</p></li>
</ul>
<p>If you generate bio-imaging data and want to make your data FAIR by openly depositing it in the BioImage Archive,
this recipe is for you.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This recipe is one of two recipe dealing with image data deposition in the BioImage Archive.
The one you are reading now deals with data submission based on regular <strong>epifluorescence and confocal microscopy data</strong>
and focuses on data organisation of datasets containing more than one imaging technique. It also provides step-by-step
instructions on how to submit data to the BioImage Archive and how to navigate the Archive’s data submission web
interface.</p>
<p>The <a class="reference external" href="https://faircookbook.elixir-europe.org/content/recipes/applied-examples/eubopen-hcs-bioimage.html">other recipe</a>
focuses on high content screening data for compound libraries and provides more background on available data and
metadata standards and repositories, as well as details on how to structure such data and upload mechanisms for larger
datasets.</p>
</div>
</div>
<div class="section" id="ingredients">
<h2><span class="section-number">7.2. </span>Ingredients<a class="headerlink" href="#ingredients" title="Permalink to this headline">¶</a></h2>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Tools</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><a class="reference external" href="https://www.ebi.ac.uk/bioimage-archive/">BioImage Archive</a></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://www.ebi.ac.uk/biostudies/">BioStudies</a></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://orcid.org/">ORCID</a></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://filezilla-project.org/">Filezilla</a></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://www.ibm.com/products/aspera">Aspera</a></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://www.ebi.ac.uk/ols4/ontologies/fbbi">FBbi</a></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://bioportal.bioontology.org/ontologies/EDAM-BIOIMAGING/?p=classes&amp;conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_Widefield_microscopy&amp;jump_to_nav=true">EDAM-Bioimaging</a></p></td>
</tr>
</tbody>
</table>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Knowledge requirements</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><a class="reference external" href="https://www.nature.com/articles/s41592-021-01166-8">REMBI</a></p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="chosing-repositories-for-bioimaging-data">
<h2><span class="section-number">7.3. </span>Chosing repositories for bioimaging data<a class="headerlink" href="#chosing-repositories-for-bioimaging-data" title="Permalink to this headline">¶</a></h2>
<p>Over the last decade, the field of bioimaging has sought to develop a robust and effective bioimaging data ecosystem by
creating bioimaging repositories and working towards their widespread adoption. There are a number of resources
available to learn more about the existing repositories in each domain and their respective scopes. Here is a short
overview of some repositories for bioimaging data:</p>
<ul class="simple">
<li><p>A central, primary archive for bioimaging data is available through the <a class="reference external" href="https://www.ebi.ac.uk/bioimage-archive/"><strong>BioImage Archive</strong></a>
(Hartley et al. 2022), which hosts data from all imaging modalities associated to a peer-reviewed publication where no
more specialised resource exists.</p></li>
<li><p>In contrast, the <a class="reference external" href="https://idr.openmicroscopy.org/"><strong>Image Data Resource (IDR)</strong></a> (Williams et al., 2017) is an
added-value database for several microscopic imaging modalities with highly curated metadata, including high-content
screening data. The objective is to link the imaging data with other databases, such as those for genetic and chemical
information, as well as cell and tissue phenotypes.</p></li>
<li><p>The <a class="reference external" href="https://www.ebi.ac.uk/empiar/"><strong>Electron Microscopy Public Image ARchive (EMPIAR)</strong></a> (Iudin et al. 2023)
publicly archives 2D electron microscopy raw data underlying 3D cryo-EM protein structures and data from 3D volume
EM experiments.</p></li>
<li><p>The <a class="reference external" href="https://ssbd.qbic.riken.jp/"><strong>Systems Science of Biological Dynamics Repository and Database (SSBD)</strong></a>
(Tohsato et al., 2016) is a repository and database pair comprising a primary archive and added-value database for
quantitative data of spatiotemporal dynamics of biological objects primarily obtained from microscopy. According to the
FAIR principles, data should be shared as openly as possible, but as closed as necessary. This principle is especially
important for sensitive and biomedical data that cannot be fully openly shared and deposited.</p></li>
<li><p>Thus, restricted-access and topic-specific repositories for bioimaging data include for example
<a class="reference external" href="https://www.cancerimagingarchive.net/"><strong>The Cancer Imaging Archive</strong></a> (Clark et al. 2013).</p></li>
</ul>
<p>For more detailed information on repositories and FAIR practices in bioimaging data, we recommend consulting
the <a class="reference external" href="https://rdmkit.elixir-europe.org/bioimaging_data">RDMkit pages on bioimage data</a> or the <a class="reference external" href="https://www.eurobioimaging.eu/data-services/fair-image-data/">repository overview from Euro-BioImaging</a>.</p>
</div>
<div class="section" id="the-example-dataset">
<h2><span class="section-number">7.4. </span>The example dataset<a class="headerlink" href="#the-example-dataset" title="Permalink to this headline">¶</a></h2>
<p>This recipe will detail the data preparation and deposition process on the following real-life example:</p>
<p>Beucher, Guillaume et al. <strong>“Bronchial epithelia from adults and children: SARS-CoV-2 spread via syncytia formation and
type III interferon infectivity restriction.”</strong> Proceedings of the National Academy of Sciences of the United States of
America vol. 119,28 (2022): e2202370119. <a class="reference external" href="https://www.pnas.org/doi/full/10.1073/pnas.2202370119">doi:10.1073/pnas.2202370119</a></p>
<p>To study the spread of SARS-CoV-2 infection, the authors performed light and electron microscopy on bronchial epithelia
that were reconstructed from infected adult and child donors. The data from the example dataset comprises light
microscopy and some transmission electron microscopy published in a peer-reviewed publication and does not contain
person-identifiable images. Therefore, the <strong>BioImage Archive</strong> is the optimal choice for the deposition of this dataset.</p>
</div>
<div class="section" id="step-by-step-recipe-for-bioimage-data-deposition-to-bioimage-archive">
<h2><span class="section-number">7.5. </span>Step-by-step recipe for bioimage data deposition to BioImage Archive<a class="headerlink" href="#step-by-step-recipe-for-bioimage-data-deposition-to-bioimage-archive" title="Permalink to this headline">¶</a></h2>
<div class="section" id="step-1-familiarize-with-the-bioimage-archive">
<h3><span class="section-number">7.5.1. </span>Step 1: Familiarize with the BioImage Archive<a class="headerlink" href="#step-1-familiarize-with-the-bioimage-archive" title="Permalink to this headline">¶</a></h3>
<p>The first step in any deposition is to get to know the repository, to assess whether the
<a class="reference external" href="https://www.ebi.ac.uk/bioimage-archive/scope/">scope</a> is really appropriate, and to get an overview of the requirements
and the <a class="reference external" href="https://www.ebi.ac.uk/bioimage-archive/submit/">deposition process</a>. For the BioImage Archive, this information
is also summarised in their corresponding <a class="reference external" href="https://www.ebi.ac.uk/training/online/courses/bioimage-archive-quick-tour/">Quick-Tour</a>.</p>
<p>Furthermore, it is important to gain insight into the general architecture of the repositories entries to understand
what is possible for specific cases. <a class="reference external" href="https://www.ebi.ac.uk/biostudies/BioImages/studies?facet.collection=bioimages">Browse</a>
the archive to see some examples of current entries.</p>
<p>The completed <a class="reference external" href="https://www.ebi.ac.uk/biostudies/BioImages/studies/S-BIAD677">entry from the example dataset</a>, generated
through the steps outlined in this recipe, is provided here for reference:</p>
<details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="bia-entry">
<a class="reference internal image-reference" href="../../../_images/BIA_1_BIA-entry.jpg"><img alt="Overview of the entry in BioImage Archive" src="../../../_images/BIA_1_BIA-entry.jpg" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 7.8 </span><span class="caption-text">Overview of the example entry in BioImage Archive</span><a class="headerlink" href="#bia-entry" title="Permalink to this image">¶</a></p>
</div>
</div>
</details><details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="file-list">
<a class="reference internal image-reference" href="../../../_images/BIA_2_file-list.jpg"><img alt="Overview of a File-list in BioImage Archive" src="../../../_images/BIA_2_file-list.jpg" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 7.9 </span><span class="caption-text">Overview of a File-list in BioImage Archive</span><a class="headerlink" href="#file-list" title="Permalink to this image">¶</a></p>
</div>
</div>
</details></div>
<div class="section" id="creating-an-account">
<h3><span class="section-number">7.5.2. </span>Creating an account<a class="headerlink" href="#creating-an-account" title="Permalink to this headline">¶</a></h3>
<p>To deposit data in the BioImage Archive, a BioStudies Account is required as the BioImage Archive currently uses the
<a class="reference external" href="https://www.ebi.ac.uk/biostudies/submissions/signup">BioStudies submission</a> system. This account is free and only
requires a valid email address. However, providing your <a class="reference external" href="https://orcid.org/">ORCID</a> will improve the findability of
your submissions.</p>
<details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="biostudies">
<a class="reference internal image-reference" href="../../../_images/BIA_3_biostudies_registration.jpg"><img alt="Registration for BioStudies Account" src="../../../_images/BIA_3_biostudies_registration.jpg" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 7.10 </span><span class="caption-text">Registration for BioStudies Account</span><a class="headerlink" href="#biostudies" title="Permalink to this image">¶</a></p>
</div>
</div>
</details></div>
<div class="section" id="step-2-familiarize-with-rembi">
<h3><span class="section-number">7.5.3. </span>Step 2: Familiarize with REMBI<a class="headerlink" href="#step-2-familiarize-with-rembi" title="Permalink to this headline">¶</a></h3>
<p>Metadata is essential to making data FAIR, as it enables researchers to understand the data. To ensure basic metadata
quality yet enable routine deposition, the BioImage Archive uses minimal metadata reporting according to the REMBI
guidelines. The imaging community has come together to create a standard for describing biological images and their
associated studies and specimens: Recommended Metadata for Biological Images (<a class="reference external" href="https://www.nature.com/articles/s41592-021-01166-8">REMBI</a>).
It improves the sharing of image data and enables automated data harvesting. REMBI has a high-level 8-component
structure which can be further specified and adapted to cover different biological imaging methods and subdomains.
Take a look at which metadata information needs to be included for each of the <a class="reference external" href="https://www.ebi.ac.uk/bioimage-archive/rembi-help-lab/">REMBI components</a>
in the BioImage Archive <a class="footnote-reference brackets" href="#sarkans2021" id="id1">1</a> .</p>
<details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="rembi">
<a class="reference internal image-reference" href="../../../_images/BIA_4_REMBI.jpg"><img alt="Overview of REMBI components" src="../../../_images/BIA_4_REMBI.jpg" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 7.11 </span><span class="caption-text">Overview of REMBI components</span><a class="headerlink" href="#rembi" title="Permalink to this image">¶</a></p>
</div>
</div>
</details></div>
<div class="section" id="step-3-organize-the-data">
<h3><span class="section-number">7.5.4. </span>Step 3: Organize the data<a class="headerlink" href="#step-3-organize-the-data" title="Permalink to this headline">¶</a></h3>
<p>Before uploading, it is important to organise the data. This includes organising it into folders as well as thinking
about organising the BioImage Archive entry into Study Components (see 3.2). The two do not necessarily have to overlap
completely, but it is helpful nevertheless to also think about the Study Components beforehand which can guide the
folder organisation.</p>
<div class="section" id="data-organization-in-folders">
<h4><span class="section-number">7.5.4.1. </span>Data organization in folders<a class="headerlink" href="#data-organization-in-folders" title="Permalink to this headline">¶</a></h4>
<p>Data files (images, raw and processed, and associated data files) to be included in the BioImage Archive entry should be
organised in one or several folders, each with as many subfolders as necessary, following a logical and hierarchical
file structure. The REMBI framework is useful for planning what to include and how to organise the data. It is
recommended that the dataset is organised by grouping imaging modalities and/or experiments (i.e. Study Components) in
one folder, although in some cases it may also be appropriate to organise the data according to the structure of the
underlying paper.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The file paths will not be visible on the submission page. For this reason, if the directory structure contains metadata,
please include it in the File-list (Step 5).</p>
</div>
</div>
<div class="section" id="define-the-study-components">
<h4><span class="section-number">7.5.4.2. </span>Define the Study Components<a class="headerlink" href="#define-the-study-components" title="Permalink to this headline">¶</a></h4>
<p>As the BioImage Archive uses the REMBI scheme for metadata, the data organisation follows the REMBI scheme and is
therefore structured in so-called <strong>Study Components</strong>. A submission may contain one or more Study Components and there
is the possibility to mix and duplicate several REMBI category components in the same Study Component. In this way,
the structure of the experimental layout can be freely designed and there are several ways and <a class="reference external" href="https://www.ebi.ac.uk/bioimage-archive/rembi-help-examples/">examples</a>
on how to do so for different types of studies.</p>
<p>In the simplest case, a study contains only one Study Component, because only one experimental unit was performed, and
each REMBI component appears only once, because a single specimen was taken from a biosample and a single imaging
protocol was performed.</p>
<p>The structure of the example study is a more complex as it contains two basic variables: the imaging technologies and
the experimental sample. This is the overview scheme of the example dataset and a preview of its organisation in the
finished entry.</p>
<details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="study-components">
<a class="reference internal image-reference" href="../../../_images/BIA_5_study_component_illustration.jpg"><img alt="Overview of study component logic" src="../../../_images/BIA_5_study_component_illustration.jpg" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 7.12 </span><span class="caption-text">Overview of study component logic</span><a class="headerlink" href="#study-components" title="Permalink to this image">¶</a></p>
</div>
</div>
</details><details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="components-example">
<a class="reference internal image-reference" href="../../../_images/BIA_6_study_components_example.jpg"><img alt="sexample of multiple study components" src="../../../_images/BIA_6_study_components_example.jpg" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 7.13 </span><span class="caption-text">example of multiple study components</span><a class="headerlink" href="#components-example" title="Permalink to this image">¶</a></p>
</div>
</div>
</details><p>Two basic variants of “biosample” were imaged: native bronchial epithelia and genetically altered epithelia (CRISPR
knockout of interferon genes). This means that we chose to represent that by two Study Components to distinguish those
two biosamples.</p>
<p>For native bronchial epithelia, we have also performed different types of imaging: light microscopy including widefield
epifluorescence and confocal microscopy plus some amount of transmission electron microscopy. As each of the three
imaging modalities has its own image acquisition procedure, we will require three image acquisition components. The two
light microscopy methods share the same sample preparation, so we are grouping them into one Study Component with one
‘biosample’ and one ‘specimen’ but with two ‘image acquisition’ components.</p>
<p>The electron microscopy shares the same ‘biosample’ as light microscopy but does not share the sample preparation, so we
have decided to present the electron microscopy as a separate Study Component rather than combining all three imaging
modalities into a single Study Component. In this way it is easier to understand the data and to distinguish the
electron microscopy component more clearly, as we can give it its own name and description.</p>
<p>The bronchial epithelia with genetic alteration were only imaged using the two light microscopy methods. So the third
Study Component contains again two ‘image acquisition’ modules and share the other modules.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This is not the only possible organisation of the dataset, and alternative structures may also have been equally
suitable. For example, this dataset could have been described by fewer or even more Study Components. It’s simply a
matter of providing all the metadata and ensuring a structure that others can understand.</p>
</div>
<p>This process of data organisation is crucial as it determines which metadata needs to be input where and how the
File-list (Step 6) will look. Ultimately, it’s essential that all the relevant metadata information is included,
not where it is represented. Therefore, already at this stage, you can design what you would like to include in the
File-list(s). In a nutshell, a File-list contains all the information about what is different between the files in each
Study Component. In our dataset this would for example include whether the images were taken with confocal or
epifluorescence microscopy or to which donor the epithelial samples belonged. More details on the File-lists can be
found in Step 6.</p>
<p>Now there are two ways to proceed. Either you can follow this recipe and start the deposition by creating a new
submission and entering the metadata first and then uploading the files and creating the File-lists. Or you can upload
the organised data first (Step 5), then think about and create the File-lists (Step 6), and then afterwards create the
submission and enter the missing metadata (Step 4).</p>
</div>
</div>
<div class="section" id="step-4-upload-the-data">
<h3><span class="section-number">7.5.5. </span>Step 4: Upload the data<a class="headerlink" href="#step-4-upload-the-data" title="Permalink to this headline">¶</a></h3>
<p>Several data upload methods are available in the BioImage Archive and different methods are recommended for different
data size ranges:</p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Size</p></th>
<th class="head"><p>Submission Method</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>less than 50 GB total size &amp; <br> less than 20 GB per individual file</p></td>
<td><p>Submission tool upload</p></td>
</tr>
<tr class="row-odd"><td><p>up to 1 TB total size</p></td>
<td><p>FTP</p></td>
</tr>
<tr class="row-even"><td><p>larger than 1 TB</p></td>
<td><p>Aspera</p></td>
</tr>
</tbody>
</table>
<p>Once in the BioStudies user interface a <strong>‘secret directory’</strong> will be created for you as a place to upload your data
prior to submission.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The secret directory is not intended for long-term storage of your data. Make sure to only upload data that you plan to
submit in the near future.</p>
</div>
<p>To upload data using the submission portal, simply click on ‘File Upload’. This will allow you to upload the folder(s)
you organised in Step 3.</p>
<details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="file-upload">
<a class="reference internal image-reference" href="../../../_images/BIA_7_submission_file-upload.jpg"><img alt="the file upload interface of BioImage Archive" src="../../../_images/BIA_7_submission_file-upload.jpg" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 7.14 </span><span class="caption-text">the file upload interface of BioImage Archive</span><a class="headerlink" href="#file-upload" title="Permalink to this image">¶</a></p>
</div>
</div>
</details><p>For the example dataset it is recommended to use file transfer via FTP. This requires an FTP client and, we will use the
freely available <a class="reference external" href="https://filezilla-project.org/">Filezilla</a>. For instructions and login details for FTP and Aspera
uploads, please select the ‘FTP/Aspera’ button. To connect to the server the credentials provided under this tab
(<strong>host, username, password</strong>) should be entered in the FTP client and the files will be uploaded to the <strong>remote site</strong>
, which is unique for every user.</p>
</div>
<div class="section" id="step-5-create-the-submission">
<h3><span class="section-number">7.5.6. </span>Step 5: Create the submission<a class="headerlink" href="#step-5-create-the-submission" title="Permalink to this headline">¶</a></h3>
<p>To create a new submission, click on “New Submission” in the top right of the Biostudies account and select
“BioImage Archive Study”. You will then be able to enter the relevant metadata for the new submission.</p>
<details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="submission">
<a class="reference internal image-reference" href="../../../_images/BIA_8_new_submission.jpg"><img alt="new submission in submission tool" src="../../../_images/BIA_8_new_submission.jpg" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 7.15 </span><span class="caption-text">new submission in submission tool</span><a class="headerlink" href="#submission" title="Permalink to this image">¶</a></p>
</div>
</div>
</details><details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="submission-bia-entry">
<a class="reference internal image-reference" href="../../../_images/BIA_9_new_submission_bia_entry.jpg"><img alt="select BioImage Archive study" src="../../../_images/BIA_9_new_submission_bia_entry.jpg" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 7.16 </span><span class="caption-text">select BioImage Archive study</span><a class="headerlink" href="#submission-bia-entry" title="Permalink to this image">¶</a></p>
</div>
</div>
</details><div class="tip admonition">
<p class="admonition-title">Tip</p>
<p>It is also possible to generate and submit the study-level and Study Component metadata using alternative methods to the
submission portal, for example YAML files. Further details can be found on the <a class="reference external" href="https://github.com/BioImage-Archive">BioImage Archive GitHub page</a>
or by contacting the Archive <a class="reference external" href="https://www.ebi.ac.uk/bioimage-archive/contact-us/">directly</a></p>
</div>
<div class="section" id="input-the-study-level-metadata">
<h4><span class="section-number">7.5.6.1. </span>Input the Study-level metadata<a class="headerlink" href="#input-the-study-level-metadata" title="Permalink to this headline">¶</a></h4>
<p>BioImage Archive employs REMBI, so the submission interface is structured accordingly. The first component of REMBI is
the ‘Study’, which also is the first section of every BioImage Archive entry. This includes details about the current
submission, authors, and the corresponding paper. Some fields will have free-text boxes, while others will have dropdown
menus. If a particular item is not available in the dropdown menu, you can enter free text there instead.</p>
<details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="submission-portal">
<a class="reference internal image-reference" href="../../../_images/BIA_10_submission_portal.jpg"><img alt="Overview of submission portal" src="../../../_images/BIA_10_submission_portal.jpg" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 7.17 </span><span class="caption-text">Overview of submission portal</span><a class="headerlink" href="#submission-portal" title="Permalink to this image">¶</a></p>
</div>
</div>
</details><p>Ensure that every dataset author is correctly associated with the appropriate roles. It is also important to acknowledge
the contributions of core facility staff involved in the generation of the data and data stewards involved in the
submission process, even if they did not become authors of the paper.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The study persons refer to people involved in image data generation, analysis and submission and do not have to be the same as the authors of the corresponding paper.</p>
</div>
<details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="author-role-submission">
<a class="reference internal image-reference" href="../../../_images/BIA_11_submission_author_role.jpg"><img alt="author roles in BioImage Archive" src="../../../_images/BIA_11_submission_author_role.jpg" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 7.18 </span><span class="caption-text">author roles in BioImage Archive</span><a class="headerlink" href="#author-role-submission" title="Permalink to this image">¶</a></p>
</div>
</div>
</details><details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="author-role">
<a class="reference internal image-reference" href="../../../_images/BIA_12_dataset_author_role.jpg"><img alt="author roles in BioImage Archive" src="../../../_images/BIA_12_dataset_author_role.jpg" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 7.19 </span><span class="caption-text">author roles in BioImage Archive</span><a class="headerlink" href="#author-role" title="Permalink to this image">¶</a></p>
</div>
</div>
</details></div>
<div class="section" id="input-the-experiment-metadata">
<h4><span class="section-number">7.5.6.2. </span>Input the Experiment metadata<a class="headerlink" href="#input-the-experiment-metadata" title="Permalink to this headline">¶</a></h4>
<p>Once you have entered the study-level metadata, the other REMBI components will be listed at the bottom of the page.
For studies with a single linear assay, the experiment metadata can be entered in each REMBI component as is.</p>
<details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="subsections">
<a class="reference internal image-reference" href="../../../_images/BIA_13_submission_subsections.jpg"><img alt="subsections of REMBI components" src="../../../_images/BIA_13_submission_subsections.jpg" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 7.20 </span><span class="caption-text">subsections of REMBI components</span><a class="headerlink" href="#subsections" title="Permalink to this image">¶</a></p>
</div>
</div>
</details><p>As a general guide for metadata, give as much information as necessary to understand what you have done in order to
arrive at the deposited images. If possible describe your procedures, materials and outcomes using terms from controlled
vocabulary and according to an <a class="reference external" href="https://rdmkit.elixir-europe.org/metadata_management#how-do-you-find-appropriate-vocabularies-or-ontologies">ontology</a>
so that findability and interoperability are ensured.</p>
<p>As for the study-level metadata, information can be provided in free-text boxes or dropdown menus and should particular
items not be available in the dropdown free-text can be also entered there. Please keep in mind that the wording of the
dropdown menu is already in accordance with controlled vocabularies or ontologies and therefore should be preferred.</p>
<details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="image-aqusition">
<a class="reference internal image-reference" href="../../../_images/BIA_14_submission_image_aquisition.jpg"><img alt="drop-down menu of imaging method" src="../../../_images/BIA_14_submission_image_aquisition.jpg" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 7.21 </span><span class="caption-text">drop-down menu of imaging method</span><a class="headerlink" href="#image-aqusition" title="Permalink to this image">¶</a></p>
</div>
</div>
</details></div>
<div class="section" id="duplicating-rembi-components">
<h4><span class="section-number">7.5.6.3. </span>Duplicating REMBI components<a class="headerlink" href="#duplicating-rembi-components" title="Permalink to this headline">¶</a></h4>
<p>If the record contains multiple Study Components and/or multiple variants of REMBI components, it is important to ensure
that this complex organisation is properly reflected in the submission interface. This is the case in the current example,
where we have three Study Components and within each of these, some duplicated REMBI components.
To add components click on “add” in the top left corner and select the section to duplicate.
In this case we selected “Image acquisition” as this will give us another blank copy of the image acquisition sheet.
We enter here the information of the second image acquisition parameters, in this case the epifluorescence microscopy.</p>
<details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="add-components">
<a class="reference internal image-reference" href="../../../_images/BIA_15_submission_add_components.jpg"><img alt="add study components in submission portal" src="../../../_images/BIA_15_submission_add_components.jpg" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 7.22 </span><span class="caption-text">add study components in submission portal</span><a class="headerlink" href="#add-components" title="Permalink to this image">¶</a></p>
</div>
</div>
</details><p>For the example, we need in total 3 Biosample components, 3 Specimen components, 3 Image acquisition components
(the image acquisition protocol is the same between the native and the genetically altered epithelia, otherwise we would
have needed 5 image acquisition components) and 3 Image analysis components.</p>
</div>
<div class="section" id="study-component-associations">
<h4><span class="section-number">7.5.6.4. </span>Study Component associations<a class="headerlink" href="#study-component-associations" title="Permalink to this headline">¶</a></h4>
<p>Now we need to bring everything together inside the Study Components. Select the Study Component tab and enter a
meaningful name and description for each Study Component. It is now necessary to assign the corresponding REMBI
components to the previously defined Study Components (Step 3). If you have only one Study Component, simply assign all
REMBI components to the already existing Study Component. In the example dataset we decided on three Study Components so
we also need to create two more Study Components which can be done like for any other REMBI component as described above
(Step 3).</p>
<details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="id2">
<a class="reference internal image-reference" href="../../../_images/BIA_16_submission_study_components.jpg"><img alt="study component information" src="../../../_images/BIA_16_submission_study_components.jpg" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 7.23 </span><span class="caption-text">study component information</span><a class="headerlink" href="#id2" title="Permalink to this image">¶</a></p>
</div>
</div>
</details><div class="admonition note">
<p class="admonition-title">Note</p>
<p>We leave the File-list empty for now since we are only generating those in the next Step. If you have already created them because you chose the other route then you can also already associate the corresponding File-list.</p>
</div>
<p>If you have duplicate REMBI components in one Study Component, such as confocal and epifluorescence microscopy in the
example dataset, you can add association rows to match this structure.</p>
<details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="id3">
<a class="reference internal image-reference" href="../../../_images/BIA_17_submission_study_component_associations.jpg"><img alt="study component associations" src="../../../_images/BIA_17_submission_study_component_associations.jpg" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 7.24 </span><span class="caption-text">study component associations</span><a class="headerlink" href="#id3" title="Permalink to this image">¶</a></p>
</div>
</div>
</details></div>
</div>
<div class="section" id="step-6-generate-the-file-list">
<h3><span class="section-number">7.5.7. </span>Step 6: Generate the File-list<a class="headerlink" href="#step-6-generate-the-file-list" title="Permalink to this headline">¶</a></h3>
<p>One of the key elements of a BioImage Archive submission is the <a class="reference external" href="https://www.ebi.ac.uk/bioimage-archive/help-file-list/"><strong>File-list</strong></a>,
which acts as a “table of contents” for the files included in the submission.
This encompasses both image files and other supporting files, such as analysis results.
The File-list is crucial because it provides detailed metadata about each data file, enabling others to easily
understand the context and significance of each file. There must be exactly one File-list for each Study Component.
For this reason, it is practical to organise the data so that there is one folder per Study Component.
The first column of this tab-delimited file (.tsv, .xlsx) lists the file names and their relative paths, so that there
is exactly one row for each file included in a Study Component. Further columns will then detail the file-level metadata.</p>
<p>When all data is uploaded, the File-list template(s) can be automatically generated over the <strong>file upload portal</strong>.
For this click again on “File upload” in the top bar and for each folder, representing one Study Component, you can now
download a File-list template which already lists all the files inside one folder.</p>
<details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="file-list-download">
<a class="reference internal image-reference" href="../../../_images/BIA_18_submission_download_file_list.jpg"><img alt="tool to download File-list" src="../../../_images/BIA_18_submission_download_file_list.jpg" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 7.25 </span><span class="caption-text">tool to download File-list</span><a class="headerlink" href="#file-list-download" title="Permalink to this image">¶</a></p>
</div>
</div>
</details><div class="tip admonition">
<p class="admonition-title">Tip</p>
<p>It is also possible to have a File-list that collects files from a number of different folders, as may be necessary if a
Study Component is not inside a single folder but spread across multiple folders. In this case, however,
the File-list must be created in a different way elsewhere.</p>
</div>
<p>Once you have downloaded the empty File-list using the submission tool, you need to locally edit the File-list to
include additional columns describing file-level metadata. Therefore, you need to think about what file-level metadata
is essential to include so that others can understand the differences between each file in a Study Component.
For each  metadata item create one extra column in the File-list. You can either add more columns directly to the
generated File-list or, alternatively, copy and paste the “Files” column into another template.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This submission tool only generates a File-list with a single column listing all files (recursively, i.e. all files in
all subdirectories) and their paths, but no suggestions for additional metadata columns.</p>
</div>
<p>It is helpful to look at some <a class="reference external" href="https://www.ebi.ac.uk/bioimage-archive/help-file-list/">example File-lists</a> for different
types of studies to get suggestions for metadata in different studies and further guidance on File-lists.</p>
<p>In the example dataset we have several types of metadata, which are described in more detail in the File-list.
Of course this is just an example and the amount of additional information and thus column names will greatly vary
between studies. If we had chosen for a different organization in the example dataset, some of this information may
already have been described in the general metadata section.</p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Category</p></th>
<th class="head"><p>File-list column</p></th>
<th class="head"><p>Additional description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Experiment</p></td>
<td><p>Study-component</p></td>
<td><p>which Study Component the image belongs to</p></td>
</tr>
<tr class="row-odd"><td><p>Experiment</p></td>
<td><p>data-type</p></td>
<td><p>raw / processed image</p></td>
</tr>
<tr class="row-even"><td><p>Experiment</p></td>
<td><p>linked_to_figure</p></td>
<td><p>which figure in the paper the image belongs to</p></td>
</tr>
<tr class="row-odd"><td><p>Experiment</p></td>
<td><p>associated_metadata</p></td>
<td><p>whether metadata is included in the image files</p></td>
</tr>
<tr class="row-even"><td><p>Biosample</p></td>
<td><p>donor age</p></td>
<td><p>Adult / Child</p></td>
</tr>
<tr class="row-odd"><td><p>Biosample</p></td>
<td><p>donor ID</p></td>
<td><p>A1-A12 / C1-C8</p></td>
</tr>
<tr class="row-even"><td><p>Biosample</p></td>
<td><p>infection</p></td>
<td><p>yes / no</p></td>
</tr>
<tr class="row-odd"><td><p>Biosample</p></td>
<td><p>time</p></td>
<td><p>days post infection</p></td>
</tr>
<tr class="row-even"><td><p>Biosample</p></td>
<td><p>genetic alteration</p></td>
<td><p>IFN-λ gene knockout using CRISPR/Cas9</p></td>
</tr>
<tr class="row-odd"><td><p>Biosample</p></td>
<td><p>IFN-addition</p></td>
<td><p>concentration of IFN-addition</p></td>
</tr>
<tr class="row-even"><td><p>Image Acqusition</p></td>
<td><p>microscope-type</p></td>
<td><p>confocal, widefield or electron microscopy</p></td>
</tr>
<tr class="row-odd"><td><p>Image Acqusition</p></td>
<td><p>magnification</p></td>
<td><p>from 10x to 6000x</p></td>
</tr>
<tr class="row-even"><td><p>Image Acqusition</p></td>
<td><p>x,y resolution</p></td>
<td><p>in nm</p></td>
</tr>
<tr class="row-odd"><td><p>Image Acqusition</p></td>
<td><p>z resolution</p></td>
<td><p>in mn</p></td>
</tr>
<tr class="row-even"><td><p>Image file</p></td>
<td><p>view</p></td>
<td><p>X,Y,Z-section or hyperstack</p></td>
</tr>
<tr class="row-odd"><td><p>Image file</p></td>
<td><p>channels</p></td>
<td><p>how many channels are present in the image</p></td>
</tr>
<tr class="row-even"><td><p>Image file</p></td>
<td><p>stained cell type</p></td>
<td><p>goblet, basal or ciliated cells</p></td>
</tr>
<tr class="row-odd"><td><p>Image file</p></td>
<td><p>stained antibody</p></td>
<td><p>names of antibodies</p></td>
</tr>
<tr class="row-even"><td><p>Image file</p></td>
<td><p>stained cellular component</p></td>
<td><p>Nucleus, Actin, protein N, MUC5AC</p></td>
</tr>
</tbody>
</table>
<p>Once you have decided on the additional columns for the File-list, you must fill in the corresponding values for each
file in the submission. If files do not have a value for a specific column these can be left empty.</p>
<div class="tip admonition">
<p class="admonition-title">Tip</p>
<p>You can also programmatically generate the completed File List, including the additional metadata columns and content, which is particularly beneficial for large datasets..</p>
</div>
<p>Once you have completed the File-list(s), proceed to upload them again in the submission portal.
The File-list can be either uploaded to the corresponding Study Component folders if they are available,
or alternatively, they can be uploaded without a folder.
If you have multiple File-lists, please ensure that you name them clearly and distinctively.</p>
<div class="section" id="associating-the-file-lists">
<h4><span class="section-number">7.5.7.1. </span>Associating the File-lists<a class="headerlink" href="#associating-the-file-lists" title="Permalink to this headline">¶</a></h4>
<p>After having uploaded one File-list per Study Component, we go back to the prepared submission.
For this we click again on “Submission” in the top bar. Then we will select the current submission from the “Draft” category.</p>
<p>Earlier (Step 5) we had already generated and described the Study Components present in our submission.
We go now again to each of the Study Components and associate the corresponding File-lists to each Study Component,
making sure that we have one File-list per Study Component.</p>
</div>
</div>
<div class="section" id="optional-annotations">
<h3><span class="section-number">7.5.8. </span>Optional: Annotations<a class="headerlink" href="#optional-annotations" title="Permalink to this headline">¶</a></h3>
<p>When image files are accompanied by annotations, these may also be submitted to the BioImage Archive.
In the present example, no annotations are available, however, a brief overview of this process is outlined below.</p>
<p>The metadata for annotations are provided in accordance with the <a class="reference external" href="https://www.ebi.ac.uk/bioimage-archive/mifa-overview/">MIFA</a> standard.</p>
<details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="mifa">
<a class="reference internal image-reference" href="../../../_images/BIA_19_mifa.jpg"><img alt="the MIFA standard for annotations" src="../../../_images/BIA_19_mifa.jpg" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 7.26 </span><span class="caption-text">the MIFA standard for annotations</span><a class="headerlink" href="#mifa" title="Permalink to this image">¶</a></p>
</div>
</div>
</details><p>Annotations are also one component of the submission interface; they are treated similar to whole Study Components.</p>
<details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="annotations">
<a class="reference internal image-reference" href="../../../_images/BIA_20_annotations.jpg"><img alt="annotations" src="../../../_images/BIA_20_annotations.jpg" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 7.27 </span><span class="caption-text">annotations</span><a class="headerlink" href="#annotations" title="Permalink to this image">¶</a></p>
</div>
</div>
</details><div class="admonition note">
<p class="admonition-title">Note</p>
<p>It is necessary to create a separate File-list for annotations, in the same way that a File-list is required for each Study Component.</p>
</div>
<p>The File-list for annotations should always contain a column that relates the annotations to the images you have
uploaded (i.e. column name “related_image”) and gives the path to that image.
Otherwise, you should select the most appropriate metadata for your type of annotations, as you would for images.</p>
</div>
<div class="section" id="step-8-submit-your-data">
<h3><span class="section-number">7.5.9. </span>Step 8: Submit your data<a class="headerlink" href="#step-8-submit-your-data" title="Permalink to this headline">¶</a></h3>
<p>Now that all data is uploaded, the File-list(s) are associated and all metadata is entered in the portal you are ready
for submission. When you have checked that all the information is correct, click on ‘submit’ to complete the submission.</p>
<p>Upon successful submission, a unique BioImage Archive accession number will be assigned to your dataset, which serves as
a unique identifier within the archive. Additionally, your dataset will get a <a class="reference external" href="https://www.doi.org/">DOI</a>.
You will then be directed to a confirmation window, where you will find instructions on how to access and share your
study with others.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Your submission will not be available to the public until the specified release date.
However reviewers of a manuscript may receive blind access to the data before paper publication.</p>
</div>
<div class="section" id="orcid-claim">
<h4><span class="section-number">7.5.9.1. </span>ORCID claim<a class="headerlink" href="#orcid-claim" title="Permalink to this headline">¶</a></h4>
<p>Once your dataset is public, you can associate it with your <a class="reference external" href="https://orcid.org/">ORCID</a> profile to ensure that it is attributed to you in this record.</p>
<details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="orcid">
<a class="reference internal image-reference" href="../../../_images/BIA_21_orcid_claiming.jpg"><img alt="link the deposition to researchers ORCID" src="../../../_images/BIA_21_orcid_claiming.jpg" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 7.28 </span><span class="caption-text">link the deposition to researchers ORCID</span><a class="headerlink" href="#orcid" title="Permalink to this image">¶</a></p>
</div>
</div>
</details></div>
</div>
</div>
<div class="section" id="conclusion">
<h2><span class="section-number">7.6. </span>Conclusion<a class="headerlink" href="#conclusion" title="Permalink to this headline">¶</a></h2>
<p>By depositing your image data in the BioImage Archive, it will be findable and accessible as your data will be assigned
a persistent identifier and will be downloadable using a standard access protocol.
Depending on the depth and quality of the metadata provided, your dataset will now be FAIR.
By making your data openly available for reuse you have made an important contribution to open science and scientific
progress. It is also great for you, because now people will be able to find your data from the paper and vice versa,
giving you more recognition for your work as a scientist.</p>
<div class="section" id="what-to-read-next">
<h3><span class="section-number">7.6.1. </span>What to read next?<a class="headerlink" href="#what-to-read-next" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p><a class="reference internal" href="../accessibility/sftp.html#fcb-sftp"><span class="std std-ref">Transferring data with SFTP</span></a></p></li>
<li><p><a class="reference internal" href="../accessibility/aspera.html#fcb-access-aspera"><span class="std std-ref">Downloading data with Aspera</span></a></p></li>
<li><p><a class="reference internal" href="../interoperability/introduction-terminologies-ontologies.html#fcb-introduction-terminologies-ontologies"><span class="std std-ref">Introduction to terminologies and ontologies</span></a></p></li>
</ul>
<div class="sphinx-bs container pb-4 docutils">
<div class="row docutils">
<div class="d-flex col-md-4 docutils">
<div class="card w-100 shadow docutils">
<div class="card-body p-0 docutils">
<div class="card-header bg-primary pa_dark docutils text-center">
<img alt="RDMkit logo" height="40px" src="/_static/images/logo/RDMkit_logo.svg">
</div>
<div class="card-body docutils">
<p class="mb-0">Learn more about:</p>
<ul><li><a href="https://rdmkit.elixir-europe.org/bioimaging_data" target="_blank">Bioimaging data</a></li></ul>
</div>
</div>
</div>
</div>
<div class="d-flex col-md-4 docutils">
<div class="card w-100 shadow docutils">
<div class="card-body p-0 docutils">
</div>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="section" id="references">
<h2><span class="section-number">7.7. </span>References<a class="headerlink" href="#references" title="Permalink to this headline">¶</a></h2>
<details class="sphinx-bs dropdown card mb-3">
<summary class="summary-title card-header">
<strong>References</strong><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<p class="card-text" id="bibtex-footbibliography-content/recipes/applied-examples/BIA_submission-0"><dl class="footnote brackets">
<dt class="label" id="sarkans2021"><span class="brackets"><a class="fn-backref" href="#id1">1</a></span></dt>
<dd><p class="card-text">U. Sarkans, W. Chiu, L. Collinson, M. C. Darrow, J. Ellenberg, D. Grunwald, J. K. Hériché, A. Iudin, G. G. Martins, T. Meehan, K. Narayan, A. Patwardhan, M. R. G. Russell, H. R. Saibil, C. Strambio-De-Castillia, J. R. Swedlow, C. Tischer, V. Uhlmann, P. Verkade, M. Barlow, O. Bayraktar, E. Birney, C. Catavitello, C. Cawthorne, S. Wagner-Conrad, E. Duke, P. Paul-Gilloteaux, E. Gustin, M. Harkiolaki, P. Kankaanpää, T. Lemberger, J. McEntyre, J. Moore, A. W. Nicholls, S. Onami, H. Parkinson, M. Parsons, M. Romanchikova, N. Sofroniew, J. Swoger, N. Utz, L. M. Voortman, F. Wong, P. Zhang, G. J. Kleywegt, and A. Brazma. REMBI: Recommended Metadata for Biological Images-enabling reuse of microscopy data in biology. <em>Nat Methods</em>, 18(12):1418–1422, 12 2021.</p>
</dd>
</dl>
</p>
</div>
</details></div>
<div class="section" id="funding">
<h2><span class="section-number">7.8. </span>Funding<a class="headerlink" href="#funding" title="Permalink to this headline">¶</a></h2>
<details class="sphinx-bs dropdown card mb-3">
<summary class="summary-title card-header">
<strong>Funding</strong><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<p class="card-text">The this recipe was generated with funding from EU Horizon 2020 under grant agreement no. 101046203 (BY-COVID).</p>
</div>
</details></div>
<div class="section" id="authors">
<h2><span class="section-number">7.9. </span>Authors<a class="headerlink" href="#authors" title="Permalink to this headline">¶</a></h2>
<details class="sphinx-bs dropdown card mb-3">
<summary class="summary-title card-header">
<strong>Authors</strong><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p class="card-text">Name</p></th>
<th class="head"><p class="card-text">ORCID</p></th>
<th class="head"><p class="card-text">Affiliation</p></th>
<th class="head"><p class="card-text">Type</p></th>
<th class="head"><p class="card-text">ELIXIR Node</p></th>
<th class="head"><p class="card-text">Contribution</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p class="card-text"><div class="firstCol"><img class="avatar-style" src="https://avatars.githubusercontent.com/NONE"></img><div class="d-block">Isabel Kemmer</div></div></p></td>
<td><p class="card-text"><a target="_blank" href="https://orcid.org/0000-0002-8799-4671"><i class="fab fa-orcid fa-2x text--orange"></i></a></p></td>
<td><p class="card-text">Euro-BioImaging ERIC</p></td>
<td><p class="card-text"><i class="fas fa-graduation-cap fa-1x text--orange" alt="Academic"></i></p></td>
<td><p class="card-text"></p></td>
<td><p class="card-text">Conceptualization, Writing - Original Draft, Review &amp; Editing</p></td>
</tr>
<tr class="row-odd"><td><p class="card-text"><div class="firstCol"><img class="avatar-style" src="https://avatars.githubusercontent.com/NONE"></img><div class="d-block">Maria Mirza</div></div></p></td>
<td><p class="card-text"><a target="_blank" href="https://orcid.org/0000-0001-9158-6403"><i class="fab fa-orcid fa-2x text--orange"></i></a></p></td>
<td><p class="card-text">Euro-BioImaging ERIC</p></td>
<td><p class="card-text"><i class="fas fa-graduation-cap fa-1x text--orange" alt="Academic"></i></p></td>
<td><p class="card-text"></p></td>
<td><p class="card-text">Writing - Review &amp; Editing</p></td>
</tr>
</tbody>
</table>
</div>
</details></div>
<div class="section" id="license">
<h2><span class="section-number">7.10. </span>License<a class="headerlink" href="#license" title="Permalink to this headline">¶</a></h2>
<details class="sphinx-bs dropdown card mb-3">
<summary class="summary-title card-header">
<strong>License</strong><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<a href="https://creativecommons.org/licenses/by/4.0/">
    <img src="../../../_static/images/logo/cc.svg"  height="25"/>
    <img src="../../../_static/images/logo/by.svg"  height="25"/></a>
    <a href="https://creativecommons.org/licenses/by/4.0/"> The Creative Commons 4.0 BY license</a></br>
    <br/>
</div>
</details><hr class="footnotes docutils" />
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./content/recipes/applied-examples"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                                            </v-col>
                                            <div class="d-none d-md-block col-md-2 bd-toc show collapse custom-bdtoc pt-5">
                                                <div style="position: fixed">
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i>
            Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#overview">
   7.1. Overview
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#ingredients">
   7.2. Ingredients
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#chosing-repositories-for-bioimaging-data">
   7.3. Chosing repositories for bioimaging data
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#the-example-dataset">
   7.4. The example dataset
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#step-by-step-recipe-for-bioimage-data-deposition-to-bioimage-archive">
   7.5. Step-by-step recipe for bioimage data deposition to BioImage Archive
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#step-1-familiarize-with-the-bioimage-archive">
     7.5.1. Step 1: Familiarize with the BioImage Archive
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#creating-an-account">
     7.5.2. Creating an account
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#step-2-familiarize-with-rembi">
     7.5.3. Step 2: Familiarize with REMBI
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#step-3-organize-the-data">
     7.5.4. Step 3: Organize the data
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#data-organization-in-folders">
       7.5.4.1. Data organization in folders
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#define-the-study-components">
       7.5.4.2. Define the Study Components
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#step-4-upload-the-data">
     7.5.5. Step 4: Upload the data
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#step-5-create-the-submission">
     7.5.6. Step 5: Create the submission
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#input-the-study-level-metadata">
       7.5.6.1. Input the Study-level metadata
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#input-the-experiment-metadata">
       7.5.6.2. Input the Experiment metadata
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#duplicating-rembi-components">
       7.5.6.3. Duplicating REMBI components
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#study-component-associations">
       7.5.6.4. Study Component associations
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#step-6-generate-the-file-list">
     7.5.7. Step 6: Generate the File-list
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#associating-the-file-lists">
       7.5.7.1. Associating the File-lists
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#optional-annotations">
     7.5.8. Optional: Annotations
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#step-8-submit-your-data">
     7.5.9. Step 8: Submit your data
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#orcid-claim">
       7.5.9.1. ORCID claim
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#conclusion">
   7.6. Conclusion
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#what-to-read-next">
     7.6.1. What to read next?
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#references">
   7.7. References
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#funding">
   7.8. Funding
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#authors">
   7.9. Authors
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#license">
   7.10. License
  </a>
 </li>
</ul>

        </nav>
        </div>
                                            </div>
                                        
                                    </v-row>

                                    
                                        <v-row class="mb-5">
                                            <div class='pl-10 prev-next-bottom mr-10' style="width:82%">
                                                
<a class='left-prev' id="prev-link" href="eubopen-hcs-bioimage.html" title="previous page"><span class="section-number">6. </span>IMI EUBOPEN FAIR High-Content Screening data deposition</a>
<a class='right-next' id="next-link" href="approach-cdisc.html" title="next page"><span class="section-number">8. </span>Mapping of clinical trial data to CDISC-SDTM: a practical example based on APPROACH and ABIRISK</a>

                                            </div>
                                        </v-row>
                                    
                                </v-container>
                            </main>
                        
                    
                </v-row>
                
            </v-container>
        </v-row>
    </v-container>
</v-app>


  <script src="../../../_static/js/index.d3f166471bb80abb5163.js"></script>



<!-- Google Analytics -->
<script>
    window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
    ga('create', 'UA-171697285-1', 'auto');
    ga('set', 'anonymizeIp', true);
    ga('send', 'pageview');
</script>
<script async src='https://www.google-analytics.com/analytics.js'></script>
<!-- End Google Analytics -->

<v-container fluid>
    <v-row no-gutter>
        <v-col cols="12"><footer class="footer" style=" margin-top: 12px;  ">
    <div class="container-md mt-4">
      <div class="row" style="display:flex; justify-content: center">
          <div class="col-12 col-sm-12 col-md-4 col-lg-4 col-xl-3">
            <div class="footerImages">
              <a href="https://europa.eu/"><img src="/_static/images/logo/eu-flag.png" height="40" alt="EU flag"></a>
              <a href="https://www.imi.europa.eu/"><img src="/_static/images/logo/imi-logo.png" height="40" alt="IMI logo"></a>
              <a href="https://efpia.eu"><img src="/_static/images/logo/efpia-logo.png" height="40" alt="EFPIA logo"></a>
            </div>
            <div class="mb-3"> Grant agreement 802750 </div>
          </div>
          <div class="col-5 col-sm-2 col-xs-5 col-md-2 col-lg-2 col-xl-2">
              <ul>
                <li><a href="mailto:faircookbook-ed@elixir-europe.org"> Contact us </a></li>
                <li><a href="https://fairplus-project.eu" target="_blank"> FAIRplus </a> </li>
              </ul>
          </div>
          <div class="col-5 col-sm-2 col-xs-5 col-md-2 col-lg-2 col-xl-2">
              <ul>
                <li> <a href="https://faircookbook.elixir-europe.org/content/recipes/afterword/people.html  "> Contributors </a> </li> </li>
                <li> <a href="https://faircookbook.elixir-europe.org/content/recipes/help.html"> How to contribute </a> </li>
              </ul>
          </div>
          <div class="col-12 col-sm-4 col-xs-12 col-md-3 col-lg-3 col-xl-3 pb-5">
            <a class="mr-3" href="https://elixir-europe.org/" target="_blank"><img src="/_static/images/logo/Elixir/ELIXIR.png" height="50" alt="Elixir logo"></a>
            <a href="https://fairplus-project.eu/"><img src="https://fairplus-project.eu/images/fairplus-logo.png" alt="FAIRplus logo" style="background:white;padding:5px;height:33px;"></a>
          </div>
          <div class="col-12 pb-4">
              Recommended in the
              <a
                href="https://www.imi.europa.eu/sites/default/files/uploads/documents/resources-for-projects/IMI2_OpenAcesGuidelines_Updated2021.pdf"
                target="_blank"
                class="primary--text"
              >
                 IMI/IHI Project Guidelines
              </a> and the
              <a
                href="link to https://ec.europa.eu/info/funding-tenders/opportunities/docs/2021-2027/horizon/wp-call/2023-2024/wp-4-health_horizon-2023-2024_en.pdf"
                target="_blank"
                class="primary--text"
              >
                  Horizon Europe Work Programme for Health
              </a> as the resource for FAIR Data Management guidelines and good practices for the Life Sciences.
          </div>
      </div>
    </div>

    <div class="container-fluid" style="padding:0">
        <div class="row" style="padding:0; margin:0">
            <div class="col-12 purple-dark">
                The FAIR Cookbook is licensed under the
                <a href="https://creativecommons.org/licenses/by/4.0/" target="_blank">
                    Creative Commons Attribution 4.0 International
                </a> (CC BY 4.0) license.
            </div>
        </div>
        <div class="row extra_footer d-none">
            <script> mermaid.init() </script>
        </div>
    </div>

</footer> </v-col>
    </v-row>
</v-container>
<script type="application/javascript">
    const vuetify = new Vuetify({
        theme: {
            themes: {
                light: {
                    primary: '#421C6E',
                    secondary: '#FC7A4A',
                    accent: '#8038D1',
                    error: '#FF5252',
                    info: '#2196F3',
                    success: '#7E0038',
                    warning: '#FFC107'
                }
            }
        },
        icons: { iconfont: 'fa4' }
    })
    const my_vue = new Vue({ el: '#app', vuetify: vuetify })

</script>
  </body>
</html>