(fcb-public-kg)=
# Public knowledge graph resources for life sciences


````{panels_fairplus}
:identifier_text: FCB071
:identifier_link: 'https://w3id.org/faircookbook/FCB071'
:difficulty_level: 2
:recipe_type: guidance
:reading_time_minutes: 10
:intended_audience:  everyone 
:has_executable_code: nope
:maturity_level: 0
:maturity_indicator: 0
:recipe_name: Raising Awareness in Public Knowledge Graphs for Life Sciences
```` 

## Main Objectives

In the following sections, we will cover the following topics:

1. the known reference knowledge graphs to integrate with.
2. the different scientific contexts.
3. the impact for Pharma industry and EFPIA more broadly.

## Introduction

When it comes to research (URL_TO_INSERT_RECORD_6843 https://fairsharing.org/FAIRsharing.52b22c)  carried out by academic and industry ventures such as the Innovative Medicine Initiative of these European Union, three main scenarios can be outlined:

- Identification of new drug targets, as it the case for the [IMI Resolute](https://www.imi.europa.eu/projects-results/project-factsheets/resolute) project (URL_TO_INSERT_TERM_6844 https://fairsharing.org/search?recordType=project)  for instance, which focuses on
characterizing solute carriers and identifying potential candidates for "druggability".
- Development of new therapeutic products and techniques given a specific disease area, as in the [IMI AETIONOMY](https://www.imi.europa.eu/projects-results/project-factsheets/aetionomy) project (URL_TO_INSERT_TERM_6845 https://fairsharing.org/search?recordType=project) .
- Discovery of new drugs against pathogens as for instance the [New Drugs for Bad Bugs (ND4BB)](https://www.imi.europa.eu/projects-results/project-factsheets/nd4bb) project (URL_TO_INSERT_TERM_6846 https://fairsharing.org/search?recordType=project) .

This gives a backdrop against which work to enhance interoperability of data generated by these project (URL_TO_INSERT_TERM_6847 https://fairsharing.org/search?recordType=project) s needs to be aware of the existing data landscape.

The purpose of the present content type is to provide a basic survey of the existing, and the relevance to specific research (URL_TO_INSERT_RECORD_6852 https://fairsharing.org/FAIRsharing.52b22c)  contexts and tasks. This informat (URL_TO_INSERT_TERM_6848 https://fairsharing.org/search?recordType=model_and_format) ion may be useful when deciding on particular technical aspects such as choice of data interface, choice of terminology (URL_TO_INSERT_TERM_6850 https://fairsharing.org/search?recordType=terminology_artefact)  and controlled terminologies (URL_TO_INSERT_TERM_6851 https://fairsharing.org/search?recordType=terminology_artefact)  or, choice of data format (URL_TO_INSERT_TERM_6849 https://fairsharing.org/search?recordType=model_and_format) s.

We therefore focus the dataset scan on main entities such as Drugs / Chemicals and Disease / Clinical Research (URL_TO_INSERT_RECORD_6853 https://fairsharing.org/FAIRsharing.52b22c)  Data. Such datasets are popularly known as Knowledge Graphs (KGs).

Additionally, the availability of such informat (URL_TO_INSERT_TERM_6854 https://fairsharing.org/search?recordType=model_and_format) ion in specialized graph representations was also included as a selection criteria. Therefore, services providing SP (URL_TO_INSERT_RECORD_6857 https://fairsharing.org/FAIRsharing.s63y3p) ARQL (URL_TO_INSERT_RECORD_6856 https://fairsharing.org/FAIRsharing.87ccfd)  endpoint and Graph API for dataset access were considered and the following sections providing specific informat (URL_TO_INSERT_TERM_6855 https://fairsharing.org/search?recordType=model_and_format) ion and references for each of the resources and datasets are considered


## A catalog of SPARQL Endpoints for Life Sciences: Yummydata

Yummydata is a catalog of SP (URL_TO_INSERT_RECORD_6860 https://fairsharing.org/FAIRsharing.s63y3p) ARQL (URL_TO_INSERT_RECORD_6859 https://fairsharing.org/FAIRsharing.87ccfd)  endpoints of interest to the biomedical community, maintained by the Database (URL_TO_INSERT_TERM_6858 https://fairsharing.org/search?fairsharingRegistry=Database)  Center for Life Sciences in Kashiwa, Japan {footcite}`pmid29688370`. 

Database (URL_TO_INSERT_TERM_6861 https://fairsharing.org/search?fairsharingRegistry=Database)  URL (URL_TO_INSERT_RECORD_6862 https://fairsharing.org/FAIRsharing.9d38e2) : http://yummydata.org/

The resource can be viewed as a registry to resources as well as a monitoring services, which regularly checks the "health status" of the resources it indexes.

The "health status" is reported as a "Umaka Score (URL_TO_INSERT_RECORD_6863 https://fairsharing.org/FAIRsharing.m283c)  (URL_TO_INSERT_RECORD_6864 https://fairsharing.org/FAIRsharing.xMmOCL) " which consolidates endpoint status according to six dimensions (URL_TO_INSERT_RECORD_6865 https://fairsharing.org/FAIRsharing.9af33c) :
**Availability, Freshness, Operation, Usefulness, Validity, and Performance**.

It should be noted that Yummydata does not differentiate between semantic artefacts, such as an ontology (URL_TO_INSERT_TERM_6866 https://fairsharing.org/search?recordType=terminology_artefact) , or a dataset
represented in RDF (URL_TO_INSERT_RECORD_6867 https://fairsharing.org/FAIRsharing.p77ph9) Linked data. 

The authors of the resource highlight this point and underline that being able to distinguish between these types of graph explicitly may be required in the future.


````{dropdown} **YummyData**
:open:

```{figure} ../../../images/yummydata.png
---
name: yummydata
alt: YummyData, a registry of sparql endpoints
---
YummyData, a registry of sparql endpoints
```
````

## Resources for Druggable Targets Predictions


### 1. OpenTarget

The [OpenTarget project](https://www.opentargets.org/) is a public private partnership aiming at providing a 
central resource based on public domain data to enable target identification and prioritisation, based on functional genomics data and GWAS (URL_TO_INSERT_RECORD_6868 https://fairsharing.org/FAIRsharing.ubYn8D)  data {footcite}`10.1093/nar/gkaa1027`. 

Using a genome wide, evidence based approach, the OpenTarget project (URL_TO_INSERT_TERM_6869 https://fairsharing.org/search?recordType=project)  offers the possibility to obtain reliable informat (URL_TO_INSERT_TERM_6870 https://fairsharing.org/search?recordType=model_and_format) ion
about validated targets against which design active compounds. Such approach has been shown to yield higher success in progressing potential therapeutic compounds along the clinical trial pipeline.

All the data generated by the OpenTarget project (URL_TO_INSERT_TERM_6871 https://fairsharing.org/search?recordType=project)  is available for download and is accessible via 
* a GraphQL application programming interface. https://api.genetics.opentargets.org/graphql
* a Google Big Query interface. https://platform-docs.opentargets.org/data-access/google-bigquery
* a ftp download. https://platform.opentargets.org (URL_TO_INSERT_RECORD_6872 https://fairsharing.org/FAIRsharing.3f9n4y) /downloads/data

More recently, the dataset is also available from Amazon Web Services


````{dropdown} OpenTarget project in AWS
:open:
```{figure} ../../../images/opentarget-aws.png
---
name: OpenTarget project (URL_TO_INSERT_TERM_6873 https://fairsharing.org/search?recordType=project)  in AWS
alt: OpenTarget project (URL_TO_INSERT_TERM_6874 https://fairsharing.org/search?recordType=project)  in AWS
---
OpenTarget project (URL_TO_INSERT_TERM_6875 https://fairsharing.org/search?recordType=project)  in AWS
```
````

The associated code is also open sourced, under an Apache License 2.0. More informat (URL_TO_INSERT_TERM_6876 https://fairsharing.org/search?recordType=model_and_format) ion is available here: https://genetics-docs.opentargets.org/data-access/data-download

GitHub (URL_TO_INSERT_RECORD_6878 https://fairsharing.org/FAIRsharing.c55d5e)  Repository (URL_TO_INSERT_TERM_6877 https://fairsharing.org/search?recordType=repository) : https://github.com (URL_TO_INSERT_RECORD_6879 https://fairsharing.org/FAIRsharing.c55d5e) /opentargets

The OpenTarget project (URL_TO_INSERT_TERM_6880 https://fairsharing.org/search?recordType=project)  integrates data from several key resources listed in the project (URL_TO_INSERT_TERM_6881 https://fairsharing.org/search?recordType=project) 's main publication. 
From an integration scenario standpoint, it is worth noting that data are annotated with semantic resources such as:
* EFO (URL_TO_INSERT_RECORD_6884 https://fairsharing.org/FAIRsharing.1gr4tz) : Experimental Factor Ontology (URL_TO_INSERT_TERM_6882 https://fairsharing.org/search?recordType=terminology_artefact)  (URL_TO_INSERT_RECORD_6883 https://fairsharing.org/FAIRsharing.1gr4tz)  and associated open target slim EFO (URL_TO_INSERT_RECORD_6885 https://fairsharing.org/FAIRsharing.1gr4tz) -OTAR
* MONDO (URL_TO_INSERT_RECORD_6886 https://fairsharing.org/FAIRsharing.b2979t) : Diseases
* HP (URL_TO_INSERT_RECORD_6887 https://fairsharing.org/FAIRsharing.kbtt7f) O: clinical signs, symptoms and phenotypes 
* ECO (URL_TO_INSERT_RECORD_6889 https://fairsharing.org/FAIRsharing.wvpgwn) : evidence code ontology (URL_TO_INSERT_TERM_6888 https://fairsharing.org/search?recordType=terminology_artefact)  to annotate confidence and reliability of findings

```bash
query genePrioritisationUsingL2G {
  studyLocus2GeneTable(studyId: "FINNGEN_R5_E4_DM2", variantId: "11_72752390_G_A") {
    rows {
      gene {
        symbol
        id
      }
      yProbaModel
      yProbaDistance
      yProbaInteraction
      yProbaMolecularQTL
      yProbaPathogenicity
      hasColoc
      distanceToLocus
    }
  }
}
```


```python
#!/usr/bin/env python3

# Import relevant libraries for HTTP request and JSON formatting
import requests
import json

# Set study_id and variant_id variables
study_id = "FINNGEN_R5_E4_DM2"
variant_id = "11_72752390_G_A"

# Build query string
query_string = """
  query genePrioritisationUsingL2G($myVariantId: String!, $myStudyId: String! ){
    studyLocus2GeneTable(studyId: $myStudyId, variantId: $myVariantId){
      rows {
        gene {
          symbol
          id
        }
        yProbaModel
        yProbaDistance
        yProbaInteraction
        yProbaMolecularQTL
        yProbaPathogenicity
        hasColoc
        distanceToLocus
      }
    }
  }
"""

# Set variables object of arguments to be passed to endpoint
variables = {"myVariantId": variant_id, "myStudyId": study_id}

# Set base URL of Genetics Portal GraphQL API endpoint
base_url = "https://api.genetics.opentargets.org/graphql"

# Perform POST request and check status code of response
r = requests.post(base_url, json={"query": query_string, "variables": variables})
print(r.status_code)

# Transform API response into JSON 
api_response_as_json = json.loads(r.text)

# Print first element of JSON response data
print(api_response_as_json["data"]["studyLocus2GeneTable"]["rows"][0])
```


### 2. Pharos

Pharos is a US NIH funded effort, which can be viewed as the counterpart to the European lead OpenTarget project (URL_TO_INSERT_TERM_6890 https://fairsharing.org/search?recordType=project)  {footcite}`10.1093/nar/gkw1072`.

Pharos comes on the back of initiative such as Illuminating the Druggable Genome (IDG) program (https://commonfund.nih.gov/idg/index) and Target Central Resource Database (URL_TO_INSERT_TERM_6891 https://fairsharing.org/search?fairsharingRegistry=Database)  (URL_TO_INSERT_RECORD_6892 https://fairsharing.org/FAIRsharing.1gn47b)  (TCRD (URL_TO_INSERT_RECORD_6893 https://fairsharing.org/FAIRsharing.1gn47b) ).

Pharos integrates data from 66 different data sources, the list of which is available [here](https://oup.silverchair-cdn.com/oup/backfile/Content_public/Journal/nar/45/D1/10.1093_nar_gkw1072/3/gkw1072_Supp.zip?Expires=1653384840&Signature=kOr49PEQRpjs-NQN2Q~Vct~BpdLLJZhi-mAUoTqG9c7zxMrO7UiuCzNh~NyGt50JEqOYRC7DIx-h5h1rFPtphFUIsHDIjBa5mpndXfXZxcL4T-mVKbV92JzOpvqfErWGHjZ4859~qHXb3h97ngUwWl~7OdKVMSa0IutzE0eSns9Kdje25lpaWHhEI-5M946e1-l-IAbqC7~QYZXXqdDPGJi2BCEXVHI2SU8I2j8c~TwhFXioVfrN8LwJpLBw-iiWYTbInocWself9GCvmCbv2poNdYTJIaE03UYpI~l7YVah7sApesGOjlBDPppRiJLvDUf5TI6Y2pjliAECS8tJyg__&Key-Pair-Id=APKAIE5G5CRDK6RD3PGA)

Pharos informat (URL_TO_INSERT_TERM_6894 https://fairsharing.org/search?recordType=model_and_format) ion is organized around:
- Diseases
- Targets
- Ligands

````{dropdown} Pharos
:open:
```{figure} ../../../images/pharos-project.png
---
width: 850px
name: Pharos
alt: Pharos
---
Pharos
```
````

All the Pharos data can be queried via a GraphQL API: https://pharos.nih.gov/api

From an integration scenario standpoint, Pharos integrates data with custom-built ontology (URL_TO_INSERT_TERM_6895 https://fairsharing.org/search?recordType=terminology_artefact) , the [Drug Target Ontology (DTO)](http://drugtargetontology.org).

The DTO (URL_TO_INSERT_RECORD_6896 https://fairsharing.org/FAIRsharing.tke3y2)  reuses terms from well established resources such as:
* DO: Disease Ontology (URL_TO_INSERT_TERM_6897 https://fairsharing.org/search?recordType=terminology_artefact)  (URL_TO_INSERT_RECORD_6898 https://fairsharing.org/FAIRsharing.8b6wfq) 
* GO (URL_TO_INSERT_RECORD_6901 https://fairsharing.org/FAIRsharing.6xq0ee) : Gene Ontology (URL_TO_INSERT_TERM_6899 https://fairsharing.org/search?recordType=terminology_artefact)  (URL_TO_INSERT_RECORD_6900 https://fairsharing.org/FAIRsharing.6xq0ee) 
* BAO (URL_TO_INSERT_RECORD_6904 https://fairsharing.org/FAIRsharing.mye76w) : BioAssay Ontology (URL_TO_INSERT_TERM_6902 https://fairsharing.org/search?recordType=terminology_artefact)  (URL_TO_INSERT_RECORD_6903 https://fairsharing.org/FAIRsharing.mye76w) 
* GPCRO: GPCR Ontology (URL_TO_INSERT_TERM_6905 https://fairsharing.org/search?recordType=terminology_artefact) 

Pharos source code is available under the MIT license, from https://spotlite.nih.gov/ncats/pharos.


### 3. Hetionet: an integrative network of disease

The main purpose of  Hetionet was  to predict **new uses for existing drugs**, as part of the "Project (URL_TO_INSERT_TERM_6906 https://fairsharing.org/search?recordType=project)  Rephetio" .

Hetionet Version 1.0 corresponds to an 'heterogeneous network' (or HetNet) and contains 47,031 nodes of 11 types and 2,250,197 relationships of 24 types, integrating data from 29 Resources to connect compounds, diseases, genes, 
anatomies, pathways, biological processes, molecular functions, cellular components, pharmacologic classes, side effects, and symptoms. 

For more details about the full list of source, consult the publication {footcite}`pmid28936969`.

[![Latest Zenodo DOI](https://zenodo.org/badge/56091268.svg)](https://zenodo.org/badge/latestdoi/56091268)

The network is available in four format (URL_TO_INSERT_TERM_6907 https://fairsharing.org/search?recordType=model_and_format) s:

+ **JSON (URL_TO_INSERT_RECORD_6908 https://fairsharing.org/FAIRsharing.5bbab9) ** — see [`hetnet/json`](https://github.com/hetio/hetionet/tree/master/hetnet/json)
+ **Neo4j** — see [`hetnet/neo4j`](https://github.com/hetio/hetionet/tree/master/hetnet/neo4j) annd [https://neo4j.het.io/browser/](https://neo4j.het.io/browser/)
+ **TSV** — see [`hetnet/tsv`](https://github.com/hetio/hetionet/tree/master/hetnet/tsv)
+ **matrix** — see [`hetnet/matrix`](https://github.com/hetio/hetionet/tree/master/hetnet/matrix)

GitHb Repository (URL_TO_INSERT_TERM_6909 https://fairsharing.org/search?recordType=repository) : https://github.com (URL_TO_INSERT_RECORD_6910 https://fairsharing.org/FAIRsharing.c55d5e) /hetio/hetionet

license: [CC0](https://creativecommons.org/publicdomain/zero/1.0/ "CC0 1.0 Universal · Public Domain Dedication")

From an integration scenario standpoint, Hetionet integrates knowledge semantically by anchoring entities to several ontology (URL_TO_INSERT_TERM_6911 https://fairsharing.org/search?recordType=terminology_artefact)  "slims", which are simplified versions of larger semantic artefacts.

* DO: disease ontology (URL_TO_INSERT_TERM_6912 https://fairsharing.org/search?recordType=terminology_artefact)  (URL_TO_INSERT_RECORD_6913 https://fairsharing.org/FAIRsharing.8b6wfq)  slim, with 137 terms considered.
* MESH (URL_TO_INSERT_RECORD_6914 https://fairsharing.org/FAIRsharing.qnkw45) : for signs and symptoms, with 438 terms considered
* SIDER (URL_TO_INSERT_RECORD_6915 https://fairsharing.org/FAIRsharing.zzaykv) : for side effects
* GO (URL_TO_INSERT_RECORD_6916 https://fairsharing.org/FAIRsharing.6xq0ee) : Biological processes, cellular components, and molecular functions.
* UBERON (URL_TO_INSERT_RECORD_6917 https://fairsharing.org/FAIRsharing.4c0b6b) : for anatomical terms.

Map (URL_TO_INSERT_RECORD_6920 https://fairsharing.org/FAIRsharing.53edcc) pings provide connection to chemical informat (URL_TO_INSERT_TERM_6918 https://fairsharing.org/search?recordType=model_and_format) ion via the use of InChI (URL_TO_INSERT_RECORD_6919 https://fairsharing.org/FAIRsharing.ddk9t9)  keys to unambiguously specific chemical structures. 

### 4. Drug Repurposing Knowledge graph

Drug Repurposing Knowledge Graph (DRKG) is a comprehensive biological KG relating genes, compounds, diseases, biological processes, side effects, and symptoms.

DRKG includes informat (URL_TO_INSERT_TERM_6922 https://fairsharing.org/search?recordType=model_and_format) ion from six existing database (URL_TO_INSERT_TERM_6921 https://fairsharing.org/search?fairsharingRegistry=Database) s including DrugBank (URL_TO_INSERT_RECORD_6924 https://fairsharing.org/FAIRsharing.353yat) , Hetionet, GNBR, String (URL_TO_INSERT_RECORD_6925 https://fairsharing.org/FAIRsharing.9b7wvk) , IntAct (URL_TO_INSERT_RECORD_6923 https://fairsharing.org/FAIRsharing.d05nwx)  and DGIdb, and data collected from recent publications particularly related to Covid-19. 

It includes 97,238 entities belonging to 13 entity-types; and 5,874,261 triples belonging to 107 edge-types.

GitHb Repository (URL_TO_INSERT_TERM_6926 https://fairsharing.org/search?recordType=repository) : https://github.com (URL_TO_INSERT_RECORD_6927 https://fairsharing.org/FAIRsharing.c55d5e) /gnn4dr/DRKG/

A number of jupyter notebooks are available from the repository (URL_TO_INSERT_TERM_6928 https://fairsharing.org/search?recordType=repository) . They show how to use the KGs to explore drug repurposing possibilities with help of a pretrained model (URL_TO_INSERT_TERM_6929 https://fairsharing.org/search?recordType=model_and_format) :

Knowledge Graph training notebook: https://github.com (URL_TO_INSERT_RECORD_6930 https://fairsharing.org/FAIRsharing.c55d5e) /gnn4dr/DRKG/blob/master/drug_repurpose/COVID-19_drug_repurposing_via_genes.ipynb

### 5. OpenBiolink

OpenBioLink is an open benchmark dataset that serve as training or evaluation datasets for link prediction model (URL_TO_INSERT_TERM_6931 https://fairsharing.org/search?recordType=model_and_format) s {footcite}`rpath_paper_ddomingo`, {footcite}`drug2ways_ddomingo`. It is an ensembl (URL_TO_INSERT_RECORD_6932 https://fairsharing.org/FAIRsharing.fx0mw7)  of large independent resource connecting drugs to disease via proteins. These resources include STRING (URL_TO_INSERT_RECORD_6940 https://fairsharing.org/FAIRsharing.9b7wvk) , GO (URL_TO_INSERT_RECORD_6934 https://fairsharing.org/FAIRsharing.6xq0ee) , DisGeNet (URL_TO_INSERT_RECORD_6941 https://fairsharing.org/FAIRsharing.fssydn) , HP (URL_TO_INSERT_RECORD_6937 https://fairsharing.org/FAIRsharing.kbtt7f) O, STITCH (URL_TO_INSERT_RECORD_6933 https://fairsharing.org/FAIRsharing.zhwa8x) , CTD (URL_TO_INSERT_RECORD_6936 https://fairsharing.org/FAIRsharing.h3tjtr) , HP (URL_TO_INSERT_RECORD_6938 https://fairsharing.org/FAIRsharing.kbtt7f) O, DrugCentral (URL_TO_INSERT_RECORD_6939 https://fairsharing.org/FAIRsharing.3me82d) , and SIDER (URL_TO_INSERT_RECORD_6935 https://fairsharing.org/FAIRsharing.zzaykv) . 

Taking a deeper dive into the KG statistics, there are 24,806 entities that are either drugs, proteins, indications or phenotype and 2,35.881 triples that are part of 4 edge-types.

GitHb Repository (URL_TO_INSERT_TERM_6942 https://fairsharing.org/search?recordType=repository) : https://openbiolink.github.io/


```python
from openbiolink.obl2021 import OBL2021Dataset

dl = OBL2021Dataset()

train = dl.training # torch.tensor of shape(num_train,3)
valid = dl.validation # torch.tensor of shape(num_val,3)
```
{footcite}`pmid32339214`


### 6. Clinical Knowledge Graph

The Clinical Knowledge Graph is an open source platform currently comprised of more than 16 million nodes and 220 million relationships to represent relevant experimental data, public database (URL_TO_INSERT_TERM_6943 https://fairsharing.org/search?fairsharingRegistry=Database) s and the literature {footcite}`Santos2020.05.09.084897`.

Documentation: https://ckg.readthedocs.io/en/latest/INTRO.html

From an integration scenario standpoint, the CKG integrates data using the following semantic resources:
* DO: Disease Ontology (URL_TO_INSERT_TERM_6944 https://fairsharing.org/search?recordType=terminology_artefact)  (URL_TO_INSERT_RECORD_6945 https://fairsharing.org/FAIRsharing.8b6wfq) 
* BT (URL_TO_INSERT_RECORD_6949 https://fairsharing.org/FAIRsharing.6s2sfz) O (URL_TO_INSERT_RECORD_6947 https://fairsharing.org/FAIRsharing.1414v8) : Brenda Tissue Ontology (URL_TO_INSERT_TERM_6946 https://fairsharing.org/search?recordType=terminology_artefact)  for anatomical entities and tissues (URL_TO_INSERT_RECORD_6948 https://fairsharing.org/FAIRsharing.yXPvpU) .
* EFO (URL_TO_INSERT_RECORD_6952 https://fairsharing.org/FAIRsharing.1gr4tz) : Experimental Factor Ontology (URL_TO_INSERT_TERM_6950 https://fairsharing.org/search?recordType=terminology_artefact)  (URL_TO_INSERT_RECORD_6951 https://fairsharing.org/FAIRsharing.1gr4tz) 
* HP (URL_TO_INSERT_RECORD_6955 https://fairsharing.org/FAIRsharing.kbtt7f) O: Human Phenotype Ontology (URL_TO_INSERT_TERM_6953 https://fairsharing.org/search?recordType=terminology_artefact)  (URL_TO_INSERT_RECORD_6954 https://fairsharing.org/FAIRsharing.kbtt7f)  
* GO (URL_TO_INSERT_RECORD_6958 https://fairsharing.org/FAIRsharing.6xq0ee) : Gene Ontology (URL_TO_INSERT_TERM_6956 https://fairsharing.org/search?recordType=terminology_artefact)  (URL_TO_INSERT_RECORD_6957 https://fairsharing.org/FAIRsharing.6xq0ee) 
* SNOMED-CT: which defines clinical terms and their associative relationships.

The code for CKG is available from GitHub (URL_TO_INSERT_RECORD_6959 https://fairsharing.org/FAIRsharing.c55d5e)  at: https://github.com (URL_TO_INSERT_RECORD_6960 https://fairsharing.org/FAIRsharing.c55d5e) /MannLabs/CKG under a MIT License.

CKG can be loaded and queried via a Neo4j graph database (URL_TO_INSERT_TERM_6961 https://fairsharing.org/search?fairsharingRegistry=Database) , a data dump of which is available from: https://data.mendeley.com (URL_TO_INSERT_RECORD_6962 https://fairsharing.org/FAIRsharing.3epmpp) /datasets/mrcf7f4tc2/1.

### 7. AstraZeneca Biological Insights Knowledge Graph BIKG

The BIKG is a resource developed (URL_TO_INSERT_RECORD_6963 https://fairsharing.org/FAIRsharing.31385c)  by AstraZeneca to combine both public and internal data to facilitate knowledge discovery using machine learning over KG.

To build it, AstraZeneca developers created dedicated informat (URL_TO_INSERT_TERM_6965 https://fairsharing.org/search?recordType=model_and_format) ion extraction pipeline for mining unstructured text such as full text publications using Natural Language Processing (NLP) techniques, as well as informat (URL_TO_INSERT_TERM_6966 https://fairsharing.org/search?recordType=model_and_format) ion obtained from specialized database (URL_TO_INSERT_TERM_6964 https://fairsharing.org/search?fairsharingRegistry=Database) s such as ChEMBL (URL_TO_INSERT_RECORD_6968 https://fairsharing.org/FAIRsharing.m3jtpg)  or Ensembl (URL_TO_INSERT_RECORD_6967 https://fairsharing.org/FAIRsharing.fx0mw7)  for chemical and gene products respectively, but also other datasets such as Hetionet and Open Target.

The latest version of the BIKG consists of 10.9m nodes (of 22 types) and over 118 million unique edges (of 59 types, forming 398 different triples)" {footcite}`Geleta2021.10.28.466262`.

Interestingly, the BIKG pipelines also allow third-party datasets to be integrated. This step involves an ETL into the BIKG semantic model (URL_TO_INSERT_TERM_6969 https://fairsharing.org/search?recordType=model_and_format)  and the BIKG upper ontology (URL_TO_INSERT_TERM_6970 https://fairsharing.org/search?recordType=terminology_artefact)  (BIKG ULO).

For integration purpose, the BIKG ULO is compatible with ontologies (URL_TO_INSERT_TERM_6971 https://fairsharing.org/search?recordType=terminology_artefact)  such as:
* UBERON (URL_TO_INSERT_RECORD_6972 https://fairsharing.org/FAIRsharing.4c0b6b) : for anatomical terms
* GO (URL_TO_INSERT_RECORD_6975 https://fairsharing.org/FAIRsharing.6xq0ee) :  Gene Ontology (URL_TO_INSERT_TERM_6973 https://fairsharing.org/search?recordType=terminology_artefact)  (URL_TO_INSERT_RECORD_6974 https://fairsharing.org/FAIRsharing.6xq0ee)  for Biological processes, cellular components, and molecular functions.


BIKG can be accessed via a GraphQL API and the underlying data can be exported in 2 different graph format (URL_TO_INSERT_TERM_6976 https://fairsharing.org/search?recordType=model_and_format) s, RDF (URL_TO_INSERT_RECORD_6977 https://fairsharing.org/FAIRsharing.p77ph9)  or Neo4j/CSV for linked data graph or property graph representations respectively.

Interestingly, two use cases for using the BIKG are discussed in the associated publications and clearly outline FAIR (URL_TO_INSERT_RECORD_6978 https://fairsharing.org/FAIRsharing.WWI10U) ification goals. These use cases are:
> Target Identifications
>
> CRISP (URL_TO_INSERT_RECORD_6979 https://fairsharing.org/FAIRsharing.1evfpc) R Screen Hit ranking


## PyKEEN

PyKEEN (Python KnowlEdge EmbeddiNgs) is a Python package designed to train and evaluate knowledge graph embedding model (URL_TO_INSERT_TERM_6980 https://fairsharing.org/search?recordType=model_and_format) s (KGEM) (incorporating multi-modal informat (URL_TO_INSERT_TERM_6981 https://fairsharing.org/search?recordType=model_and_format) ion) {footcite}`pmid34735335`, {footcite}`pmid30768158`.

GitHub (URL_TO_INSERT_RECORD_6983 https://fairsharing.org/FAIRsharing.c55d5e)  Repository (URL_TO_INSERT_TERM_6982 https://fairsharing.org/search?recordType=repository) : https://github.com (URL_TO_INSERT_RECORD_6984 https://fairsharing.org/FAIRsharing.c55d5e) /pykeen/pykeen

So, as such, PyKEEN is more a tool than a KG, however PyKEEN contains many reference KGs and datasets which can be used for a variety of data science tasks and machine learning applications. PyKEEN is mainly geared towards helping developers train KGEM for tasks such as link prediction.

The following figure shows a snippet of the various KGs available from the python package.

````{dropdown} PyKEEN datasets
:open:
```{figure} ../../../images/pykeen_datasets.png
---
width: 750px
name: pykeen
alt: pykeen
---
A screenshot of the README on the PyKEEN GitHub (URL_TO_INSERT_RECORD_6986 https://fairsharing.org/FAIRsharing.c55d5e)  repository (URL_TO_INSERT_TERM_6985 https://fairsharing.org/search?recordType=repository) 
```

pykeen
````

This is one of the main reason why we mention this library as it provides a context to data integration scenarios along with the datasets and tooling to do so.

```python
from pykeen.pipeline import pipeline

result = pipeline(
    model='TransE',
    dataset='nations',
)
```

To see an example of how to use the library, we highly recommended reading the [Medium blog post](https://towardsdatascience.com/knowledge-graph-completion-with-pykeen-and-neo4j-6bca734edf43) and consult the associated jupyter notebook, which shows how to use your own dataset, understand the evaluation, and make novel link predictions:

Link Prediction example notebook: https://github.com (URL_TO_INSERT_RECORD_6987 https://fairsharing.org/FAIRsharing.c55d5e) /tomasonjo/blogs/blob/master/pykeen/Hetionet%20-%20RotatE.ipynb

---
## Conclusion

With this recipe, the intent was to raise awareness in the wealth of Life Science informat (URL_TO_INSERT_TERM_6988 https://fairsharing.org/search?recordType=model_and_format) ion available as Knowledge Graph and ongoing integrative efforts.
We have also tried to convey informat (URL_TO_INSERT_TERM_6989 https://fairsharing.org/search?recordType=model_and_format) ion about the license for each of the resources as well as some insights into the curation protocols specific to each the KG.
Finally, we wanted to highlight how some KGs are already used to train model (URL_TO_INSERT_TERM_6990 https://fairsharing.org/search?recordType=model_and_format) s and make predictions in the context of drug repurposing for instance. 
This is of particular interest in the context of the Innovative Medicine Initiative but not only, obviously.
It certainly should provide food for thoughts when engaging with a FAIR (URL_TO_INSERT_RECORD_6991 https://fairsharing.org/FAIRsharing.WWI10U) ification process and should help define FAIR (URL_TO_INSERT_RECORD_6992 https://fairsharing.org/FAIRsharing.WWI10U) ification goals as well as the process to perform in order to be able to be compatible with specific resources for a specific task at hand.

### What to read next?

* [Knowledge graph completion with PyKEEN and neo4j](https://towardsdatascience.com/knowledge-graph-completion-with-pykeen-and-neo4j-6bca734edf43)
* [Implications of Topological Imbalance for Representation Learning on Biomedical Knowledge Graphs](https://arxiv.org/abs/2112.06567) {footcite}`DBLP:journal (URL_TO_INSERT_TERM_6993 https://fairsharing.org/search?recordType=journal) s/corr/abs-2112-06567`
* [A Review of Biomedical Datasets Relating to Drug Discovery: A Knowledge Graph Perspective](https://doi.org/10.48550/arXiv.2102.10062) {footcite}`https://doi.org/10.48550/arxiv.2102.10062`

````{rdmkit_panel}
````

---

## References

````{dropdown} **Reference**
```{footbibliography}
```
````

## Authors
````{authors_fairplus}
Philippe: Writing - Original Draft
Yojana: Writing - Review & Editing
````

## License

````{license_fairplus}
CC-BY-4.0
````
