
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>15. Extraction, transformation, and loading process</title>

  <link rel="stylesheet" href="../../../_static/css/index.f658d18f9b420779cfdf24aa0a7e2d77.css">


  <link rel="stylesheet"
    href="../../../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">



  <link rel="stylesheet"
    href="../../../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../../../_static/vendor/lato_latin-ext/1.44.1/index.css">


    <link rel="stylesheet" href="../../../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../../../_static/sphinx-book-theme.e7340bb3dbd8dde6db86f25597f54a1b.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/vuetify.min.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/custom.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/animate.min.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/search-wizard.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/search.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/materialdesignicons.min.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />

  <link rel="preload" as="script" href="../../../_static/js/index.d3f166471bb80abb5163.js">

    <script id="documentation_options" data-url_root="../../../" src="../../../_static/documentation_options.js"></script>
    <script src="../../../_static/jquery.js"></script>
    <script src="../../../_static/underscore.js"></script>
    <script src="../../../_static/doctools.js"></script>
    <script src="../../../_static/togglebutton.js"></script>
    <script src="../../../_static/clipboard.min.js"></script>
    <script src="../../../_static/copybutton.js"></script>
    <script src="../../../_static/js/mermaid.core.js"></script>
    <script src="../../../_static/js/vue.min.js"></script>
    <script src="../../../_static/js/searchNodal.js"></script>
    <script src="../../../_static/js/mermaid.min.js"></script>
    <script src="../../../_static/js/mermaid.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../../../_static/sphinx-book-theme.7d483ff0a819d6edff12ce0b1ead3928.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../../../_static/sphinx-thebe.js"></script>
    <link rel="canonical" href="https://fairplus.github.io/the-fair-cookbook/content/recipes/interoperability/ETL-tools.html" />
    <link rel="shortcut icon" href="../../../_static/favicon.ico"/>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" />
    <link rel="next" title="16. Converting VCF file to HL7 FHIR JSON Genomic Report profile" href="VCF-to-FHIR-conversion.html" />
    <link rel="prev" title="14. An inventory of tools for converting data to RDF" href="rdf-conversion.html" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<meta name="docsearch:language" content="en" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/vuetify/2.6.9/vuetify.js"
        integrity="sha512-cgrSlRKhr/ovJePDoGIdcfZ3wiGcc0fT4Q7hAo0x3P1lyLwUmv2NgJtCo2gzFbxqFjKvNnCfVUjZ1XADm0/ZdQ=="
        crossorigin="anonymous" referrerpolicy="no-referrer">
</script>

<!-- Opengraph tags -->
<meta property="og:url"         content="https://fairplus.github.io/the-fair-cookbook/content/recipes/interoperability/ETL-tools.html" />
<meta property="og:type"        content="article" />
<meta property="og:title"       content="Extraction, transformation, and loading process" />
<meta property="og:description" content="An online, open and live resource for the Life Sciences with recipes that help you to make and keep data Findable, Accessible, Interoperable and Reusable; in one word FAIR." />
 <meta property="og:image"       content="/_static/images/homepage.png" /> 
<meta name="twitter:card" content="summary" />
  </head>
<body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">



<v-app id="app" class="pa-0">
    <v-container fluid class="pa-0">
        <v-row class="pa-0" no-gutters>
            <v-container fluid class="pa-0">
                <v-row>
                    <div class="topbar container-xl fixed-top pl-0 primary lighten-1 py-2" style="height:65px !important;">
    <div class="topbar-contents row" style="margin: initial !important;">
        <div class="col topbar-main d-flex align-center">
            
                <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                        data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                        aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                        title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                    <i class="fas fa-bars"></i>
                    <i class="fas fa-arrow-left"></i>
                    <i class="fas fa-arrow-up"></i>
                </button>
            

            
                <a href="../../../index.html" style="width: 300px">
                    <img
                        src="../../../_static/cookbook-logo-small.png"
                        class="logo"
                        alt="The FAIR Cookbook"
                        style="width: 100%"
                    >
                </a>
            

            <v-spacer> </v-spacer>
            <div class="mr-4 extraButtons">
                <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger d-flex justify-center align-center" style="height:100%">

    <v-menu offset-y content-class="elevation-0" open-on-hover>
        <template v-slot:activator="{ on, attrs }">
            <v-btn
              id="dropdown-buttons-trigger"
              color="white"
              outlined
              v-bind="attrs"
              v-on="on"
              flat
              solo
              rounded
              :fab="$vuetify.breakpoint.mdAndDown"
              :small="$vuetify.breakpoint.mdAndDown"
              :class="$vuetify.breakpoint.mdAndDown ? 'pa-0' : ''"
            >
                <v-icon :class="$vuetify.breakpoint.lgAndUp ? 'mr-3' : 'mx-1'" left>fab fa-github</v-icon>
                <span v-if="$vuetify.breakpoint.lgAndUp"> GitHub </span>
            </v-btn>
        </template>
        <div class="homepageMenu d-flex flex-column">
            
            <v-btn
              color="grey darken-1"
              dark
              flat
              solo
              rounded
              href="https://github.com/FAIRplus/the-fair-cookbook"
              class="mt-1"
              target="_blank"
            >
                <v-icon class="mr-3" left>fab fa-github</v-icon>
                repository
            </v-btn>
            

            
            <v-btn
              color="grey darken-1"
              dark
              flat
              solo
              rounded
              href="https://github.com/FAIRplus/the-fair-cookbook/issues/new?title=Issue%20on%20page%20%2Fcontent/recipes/interoperability/ETL-tools.html&body=Your%20issue%20content%20here."
              class="mt-1"
              target="_blank"
            >
                <v-icon class="mr-3" left>fas fa-lightbulb</v-icon>
                open issue
            </v-btn>
            

            
            <v-btn
              color="grey darken-1"
              dark
              flat
              solo
              rounded
              href="https://github.com/FAIRplus/the-fair-cookbook/edit/main/content/recipes/interoperability/ETL-tools.md"
              class="mt-1"
              target="_blank"
            >
                <v-icon class="mx-3" left>fas fa-pencil-alt</v-icon>
                suggest edit
            </v-btn>
            
        </div>
    </v-menu>
</div>

                <!-- Launch buttons -->

            </div>
                
                    <script>
    Vue.component("Searchfield", {
      data () { return {
        query: null
      } },
      methods: {
        search() {
            if (this.query) window.location.href = `../../search-wizard.html?q=` + this.query
            else window.location.href = `../../search-wizard.html`
        }
      },
      template: `
      <div style="width:250px">
            <v-text-field
              v-model="query"
              append-icon="fa-search"
              label="Search Wizard..."
              dense
              rounded
              single-line
              solo
              hide-details
              class="white primary--text"
              @click:append="search"
              @keyup.enter="search"
            />
      </div>
      `
    })
</script>

<Searchfield> </Searchfield>






                
                
                
        </div>
    </div>
</div>
                </v-row>
                
                    <v-row class="d-flex">
                    
                        <div
                            class=" bd-sidebar site-navigation show"
                            id="site-navigation"
                            style="width:350px"
                        >
                            
                                <div class="nav-drawer d-flex flex-column grey lighten-3" style="height:100%;padding-top: 40px;">

    <div class="testTOC">
        <script>
    Vue.component("Navigation", {
        data() { return { recipes: {} }},
        methods: {
            async getData(){
                const recipes = await fetch('/_static/recipes.json')
                this.recipes = await recipes.json();
                this.parseDom()
            },
            parseDom() {
                const links = Array.prototype.slice.call(document.getElementsByClassName("reference internal"))
                const icon = '<i class="fas fa-play-circle grey lighten-3 secondary--text ml-2"></i>'
                links.forEach(link => {
                    const recipeSplit = link.innerText.split(". ")
                    const recipeName = recipeSplit[recipeSplit.length - 1].trim()
                    recipeSplit.pop()
                    recipeSplit.forEach(pref => { pref = pref.trim() })
                    const prefix = recipeSplit.length > 0 ? recipeSplit.join(".") + "." : null
                    const content = prefix ? `${prefix} ${recipeName}` : recipeName
                    if (Object.keys(this.recipes).includes(recipeName)) {
                        if (this.recipes[recipeName].executable === 'Yes') link.innerHTML = content + icon
                    }
                })
            }
        },
        mounted() { this.getData() }
    })
</script>


<Navigation> </Navigation>

    </div>

    <div class="nav-tree grey px-5 lighten-3 pb-5" style="flex-grow: 1; border-bottom: 1px solid #ccc !important">
        
        
            <nav class="bd-links pt-10" id="bd-docs-nav" aria-label="Main navigation">
    <p class="caption collapsible-parent">
 <span class="caption-text">
  FOREWORD
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/FAIR-cookbook-audience.html">
   Introduction
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/brief-FAIR-principles.html">
   Introducing the FAIR Principles
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/FAIRplus-values.html">
   Reflecting on the ethical values of FAIR
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/fairification-process.html">
   Introducing our FAIRification framework
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/priorization.html">
   Prioritizing projects for FAIRification
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/metadata-fair.html">
   Framing FAIR and the notion of metadata
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/FAIR-and-knowledge-graphs.html">
   Understanding the relation between FAIR and Knowledge Graphs
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/training-datasets.html">
   Training for FAIRification with open or synthetic biomedical datasets
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/public-kg-resource-integration.html">
   Raising Awareness in Public Knowledge Graphs for Life Sciences
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/fair-cro.html">
   Reflecting on Practical Considerations for CROs to play FAIR
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/dpia.html">
   Data Protection Impact Assessment and Data Privacy
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/glossary-fair-terms.html">
   Glossary
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Recipes at a Glance
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../../search-wizard.html">
   All Recipes In a Table
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  FAIR RECIPES
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../findability.html">
   Findability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/identifiers.html">
     1. Introducing unique, persistent identifiers
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/chemical-identities.html">
     2. Creating InChI &amp; SMILES identifiers for chemical structures
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/iupac-names.html">
     3. Creating InChIKeys for IUPAC names
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/identifiers-minids.html">
     4. Minting identifiers with Globus Minid client
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/zenodo-deposition.html">
     5. Depositing to generic repositories - Zenodo use case
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/registeringDatasets.html">
     6. Registering datasets with Wikidata
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/checksum-create.html">
     7. Creating file checksums
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/checksum-validate.html">
     8. Validating checksums to verify file integrity
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../findability/seo.html">
     9. Introducing Search Engine Optimization (SEO)
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../findability/seo/bioschemas-data-page.html">
       9.9.1. Marking up Data pages with Schema.org &amp; Bioschemas for SEO
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../findability/seo/bioschemas-dataset.html">
       9.9.2. Marking up Dataset page with Schema.org &amp; Bioschemas for SEO
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../findability/seo/bioschemas-datacatalog.html">
       9.9.3. Marking up Data Catalogue page with Schema.org &amp; Bioschemas for SEO
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../accessibility.html">
   Accessibility
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../accessibility/sftp.html">
     1. Transferring data with SFTP protocol
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../accessibility/aspera.html">
     2. Downloading data with Aspera protocol
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="../interoperability.html">
   Interoperability
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="registering-swisslipids-identifiers.html">
     1. Registering SwissLipids identifiers in Wikidata
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="identifier-mapping.html">
     2. Interlinking data from different sources
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="bridgedb-recipe.html">
     3. Mapping identifiers with BridgeDb
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="bridgedb/bridgedb-example.html">
       3.11.1. Mapping identifiers using BridgeDb web services
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="introduction-terminologies-ontologies.html">
     4. Introducing terminologies and ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="selecting-ontologies.html">
     5. Selecting terminologies and ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ontology-new-term-request-recipe.html">
     6. Requesting new terms from terminologies and ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ontology-operations-tools.html">
     7. Introducing ontology-related tools and services
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="ontology-robot-recipe.html">
     8. Building an application ontology with ROBOT
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="ontology-robot-recipe/MSIO-robot-build-process.html">
       8.11.1. Building an application ontology for metabolomics - MSIO
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="ontology-robot-recipe/competency-questions.html">
       8.11.2. Defining competency questions for the Ontology ROBOT use case
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ontology-alignment-oxo.html">
     9. Mapping Ontologies with OxO, EBI Ontology Xref Service
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="creating-data-dictionary.html">
     10. Creating a data/variable dictionary
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="creating-minimal-metadata-profiles.html">
     11. Creating a metadata profile
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="transcriptomics-metadata.html">
       11.5.1. Outlining a metadata profile for transcriptomics
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="covid19-sample-metadata-profile-shex-use-case.html">
       11.5.2. Building a community compliant metadata profile - The Covid19 sample profile use case
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="bioactivity-profile.html">
       11.5.3. Outlining a metadata profile for Bioactivity data
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="c4c-clinical-trials.html">
       11.5.4. Creating a metadata profile for clinical trial protocols
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="from-proprietary-to-open-standard-mzml-exemplar.html">
     12. Converting from proprietary to open format
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="fastq-file-format-validators.html">
     13. Validating file format - FASTQ example
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="rdf-conversion.html">
     14. Inventorying tools for converting data to RDF
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     15. Surveying extraction, transformation, load (ETL) tools
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="VCF-to-FHIR-conversion.html">
     16. Expressing Clinical Genetic Information as FHIR JSON
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="vcf2fhir/FAIRcookbook-vcf2fhir.html">
       16.9.1. Converting VCF file to FHIR JSON
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="creating-knowledge-graph-from-unstructured-text.html">
     17. Creating knowledge graphs from unstructured text
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="nlp2kg/creating-knowledge-graph-from-text.html">
       17.12.1. unstructured text to graph as executable notebook
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../reusability.html">
   Reusability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/ATI-licensing.html">
     1. Licensing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/ATI_licensing_software.html">
     2. Licensing Software
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/ATI_licensing_data.html">
     3. Licensing Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/expressing-data-use.html">
     4. Declaring data permitted uses
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/provenance.html">
     5. Introducing Provenance Information
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../infrastructure.html">
   Infrastructure
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/id-resolution.html">
     1. Introducing identifier resolution services
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/gupri.html">
     2. Creating resolvable identifiers
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/data-catalog.html">
     3. Building a catalogue of datasets
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/dats-model.html">
     4. Introducing the DATS model
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/data-catalog-deployment.html">
     5. Deploying a data catalogue - The IMI data catalogue example
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../infrastructure/vocabularies.html">
     6. Vocabulary management
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../infrastructure/vocabularies/technical-and-architectural-selection-criteria-of-ontology-lookup-services.html">
       6.1. Introducing vocabulary portals and lookup services
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../infrastructure/vocabularies/Selecting_and_using_ontology_lookup_services.html">
       6.2. Selecting an ontology lookup service
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../infrastructure/vocabularies/UC3_R13_local_ontology_services.html">
       6.3. Deploying EBI Ontology Loopkup Service
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/openrefine-nvt.html">
     7. Using OpenRefine and Karma for FAIRification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/fair-api.html">
     8. Developing FAIR API for the Web
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../assessing-fairness.html">
   Assessment
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../assessing-fairness/fair-assessment-fairevaluator.html">
     1. Assessing with FAIR Evaluator
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../assessing-fairness/fair-assessment-fairshake.html">
     2. Assessing with FAIRshake
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  FAIR Maturity
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../maturity.html">
   Changing culture with the Dataset Maturity Model
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/miappe.html">
     Improving dataset maturity - MIAPPE-compliant submission to EMBL-EBI databases
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/plant-pheno-data-publication.html">
     Publishing plant phenotypic data
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../maturity/isa-maturity-progression.html">
     Moving through maturity levels with ISA
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../maturity/create-a-simple-ISA-descriptor.html">
       Minimal Data Maturity with ISA - ISA-Tab and free text
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../maturity/isa-api-programmatic-rebuild-of-BII-S-3.html">
       Enhanced Data Maturity with ISA - ISA-JSON and ontology markup
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../maturity/isa-json-conversion-to-rdf-linked-data.html">
       Moving to a semantically typed version - ISA-JSON-LD
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../maturity/isa-as-RO.html">
       Dissemination - Packaging ISA as a Research Object (RO)
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../applied-examples/fair-data-matrix-recipe.html">
     Making an omics data matrix FAIR
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/0-rose-metabolites-Python-data-handling.html">
       FAIRifying Data Matrices - Step1 - Starting material
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/1-rose-metabolites-Python-analysis.html">
       FAIRifying Data Matrices - Step2 - Structuring data
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/2-rose-metabolites-Python-RDF-querying-analysis.html">
       FAIRifying Data Matrices - Step3 - Exploring data with SPARQL
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/3-rose-metabolites-R-analysis.html">
       FAIRifying Data Matrices - Step4 - Integrating data
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/fair-workflows.html">
     Making Computational Workflows FAIR
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  FAIRified Datasets
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../applied-examples.html">
   Applied examples
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/etox.html">
     1. IMI eTox - toxicity datasets
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/nd4bb.html">
     2. IMI nd4bb - chemical activities datasets
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/oncotrack_raw/rawRecipe-oncotrack.html">
     3. Readying IMI Oncotrack - clinical cohort datasets for deposition to EBI Biosamples
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/resolute_raw/resolute.html">
     4. Depositing IMI ReSOLUTE transcriptomics datasets to EBI repositories
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/ehden-ohdsi.html">
     5. Enhancing discoverability of EHDEN OHDSI data with Schema.org markup
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/eubopen-hcs-bioimage.html">
     6. Depositing IMI EUBOPEN High-Content Screening data to EBI BioImage Archive
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/BIA_submission.html">
     7. Depositing epifluorescence and confocal microscopy data to EBI BioImage Archive
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/approach-cdisc.html">
     8. Mapping IMI APPROACH datasets to CDISC-SDTM standard
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  AFTERWORD
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../afterword/fair-reflections.html">
   Lessons learned from the FAIR journey and project outlook
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  GLOSSARY
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/glossary-fair-terms.html">
   Glossary of terms and abbreviations
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  JOIN US
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../community.html">
   Community
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../afterword/people.html">
     Boards and contributors
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../afterword/code_of_conduct.html">
     Code of conduct
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../afterword.html">
     Platform
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../afterword/the-turing-way.html">
       Leveraging the Turing Way Book
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../help.html">
   Contribute
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../afterword/contributing.html">
     1. How to contribute
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/how-to-create-recipe-with-gdoc.html">
     2. Add via GoogleDoc
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/how-to-create-recipe-with-hackmd.html">
     3. Add via HackMD
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/how-to-create-recipe-with-git.html">
     4. Add via Git
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/recipe-template.html">
     5. Git recipe template
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/tips-tricks.html">
     6. Tips and tricks
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/myst.html">
     7. Markdown cheatsheet
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>

</nav>
        
        
             <!-- To handle the deprecated key -->

        
    </div>

    <div class="text-center px-5 lighten-3 grey py-2">
        Powered by <a href="https://jupyterbook.org" class="secondary--text">Jupyter Book</a>
    </div>
</div>
                        </div>
                        
                            <main class="bd-content" role="main" style="width:1%; flex-grow:1;">
                                <v-container fluid id="main-content" class="pa-0">
                                    <v-row class="d-flex">
                                        
                                            <v-col class="col-xl-10 col-lg-10 col-md-9 col-12 px-10 pt-10 mt-3">
                                                
  <div class="section" id="extraction-transformation-and-loading-process">
<span id="fcb-interop-etl"></span><h1><span class="section-number">15. </span>Extraction, transformation, and loading process<a class="headerlink" href="#extraction-transformation-and-loading-process" title="Permalink to this headline">¶</a></h1>
<script type="application/ld+json"> {"@context": {"bs": "https://bioschema.org/", "dc": "http://purl.org/dc/terms/", "owl": "http://www.w3.org/2002/07/owl#", "sdo": "https://schema.org/"}, "@type": ["sdo:HowTo", "sdo:LearningResource"], "@id": "https://w3id.org/faircookbook/FCB031", "sdo:name": "Surveying extraction, transformation, load (ETL) tools", "sdo:description": "Surveying extraction, transformation, load (ETL) tools", "sdo:keywords": [], "dc:conformsTo": {"@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE", "@type": "CreativeWork"}, "sdo:inLanguage": ["en-GB"], "sdo:audience": ["Data Curator|T4FS_0000389", "Data Manager|T4FS_0000520", "Data Scientist|T4FS_0000566"], "sdo:isPartOf": [{"@type": "CreativeWork", "@id": "https://fairplus.github.io/the-fair-cookbook/", "sdo:url": "https://fairplus.github.io/the-fair-cookbook/", "sdo:name": "The FAIR Cookbook"}], "sdo:learningResourceType": "Survey / Review", "sdo:identifier": {"@type": "PropertyValue", "@id": "", "sdo:name": "FCB031", "dso:url": "https://w3id.org/faircookbook/FCB031", "owl:sameAs": "https://identifiers.org/fcb:FCB031"}, "sdo:license": {"@type": "CreativeWork", "@id": "https://creativecommons.org/licenses/by/4.0/", "sdo:url": "https://creativecommons.org/licenses/by/4.0/", "sdo:name": "CC-BY-4.0"}, "sdo:timeRequired": "PT0H15M0S", "sdo:totalTime": "PT0H15M0S", "sdo:step": [{"@type": "sdo:HowToSection", "sdo:position": "1", "sdo:name": "Overview", "sdo:url": "https://w3id.org/faircookbook/FCB031#main-objectives", "sdo:itemListElement": [{"@type": "sdo:HowToStep", "sdo:position": "1", "sdo:name": "", "sdo:itemListElement": [{"@type": "sdo:HowToDirection", "sdo:position": "1", "sdo:name": "step-by-step section", "sdo:text": "Follow the step-by-step section.", "sdo:url": "https://w3id.org/faircookbook/FCB031#main-objectives"}, {"@type": "sdo:HowToTip", "sdo:position": "2", "sdo:name": "Reporting an issue", "sdo:text": "Use our GitHub issue tracker to report any error. Thanks!", "sdo:url": "https://github.com/FAIRplus/the-fair-cookbook/issues/new/choose"}]}]}, {"@type": "sdo:HowToSection", "sdo:position": "2", "sdo:name": "Conclusion", "sdo:url": "https://w3id.org/faircookbook/FCB031#conclusion", "sdo:itemListElement": [{"@type": "sdo:HowToStep", "sdo:position": "1", "sdo:name": "", "sdo:itemListElement": [{"@type": "sdo:HowToDirection", "sdo:position": "1", "sdo:name": "conclusion section", "sdo:text": "What to read next?", "sdo:url": "https://w3id.org/faircookbook/FCB031#conclusion"}, {"@type": "sdo:HowToTip", "sdo:position": "2", "sdo:name": "Reporting an issue", "sdo:text": "Use our GitHub issue tracker to report any error. Thanks!", "sdo:url": "https://github.com/FAIRplus/the-fair-cookbook/issues/new/choose"}]}]}]} </script>
<div class="card my-10 recipe d-flex">
<div class="card-header purple-dark text--white">
<div class="overview">
<div class="overviewLabel">Recipe Overview</div>
<div class="overviewContent">
<div class="item"><div class="itemContainer">
<i class="itemIcon fa fa-clock fa-2x text--white"></i>
<div class="itemContent">
<div class="label"> Reading Time </div>
15 minutes
</div>
</div></div>
<div class="item"><div class="itemContainer">
<i class="itemIcon fa fa-play-circle fa-2x text--white"></i>
<div class="itemContent">
<div class="label"> Executable Code </div>
No
</div>
</div></div>
<div class="item" style="margin-bottom:10px"><div class="itemContainer">
<i class="itemIcon fa fa-fire fa-2x text--white" style="margin-left:4px;margin-right: 5px;"></i>
<div class="itemContent">
<div class="label" style="position:relative; left:-6px">Difficulty</div>
<i class="fa fa-fire"></i>
<i class="fa fa-fire"></i>
<i class="fa fa-fire"></i>
<i class="fa fa-fire"></i>
<i class="fa fa-fire"></i>
</div>
</div></div>
</div></div>
</div>
<div class="card-body text--purple-dark">
<div class="title">
<div style="flex-grow:1; margin-right:5px">Surveying extraction, transformation, load (ETL) tools</div>
<img src="https://fairplus.github.io/the-fair-cookbook/_static/images/fairplus-mini.png" alt="FAIRPlus logo"> </img>
</div>
<div class="section">
<i class="sectionIcon fa fa-laptop fa-2x"></i><div class="sectionContent">
<div class="label">Recipe Type</div>
<div class="sectionValue">Survey / Review</div>
</div>
</div>
<div class="section" style="flex-grow: 1;">
<i class="sectionIcon fa fa-users fa-2x"></i><div class="sectionContent">
<div class="label">Audience</div>
<div class="sectionValue"><a href="http://purl.obolibrary.org/obo/T4FS_0000389">Data Curator</a>, <a href="http://purl.obolibrary.org/obo/T4FS_0000520">Data Manager</a>, <a href="http://purl.obolibrary.org/obo/T4FS_0000566">Data Scientist</a></div>
</div>
</div>
<div class="section" style="flex-grow: 1;">
<svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="signal" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 600" class="bars not applicable">
<path class="bar5" d="M216 288h-48c-8.84 0-16 7.16-16 16v192c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V304c0-8.84-7.16-16-16-16zM88 384H40c-8.84 0-16 7.16-16 16v96c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16v-96c0-8.84-7.16-16-16-16zm256-192h-48c-8.84 0-16 7.16-16 16v288c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V208c0-8.84-7.16-16-16-16zm128-96h-48c-8.84 0-16 7.16-16 16v384c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V112c0-8.84-7.16-16-16-16zM600 0h-48c-8.84 0-16 7.16-16 16v480c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V16c0-8.84-7.16-16-16-16z"/>
<path class="bar4" d="M216 288h-48c-8.84 0-16 7.16-16 16v192c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V304c0-8.84-7.16-16-16-16zM88 384H40c-8.84 0-16 7.16-16 16v96c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16v-96c0-8.84-7.16-16-16-16zm256-192h-48c-8.84 0-16 7.16-16 16v288c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V208c0-8.84-7.16-16-16-16zm128-96h-48c-8.84 0-16 7.16-16 16v384c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V112c0-8.84-7.16-16-16-16zM600" />
<path class="bar3" d="M216 288h-48c-8.84 0-16 7.16-16 16v192c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V304c0-8.84-7.16-16-16-16zM88 384H40c-8.84 0-16 7.16-16 16v96c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16v-96c0-8.84-7.16-16-16-16zm256-192h-48c-8.84 0-16 7.16-16 16v288c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V208c0-8.84-7.16-16-16-16zm128-96h-48c-8.84" />
<path class="bar1" d="M216 288h-48c-8.84 0-16 7.16-16 16v192c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V304c0-8.84-7.16-16-16-16zM88 384H40c-8.84 0-16 7.16-16 16v96c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16v-96c0-8.84-7.16-16-16-16zm256-192h-48c-8.84"/>
<path class="bar2" d="M216 288h-48c-8.84 0-16 7.16-16 16v192c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V304c0-8.84-7.16-16-16-16zM88 384H40c-8.84"/>
</svg>
<div class="sectionContent">
<div class="label">Maturity Level & Indicator</div>
<div class="sectionValue"><v-chip-group><v-chip href="https://w3id.org/Data-Maturity/not applicable" class="secondary white--text" target="_blank" label>not applicable</v-chip></v-chip-group></div>
<div class="tooltip">hover me
<span class="tooltiptext">Tooltip text</span></div>
</div>
</div><div class="card-footer text--orange sphinx-bs.badge.badge-primary"> Cite me with 
<a href="https://w3id.org/faircookbook/FCB031" class="text--purple-dark">FCB031
</a></div>
</div>
</div>
<div class="section" id="main-objectives">
<h2><span class="section-number">15.1. </span>Main Objectives<a class="headerlink" href="#main-objectives" title="Permalink to this headline">¶</a></h2>
<p>This recipe identifies tools for <code class="docutils literal notranslate"><span class="pre">data</span> <span class="pre">extraction</span></code>, <code class="docutils literal notranslate"><span class="pre">transformation</span></code>, and <code class="docutils literal notranslate"><span class="pre">loading</span></code> (ETL).
ETL is the process of collecting data from one source to persist it in another designated system in which the data is represented differently <a class="footnote-reference brackets" href="#etl-wiki" id="id1">1</a>.
A common use case in data science is to build a scalable and portable ETL system to extract data from various sources, transform said-data into a cohesive dataset, and load the data to an internal or public database to support data exploration.</p>
<p>This recipe aims to serve as a start point for designing ETL workflows, rather than provide a comprehensive list covering all available tools.</p>
<blockquote>
<div><p>The lists of tools are generated either by manual curation, which reflects what is being used in the industry, or automatally discovered from the bio.tools repository.</p>
</div></blockquote>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>The content in these tables was generated in March 2021.
For updated content, please check the <a class="reference external" href="https://github.com/FAIRplus/WP3_FAIR_tooling">FAIR tooling repository</a>.
To provide feedback on this content or report issues, please do so via the <a class="reference external" href="https://github.com/FAIRplus/the-fair-cookbook/issues">FAIR Cookbook github issue tracker</a></p>
</div>
</div>
<div class="section" id="graphical-overview">
<h2><span class="section-number">15.2. </span>Graphical Overview<a class="headerlink" href="#graphical-overview" title="Permalink to this headline">¶</a></h2>
<p>The figure below shows different ETL-related operations and their relationships, together with related tools and recipes.</p>
<details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="overview-of-key-aspects-in-etl-process">
<a class="reference internal image-reference" href="../../../_images/ETL-tools-mermaid.png"><img alt="Overview of key aspects in ETL process" src="../../../_images/ETL-tools-mermaid.png" style="width: 800px;" /></a>
<p class="caption"><span class="caption-number">Fig. 15.1 </span><span class="caption-text">Overview of key aspects in ETL process</span><a class="headerlink" href="#overview-of-key-aspects-in-etl-process" title="Permalink to this image">¶</a></p>
</div>
</div>
</details><p>The table below is an overview of ETL tools identified.
Details of each tools are also provided in this doc.
For ETL tools for RDF data model, please check the dedicated recipe<!-- {ref}`fcb-interop-etl2rdf-tools` -->.</p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p></p></th>
<th class="head"><p></p></th>
<th class="head"><p></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Data query and retrieval</p></td>
<td><p>Data transformation</p></td>
<td><p>Data deposition</p></td>
</tr>
<tr class="row-odd"><td><p>TAMR</p></td>
<td><p>SDTM</p></td>
<td><p>REDCap</p></td>
</tr>
<tr class="row-even"><td><p>Bert</p></td>
<td><p>TriFacta</p></td>
<td><p>TransMART</p></td>
</tr>
<tr class="row-odd"><td><p>Termite</p></td>
<td><p>OMOP</p></td>
<td><p>Stardog</p></td>
</tr>
<tr class="row-even"><td><p>Oracle clinical</p></td>
<td><p>Collibra</p></td>
<td><p>Postgresql</p></td>
</tr>
<tr class="row-odd"><td><p>GeoBoost2</p></td>
<td><p>Apache Hadoop</p></td>
<td><p>Chemotion</p></td>
</tr>
<tr class="row-even"><td><p>MartView</p></td>
<td><p>Talend</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>CLOBNET</p></td>
<td><p>Informatica</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p>PDBUSQLExtractor</p></td>
<td><p>OpenRefine-metadata-extension</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>repo</p></td>
<td><p>DISQOVER</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p>SNPator</p></td>
<td><p>Query Tabular</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>PyGMQL</p></td>
<td><p>ms-data-core-api</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p>PlantPIs</p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>BAGET</p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p>NCBI Resources</p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>CPAD</p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
</tbody>
</table>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>The tools list above aims to provide a basic overview of what is available on the market. It is <em>not</em> a formal recommendation. If you think key tools are missing or the list needs an update, please contact us via the <a class="reference external" href="https://github.com/FAIRplus/the-fair-cookbook/issues">github issue tracker</a>.</p>
</div>
</div>
<div class="section" id="requirements">
<h2><span class="section-number">15.3. </span>Requirements<a class="headerlink" href="#requirements" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><p>Knowledge requirement:</p>
<ul>
<li><p>General familiarity with data cleaning, data loading tasks.</p></li>
</ul>
</li>
</ul>
</div>
<hr class="docutils" />
<div class="section" id="fairification-objectives-inputs-and-outputs">
<h2><span class="section-number">15.4. </span>FAIRification Objectives, Inputs and Outputs<a class="headerlink" href="#fairification-objectives-inputs-and-outputs" title="Permalink to this headline">¶</a></h2>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="text-align:left head"><p>Actions.Objectives.Tasks</p></th>
<th class="text-align:left head"><p>Input</p></th>
<th class="text-align:left head"><p>Output</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-align:left"><p><a class="reference external" href="http://edamontology.org/operation_2422">data retrieval</a></p></td>
<td class="text-align:left"><p></p></td>
<td class="text-align:left"><p></p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p><a class="reference external" href="http://edamontology.org/operation_2428">validation</a></p></td>
<td class="text-align:left"><p></p></td>
<td class="text-align:left"><p></p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p><a class="reference external" href="http://edamontology.org/operation_3431">data deposition</a></p></td>
<td class="text-align:left"><p></p></td>
<td class="text-align:left"><p></p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p><a class="reference external" href="http://edamontology.org/topic_0219">data submission, annotation and curation</a></p></td>
<td class="text-align:left"><p></p></td>
<td class="text-align:left"><p></p></td>
</tr>
</tbody>
</table>
</div>
<hr class="docutils" />
<div class="section" id="operations">
<h2><span class="section-number">15.5. </span>Operations<a class="headerlink" href="#operations" title="Permalink to this headline">¶</a></h2>
<!-- <h3 id=Data-query-and-retrieval>Data query and retrieval</h3> -->
<p><code class="docutils literal notranslate"><span class="pre">Data</span> <span class="pre">query</span> <span class="pre">and</span> <span class="pre">retrieval</span></code> is the process of extracting structured or unstructured data from different sources. The process of identifying and obtaining data from data management systems.</p>
<p><strong>Manually curated list of tools</strong></p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Tool</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>License</p></th>
<th class="head"><p>Topics</p></th>
<th class="head"><p>Resource type</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><a class="reference external" href="https://www.tamr.com/">TAMR</a></p></td>
<td><p>A cloud-native data mastering solution (cloud MDM) accelerate analytics through machine learning (ML), available on Google Cloud, Azure and AWS.</p></td>
<td><p>Commercial license</p></td>
<td><p>information integration, data unification</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://github.com/google-research/bert">Bert</a></p></td>
<td><p>Bidirectional Encoder Representations from Transformers (BERT) is a Transformer-based machine learning technique for natural language processing (NLP) pre-training developed by Google, aiming to extractcontext-sensitive features from an input text.</p></td>
<td><p>Apache License 2.0</p></td>
<td><p>relationship extraction</p></td>
<td><p>Data model</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://www.scibite.com/platform/termite/">Termite</a></p></td>
<td><p>TERMite (TERM identification, tagging &amp; extraction) is a fast named entity recognition (NER) and extraction engine for semantic analytics.</p></td>
<td><p>Commercial license</p></td>
<td><p>information extraction</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://www.oracle.com/uk/industries/life-sciences/clinical-research/">Oracle clinical</a></p></td>
<td><p>a single application and infrastructure for electronic data capture and clinical data management, while leveraging the renowned Oracle database. Oracle Clinical enables management of all clinical trial data in a single system, improving accuracy, visibility, and data integrity.</p></td>
<td><p>Commercial license</p></td>
<td><p>Clinical data</p></td>
<td><p></p></td>
</tr>
</tbody>
</table>
<p><strong>Automatically created list of tools by querying <a class="reference external" href="https://bio.tools">Bio.Tools</a>.</strong></p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Tool</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>License</p></th>
<th class="head"><p>Topics</p></th>
<th class="head"><p>Resource type</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><a class="reference external" href="https://bio.tools/geoboost2">GeoBoost2</a></p></td>
<td><p>A natural languageprocessing pipeline for GenBank metadata enrichment for virus phylogeography.</p></td>
<td><p>Apache-2.0 License</p></td>
<td><p>Biotechnology, Natural language processing, Workflows, Public health and epidemiology, Infectiousdisease</p></td>
<td><p>Web application</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://bio.tools/martview">MartView</a></p></td>
<td><p>Tool for data retrieval and data mining that integrates data from Ensembl. Through the web interface it allows you to apply a series of filters to create custom datasets which can be converted to several useful output formats.</p></td>
<td><p>N/A</p></td>
<td><p>Pathology, Data integration and warehousing, Database management, Molecular interactions, pathways and networks, Gene transcripts</p></td>
<td><p>Web Application</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://bio.tools/CLOBNET">CLOBNET</a></p></td>
<td><p>Cloud-based machine learning system (CLOBNET) that is an open-source, lean infrastructure for electronic health record (EHR) data integration and is capable of extract, transform, and load (ETL) processing</p></td>
<td><p>BSD-2-Clause</p></td>
<td><p>Machine learning, Medicines research and development</p></td>
<td><p>Web application (for internal use)</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://bio.tools/PDBUSQLExtractor">PDBUSQLExtractor</a></p></td>
<td><p>Scalable Extraction of Big Macromolecular Data in Azure Data Lake Environment.</p></td>
<td><p>GPL-3.0</p></td>
<td><p>Nucleic acid structure analysis</p></td>
<td><p>C# library</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://bio.tools/repo">repo</a></p></td>
<td><p>A data manager meant to avoid manual storage/retrieval of data to/from the file system. It builds one (or more) centralized repository where R objects are stored with rich annotations, including corresponding code chunks, and easily searched and retrieved.</p></td>
<td><p>GPL-3.0</p></td>
<td><p>Data submission, annotation and curation, Data management, Database management</p></td>
<td><p>R library</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://bio.tools/snpator">SNPator</a></p></td>
<td><p>Originally designed to help CeGen users to handle, retrieve, transform and analyze the genetic data generated by the genotyping facilities of the institution. However, it is also open to external users who may want to upload their own genotyped data in order to take advantage its data processing features. Users will be able to perform a set of operations which may range from very simple format transformations to some complex biostatistical calculations.</p></td>
<td><p>N/A</p></td>
<td><p>Bioinformatics, Data quality management, Genetic variation, GWAS study</p></td>
<td><p>Web Application</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://bio.tools/PyGMQL">PyGMQL</a></p></td>
<td><p>Scalable data extraction and analysis for heterogeneous genomic datasets. Based on GMQL.</p></td>
<td><p>Apache-2.0</p></td>
<td><p>Imaging, Workflows, Database management</p></td>
<td><p>Python Library</p></td>
</tr>
</tbody>
</table>
<p><strong>Data portals with extraction functionalities discovered from <a class="reference external" href="https://bio.tools">Bio.Tools</a>.</strong></p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Tool</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>License</p></th>
<th class="head"><p>Topics</p></th>
<th class="head"><p>Resource type</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><a class="reference external" href="https://bio.tools/plantpis">PlantPIs</a></p></td>
<td><p>Web querying system for a database collecting plant protease inhibitors data.</p></td>
<td><p>Plant biology, Gene and protein families</p></td>
<td><p>Database Portal</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://bio.tools/baget">BAGET</a></p></td>
<td><p>Web service designed to facilitate extraction of specific gene and protein sequences from completely determined prokaryotic genomes. Query results can be exported as a rich text format file for printing, archival or further analysis.</p></td>
<td><p>N/A</p></td>
<td><p>DNA, Sequencing, Model organisms, Sequence analysis</p></td>
<td><p>Web application</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://bio.tools/ncbi_resources">NCBI Resources</a></p></td>
<td><p>Analysis and retrieval resources for the data in GenBank and other biological data made available through the NCBI web site.</p></td>
<td><p>Open Access</p></td>
<td><p>Data mining, Data integration and warehousing, Database management, Molecular interactions, pathways and networks, Rare diseases</p></td>
<td><p>Web Application, Database portal</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://bio.tools/cpad">CPAD</a></p></td>
<td><p>Curated Protein Aggregation Database (CPAD) is an integrated database which contains information related to amyloidogenic proteins, Aggregation prone regions, aggregation kinetics and structure of proteins relevant to aggregation.</p></td>
<td><p>N/A</p></td>
<td><p>Proteomics, Data submission, annotation and curation, Proteins, Small molecules, Literature and language</p></td>
<td><p>Data Portal</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://bio.tools/GenomicsDB">GenomicsDB</a></p></td>
<td><p>Advancing clinical cohort selection with genomics analysis on a distributed platform. Highly performant data storage in C++ for importing, querying and transforming variant data with Java/Spark. Used in gatk4.</p></td>
<td><p>MIT License</p></td>
<td><p>Exome sequencing, Systems medicine, Personalised medicine, Complementary medicine, Biobank</p></td>
<td><p>Command-line</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://bio.tools/epivizFileServer">Epiviz File Server</a></p></td>
<td><p>A library to query and transform genomic data from indexed files</p></td>
<td><p>MIT License</p></td>
<td><p>Oncology,Workflows, DNA</p></td>
<td><p>Python Library</p></td>
</tr>
</tbody>
</table>
<!-- <h3 id=Data-transformation>Data transformation</h3> -->
<div class="section" id="data-transformation">
<h3><span class="section-number">15.5.1. </span>Data transformation<a class="headerlink" href="#data-transformation" title="Permalink to this headline">¶</a></h3>
<p><code class="docutils literal notranslate"><span class="pre">Data</span> <span class="pre">transformation</span></code> is the process of converting the data format, structure, and value for a specific goal, which may include meeting a specific data standards.</p>
<p>Below are listed common operations in data transformation:</p>
<ul class="simple">
<li><p>Data annotation
The process of labeling data or metadata.</p></li>
<li><p>Data validation
The process of ensuring data has met data standards both in terms of data content and data format.</p></li>
<li><p>Data aggregation
The process of gathering data and presenting it, usually, in a summarized format.</p></li>
</ul>
<p><strong>Manually curated list of tools</strong></p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Tool</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>License</p></th>
<th class="head"><p>Topics</p></th>
<th class="head"><p>Resource type</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><a class="reference external" href="https://www.cdisc.org/standards/foundational/sdtm">SDTM</a></p></td>
<td><p>SDTM provides a standard for organizing and formatting data to streamline processes in collection management analysis and reporting.</p></td>
<td><p>N/A</p></td>
<td><p>data aggregation, data warehousing, Standard for Exchange of Non-clinical Data</p></td>
<td><p>Data standard</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://www.trifacta.com">TriFacta</a></p></td>
<td><p>A data preparation tool for data quality improvement, data transformation, and building data pipelines.</p></td>
<td><p>Commercial license</p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://www.ohdsi.org/data-standardization/the-common-data-model/">OMOP</a></p></td>
<td><p>The OMOP Common Data Model allows for the systematic analysis of disparate observational databases.</p></td>
<td><p>N/A</p></td>
<td><p>Data transformation</p></td>
<td><p>Data model</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://www.collibra.com/">Collibra</a></p></td>
<td><p>an enterprise-oriented data governance platform that provides tools for data management and stewardship.</p></td>
<td><p>Commercial license</p></td>
<td><p>Data catalogue,data governance, data linkage</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://hadoop.apache.org/">Apache Hadoop</a></p></td>
<td><p>A framework that allows for the distributed processing of large data sets across clusters of computers using simple programming models. It is designed to scale up from single servers to thousands of machines</p></td>
<td><p><a class="reference external" href="http://www.apache.org/licenses/LICENSE-2.0">Apache License 2.0</a></p></td>
<td><p>Distributed computing</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://www.talend.com/products/integrate-data/">Talend</a></p></td>
<td><p>A unified approach that combines rapid data integration, transformation, and mapping with automated quality checks to ensure trustworthy data in every step.</p></td>
<td><p>Commercial license</p></td>
<td><p>Data integration</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://www.informatica.com/">Informatica</a></p></td>
<td><p>Connect &amp; fetch data from different heterogeneous source and processing of data.</p></td>
<td><p>Commercial license</p></td>
<td><p>Data integration</p></td>
<td><p></p></td>
</tr>
</tbody>
</table>
<p><strong>Automatically created list of tools by querying <a class="reference external" href="https://bio.tools">Bio.Tools</a>.</strong></p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Tool</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>License</p></th>
<th class="head"><p>Topics</p></th>
<th class="head"><p>Resource type</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><a class="reference external" href="https://github.com/FAIRDataTeam/OpenRefine-metadata-extension">OpenRefine-metadata-extension</a></p></td>
<td><p>Enables a post-hoc FAIRification workflow: load an existing dataset, perform data wrangling tasks, add FAIR attributes to the data, generate a linked data version of the data and, finally, push the result to an online FAIR data infrastructure to make it accessible and discoverable.</p></td>
<td><p>MIT License</p></td>
<td><p>Data submission, Annotation and curation, Data identity and mapping</p></td>
<td><p>Extension</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://bio.tools/discover">DISQOVER</a></p></td>
<td><p>Data integration platform for public, licensed and internal data. The Data Ingestion Engine enables transforming data into Linked Data which can be searched, navigated and analysed via the user interface and the API.</p></td>
<td><p>N/A</p></td>
<td><p>Biomedical science, Omics, Biology, Chemistry, Medicine</p></td>
<td><p>–</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://bio.tools/pga">PGA</a></p></td>
<td><p>Construction of customized protein databases based on RNA-Seq data with/without genome guided, database searching, post-processing and report generation.</p></td>
<td><p>GPL-2.0</p></td>
<td><p>Proteomics, Proteomics experiment, Data submission, annotation and curation, Data integration and warehousing, Database management, Data governance, RNA-seq</p></td>
<td><p>R library</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://bio.tools/Query_Tabular">Query Tabular</a></p></td>
<td><p>Galaxy-based tool which manipulates tabular files. Query Tabular automatically creates a SQLite database directly from a tabular file within a Galaxy workflow. The SQLite database can be saved to the Galaxy history, and further process to generate tabular outputs containing desired information and formatting.</p></td>
<td><p>CC-BY-4.0</p></td>
<td><p>Bioinformatics,Workflows</p></td>
<td><p>Galaxy-based</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://bio.tools/ms-data-core-api">ms-data-core-api</a></p></td>
<td><p>The primary purpose of ms-data-core-api library is to provide commonly used classes and Object Model for Proteomics Experiments. You may also find it useful for your own computational proteomics projects.</p></td>
<td><p>Apache-2.0</p></td>
<td><p>Proteomics, Proteomics experiment, Software engineering</p></td>
<td><p>Java Library</p></td>
</tr>
</tbody>
</table>
<!-- <h3 id=Data-deposition>Data deposition</h3> -->
</div>
<div class="section" id="data-deposition">
<h3><span class="section-number">15.5.2. </span>Data Deposition<a class="headerlink" href="#data-deposition" title="Permalink to this headline">¶</a></h3>
<p><code class="docutils literal notranslate"><span class="pre">Data</span> <span class="pre">deposition</span></code> is the process of loading data to hosting (end) destinations, such as public data archives and data warehouses.</p>
<p><strong>Manually curated list of tools</strong></p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Tool</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>License</p></th>
<th class="head"><p>Topics</p></th>
<th class="head"><p>Resource type</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><a class="reference external" href="https://www.project-redcap.org/">REDCap</a></p></td>
<td><p>REDCap is a secure web application for building and managing online surveys and databases. While REDCap can be used to collect virtually any type of data in any environment</p></td>
<td><p><a class="reference external" href="https://projectredcap.org/partners/termsofuse/">REDCap License Terms</a></p></td>
<td><p>Cloud migration</p></td>
<td><p>Web application</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://github.com/transmart">TransMART</a></p></td>
<td><p>An open-source data warehouse designed to store large amounts of clinical data from clinical trials, as well as data from basic research</p></td>
<td><p><a class="reference external" href="https://github.com/transmart/transmartApp/blob/master/LICENSE.TXT">GPL-3.0 License</a></p></td>
<td><p>Data storage, clinical data</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="https://www.stardog.com/">Stardog</a></p></td>
<td><p>Triple Store Database, Provide an enterprise knowledge graph as FAIR+ data catalogue</p></td>
<td><p>Commercial license</p></td>
<td><p>Data integration,data catalog</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="https://www.postgresql.org/">Postgresql</a></p></td>
<td><p>A free and open-source relational database management system (RDBMS) emphasizing extensibility and SQL compliance.</p></td>
<td><p><a class="reference external" href="https://www.postgresql.org/about/licence/">The PostgreSQL Licence</a></p></td>
<td><p>Relational database</p></td>
<td><p></p></td>
</tr>
</tbody>
</table>
<p><strong>Automatically created list of tools by querying <a class="reference external" href="https://bio.tools">Bio.Tools</a>.</strong></p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Tool</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>License</p></th>
<th class="head"><p>Topics</p></th>
<th class="head"><p>Resource type</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><a class="reference external" href="https://bio.tools/chemotion">Chemotion</a></p></td>
<td><p>Repository for chemistry research data that provides solutions for current challenges to store research data in a feasible manner, allowing the conservation of domain specific information in a machine readable format.</p></td>
<td><p>CC-BY-4.0</p></td>
<td><p>Data submission, annotation and curation, Workflows, Chemistry</p></td>
<td><p>Data repository</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="example-use-case">
<h3><span class="section-number">15.5.3. </span>Example use case:<a class="headerlink" href="#example-use-case" title="Permalink to this headline">¶</a></h3>
<p>To show how some of these tools may be used, the following related recipes provide further guidance with real life examples.</p>
<ol class="simple">
<li><p><a class="reference internal" href="fastq-file-format-validators.html#fcb-interop-fastqval"><span class="std std-ref">File format validation, FASTQ example</span></a></p></li>
</ol>
<!--
2. OMOP ETL [link]( TODO link to recipe) (*in prepartion*)
3. RDF ETL [link]( TODO link to recipe)  (*in prepartion*)
-->
</div>
</div>
<div class="section" id="references">
<h2><span class="section-number">15.6. </span>References<a class="headerlink" href="#references" title="Permalink to this headline">¶</a></h2>
<details class="sphinx-bs dropdown card mb-3">
<summary class="summary-title card-header">
<strong>References</strong><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<p class="card-text" id="bibtex-footbibliography-content/recipes/interoperability/ETL-tools-0"><dl class="footnote brackets">
<dt class="label" id="etl-wiki"><span class="brackets"><a class="fn-backref" href="#id1">1</a></span></dt>
<dd><p class="card-text">Wikipedia. Extract, transform, load. URL: <a class="reference external" href="https://en.wikipedia.org/wiki/Extract,_transform,_load">https://en.wikipedia.org/wiki/Extract,_transform,_load</a>.</p>
</dd>
</dl>
</p>
</div>
</details><div class="section" id="what-to-read-next">
<h3><span class="section-number">15.6.1. </span>What to read next?<a class="headerlink" href="#what-to-read-next" title="Permalink to this headline">¶</a></h3>
<div class="card w-100 border-2 col-md-4 p-0 docutils">
<div class="card-header bg-primary pa_dark docutils text-center">
<img alt="FAIRsharing logo" height="40px" src="/_static/images/logo/fairsharing-logo.png">
</div>
<div class="card-body docutils">
<p class="mb-0">FAIRsharing records appearing in this recipe:</p>
<ul><li><a href="https://doi.org/10.25504/FAIRsharing.fx0mw7" target="_blank">Ensembl</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.9kahy4" target="_blank">GenBank Nucleotide Sequence Database (GenBank)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.364323" target="_blank">Rich Text Format (RTF)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.WWI10U" target="_blank">The FAIR Principles (FAIR)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.63520c" target="_blank">bio.tools</a></li></ul>
</div>
</div>
</div>
</div>
<div class="section" id="authors">
<h2><span class="section-number">15.7. </span>Authors<a class="headerlink" href="#authors" title="Permalink to this headline">¶</a></h2>
<details class="sphinx-bs dropdown card mb-3">
<summary class="summary-title card-header">
<strong>Authors</strong><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p class="card-text">Name</p></th>
<th class="head"><p class="card-text">ORCID</p></th>
<th class="head"><p class="card-text">Affiliation</p></th>
<th class="head"><p class="card-text">Type</p></th>
<th class="head"><p class="card-text">ELIXIR Node</p></th>
<th class="head"><p class="card-text">Contribution</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p class="card-text"><div class="firstCol"><a target="_blank" href="https://github.com/{github_handle}"><img class="avatar-style" src="https://avatars.githubusercontent.com/fuqix"></img><div class="d-block">Fuqi Xu</div></a></div></p></td>
<td><p class="card-text"><a target="_blank" href="https://orcid.org/0000-0002-5923-3859"><i class="fab fa-orcid fa-2x text--orange"></i></a></p></td>
<td><p class="card-text">EMBL-EBI</p></td>
<td><p class="card-text"><i class="fas fa-graduation-cap fa-1x text--orange" alt="Academic"></i></p></td>
<td><p class="card-text"><img class="elixir-style" src="../../../_static/images/logo/Elixir/embl.png" ></img></p></td>
<td><p class="card-text">Writing - Original Draft</p></td>
</tr>
<tr class="row-odd"><td><p class="card-text"><div class="firstCol"><a target="_blank" href="https://github.com/{github_handle}"><img class="avatar-style" src="https://avatars.githubusercontent.com/evaMart"></img><div class="d-block">Eva Marin del Pico</div></a></div></p></td>
<td><p class="card-text"><a target="_blank" href="https://orcid.org/0000-0001-8324-2897"><i class="fab fa-orcid fa-2x text--orange"></i></a></p></td>
<td><p class="card-text">Barcelona Supercomputing Centre</p></td>
<td><p class="card-text"><i class="fas fa-graduation-cap fa-1x text--orange" alt="Academic"></i></p></td>
<td><p class="card-text"><img class="elixir-style" src="../../../_static/images/logo/Elixir/ELIXIR-ES.svg" ></img></p></td>
<td><p class="card-text">Writing - Original Draft</p></td>
</tr>
<tr class="row-even"><td><p class="card-text"><div class="firstCol"><img class="avatar-style" src="https://avatars.githubusercontent.com/NONE"></img><div class="d-block">Sukhi Singh</div></div></p></td>
<td><p class="card-text"><a target="_blank" href="https://orcid.org/0000-0001-8324-2897"><i class="fab fa-orcid fa-2x text--orange"></i></a></p></td>
<td><p class="card-text">The Hyve</p></td>
<td><p class="card-text"><i class="fas fa-project-diagram fa-1x" style="color:#300861;" alt="SME"></i></p></td>
<td><p class="card-text"></p></td>
<td><p class="card-text">Tool curation</p></td>
</tr>
<tr class="row-odd"><td><p class="card-text"><div class="firstCol"><a target="_blank" href="https://github.com/{github_handle}"><img class="avatar-style" src="https://avatars.githubusercontent.com/proccaserra"></img><div class="d-block">Philippe Rocca-Serra</div></a></div></p></td>
<td><p class="card-text"><a target="_blank" href="https://orcid.org/0000-0001-9853-5668"><i class="fab fa-orcid fa-2x text--orange"></i></a></p></td>
<td><p class="card-text">University of Oxford</p></td>
<td><p class="card-text"><i class="fas fa-graduation-cap fa-1x text--orange" alt="Academic"></i></p></td>
<td><p class="card-text"><img class="elixir-style" src="../../../_static/images/logo/Elixir/ELIXIR-UK.svg" ></img></p></td>
<td><p class="card-text">Writing - Review &amp; Editing</p></td>
</tr>
</tbody>
</table>
</div>
</details></div>
<div class="section" id="license">
<h2><span class="section-number">15.8. </span>License<a class="headerlink" href="#license" title="Permalink to this headline">¶</a></h2>
<details class="sphinx-bs dropdown card mb-3">
<summary class="summary-title card-header">
<strong>License</strong><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<a href="https://creativecommons.org/licenses/by/4.0/">
    <img src="../../../_static/images/logo/cc.svg"  height="25"/>
    <img src="../../../_static/images/logo/by.svg"  height="25"/></a>
    <a href="https://creativecommons.org/licenses/by/4.0/"> The Creative Commons 4.0 BY license</a></br>
    <br/>
</div>
</details><hr class="footnotes docutils" />
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./content/recipes/interoperability"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                                            </v-col>
                                            <div class="d-none d-md-block col-md-2 bd-toc show collapse custom-bdtoc pt-5">
                                                <div style="position: fixed">
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i>
            Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#main-objectives">
   15.1. Main Objectives
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#graphical-overview">
   15.2. Graphical Overview
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#requirements">
   15.3. Requirements
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#fairification-objectives-inputs-and-outputs">
   15.4. FAIRification Objectives, Inputs and Outputs
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#operations">
   15.5. Operations
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#data-transformation">
     15.5.1. Data transformation
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#data-deposition">
     15.5.2. Data Deposition
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#example-use-case">
     15.5.3. Example use case:
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#references">
   15.6. References
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#what-to-read-next">
     15.6.1. What to read next?
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#authors">
   15.7. Authors
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#license">
   15.8. License
  </a>
 </li>
</ul>

        </nav>
        </div>
                                            </div>
                                        
                                    </v-row>

                                    
                                        <v-row class="mb-5">
                                            <div class='pl-10 prev-next-bottom mr-10' style="width:82%">
                                                
<a class='left-prev' id="prev-link" href="rdf-conversion.html" title="previous page"><span class="section-number">14. </span>An inventory of tools for converting data to RDF</a>
<a class='right-next' id="next-link" href="VCF-to-FHIR-conversion.html" title="next page"><span class="section-number">16. </span>Converting VCF file to HL7 FHIR JSON Genomic Report profile</a>

                                            </div>
                                        </v-row>
                                    
                                </v-container>
                            </main>
                        
                    
                </v-row>
                
            </v-container>
        </v-row>
    </v-container>
</v-app>


  <script src="../../../_static/js/index.d3f166471bb80abb5163.js"></script>



<!-- Google Analytics -->
<script>
    window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
    ga('create', 'UA-171697285-1', 'auto');
    ga('set', 'anonymizeIp', true);
    ga('send', 'pageview');
</script>
<script async src='https://www.google-analytics.com/analytics.js'></script>
<!-- End Google Analytics -->

<v-container fluid>
    <v-row no-gutter>
        <v-col cols="12"><footer class="footer" style=" margin-top: 12px;  ">
    <div class="container-md mt-4">
      <div class="row" style="display:flex; justify-content: center">
          <div class="col-12 col-sm-12 col-md-4 col-lg-4 col-xl-3">
            <div class="footerImages">
              <a href="https://europa.eu/"><img src="/_static/images/logo/eu-flag.png" height="40" alt="EU flag"></a>
              <a href="https://www.imi.europa.eu/"><img src="/_static/images/logo/imi-logo.png" height="40" alt="IMI logo"></a>
              <a href="https://efpia.eu"><img src="/_static/images/logo/efpia-logo.png" height="40" alt="EFPIA logo"></a>
            </div>
            <div class="mb-3"> Grant agreement 802750 </div>
          </div>
          <div class="col-5 col-sm-2 col-xs-5 col-md-2 col-lg-2 col-xl-2">
              <ul>
                <li><a href="mailto:faircookbook-ed@elixir-europe.org"> Contact us </a></li>
                <li><a href="https://fairplus-project.eu" target="_blank"> FAIRplus </a> </li>
              </ul>
          </div>
          <div class="col-5 col-sm-2 col-xs-5 col-md-2 col-lg-2 col-xl-2">
              <ul>
                <li> <a href="https://faircookbook.elixir-europe.org/content/recipes/afterword/people.html  "> Contributors </a> </li> </li>
                <li> <a href="https://faircookbook.elixir-europe.org/content/recipes/help.html"> How to contribute </a> </li>
              </ul>
          </div>
          <div class="col-12 col-sm-4 col-xs-12 col-md-3 col-lg-3 col-xl-3 pb-5">
            <a class="mr-3" href="https://elixir-europe.org/" target="_blank"><img src="/_static/images/logo/Elixir/ELIXIR.png" height="50" alt="Elixir logo"></a>
            <a href="https://fairplus-project.eu/"><img src="https://fairplus-project.eu/images/fairplus-logo.png" alt="FAIRplus logo" style="background:white;padding:5px;height:33px;"></a>
          </div>
          <div class="col-12 pb-4">
              Recommended in the
              <a
                href="https://www.imi.europa.eu/sites/default/files/uploads/documents/resources-for-projects/IMI2_OpenAcesGuidelines_Updated2021.pdf"
                target="_blank"
                class="primary--text"
              >
                 IMI/IHI Project Guidelines
              </a> and the
              <a
                href="link to https://ec.europa.eu/info/funding-tenders/opportunities/docs/2021-2027/horizon/wp-call/2023-2024/wp-4-health_horizon-2023-2024_en.pdf"
                target="_blank"
                class="primary--text"
              >
                  Horizon Europe Work Programme for Health
              </a> as the resource for FAIR Data Management guidelines and good practices for the Life Sciences.
          </div>
      </div>
    </div>

    <div class="container-fluid" style="padding:0">
        <div class="row" style="padding:0; margin:0">
            <div class="col-12 purple-dark">
                The FAIR Cookbook is licensed under the
                <a href="https://creativecommons.org/licenses/by/4.0/" target="_blank">
                    Creative Commons Attribution 4.0 International
                </a> (CC BY 4.0) license.
            </div>
        </div>
        <div class="row extra_footer d-none">
            <script> mermaid.init() </script>
        </div>
    </div>

</footer> </v-col>
    </v-row>
</v-container>
<script type="application/javascript">
    const vuetify = new Vuetify({
        theme: {
            themes: {
                light: {
                    primary: '#421C6E',
                    secondary: '#FC7A4A',
                    accent: '#8038D1',
                    error: '#FF5252',
                    info: '#2196F3',
                    success: '#7E0038',
                    warning: '#FFC107'
                }
            }
        },
        icons: { iconfont: 'fa4' }
    })
    const my_vue = new Vue({ el: '#app', vuetify: vuetify })

</script>
  </body>
</html>