
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>From proprietary to open standard data format</title>
    
  <link rel="stylesheet" href="../../../_static/css/index.f658d18f9b420779cfdf24aa0a7e2d77.css">

    
  <link rel="stylesheet"
    href="../../../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../../../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../../../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../../../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../../../_static/sphinx-book-theme.e7340bb3dbd8dde6db86f25597f54a1b.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/custom.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../../../_static/js/index.d3f166471bb80abb5163.js">

    <script id="documentation_options" data-url_root="../../../" src="../../../_static/documentation_options.js"></script>
    <script src="../../../_static/jquery.js"></script>
    <script src="../../../_static/underscore.js"></script>
    <script src="../../../_static/doctools.js"></script>
    <script src="../../../_static/togglebutton.js"></script>
    <script src="../../../_static/clipboard.min.js"></script>
    <script src="../../../_static/copybutton.js"></script>
    <script src="../../../_static/mermaid.js"></script>
    <script src="../../../_static/mermaid.min.js"></script>
    <script src="../../../_static/mermaid.core.js"></script>
    <script async="async" kind="hypothesis" src="https://hypothes.is/embed.js"></script>
    <script kind="utterances">

    var commentsRunWhenDOMLoaded = cb => {
    if (document.readyState != 'loading') {
        cb()
    } else if (document.addEventListener) {
        document.addEventListener('DOMContentLoaded', cb)
    } else {
        document.attachEvent('onreadystatechange', function() {
        if (document.readyState == 'complete') cb()
        })
    }
}

var addUtterances = () => {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src = "https://utteranc.es/client.js";
    script.async = "async";

    script.setAttribute("repo", "FAIRplus/the-fair-cookbook");
    script.setAttribute("issue-term", "pathname");
    script.setAttribute("theme", "github-light");
    script.setAttribute("label", "üí¨ comment");
    script.setAttribute("crossorigin", "anonymous");

    sections = document.querySelectorAll("div.section");
    if (sections !== null) {
        section = sections[sections.length-1];
        section.appendChild(script);
    }
}
commentsRunWhenDOMLoaded(addUtterances);
</script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../../../_static/sphinx-book-theme.7d483ff0a819d6edff12ce0b1ead3928.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../../../_static/sphinx-thebe.js"></script>
    <link rel="canonical" href="https://fairplus.github.io/the-fair-cookbook/content/recipes/interoperability/from-proprietary-to-open-standard-mzml-exemplar.html" />
    <link rel="shortcut icon" href="../../../_static/favicon.ico"/>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" />
    <link rel="next" title="File format validation, an example with FASTQ files" href="fastq-file-format-validators.html" />
    <link rel="prev" title="Metadata Profile Validation in RDF with Shape Expression:" href="covid19-sample-metadata-profile-shex-use-case.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />


<!-- Opengraph tags -->
<meta property="og:url"         content="https://fairplus.github.io/the-fair-cookbook/content/recipes/interoperability/from-proprietary-to-open-standard-mzml-exemplar.html" />
<meta property="og:type"        content="article" />
<meta property="og:title"       content="From proprietary to open standard data format" />
<meta property="og:description" content="From proprietary to open standard data format    &lt;br/&gt;     &lt;i class=&#34;fa fa-qrcode fa-2x&#34; style=&#34;color:#7e0038;&#34;&gt;&lt;/i&gt;  &lt;h4&gt;&lt;b&gt;Recipe metadata&lt;/b&gt;&lt;/h4&gt;  identifie" />
<meta property="og:image"       content="https://fairplus.github.io/the-fair-cookbook/_static/cookbook-logo-small.png" />

<meta name="twitter:card" content="summary" />


  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../../../index.html">
  
  <img src="../../../_static/cookbook-logo-small.png" class="logo" alt="logo">
  
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../../../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../../home.html">
   FAIR Cookbook
  </a>
 </li>
</ul>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../introduction.html">
   Introduction
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../introduction/ATI-turingway.html">
     1. The Turing Way Book of Data Science
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../introduction/FAIR-cookbook-audience.html">
     2. Who should read the FAIR Cookbook?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../introduction/brief-FAIR-principles.html">
     3. What are the FAIR Principles?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../introduction/what-is-IMI.html">
     4. What is IMI?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../introduction/FAIR-in-IMI-context.html">
     5. FAIR in the context of IMI
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../introduction/IMI-Data-Access-Request-Process.html">
     6. Engaging with IMI Projects
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../introduction/FAIRplus-values.html">
     7. The Ethical Values behind FAIR and the FAIRPlus project
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../introduction/data-sharing-models.html">
     8. High level data sharing models
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../assessing-fairness.html">
   Assessing FAIR
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../assessing-fairness/cmmi/Data-Management-Maturity-Model.html">
     1. Data Management Maturity Model
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../assessing-fairness/cmmi/Datasets-Maturity-Model.html">
     2. FAIRplus Datasets Maturity Model
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../assessing-fairness/cmmi/Data-Usage-Areas.html">
     3. Data Usage Areas
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../assessing-fairness/fair-indicators.html">
     4. FAIR Indicators
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../assessing-fairness/cmmi/RDA-Indicators.html">
       4.4.1. RDA Indicators
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../assessing-fairness/cmmi/FAIRPlus-Indicators.html">
       4.4.2. IMI FAIRPlus Indicators
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../assessing-fairness/manual-fair-assessment.html">
     5. FAIR assessment by hand
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../assessing-fairness/cmmi/evaluation-method.html">
       5.4.1. Evaluation Method
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../assessing-fairness/cmmi/rda-evaluation.html">
       5.4.2. RDA Evaluation Methods
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../assessing-fairness/automatic-fair-assessment.html">
     6. FAIR assessment by software
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../assessing-fairness/fair-assessment-fairevaluator.html">
       6.4.1. Assessing FAIRness with FAIREvaluator
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../assessing-fairness/fair-assessment-fairshake.html">
       6.4.2. FAIR assessment with FAIRShake
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../infrastructure.html">
   Infrastructure for FAIR
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/id-resolution.html">
     1. Identifier resolution services
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/data-catalog.html">
     2. Building a Dataset Catalogue
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/data-catalog-deployment.html">
     3. Deploying the IMI data catalogue
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../infrastructure/vocabularies.html">
     4. Vocabularies and Terminology Infrastructure
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../infrastructure/vocabularies/technical-and-architectural-selection-criteria-of-ontology-lookup-services.html">
       4.1. Technical &amp; architectural selection criteria of ontology lookup services
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../infrastructure/vocabularies/Selecting_and_using_ontology_lookup_services.html">
       4.2. Selection and exploitation of Ontology Lookup Services
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../infrastructure/vocabularies/UC3_R13_local_ontology_services.html">
       4.3. How to deploy a terminology service - EBI Ontology Lookup Service
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../findability.html">
   Improving Findability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/identifiers.html">
     1. How to generate globally unique, resolvable and persistent identifiers
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/chemical-identities.html">
     2. Chemical Identity - InChi and SMILES
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/identifiers-minids.html">
     3. Identifier Minting Service with Minid Client
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/zenodo-deposition.html">
     4. Deposition to Zenodo Archive
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../findability/seo.html">
     5. Findability: Search Engine Optimization
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../findability/seo/bioschemas-data-page.html">
       5.8.1. Findability: Data Page Markup with Bioschemas
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../findability/seo/bioschemas-dataset.html">
       5.8.2. Findability: Dataset Markup
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../findability/seo/bioschemas-datacatalog.html">
       5.8.3. Findability: Data Catalog Markup
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../accessibility.html">
   Improving Accessibility
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../accessibility/anonymization.html">
     1. Anonymizing Patient Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../accessibility/how-to-use-sftp-for-transfer.html">
     2. How to use SFTP to transfer data files between institutions.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../accessibility/aspera.html">
     3. How to download files with Aspera
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="../interoperability.html">
   Improving Interoperability
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="identifier-mapping.html">
     How to interlink data from different sources?
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="bridgedb-recipe.html">
     Identifier mapping with BridgeDB
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="bridgedb/bridgedb-example.html">
       Identifier mapping using BridgeDb WS
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="selecting-ontologies.html">
     Which vocabulary to use?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ontology-new-term-request-recipe.html">
     Requesting terms addition to terminology artefacts
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="ontology-robot-recipe.html">
     Building an application ontology with Robot
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="ontology-robot/MSIO-robot-build-process.html">
       MSIO, an application ontology for Metabolomics
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="creating-minimal-metadata-profiles.html">
     Creating a Metadata Profile
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="transcriptomics-metadata.html">
       Establishing a Metadata Profile for Transcriptomics
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="covid19-sample-metadata-profile-shex-use-case.html">
       Metadata Profile Validation in RDF with Shape Expression:
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     From proprietary to open standard data format
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="fastq-file-format-validators.html">
     File format validation, an example with FASTQ files
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../reusability.html">
   Improving Reusability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/ATI-licensing.html">
     1. Licensing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/ATI_licensing_software.html">
     2. Software Licenses
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/ATI_licensing_data.html">
     3. Data Licenses
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/expressing-data-use.html">
     4. Patterns and practice for expressing allowed data usage
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/provenance.html">
     5. Provenance Information
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/community-standards.html">
     6. Compliance with community annotation standards
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../applied-examples.html">
   Making IMI data FAIR
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../applied-examples/fair-data-matrix-recipe.html">
     Making a Data Matrix FAIR
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/0-rose-metabolites-Python-data-handling.html">
       Bibliographic Reference:
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/1-rose-metabolites-Python-analysis.html">
       Exploring and comparing Rose Scent profiles stored as tabular data packages with plotnine, a python port of R ggplot2
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/2-rose-metabolites-Python-RDF-querying-analysis.html">
       Exploring a rose garden with SPARQL
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/3-rose-metabolites-R-analysis.html">
       Exploring and comparing Rose Scent profiles stored as tabular data packages with ggplot2 R library
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/etox_raw/FAIRplus_Recipe_Ontology_mapping_the_eTox_dataset_scenario.html">
     Data normalization and integration through entity linking
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/nd4bb_raw/FAIRification_CookBook_Recipe1_V02.html">
     ND4BB dataset FAIRification recipe
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/oncotrack_raw/rawRecipe-oncotrack.html">
     Oncotrack FAIRification recipe
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/resolute_raw/rawRecipe_resolute.html">
     RESOLUTE FAIRification Recipe
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../glossary-fair-terms.html">
   Glossary of FAIR things
  </a>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../help.html">
   Help
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../help/tips-tricks.html">
     1. Recipe Tips and Tricks
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/recipe-template.html">
     2. Recipe Template
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/how-to-create-recipe-with-gdoc.html">
     3. Using GoogleDocs to contribute a FAIR Cookbook Recipe
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/how-to-create-recipe-with-hackmd.html">
     4. How to use HackMD to write a FAIR Cookbook Recipe
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/how-to-create-recipe-with-git.html">
     5. How to create a FAIRPlus Coobook Recipe ?
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../boilerplate.html">
   Miscellaneous
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../boilerplate/license.html">
     1. License for this book
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../boilerplate/people.html">
     2. FAIR Cookbook Steering Council and Institutional Partners
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../boilerplate/governance.html">
     3. FAIR Cookbook Governance
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../boilerplate/code_of_conduct.html">
     4. FAIR Cookbook Code of Conduct
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../boilerplate/project_information.html">
     5. FAIR CookBook BoilerPlate
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../boilerplate/roadmap.html">
     6. FAIR cookbook development roadmap
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../boilerplate/contributing.html">
     7. Contributing to the FAIR cookbook
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../boilerplate/disclaimer.html">
     8. Disclaimer
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../../../_sources/content/recipes/interoperability/from-proprietary-to-open-standard-mzml-exemplar.md"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.md</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/FAIRplus/the-fair-cookbook"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/FAIRplus/the-fair-cookbook/issues/new?title=Issue%20on%20page%20%2Fcontent/recipes/interoperability/from-proprietary-to-open-standard-mzml-exemplar.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        <a class="edit-button" href="https://github.com/FAIRplus/the-fair-cookbook/edit/migrating/content/recipes/interoperability/from-proprietary-to-open-standard-mzml-exemplar.md"><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Edit this page"><i class="fas fa-pencil-alt"></i>suggest edit</button></a>
    </div>
</div>


            <!-- Full screen (wrap in <a> to have style consistency -->
            <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                    data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                    title="Fullscreen mode"><i
                        class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i>
            Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#main-objectives">
   Main Objectives
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#graphical-overview">
   Graphical Overview
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#capability-maturity-table">
   Capability &amp; Maturity Table
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#fairification-objectives-inputs-and-outputs">
   FAIRification Objectives, Inputs and Outputs
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#table-of-data-standards">
   Table of Data Standards
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#ingredients">
   Ingredients
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#converting-mass-spectrometry-data-to-mzml-format-a-step-by-step-process">
   Converting Mass Spectrometry data to mzML format: a Step by Step Process.
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#step-1-obtain-the-dataset">
     Step 1: obtain the dataset
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#step-2-inspect-the-content-of-the-archive">
     Step 2: inspect the content of the archive
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#i-copying-the-archive-to-a-working-directory">
       i. copying the archive to a working directory
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#ii-expand-the-archive">
       ii. expand the archive
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#iii-inspect-the-folder">
       iii. inspect the folder
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#step-3-convert-vendor-specific-format-to-an-open-format">
     Step 3: Convert vendor specific format to an open format
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#install-docker">
       1. install docker:
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#start-and-sign-in-to-docker">
       2. start and sign in to docker
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#sign-up-and-or-login-to-https-hub-docker-com">
       3. sign-up and/or login to https://hub.docker.com/
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#pull-the-docker-container-for-proteowizard">
       4. pull the docker container for ProteoWizard:
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#testing-and-processing-the-resulting-mzml-files">
       5. Testing and processing the resulting mzML files:
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#conclusion">
   Conclusion
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#what-to-read-next">
     What to read next
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#references">
   References
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#authors">
   Authors
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#license">
   License
  </a>
 </li>
</ul>

        </nav>
        
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="from-proprietary-to-open-standard-data-format">
<span id="fcb-interop-convertopen"></span><h1>From proprietary to open standard data format<a class="headerlink" href="#from-proprietary-to-open-standard-data-format" title="Permalink to this headline">¬∂</a></h1>
<br/>
<hr class="docutils" />
<div class="sphinx-bs container-lg pb-3 docutils">
<div class="row docutils">
<div class="d-flex col-lg-3 col-md-4 col-sm-6 col-xs-12 p-1 docutils">
<div class="card w-100 rounded docutils">
<div class="card-header docutils">
<p class="card-text"><i class="fa fa-qrcode fa-2x" style="color:#7e0038;"></i></p>
</div>
<div class="card-body docutils">
<h4><b>Recipe metadata</b></h4>
 identifier: <a href="">RX.X</a> 
 version: <a href="">v1.0</a>
</div>
</div>
</div>
<div class="d-flex col-lg-3 col-md-4 col-sm-6 col-xs-12 p-1 docutils">
<div class="card w-100 rounded docutils">
<div class="card-header docutils">
<p class="card-text"><i class="fa fa-fire fa-2x" style="color:#7e0038;"></i></p>
</div>
<div class="card-body docutils">
<h4><b>Difficulty level</b></h4>
<i class="fa fa-fire fa-lg" style="color:#7e0038;"></i>
<i class="fa fa-fire fa-lg" style="color:#7e0038;"></i>
<i class="fa fa-fire fa-lg" style="color:l#7e0038;"></i>
<i class="fa fa-fire fa-lg" style="color:lightgrey"></i>
<i class="fa fa-fire fa-lg" style="color:lightgrey"></i>
</div>
</div>
</div>
<div class="d-flex col-lg-3 col-md-4 col-sm-6 col-xs-12 p-1 docutils">
<div class="card w-100 rounded docutils">
<div class="card-header docutils">
<p class="card-text"><i class="fas fa-clock fa-2x" style="color:#7e0038;"></i></p>
</div>
<div class="card-body docutils">
<h4><b>Reading Time</b></h4>
<i class="fa fa-clock fa-lg" style="color:#7e0038;"></i> 20 minutes
<h4><b>Recipe Type</b></h4>
<i class="fa fa-laptop fa-lg" style="color:#7e0038;"></i> Hands-on
<h4><b>Executable Code</b></h4>
<i class="fa fa-play-circle fa-lg" style="color:#7e0038;"></i> Yes
</div>
</div>
</div>
<div class="d-flex col-lg-3 col-md-4 col-sm-6 col-xs-12 p-1 docutils">
<div class="card w-100 rounded docutils">
<div class="card-header docutils">
<p class="card-text"><i class="fa fa-users fa-2x" style="color:#7e0038;"></i></p>
</div>
<div class="card-body docutils">
<h4><b>Intended Audience</b></h4>
<p> <i class="fa fa-user-md fa-lg" style="color:#7e0038;"></i> Principal Investigator </p>
<p> <i class="fa fa-database fa-lg" style="color:#7e0038;"></i> Data Manager </p>
<p> <i class="fa fa-wrench fa-lg" style="color:#7e0038;"></i> Data Scientist </p>
</div>
</div>
</div>
</div>
</div>
<hr class="docutils" />
<div class="section" id="main-objectives">
<h2>Main Objectives<a class="headerlink" href="#main-objectives" title="Permalink to this headline">¬∂</a></h2>
<ul class="simple">
<li><p>Document how to convert raw data from a propriatory, vendor specific format to an open standard format.</p></li>
<li><p>Apply the approach to an IMI dataset, more specifically  a targeted metabolic profiling using Biocrates kit produced by IMI Resolute project.</p></li>
</ul>
</div>
<hr class="docutils" />
<div class="section" id="graphical-overview">
<h2>Graphical Overview<a class="headerlink" href="#graphical-overview" title="Permalink to this headline">¬∂</a></h2>
<div class="figure align-default" id="converting-to-an-open-standard-file-format">
<a class="reference internal image-reference" href="../../../_images/from-proprietary.png"><img alt="Converting to an open standard file format" src="../../../_images/from-proprietary.png" style="width: 450px;" /></a>
<p class="caption"><span class="caption-number">Fig. 16 </span><span class="caption-text">Converting to an open standard file format.</span><a class="headerlink" href="#converting-to-an-open-standard-file-format" title="Permalink to this image">¬∂</a></p>
</div>
</div>
<hr class="docutils" />
<div class="section" id="capability-maturity-table">
<h2>Capability &amp; Maturity Table<a class="headerlink" href="#capability-maturity-table" title="Permalink to this headline">¬∂</a></h2>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="text-align:left head"><p>Capability</p></th>
<th class="text-align:left head"><p>Initial Maturity Level</p></th>
<th class="text-align:left head"><p>Final Maturity Level</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-align:left"><p>Interoperability</p></td>
<td class="text-align:left"><p>minimal</p></td>
<td class="text-align:left"><p>repeatable</p></td>
</tr>
</tbody>
</table>
</div>
<hr class="docutils" />
<div class="section" id="fairification-objectives-inputs-and-outputs">
<h2>FAIRification Objectives, Inputs and Outputs<a class="headerlink" href="#fairification-objectives-inputs-and-outputs" title="Permalink to this headline">¬∂</a></h2>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="text-align:left head"><p>Actions.Objectives.Tasks</p></th>
<th class="text-align:left head"><p>Input</p></th>
<th class="text-align:left head"><p>Output</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-align:left"><p><a class="reference external" href="http://edamontology.org/operation_3438">formatting</a></p></td>
<td class="text-align:left"><p><a class="reference internal" href="../interoperability.html"><span class="doc std std-doc">Waters MS format</span></a></p></td>
<td class="text-align:left"><p><a class="reference external" href="https://fairsharing.org/FAIRsharing.26dmba">mzML</a></p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p><a class="reference external" href="http://edamontology.org/operation_3778">text annotation</a></p></td>
<td class="text-align:left"><p><a class="reference external" href="https://fairsharing.org/FAIRsharing.284e1z">PSI-MS</a></p></td>
<td class="text-align:left"><p><a class="reference external" href="http://edamontology.org/data_3779">annotated text</a></p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="table-of-data-standards">
<h2>Table of Data Standards<a class="headerlink" href="#table-of-data-standards" title="Permalink to this headline">¬∂</a></h2>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="text-align:left head"><p>Data Formats</p></th>
<th class="text-align:left head"><p>Terminologies</p></th>
<th class="text-align:left head"><p>Models</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-align:left"><p><a class="reference external" href="https://fairsharing.org/FAIRsharing.26dmba">mzML</a></p></td>
<td class="text-align:left"><p><a class="reference external" href="https://fairsharing.org/FAIRsharing.284e1z">PSI-MS</a></p></td>
<td class="text-align:left"><p></p></td>
</tr>
</tbody>
</table>
</div>
<hr class="docutils" />
<div class="section" id="ingredients">
<h2>Ingredients<a class="headerlink" href="#ingredients" title="Permalink to this headline">¬∂</a></h2>
<p>Tools and  Software:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://github.com/">github</a></p></li>
<li><p><a class="reference external" href="https://www.docker.com/">docker</a></p></li>
<li><p><a class="reference external" href="https://www.python.org/">python</a></p></li>
</ul>
<!-- - [zenodo API](https://developers.zenodo.org/) -->
<!-- - [pandas](https://pandas.pydata.org/) -->
<!-- - [rdflib](https://rdflib.readthedocs.io/en/stable/)
- [jupyter notebook](https://jupyter.org/)
- [sparql](https://www.w3.org/TR/sparql11-query/) -->
</div>
<div class="section" id="converting-mass-spectrometry-data-to-mzml-format-a-step-by-step-process">
<h2>Converting Mass Spectrometry data to mzML format: a Step by Step Process.<a class="headerlink" href="#converting-mass-spectrometry-data-to-mzml-format-a-step-by-step-process" title="Permalink to this headline">¬∂</a></h2>
<div class="section" id="step-1-obtain-the-dataset">
<h3>Step 1: obtain the dataset<a class="headerlink" href="#step-1-obtain-the-dataset" title="Permalink to this headline">¬∂</a></h3>
<p>In the case of the <a class="reference internal" href="../interoperability.html"><span class="doc std std-doc">IMI RESOLUTE</span></a> project, the data is released via the <a class="reference internal" href="../interoperability.html"><span class="doc std std-doc">University of Luxembourg</span></a> server (assuming you have access resolved):</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$&gt; sftp fairplus@NNN.000.000.NNN
&gt;get RESOLUTE_Targted_Metabolomics_of_parental_cell_lines.tar.gz
&gt;exit
$&gt;
</pre></div>
</div>
<p>:warning: <code class="docutils literal notranslate"><span class="pre">NNN.000.000.NNN</span></code> should be replaced with the proper IP address to the University of Luxembourg server once users have obtained data access clearance.</p>
</div>
<div class="section" id="step-2-inspect-the-content-of-the-archive">
<h3>Step 2: inspect the content of the archive<a class="headerlink" href="#step-2-inspect-the-content-of-the-archive" title="Permalink to this headline">¬∂</a></h3>
<div class="section" id="i-copying-the-archive-to-a-working-directory">
<h4>i. copying the archive to a working directory<a class="headerlink" href="#i-copying-the-archive-to-a-working-directory" title="Permalink to this headline">¬∂</a></h4>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$&gt;mv RESOLUTE_Targted_Metabolomics_of_parental_cell_lines.tar.gz /IMI-FAIR+/RESOLUTE
$&gt;cd /IMI-FAIR+/RESOLUTE/
</pre></div>
</div>
</div>
<div class="section" id="ii-expand-the-archive">
<h4>ii. expand the archive<a class="headerlink" href="#ii-expand-the-archive" title="Permalink to this headline">¬∂</a></h4>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$&gt;gunzip RESOLUTE_Targted_Metabolomics_of_parental_cell_lines.tar.gz
$&gt;tar -xvf RESOLUTE_Targted_Metabolomics_of_parental_cell_lines.tar
$&gt;cd RESOLUTE_Targted_Metabolomics_of_parental_cell_lines
</pre></div>
</div>
</div>
<div class="section" id="iii-inspect-the-folder">
<h4>iii. inspect the folder<a class="headerlink" href="#iii-inspect-the-folder" title="Permalink to this headline">¬∂</a></h4>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>&gt;<span class="nv">$anaconda3</span>-2019.10<span class="o">)</span> bob-MacBook-3:RESOLUTE_Targted_Metabolomics_of_parental_cell_lines philippe$ ls -la
total <span class="m">1400</span>
drwxr-xr-x   <span class="m">10</span> bob  staff     <span class="m">320</span> <span class="m">14</span> Jan <span class="m">16</span>:05 .
drwxr-xr-x    <span class="m">8</span> bob  staff     <span class="m">256</span> <span class="m">14</span> Jan <span class="m">10</span>:29 ..
-rwxr-xr-x@   <span class="m">1</span> bob  staff  <span class="m">520170</span> <span class="m">15</span> Nov <span class="m">14</span>:02 Data_Release_Proposal_20191115_Metabolome_of_parental_cell_lines.docx
-rw-r--r--@   <span class="m">1</span> bob  staff   <span class="m">72868</span>  <span class="m">5</span> Nov <span class="m">20</span>:49 EX0003_processed_data.txt
-rw-r--r--@   <span class="m">1</span> bob  staff    <span class="m">6907</span> <span class="m">23</span> Oct <span class="m">16</span>:06 EX0003_sample_metadata.tsv
drwxr-xr-x@   <span class="m">7</span> bob  staff     <span class="m">224</span> <span class="m">14</span> Jan <span class="m">10</span>:43 Raw
drwxr-xr-x  <span class="m">118</span> bob  staff    <span class="m">3776</span> <span class="m">14</span> Jan <span class="m">15</span>:53 data
-rw-r--r--@   <span class="m">1</span> bob  staff   <span class="m">95677</span>  <span class="m">7</span> Nov <span class="m">11</span>:02 p180_metabolites_metadata.tsv
-rw-r--r--@   <span class="m">1</span> bob  staff     <span class="m">162</span> <span class="m">14</span> Jan <span class="m">10</span>:59 ~<span class="nv">$ta_Release_Proposal_20191115_Metabolome_of_parental_cell_lines</span>.doc
</pre></div>
</div>
<blockquote>
<div><p>:octopus: The archive would have benefitted from having a manifest file listing all the files and their associated checksums. In so doing, it would have allowed validation and verification that no corruption happened during file transfer.</p>
</div></blockquote>
<blockquote>
<div><p>:octopus: Refer to the recipe: ‚Äú<a class="reference internal" href="../interoperability.html"><span class="doc std std-doc">How to calculate file checksums</span></a>‚Äù</p>
</div></blockquote>
<blockquote>
<div><p>:octopus: Refer to the recipe: ‚Äú<a class="reference internal" href="../interoperability.html"><span class="doc std std-doc">How to package data for shipping with BDbags</span></a>‚Äù</p>
</div></blockquote>
</div>
</div>
<div class="section" id="step-3-convert-vendor-specific-format-to-an-open-format">
<h3>Step 3: Convert vendor specific format to an open format<a class="headerlink" href="#step-3-convert-vendor-specific-format-to-an-open-format" title="Permalink to this headline">¬∂</a></h3>
<p>One can consult the Elixir-UK <a class="reference external" href="https://fairsharing.org/">FAIRsharing catalog</a> of standards and resources to discover if an open specification exists in the domain of mass spectrometry. In this case, there is as shown below. Note that every records in the catalog has a digital object identifier (DOI), <a class="reference external" href="https://fairsharing.org/FAIRsharing.26dmba">https://fairsharing.org/FAIRsharing.26dmba</a> for HUPO-PSI mzML specifications.</p>
<!-- <div>
  <img src="https://i.imgur.com/AWOWTbr.png" width="750" link="https://fairsharing.org/FAIRsharing.26dmba">
</div> -->
<div class="figure align-default" id="a-standard-record-in-the-fairsharing-catalog-of-resources">
<a class="reference internal image-reference" href="https://i.imgur.com/AWOWTbr.png"><img alt="A Standard Record in the FAIRsharing catalog of resources" src="https://i.imgur.com/AWOWTbr.png" style="width: 750px;" /></a>
<p class="caption"><span class="caption-number">Fig. 17 </span><span class="caption-text">The <a class="reference external" href="https://fairsharing.org/FAIRsharing.26dmba">HUPI-PSI mzML Standard Record</a> in the <a class="reference external" href="https://fairsharing.org/">Elixir FAIRsharing catalog</a> of resources.</span><a class="headerlink" href="#a-standard-record-in-the-fairsharing-catalog-of-resources" title="Permalink to this image">¬∂</a></p>
</div>
<p>The objective here is to conversion raw data in manufacturer format to an open format, which would allow data to be used without restrictions. To achieve this, we rely on a <code class="docutils literal notranslate"><span class="pre">containerized</span></code> version of the <a class="reference external" href="https://github.com/ProteoWizard/pwiz">Proteowizard</a>.</p>
<blockquote>
<div><p><strong>requirements</strong>:</p>
</div></blockquote>
<div class="section" id="install-docker">
<h4>1. install docker:<a class="headerlink" href="#install-docker" title="Permalink to this headline">¬∂</a></h4>
<p>on a MacOS system, invoke the following:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>&gt;brew update
&gt;brew install docker
</pre></div>
</div>
</div>
<div class="section" id="start-and-sign-in-to-docker">
<h4>2. start and sign in to docker<a class="headerlink" href="#start-and-sign-in-to-docker" title="Permalink to this headline">¬∂</a></h4>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>&gt;docker start
&gt;docker login
</pre></div>
</div>
</div>
<div class="section" id="sign-up-and-or-login-to-https-hub-docker-com">
<h4>3. sign-up and/or login to <a class="reference external" href="https://hub.docker.com/">https://hub.docker.com/</a><a class="headerlink" href="#sign-up-and-or-login-to-https-hub-docker-com" title="Permalink to this headline">¬∂</a></h4>
</div>
<div class="section" id="pull-the-docker-container-for-proteowizard">
<h4>4. pull the docker container for ProteoWizard:<a class="headerlink" href="#pull-the-docker-container-for-proteowizard" title="Permalink to this headline">¬∂</a></h4>
<p>:warning: be sure to sign-up and login to <a class="reference external" href="https://hub.docker.com/">https://hub.docker.com/</a></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>&gt;docker pull chambm/pwiz-i-agree-to-the-vendor-licenses
</pre></div>
</div>
<ul class="simple">
<li><p>In order to be able to reach</p></li>
</ul>
<p><a class="reference external" href="https://hub.docker.com/r/chambm/pwiz-skyline-i-agree-to-the-vendor-licenses">https://hub.docker.com/r/chambm/pwiz-skyline-i-agree-to-the-vendor-licenses</a></p>
<ul class="simple">
<li><p>Run the Proteowizard <code class="docutils literal notranslate"><span class="pre">pwiz</span> <span class="pre">tool</span></code> from the container over the WATERS raw data, by issueing the following command from the terminal:</p></li>
</ul>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>&gt;docker run -it --rm -e <span class="nv">WINEDEBUG</span><span class="o">=</span>-all -v /Users/bob/Documents/IMI-FAIR+/Resolute/RESOLUTE_Targted_Metabolomics_of_parental_cell_lines/Raw/MCC025_p180_102024_102059_20190125/raw<span class="se">\ </span>data/KIT2-0-8015_1024503073/:/data chambm/pwiz-skyline-i-agree-to-the-vendor-licenses wine msconvert /data/KIT2-0-8015_1024503073_01_0_1_1_10_11000002.raw --mzML
</pre></div>
</div>
<p>The command explained:</p>
<p>By essence, the resulting mzML files generated during the conversion are syntactically valid documents as the <code class="docutils literal notranslate"><span class="pre">pwiz</span></code> performs validation against the mzml xml schema during the serialization.</p>
<p>In some situations, the conversion will fails and no mzML output will be generated. Various reasons can explain failure to convert. The most common ones are corrupted:</p>
<ul class="simple">
<li><p>raw data files</p></li>
<li><p>unsupported vendor format</p></li>
</ul>
<p>To address the former, it is good practice to compute a hash (md5, sha2) checksum fingerprinting each of the files. This allows to ensure that no file corruption has occurred during transfer and copy.</p>
<p>To address the latter, one should consult to table of compatibility:</p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="text-align:left head"><p>Format</p></th>
<th class="text-align:left head"><p>Status</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-align:left"><p>ABI T2D</p></td>
<td class="text-align:left"><p>not working</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>Agilent MHDAC (non-IMS)</p></td>
<td class="text-align:left"><p>working</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>Agilent MIDAC (IMS)</p></td>
<td class="text-align:left"><p>working</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>Bruker BAF</p></td>
<td class="text-align:left"><p>working</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>Bruker FID/YEP</p></td>
<td class="text-align:left"><p>not working</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>Bruker TDF</p></td>
<td class="text-align:left"><p>working</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>Sciex WIFF</p></td>
<td class="text-align:left"><p>working</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>Sciex WIFF2</p></td>
<td class="text-align:left"><p>working</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>Shimadzu QQQ</p></td>
<td class="text-align:left"><p>working</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>Shimadzu QTOF</p></td>
<td class="text-align:left"><p>working</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>Thermo RAW</p></td>
<td class="text-align:left"><p>working</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>Waters RAW</p></td>
<td class="text-align:left"><p>working</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>Waters UNIFI</p></td>
<td class="text-align:left"><p>not working</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="testing-and-processing-the-resulting-mzml-files">
<h4>5. Testing and processing the resulting mzML files:<a class="headerlink" href="#testing-and-processing-the-resulting-mzml-files" title="Permalink to this headline">¬∂</a></h4>
<p>for users unfamiliar with format, a search via popular search engine will yield options. Alternately, users may consult the Elixir Biotools registry for suggestions.</p>
<p>A number of libraries are available for parsing (reading and writing) <code class="docutils literal notranslate"><span class="pre">mzML</span></code> document. <code class="docutils literal notranslate"><span class="pre">mzML</span></code> is a king of <code class="docutils literal notranslate"><span class="pre">XML</span></code> format for which an XML schema has been defined and allows syntactic validation through standard library in languages such as java, c++ or python. The top hit corresponds the the <code class="docutils literal notranslate"><span class="pre">pymzml</span></code> library.</p>
<!-- <div>
<img src="https://i.imgur.com/BTs0GUS.png" width="750" border="1"/>
</div> -->
<div class="figure align-default" id="pymzml-in-biotools-registry">
<a class="reference internal image-reference" href="https://i.imgur.com/BTs0GUS.png"><img alt="pymzml in Biotools registry" src="https://i.imgur.com/BTs0GUS.png" style="width: 750px;" /></a>
<p class="caption"><span class="caption-number">Fig. 18 </span><span class="caption-text">The Python <code class="docutils literal notranslate"><span class="pre">pymzml</span></code> library entry in the <a class="reference external" href="https://biotools.org/">Elixir Biotools catalog</a> of resources.</span><a class="headerlink" href="#pymzml-in-biotools-registry" title="Permalink to this image">¬∂</a></p>
</div>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>import os
import sys
import pymzml
from pymzml.plot import Factory

<span class="nv">mzml_file</span> <span class="o">=</span><span class="s2">&quot;KIT3-0-8015_1024503088_42_0_1_1_00_1024502932.mzML&quot;</span>
<span class="nv">msrun</span> <span class="o">=</span> pymzml.run.Reader<span class="o">(</span>mzml_file<span class="o">)</span>

<span class="nv">mzml_basename</span> <span class="o">=</span> os.path.basename<span class="o">(</span>mzml_file<span class="o">)</span>
<span class="nv">pf</span> <span class="o">=</span> Factory<span class="o">()</span>
pf.new_plot<span class="o">()</span>
pf.add<span class="o">(</span>msrun<span class="o">[</span><span class="s2">&quot;TIC&quot;</span><span class="o">]</span>.peaks<span class="o">()</span>, <span class="nv">color</span><span class="o">=(</span><span class="m">0</span>, <span class="m">0</span>, <span class="m">0</span><span class="o">)</span>, <span class="nv">style</span><span class="o">=</span><span class="s2">&quot;lines&quot;</span>, <span class="nv">title</span><span class="o">=</span>mzml_basename<span class="o">)</span>
pf.save<span class="o">(</span>
        <span class="s2">&quot;chromatogram_{0}.html&quot;</span>.format<span class="o">(</span>mzml_basename<span class="o">)</span>,
        <span class="nv">layout</span><span class="o">={</span><span class="s2">&quot;xaxis&quot;</span>: <span class="o">{</span><span class="s2">&quot;title&quot;</span>: <span class="s2">&quot;Retention time&quot;</span><span class="o">}</span>, <span class="s2">&quot;yaxis&quot;</span>: <span class="o">{</span><span class="s2">&quot;title&quot;</span>: <span class="s2">&quot;TIC&quot;</span><span class="o">}}</span>,
<span class="o">)</span>
</pre></div>
</div>
<!-- <div>
<img src="https://i.imgur.com/caaqwFo.png" width="650"/>
</div> -->
<div class="figure align-default" id="pymzml-rendered-msrun-profile">
<a class="reference internal image-reference" href="https://i.imgur.com/caaqwFo.png"><img alt="pymzml rendered msrun profile" src="https://i.imgur.com/caaqwFo.png" style="width: 650px;" /></a>
<p class="caption"><span class="caption-number">Fig. 19 </span><span class="caption-text">The Python <code class="docutils literal notranslate"><span class="pre">pymzml</span></code> library rendering of a spectrum as extracted from an mzml formatted mass spectrum data file.</span><a class="headerlink" href="#pymzml-rendered-msrun-profile" title="Permalink to this image">¬∂</a></p>
</div>
<p>In the follow-up recipe, we will show how to boostrap the creation of an ISA metadata file from a bag of mzML documents. But in the following, we‚Äôll show how to read an mzML file using a python library.</p>
</div>
</div>
</div>
<div class="section" id="conclusion">
<h2>Conclusion<a class="headerlink" href="#conclusion" title="Permalink to this headline">¬∂</a></h2>
<p>In this recipe, we have shown how to convert a proprietary file format to an open standard format, using the exemplar situation of mass spectrometry data. Of course, there are many domain specific data formats and unfortunately not all benefit from the support of open source / open format communities. However by consulting the <a class="reference external" href="https://Elixirhug.org">Elixir UK</a> <a class="reference external" href="https://fairsharing.org">FAIRsharing registry</a>, it is possible to identify if such open format specifications are available.
Then, interrogating the <a class="reference external" href="https://bio.tools/">Biotools catalog</a>, it may well be also possible to retrieve libraries and software components allowing manipulations of such format.</p>
<div class="section" id="what-to-read-next">
<h3>What to read next<a class="headerlink" href="#what-to-read-next" title="Permalink to this headline">¬∂</a></h3>
<ul class="simple">
<li><p><a class="reference internal" href="../interoperability.html"><span class="doc std std-doc">How to calculate file checksums</span></a></p></li>
<li><p><a class="reference internal" href="../interoperability.html"><span class="doc std std-doc">How to package data for shipping with BDbags</span></a></p></li>
<li><p><a class="reference internal" href="../interoperability.html"><span class="doc std std-doc">How to produce an ISA metadata file from a set of mzML fles</span></a></p></li>
<li><p><a class="reference internal" href="../interoperability.html"><span class="doc std std-doc">How to deposit data to Zenodo</span></a></p></li>
</ul>
</div>
</div>
<hr class="docutils" />
<div class="section" id="references">
<h2>References<a class="headerlink" href="#references" title="Permalink to this headline">¬∂</a></h2>
<ol class="simple">
<li><p>Chambers, M., Maclean, B., Burke, R. et al. A cross-platform toolkit for mass spectrometry and proteomics. Nat Biotechnol 30, 918‚Äì920 (2012). <a class="reference external" href="https://doi.org/10.1038/nbt.237">https://doi.org/10.1038/nbt.237</a></p></li>
<li><p>Bald T, Barth J, Niehues A, et al. pymzML‚ÄìPython module for high-throughput bioinformatics on mass spectrometry data. Bioinformatics (Oxford, England). 2012 Apr;28(7):1052-1053. DOI: 10.1093/bioinformatics/bts066.</p></li>
</ol>
</div>
<div class="section" id="authors">
<h2>Authors<a class="headerlink" href="#authors" title="Permalink to this headline">¬∂</a></h2>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="text-align:left head"><p>Name</p></th>
<th class="text-align:left head"><p>Affiliation</p></th>
<th class="text-align:left head"><p>orcid</p></th>
<th class="text-align:left head"><p>CrediT role</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-align:left"><p>Philippe Rocca-Serra</p></td>
<td class="text-align:left"><p>University of Oxford, Data Readiness Group</p></td>
<td class="text-align:left"><p><a class="reference external" href="https://orcid.org/orcid.org/0000-0001-9853-5668">0000-0001-9853-5668</a></p></td>
<td class="text-align:left"><p>Writing - Original Draft</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>Susanna-Assunta Sansone</p></td>
<td class="text-align:left"><p>University of Oxford, Data Readiness Group</p></td>
<td class="text-align:left"><p></p></td>
<td class="text-align:left"><p>Writing - Review &amp; Editing, Funding acquisition</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p></p></td>
<td class="text-align:left"><p></p></td>
<td class="text-align:left"><p></p></td>
<td class="text-align:left"><p>Review</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p></p></td>
<td class="text-align:left"><p></p></td>
<td class="text-align:left"><p></p></td>
<td class="text-align:left"><p>Contribution</p></td>
</tr>
</tbody>
</table>
</div>
<hr class="docutils" />
<div class="section" id="license">
<h2>License<a class="headerlink" href="#license" title="Permalink to this headline">¬∂</a></h2>
<p>This page is released under the Creative Commons 4.0 BY license.</p>
<p><a href="https://creativecommons.org/licenses/by/4.0/"><img src="https://mirrors.creativecommons.org/presskit/buttons/80x15/png/by.png" height="20"/></a></p>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./content/recipes/interoperability"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        
        <div class='prev-next-bottom'>
            
    <a class='left-prev' id="prev-link" href="covid19-sample-metadata-profile-shex-use-case.html" title="previous page">Metadata Profile Validation in RDF with Shape Expression:</a>
    <a class='right-next' id="next-link" href="fastq-file-format-validators.html" title="next page">File format validation, an example with FASTQ files</a>

        </div>
        
        </div>
    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By The FAIRplus project members<br/>
        
            &copy; Copyright 2020 (CC-BY 4.0).<br/>
          <div class="extra_footer">
            <script>mermaid.init();</script> <div class="row_footer"> This project has received funding from the <a href="https://www.imi.europa.eu/">Innovative Medicines Initiative</a> 2 Joint Undertaking under grant agreement No 802750. This Joint Undertaking receives support from the European Union‚Äôs Horizon 2020 research and innovation programme and EFPIA Companies. This communication reflects the views of the authors and neither IMI nor the European Union, EFPIA or any Associated Partners are liable for any use that may be made of the information contained herein. </div>
<div class="row_footer"> This page was created by <a href="https://github.com/jupyter/jupyter-book/graphs/contributors">The Jupyter Book Community</a> </div>
<div class="row_footer"> <div class="column" style="center"><a href="https://europa.eu/"><img src="https://raw.githubusercontent.com/FAIRplus/the-fair-cookbook/migrating/_static/images/logo/eu-flag-recolored-fairplus.png" height="20" alt="EU flag" style="float:center""></a></div> <div class="column"><a href="https://www.imi.europa.eu/"><img src="https://raw.githubusercontent.com/FAIRplus/the-fair-cookbook/migrating/_static/images/logo/imi-logo-fairplus-v1.png" height="20" alt="IMI logo" style="float:center""></a></div> <div class="column"><a href="https://efpia.eu"><img src="https://raw.githubusercontent.com/FAIRplus/the-fair-cookbook/migrating/_static/images/logo/efpia-logo-fairplus2.png" height="20" alt="EFPIA logo" style="float:center""></a></div> <br/> </div>
          </div>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../../../_static/js/index.d3f166471bb80abb5163.js"></script>


    
    <!-- Google Analytics -->
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-171697285-1', 'auto');
      ga('set', 'anonymizeIp', true);
      ga('send', 'pageview');
    </script>
    <script async src='https://www.google-analytics.com/analytics.js'></script>
    <!-- End Google Analytics -->
    
  </body>
</html>