
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>8. Building an application ontology with ROBOT</title>

  <link rel="stylesheet" href="../../../_static/css/index.f658d18f9b420779cfdf24aa0a7e2d77.css">


  <link rel="stylesheet"
    href="../../../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">



  <link rel="stylesheet"
    href="../../../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../../../_static/vendor/lato_latin-ext/1.44.1/index.css">


    <link rel="stylesheet" href="../../../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../../../_static/sphinx-book-theme.e7340bb3dbd8dde6db86f25597f54a1b.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/materialdesignicons.min.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/vuetify.min.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/animate.min.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/custom.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/search-wizard.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/search.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />

  <link rel="preload" as="script" href="../../../_static/js/index.d3f166471bb80abb5163.js">

    <script id="documentation_options" data-url_root="../../../" src="../../../_static/documentation_options.js"></script>
    <script src="../../../_static/jquery.js"></script>
    <script src="../../../_static/underscore.js"></script>
    <script src="../../../_static/doctools.js"></script>
    <script src="../../../_static/togglebutton.js"></script>
    <script src="../../../_static/clipboard.min.js"></script>
    <script src="../../../_static/copybutton.js"></script>
    <script src="../../../_static/js/mermaid.min.js"></script>
    <script src="../../../_static/js/vue.min.js"></script>
    <script src="../../../_static/js/searchNodal.js"></script>
    <script src="../../../_static/js/mermaid.js"></script>
    <script src="../../../_static/js/mermaid.core.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../../../_static/sphinx-book-theme.7d483ff0a819d6edff12ce0b1ead3928.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../../../_static/sphinx-thebe.js"></script>
    <link rel="canonical" href="https://fairplus.github.io/the-fair-cookbook/content/recipes/interoperability/ontology-robot-recipe.html" />
    <link rel="shortcut icon" href="../../../_static/favicon.ico"/>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" />
    <link rel="next" title="8.11.1. Application ontology for metabolomics" href="ontology-robot-recipe/MSIO-robot-build-process.html" />
    <link rel="prev" title="7. Ontology-related tools and services" href="ontology-operations-tools.html" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<meta name="docsearch:language" content="en" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/vuetify/2.6.9/vuetify.js"
        integrity="sha512-cgrSlRKhr/ovJePDoGIdcfZ3wiGcc0fT4Q7hAo0x3P1lyLwUmv2NgJtCo2gzFbxqFjKvNnCfVUjZ1XADm0/ZdQ=="
        crossorigin="anonymous" referrerpolicy="no-referrer">
</script>

<!-- Opengraph tags -->
<meta property="og:url"         content="https://fairplus.github.io/the-fair-cookbook/content/recipes/interoperability/ontology-robot-recipe.html" />
<meta property="og:type"        content="article" />
<meta property="og:title"       content="Building an application ontology with ROBOT" />
<meta property="og:description" content="An online, open and live resource for the Life Sciences with recipes that help you to make and keep data Findable, Accessible, Interoperable and Reusable; in one word FAIR." />
 <meta property="og:image"       content="/_static/images/homepage.png" /> 
<meta name="twitter:card" content="summary" />
  </head>
<body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">



<v-app id="app" class="pa-0">
    <v-container fluid class="pa-0">
        <v-row class="pa-0" no-gutters>
            <v-container fluid class="pa-0">
                <v-row>
                    <div class="topbar container-xl fixed-top pl-0 primary lighten-1 py-2" style="height:65px !important;">
    <div class="topbar-contents row" style="margin: initial !important;">
        <div class="col topbar-main d-flex align-center">
            
                <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                        data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                        aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                        title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                    <i class="fas fa-bars"></i>
                    <i class="fas fa-arrow-left"></i>
                    <i class="fas fa-arrow-up"></i>
                </button>
            

            
                <a href="../../../index.html" style="width: 300px">
                    <img
                        src="../../../_static/cookbook-logo-small.png"
                        class="logo"
                        alt="The FAIR Cookbook"
                        style="width: 100%"
                    >
                </a>
            

            <v-spacer> </v-spacer>
            <div class="mr-4 extraButtons">
                <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger d-flex justify-center align-center" style="height:100%">

    <v-menu offset-y content-class="elevation-0" open-on-hover>
        <template v-slot:activator="{ on, attrs }">
            <v-btn
              id="dropdown-buttons-trigger"
              color="white"
              outlined
              v-bind="attrs"
              v-on="on"
              flat
              solo
              rounded
              :fab="$vuetify.breakpoint.mdAndDown"
              :small="$vuetify.breakpoint.mdAndDown"
              :class="$vuetify.breakpoint.mdAndDown ? 'pa-0' : ''"
            >
                <v-icon :class="$vuetify.breakpoint.lgAndUp ? 'mr-3' : 'mx-1'" left>fab fa-github</v-icon>
                <span v-if="$vuetify.breakpoint.lgAndUp"> GitHub </span>
            </v-btn>
        </template>
        <div class="homepageMenu d-flex flex-column">
            
            <v-btn
              color="grey darken-1"
              dark
              flat
              solo
              rounded
              href="https://github.com/FAIRplus/the-fair-cookbook"
              class="mt-1"
              target="_blank"
            >
                <v-icon class="mr-3" left>fab fa-github</v-icon>
                repository
            </v-btn>
            

            
            <v-btn
              color="grey darken-1"
              dark
              flat
              solo
              rounded
              href="https://github.com/FAIRplus/the-fair-cookbook/issues/new?title=Issue%20on%20page%20%2Fcontent/recipes/interoperability/ontology-robot-recipe.html&body=Your%20issue%20content%20here."
              class="mt-1"
              target="_blank"
            >
                <v-icon class="mr-3" left>fas fa-lightbulb</v-icon>
                open issue
            </v-btn>
            

            
            <v-btn
              color="grey darken-1"
              dark
              flat
              solo
              rounded
              href="https://github.com/FAIRplus/the-fair-cookbook/edit/main/content/recipes/interoperability/ontology-robot-recipe.md"
              class="mt-1"
              target="_blank"
            >
                <v-icon class="mx-3" left>fas fa-pencil-alt</v-icon>
                suggest edit
            </v-btn>
            
        </div>
    </v-menu>
</div>

                <!-- Launch buttons -->

            </div>
                
                    <script>
    Vue.component("Searchfield", {
      data () { return {
        query: null
      } },
      methods: {
        search() {
            if (this.query) window.location.href = `../../search-wizard.html?q=` + this.query
            else window.location.href = `../../search-wizard.html`
        }
      },
      template: `
      <div style="width:250px">
            <v-text-field
              v-model="query"
              append-icon="fa-search"
              label="Search Wizard..."
              dense
              rounded
              single-line
              solo
              hide-details
              class="white primary--text"
              @click:append="search"
              @keyup.enter="search"
            />
      </div>
      `
    })
</script>

<Searchfield> </Searchfield>






                
                
                
        </div>
    </div>
</div>
                </v-row>
                
                    <v-row class="d-flex">
                    
                        <div
                            class=" bd-sidebar site-navigation show"
                            id="site-navigation"
                            style="width:350px"
                        >
                            
                                <div class="nav-drawer d-flex flex-column grey lighten-3" style="height:100%;padding-top: 40px;">

    <div class="testTOC">
        <script>
    Vue.component("Navigation", {
        data() { return { recipes: {} }},
        methods: {
            async getData(){
                const recipes = await fetch('/_static/recipes.json')
                this.recipes = await recipes.json();
                this.parseDom()
            },
            parseDom() {
                const links = Array.prototype.slice.call(document.getElementsByClassName("reference internal"))
                const icon = '<i class="fas fa-play-circle grey lighten-3 secondary--text ml-2"></i>'
                links.forEach(link => {
                    const recipeSplit = link.innerText.split(". ")
                    const recipeName = recipeSplit[recipeSplit.length - 1].trim()
                    recipeSplit.pop()
                    recipeSplit.forEach(pref => { pref = pref.trim() })
                    const prefix = recipeSplit.length > 0 ? recipeSplit.join(".") + "." : null
                    const content = prefix ? `${prefix} ${recipeName}` : recipeName
                    if (Object.keys(this.recipes).includes(recipeName)) {
                        if (this.recipes[recipeName].executable === 'Yes') link.innerHTML = content + icon
                    }
                })
            }
        },
        mounted() { this.getData() }
    })
</script>


<Navigation> </Navigation>

    </div>

    <div class="nav-tree grey px-5 lighten-3 pb-5" style="flex-grow: 1; border-bottom: 1px solid #ccc !important">
        
        
            <nav class="bd-links pt-10" id="bd-docs-nav" aria-label="Main navigation">
    <p class="caption collapsible-parent">
 <span class="caption-text">
  FOREWORD
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/FAIR-cookbook-audience.html">
   Introduction
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/brief-FAIR-principles.html">
   Introducing the FAIR Principles
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/FAIRplus-values.html">
   Reflecting on the ethical values of FAIR
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/fairification-process.html">
   Introducing our FAIRification framework
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/priorization.html">
   Prioritizing projects for FAIRification
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/metadata-fair.html">
   Framing FAIR and the notion of metadata
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/FAIR-and-knowledge-graphs.html">
   Understanding the relation between FAIR and Knowledge Graphs
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/training-datasets.html">
   Training for FAIRification with open or synthetic biomedical datasets
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/public-kg-resource-integration.html">
   Raising Awareness in Public Knowledge Graphs for Life Sciences
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/fair-cro.html">
   Reflecting on Practical Considerations for CROs to play FAIR
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/dpia.html">
   Data Protection Impact Assessment and Data Privacy
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/glossary-fair-terms.html">
   Glossary
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Recipes at a Glance
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../../search-wizard.html">
   All Recipes In a Table
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  FAIR RECIPES
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../findability.html">
   Findability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/identifiers.html">
     1. Introducing unique, persistent identifiers
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/chemical-identities.html">
     2. Creating InChI &amp; SMILES identifiers for chemical structures
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/iupac-names.html">
     3. Creating InChIKeys for IUPAC names
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/identifiers-minids.html">
     4. Minting identifiers with Globus Minid client
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/zenodo-deposition.html">
     5. Depositing to generic repositories - Zenodo use case
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/registeringDatasets.html">
     6. Registering datasets with Wikidata
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/checksum-create.html">
     7. Creating file checksums
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/checksum-validate.html">
     8. Validating checksums to verify file integrity
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../findability/seo.html">
     9. Introducing Search Engine Optimization (SEO)
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../findability/seo/bioschemas-data-page.html">
       9.9.1. Marking up Data pages with Schema.org &amp; Bioschemas for SEO
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../findability/seo/bioschemas-dataset.html">
       9.9.2. Marking up Dataset page with Schema.org &amp; Bioschemas for SEO
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../findability/seo/bioschemas-datacatalog.html">
       9.9.3. Marking up Data Catalogue page with Schema.org &amp; Bioschemas for SEO
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../accessibility.html">
   Accessibility
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../accessibility/sftp.html">
     1. Transferring data with SFTP protocol
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../accessibility/aspera.html">
     2. Downloading data with Aspera protocol
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="../interoperability.html">
   Interoperability
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="registering-swisslipids-identifiers.html">
     1. Registering SwissLipids identifiers in Wikidata
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="identifier-mapping.html">
     2. Interlinking data from different sources
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="bridgedb-recipe.html">
     3. Mapping identifiers with BridgeDb
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="bridgedb/bridgedb-example.html">
       3.11.1. Mapping identifiers using BridgeDb web services
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="introduction-terminologies-ontologies.html">
     4. Introducing terminologies and ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="selecting-ontologies.html">
     5. Selecting terminologies and ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ontology-new-term-request-recipe.html">
     6. Requesting new terms from terminologies and ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ontology-operations-tools.html">
     7. Introducing ontology-related tools and services
    </a>
   </li>
   <li class="toctree-l2 current active collapsible-parent">
    <a class="current reference internal" href="#">
     8. Building an application ontology with ROBOT
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="ontology-robot-recipe/MSIO-robot-build-process.html">
       8.11.1. Building an application ontology for metabolomics - MSIO
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="ontology-robot-recipe/competency-questions.html">
       8.11.2. Defining competency questions for the Ontology ROBOT use case
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ontology-alignment-oxo.html">
     9. Mapping Ontologies with OxO, EBI Ontology Xref Service
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="creating-data-dictionary.html">
     10. Creating a data/variable dictionary
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="creating-minimal-metadata-profiles.html">
     11. Creating a metadata profile
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="transcriptomics-metadata.html">
       11.5.1. Outlining a metadata profile for transcriptomics
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="covid19-sample-metadata-profile-shex-use-case.html">
       11.5.2. Building a community compliant metadata profile - The Covid19 sample profile use case
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="bioactivity-profile.html">
       11.5.3. Outlining a metadata profile for Bioactivity data
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="c4c-clinical-trials.html">
       11.5.4. Creating a metadata profile for clinical trial protocols
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="from-proprietary-to-open-standard-mzml-exemplar.html">
     12. Converting from proprietary to open format
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="fastq-file-format-validators.html">
     13. Validating file format - FASTQ example
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="rdf-conversion.html">
     14. Inventorying tools for converting data to RDF
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ETL-tools.html">
     15. Surveying extraction, transformation, load (ETL) tools
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="VCF-to-FHIR-conversion.html">
     16. Expressing Clinical Genetic Information as FHIR JSON
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="vcf2fhir/FAIRcookbook-vcf2fhir.html">
       16.9.1. Converting VCF file to FHIR JSON
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="creating-knowledge-graph-from-unstructured-text.html">
     17. Creating knowledge graphs from unstructured text
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="nlp2kg/creating-knowledge-graph-from-text.html">
       17.12.1. unstructured text to graph as executable notebook
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../reusability.html">
   Reusability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/ATI-licensing.html">
     1. Licensing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/ATI_licensing_software.html">
     2. Licensing Software
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/ATI_licensing_data.html">
     3. Licensing Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/expressing-data-use.html">
     4. Declaring data permitted uses
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/provenance.html">
     5. Introducing Provenance Information
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../infrastructure.html">
   Infrastructure
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/id-resolution.html">
     1. Introducing identifier resolution services
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/gupri.html">
     2. Creating resolvable identifiers
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/data-catalog.html">
     3. Building a catalogue of datasets
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/dats-model.html">
     4. Introducing the DATS model
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/data-catalog-deployment.html">
     5. Deploying a data catalogue - The IMI data catalogue example
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../infrastructure/vocabularies.html">
     6. Vocabulary management
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../infrastructure/vocabularies/technical-and-architectural-selection-criteria-of-ontology-lookup-services.html">
       6.1. Introducing vocabulary portals and lookup services
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../infrastructure/vocabularies/Selecting_and_using_ontology_lookup_services.html">
       6.2. Selecting an ontology lookup service
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../infrastructure/vocabularies/UC3_R13_local_ontology_services.html">
       6.3. Deploying EBI Ontology Loopkup Service
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/openrefine-nvt.html">
     7. Using OpenRefine and Karma for FAIRification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/fair-api.html">
     8. Developing FAIR API for the Web
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../assessing-fairness.html">
   Assessment
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../assessing-fairness/fair-assessment-fairevaluator.html">
     1. Assessing with FAIR Evaluator
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../assessing-fairness/fair-assessment-fairshake.html">
     2. Assessing with FAIRshake
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  FAIR Maturity
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../maturity.html">
   Changing culture with the Dataset Maturity Model
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/miappe.html">
     Improving dataset maturity - MIAPPE-compliant submission to EMBL-EBI databases
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/plant-pheno-data-publication.html">
     Publishing plant phenotypic data
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../maturity/isa-maturity-progression.html">
     Moving through maturity levels with ISA
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../maturity/create-a-simple-ISA-descriptor.html">
       Minimal Data Maturity with ISA - ISA-Tab and free text
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../maturity/isa-api-programmatic-rebuild-of-BII-S-3.html">
       Enhanced Data Maturity with ISA - ISA-JSON and ontology markup
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../maturity/isa-json-conversion-to-rdf-linked-data.html">
       Moving to a semantically typed version - ISA-JSON-LD
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../maturity/isa-as-RO.html">
       Dissemination - Packaging ISA as a Research Object (RO)
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../applied-examples/fair-data-matrix-recipe.html">
     Making an omics data matrix FAIR
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/0-rose-metabolites-Python-data-handling.html">
       FAIRifying Data Matrices - Step1 - Starting material
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/1-rose-metabolites-Python-analysis.html">
       FAIRifying Data Matrices - Step2 - Structuring data
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/2-rose-metabolites-Python-RDF-querying-analysis.html">
       FAIRifying Data Matrices - Step3 - Exploring data with SPARQL
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/3-rose-metabolites-R-analysis.html">
       FAIRifying Data Matrices - Step4 - Integrating data
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/fair-workflows.html">
     Making Computational Workflows FAIR
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  FAIRified Datasets
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../applied-examples.html">
   Applied examples
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/etox.html">
     1. IMI eTox - toxicity datasets
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/nd4bb.html">
     2. IMI nd4bb - chemical activities datasets
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/oncotrack_raw/rawRecipe-oncotrack.html">
     3. Readying IMI Oncotrack - clinical cohort datasets for deposition to EBI Biosamples
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/resolute_raw/resolute.html">
     4. Depositing IMI ReSOLUTE transcriptomics datasets to EBI repositories
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/ehden-ohdsi.html">
     5. Enhancing discoverability of EHDEN OHDSI data with Schema.org markup
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/eubopen-hcs-bioimage.html">
     6. Depositing IMI EUBOPEN High-Content Screening data to EBI BioImage Archive
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/approach-cdisc.html">
     7. Mapping IMI APPROACH datasets to CDISC-SDTM standard
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  AFTERWORD
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../afterword/fair-reflections.html">
   Lessons learned from the FAIR journey and project outlook
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  GLOSSARY
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/glossary-fair-terms.html">
   Glossary of terms and abbreviations
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  JOIN US
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../community.html">
   Community
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../afterword/people.html">
     Boards and contributors
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../afterword/code_of_conduct.html">
     Code of conduct
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../afterword.html">
     Platform
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../afterword/the-turing-way.html">
       Leveraging the Turing Way Book
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../help.html">
   Contribute
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../afterword/contributing.html">
     1. How to contribute
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/how-to-create-recipe-with-gdoc.html">
     2. Add via GoogleDoc
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/how-to-create-recipe-with-hackmd.html">
     3. Add via HackMD
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/how-to-create-recipe-with-git.html">
     4. Add via Git
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/recipe-template.html">
     5. Git recipe template
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/tips-tricks.html">
     6. Tips and tricks
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/myst.html">
     7. Markdown cheatsheet
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>

</nav>
        
        
             <!-- To handle the deprecated key -->

        
    </div>

    <div class="text-center px-5 lighten-3 grey py-2">
        Powered by <a href="https://jupyterbook.org" class="secondary--text">Jupyter Book</a>
    </div>
</div>
                        </div>
                        
                            <main class="bd-content" role="main" style="width:1%; flex-grow:1;">
                                <v-container fluid id="main-content" class="pa-0">
                                    <v-row class="d-flex">
                                        
                                            <v-col class="col-xl-10 col-lg-10 col-md-9 col-12 px-10 pt-10 mt-3">
                                                
  <div class="section" id="building-an-application-ontology-with-robot">
<span id="fcb-interop-ontorobot"></span><h1><span class="section-number">8. </span>Building an application ontology with ROBOT<a class="headerlink" href="#building-an-application-ontology-with-robot" title="Permalink to this headline">¶</a></h1>
<script type="application/ld+json"> {"@context": {"bs": "https://bioschema.org/", "dc": "http://purl.org/dc/terms/", "owl": "http://www.w3.org/2002/07/owl#", "sdo": "https://schema.org/"}, "@type": ["sdo:HowTo", "sdo:LearningResource"], "@id": "https://w3id.org/faircookbook/FCB023", "sdo:name": "Building an application ontology with ROBOT", "sdo:description": "Building an application ontology with ROBOT", "sdo:keywords": [], "dc:conformsTo": {"@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE", "@type": "CreativeWork"}, "sdo:inLanguage": ["en-GB"], "sdo:audience": ["Terminology Manager|T4FS_0000568", "Data Manager|T4FS_0000520", "Data Scientist|T4FS_0000566", "Ontologist|T4FS_0000567"], "sdo:isPartOf": [{"@type": "CreativeWork", "@id": "https://fairplus.github.io/the-fair-cookbook/", "sdo:url": "https://fairplus.github.io/the-fair-cookbook/", "sdo:name": "The FAIR Cookbook"}], "sdo:learningResourceType": "Hands-on", "sdo:identifier": {"@type": "PropertyValue", "@id": "", "sdo:name": "FCB023", "dso:url": "https://w3id.org/faircookbook/FCB023", "owl:sameAs": "https://identifiers.org/fcb:FCB023"}, "sdo:license": {"@type": "CreativeWork", "@id": "https://creativecommons.org/licenses/by/4.0/", "sdo:url": "https://creativecommons.org/licenses/by/4.0/", "sdo:name": "CC-BY-4.0"}, "sdo:timeRequired": "PT0H60M0S", "sdo:totalTime": "PT0H60M0S", "sdo:step": [{"@type": "sdo:HowToSection", "sdo:position": "1", "sdo:name": "Overview", "sdo:url": "https://w3id.org/faircookbook/FCB023#main-objectives", "sdo:itemListElement": [{"@type": "sdo:HowToStep", "sdo:position": "1", "sdo:name": "", "sdo:itemListElement": [{"@type": "sdo:HowToDirection", "sdo:position": "1", "sdo:name": "step-by-step section", "sdo:text": "Follow the step-by-step section.", "sdo:url": "https://w3id.org/faircookbook/FCB023#main-objectives"}, {"@type": "sdo:HowToTip", "sdo:position": "2", "sdo:name": "Reporting an issue", "sdo:text": "Use our GitHub issue tracker to report any error. Thanks!", "sdo:url": "https://github.com/FAIRplus/the-fair-cookbook/issues/new/choose"}]}]}, {"@type": "sdo:HowToSection", "sdo:position": "2", "sdo:name": "Conclusion", "sdo:url": "https://w3id.org/faircookbook/FCB023#conclusion", "sdo:itemListElement": [{"@type": "sdo:HowToStep", "sdo:position": "1", "sdo:name": "", "sdo:itemListElement": [{"@type": "sdo:HowToDirection", "sdo:position": "1", "sdo:name": "conclusion section", "sdo:text": "What to read next?", "sdo:url": "https://w3id.org/faircookbook/FCB023#conclusion"}, {"@type": "sdo:HowToTip", "sdo:position": "2", "sdo:name": "Reporting an issue", "sdo:text": "Use our GitHub issue tracker to report any error. Thanks!", "sdo:url": "https://github.com/FAIRplus/the-fair-cookbook/issues/new/choose"}]}]}]} </script>
<div class="card my-10 recipe d-flex">
<div class="card-header purple-dark text--white">
<div class="overview">
<div class="overviewLabel">Recipe Overview</div>
<div class="overviewContent">
<div class="item"><div class="itemContainer">
<i class="itemIcon fa fa-clock fa-2x text--white"></i>
<div class="itemContent">
<div class="label"> Reading Time </div>
60 minutes
</div>
</div></div>
<div class="item"><div class="itemContainer">
<i class="itemIcon fa fa-play-circle fa-2x text--white"></i>
<div class="itemContent">
<div class="label"> Executable Code </div>
Yes
</div>
</div></div>
<div class="item" style="margin-bottom:10px"><div class="itemContainer">
<i class="itemIcon fa fa-fire fa-2x text--white" style="margin-left:4px;margin-right: 5px;"></i>
<div class="itemContent">
<div class="label" style="position:relative; left:-6px">Difficulty</div>
<i class="fa fa-fire"></i>
<i class="fa fa-fire"></i>
<i class="fa fa-fire"></i>
<i class="fa fa-fire"></i>
<i class="fa fa-fire" style="color:grey"></i>
</div>
</div></div>
</div></div>
</div>
<div class="card-body text--purple-dark">
<div class="title">
<div style="flex-grow:1; margin-right:5px">Building an application ontology with ROBOT</div>
<img src="https://fairplus.github.io/the-fair-cookbook/_static/images/fairplus-mini.png" alt="FAIRPlus logo"> </img>
</div>
<div class="section">
<i class="sectionIcon fa fa-laptop fa-2x"></i><div class="sectionContent">
<div class="label">Recipe Type</div>
<div class="sectionValue">Hands-on</div>
</div>
</div>
<div class="section" style="flex-grow: 1;">
<i class="sectionIcon fa fa-users fa-2x"></i><div class="sectionContent">
<div class="label">Audience</div>
<div class="sectionValue"><a href="http://purl.obolibrary.org/obo/T4FS_0000568">Terminology Manager</a>, <a href="http://purl.obolibrary.org/obo/T4FS_0000520">Data Manager</a>, <a href="http://purl.obolibrary.org/obo/T4FS_0000566">Data Scientist</a>, <a href="http://purl.obolibrary.org/obo/T4FS_0000567">Ontologist</a></div>
</div>
</div>
<div class="section" style="flex-grow: 1;">
<svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="signal" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 600" class="bars level-3">
<path class="bar5" d="M216 288h-48c-8.84 0-16 7.16-16 16v192c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V304c0-8.84-7.16-16-16-16zM88 384H40c-8.84 0-16 7.16-16 16v96c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16v-96c0-8.84-7.16-16-16-16zm256-192h-48c-8.84 0-16 7.16-16 16v288c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V208c0-8.84-7.16-16-16-16zm128-96h-48c-8.84 0-16 7.16-16 16v384c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V112c0-8.84-7.16-16-16-16zM600 0h-48c-8.84 0-16 7.16-16 16v480c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V16c0-8.84-7.16-16-16-16z"/>
<path class="bar4" d="M216 288h-48c-8.84 0-16 7.16-16 16v192c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V304c0-8.84-7.16-16-16-16zM88 384H40c-8.84 0-16 7.16-16 16v96c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16v-96c0-8.84-7.16-16-16-16zm256-192h-48c-8.84 0-16 7.16-16 16v288c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V208c0-8.84-7.16-16-16-16zm128-96h-48c-8.84 0-16 7.16-16 16v384c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V112c0-8.84-7.16-16-16-16zM600" />
<path class="bar3" d="M216 288h-48c-8.84 0-16 7.16-16 16v192c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V304c0-8.84-7.16-16-16-16zM88 384H40c-8.84 0-16 7.16-16 16v96c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16v-96c0-8.84-7.16-16-16-16zm256-192h-48c-8.84 0-16 7.16-16 16v288c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V208c0-8.84-7.16-16-16-16zm128-96h-48c-8.84" />
<path class="bar1" d="M216 288h-48c-8.84 0-16 7.16-16 16v192c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V304c0-8.84-7.16-16-16-16zM88 384H40c-8.84 0-16 7.16-16 16v96c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16v-96c0-8.84-7.16-16-16-16zm256-192h-48c-8.84"/>
<path class="bar2" d="M216 288h-48c-8.84 0-16 7.16-16 16v192c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V304c0-8.84-7.16-16-16-16zM88 384H40c-8.84"/>
</svg>
<div class="sectionContent">
<div class="label">Maturity Level & Indicator</div>
<div class="sectionValue"><v-chip-group><v-chip href="https://w3id.org/Data-Maturity/DSM-3-C4" class="secondary white--text" target="_blank" label>DSM-3-C4</v-chip></v-chip-group></div>
<div class="tooltip">hover me
<span class="tooltiptext">Tooltip text</span></div>
</div>
</div><div class="card-footer text--orange sphinx-bs.badge.badge-primary"> Cite me with 
<a href="https://w3id.org/faircookbook/FCB023" class="text--purple-dark">FCB023
</a></div>
</div>
</div>
<div class="section" id="main-objectives">
<h2><span class="section-number">8.1. </span>Main Objectives<a class="headerlink" href="#main-objectives" title="Permalink to this headline">¶</a></h2>
<blockquote>
<div><p>The main purpose of this recipe is building an application ontology from source ontologies using <code class="docutils literal notranslate"><span class="pre">ROBOT</span></code> <a class="footnote-reference brackets" href="#pmid31357927" id="id1">2</a> via a sustainable dynamic pipeline to allow seamless integration of source ontology updates <a class="footnote-reference brackets" href="#arp2015" id="id2">1</a>.</p>
</div></blockquote>
<p>An application ontology is a semantic artefact which is developed to answer the needs of a specific application or focus.
Thus, it may borrow terms from a number of reference ontologies, which can be extremely large but whose broad coverage may not be required by the application ontology. Yet, it is critical to keep the <code class="docutils literal notranslate"><span class="pre">application</span> <span class="pre">ontology</span></code> synchronized with the <code class="docutils literal notranslate"><span class="pre">reference</span> <span class="pre">ontologies</span></code> that imports are made from.  We aim to document how a certain level of automation can be achieved.</p>
<p><a class="reference external" href="http://robot.obolibrary.org/">ROBOT</a> is an open source tool for ontology development. It supports ontology editing, ontology annotation, ontology format conversion, and other functions. It runs on the Java Virtual Machine (JVM) and can be used through command line tools on Windows, macOS and Linux.</p>
</div>
<div class="section" id="graphical-overview">
<h2><span class="section-number">8.2. </span>Graphical Overview<a class="headerlink" href="#graphical-overview" title="Permalink to this headline">¶</a></h2>
<details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="ontology-robot-recipe-figure1">
<img alt="Building an ontology with Robot tool" src="../../../_images/ontology-robot-recipe.md-figure1.mmd.png" />
<p class="caption"><span class="caption-number">Fig. 8.2 </span><span class="caption-text">Building an ontology with Robot tool.</span><a class="headerlink" href="#ontology-robot-recipe-figure1" title="Permalink to this image">¶</a></p>
</div>
</div>
</details></div>
<hr class="docutils" />
<div class="section" id="fairification-objectives-inputs-and-outputs">
<h2><span class="section-number">8.3. </span>FAIRification Objectives, Inputs and Outputs<a class="headerlink" href="#fairification-objectives-inputs-and-outputs" title="Permalink to this headline">¶</a></h2>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="text-align:left head"><p>Actions.Objectives.Tasks</p></th>
<th class="text-align:left head"><p>Input</p></th>
<th class="text-align:left head"><p>Output</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-align:left"><p><a class="reference external" href="http://edamontology.org/topic_0089">Ontology and terminology</a></p></td>
<td class="text-align:left"><p>[tsv,OWL]</p></td>
<td class="text-align:left"><p>OWL</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="table-of-data-standards">
<h2><span class="section-number">8.4. </span>Table of Data Standards<a class="headerlink" href="#table-of-data-standards" title="Permalink to this headline">¶</a></h2>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="text-align:left head"><p>Data Formats</p></th>
<th class="text-align:left head"><p>Terminologies</p></th>
<th class="text-align:left head"><p>Models</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-align:left"><p><a class="reference external" href="https://fairsharing.org/FAIRsharing.atygwy">OWL</a></p></td>
<td class="text-align:left"><p></p></td>
<td class="text-align:left"><p></p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p><a class="reference external" href="https://fairsharing.org/FAIRsharing.aa0eat">OBO</a></p></td>
<td class="text-align:left"><p></p></td>
<td class="text-align:left"><p></p></td>
</tr>
</tbody>
</table>
</div>
<hr class="docutils" />
<div class="section" id="ingredients">
<h2><span class="section-number">8.5. </span>Ingredients<a class="headerlink" href="#ingredients" title="Permalink to this headline">¶</a></h2>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="text-align:left head"><p>Tool Name</p></th>
<th class="text-align:left head"><p>Tool Location</p></th>
<th class="text-align:left head"><p>Tool function</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-align:left"><p>ROBOT</p></td>
<td class="text-align:left"><p><a class="reference external" href="http://robot.obolibrary.org/">http://robot.obolibrary.org/</a></p></td>
<td class="text-align:left"><p>ontology management cli</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>Ontology development kit</p></td>
<td class="text-align:left"><p><a class="reference external" href="https://github.com/INCATools/ontology-development-kit">https://github.com/INCATools/ontology-development-kit</a> (comes with ROBOT included)</p></td>
<td class="text-align:left"><p>ontology management environment</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>Protégé/other ontology editor</p></td>
<td class="text-align:left"><p><a class="reference external" href="https://protege.stanford.edu/">https://protege.stanford.edu/</a></p></td>
<td class="text-align:left"><p>ontology editor</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>SPARQL</p></td>
<td class="text-align:left"><p><a class="reference external" href="https://www.w3.org/TR/sparql11-query/">https://www.w3.org/TR/sparql11-query/</a></p></td>
<td class="text-align:left"><p>ontology query language</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p>ELK</p></td>
<td class="text-align:left"><p><a class="reference external" href="https://www.cs.ox.ac.uk/isg/tools/ELK/">https://www.cs.ox.ac.uk/isg/tools/ELK/</a></p></td>
<td class="text-align:left"><p>ontology reasoner</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p>Hermit</p></td>
<td class="text-align:left"><p><a class="reference external" href="http://www.hermit-reasoner.com/">http://www.hermit-reasoner.com/</a></p></td>
<td class="text-align:left"><p>ontology reasoner</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="step-by-step-process">
<h2><span class="section-number">8.6. </span>Step-by-step process<a class="headerlink" href="#step-by-step-process" title="Permalink to this headline">¶</a></h2>
<div class="section" id="preliminary-requirements">
<h3><span class="section-number">8.6.1. </span>Preliminary requirements<a class="headerlink" href="#preliminary-requirements" title="Permalink to this headline">¶</a></h3>
<p>The development of an <code class="docutils literal notranslate"><span class="pre">application</span> <span class="pre">ontology</span></code> requires joint contributions from <code class="docutils literal notranslate"><span class="pre">domain</span> <span class="pre">experts</span></code>, <code class="docutils literal notranslate"><span class="pre">use</span> <span class="pre">case</span> <span class="pre">owners</span></code> and <code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">developers</span></code>. The domain expert provides essential domain knowledge. The use case owners defines the <code class="docutils literal notranslate"><span class="pre">competency</span> <span class="pre">questions</span></code> of the application ontology. And the ontology developers are <code class="docutils literal notranslate"><span class="pre">IT</span> <span class="pre">specialists</span></code> working on the construction of the application ontology.</p>
</div>
<div class="section" id="step-1-define-the-goal-of-the-application-ontology">
<h3><span class="section-number">8.6.2. </span>Step 1: Define the goal of the application ontology<a class="headerlink" href="#step-1-define-the-goal-of-the-application-ontology" title="Permalink to this headline">¶</a></h3>
<p>The development of an <code class="docutils literal notranslate"><span class="pre">application</span> <span class="pre">ontology</span></code> is driven by specific use cases and target datasets. The first step in application ontology development is to determine the subject and the aim of the application ontology.</p>
<p>In this recipe, we demonstrate the workflow of building an application ontology for patient metadata and patient sequencing data investigations. <code class="docutils literal notranslate"><span class="pre">Competency</span> <span class="pre">questions</span></code> of this ontology are provided below:</p>
<!-- [here](https://github.com/FAIRplus/the-fair-cookbook/blob/ontology_robot_recipe/docs/content/recipes/ontology-robot/competency_questions.md).  -->
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="text-align:left head"><p><strong>Theme</strong></p></th>
<th class="text-align:left head"><p><strong>Competency Questions</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td class="text-align:left"><p><strong>General Questions</strong></p></td>
<td class="text-align:left"><p></p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p></p></td>
<td class="text-align:left"><p>➕ How to identify relevant public domain ontologies suiting our needs?</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p></p></td>
<td class="text-align:left"><p>➕ How to establish an ontology covering all terms that are included in the actual data to be represented?</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p></p></td>
<td class="text-align:left"><p>➕ How to remove terms from the resulting ontology that are not needed?</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p></p></td>
<td class="text-align:left"><p>➕ How to guarantee consistency of the final ontology?</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p></p></td>
<td class="text-align:left"><p>➕ How to identify differences in comparison to a previous version of the resulting ontology?</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p><strong>Questions without specifying compounds or genes for the <a class="reference download internal" download="" href="../../../_downloads/5591660813f83203f617deec6d135104/ExternalStudiesKQ.xlsx"><code class="xref download docutils literal notranslate"><span class="pre">example</span> <span class="pre">dataset</span></code></a></strong></p></td>
<td class="text-align:left"><p></p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p></p></td>
<td class="text-align:left"><p>➕ Identify all data generated from tissues taken from patients suffering from a specific disease.</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p></p></td>
<td class="text-align:left"><p>➕ Identify all data generated from a specific tissues obtained from mouse models that are related to a specific disease.</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p></p></td>
<td class="text-align:left"><p>➕ Identify all data generated from lung tissue taken from patients suffering from a lung disease that is not related to oncology.</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p></p></td>
<td class="text-align:left"><p>➕ Identify all data generated from primary cells whose origin is the lung.</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p></p></td>
<td class="text-align:left"><p>➕ Identify all data generated from cell lines whose origin is the lung.</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p><strong>Questions including single genes or gene sets</strong></p></td>
<td class="text-align:left"><p></p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p></p></td>
<td class="text-align:left"><p>➕ What is the expression of PPARg / growths factors in lung tissue from IPF patients?</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p></p></td>
<td class="text-align:left"><p>➕ What is the expression of PPARg / growths factors in primary cells obtained from lung tissue from healthy subjects?</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p></p></td>
<td class="text-align:left"><p>➕ What is the expression of PPARg / growths factors in all available tissues obtained from healthy subjects?</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p><strong>Questions including compound or treatment information</strong></p></td>
<td class="text-align:left"><p></p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p></p></td>
<td class="text-align:left"><p>➕ Identify all data generated from primary cells treated with a kinase inhibitor.</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p></p></td>
<td class="text-align:left"><p>➕ Identify all data from patients treated with a specific medication.</p></td>
</tr>
<tr class="row-odd"><td class="text-align:left"><p></p></td>
<td class="text-align:left"><p>➕ Identify all data generated from cells / cell lines that have been treated with compounds targeting a member of a specific pathway.</p></td>
</tr>
<tr class="row-even"><td class="text-align:left"><p></p></td>
<td class="text-align:left"><p>➕ What is the expression of PPARg in lung tissue upon treatment with a specific compound in patients suffering from a specific disease</p></td>
</tr>
</tbody>
</table>
<p><em>Table 1</em> is a snapshot of the example dataset. The complete patient metadata example dataset is <a class="reference download internal" download="" href="../../../_downloads/5591660813f83203f617deec6d135104/ExternalStudiesKQ.xlsx"><code class="xref download docutils literal notranslate"><span class="pre">here</span></code></a>.</p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Study</p></th>
<th class="head"><p>source_id</p></th>
<th class="head"><p>sample_description</p></th>
<th class="head"><p>tissue</p></th>
<th class="head"><p>source_tissue</p></th>
<th class="head"><p>cell</p></th>
<th class="head"><p>cellline</p></th>
<th class="head"><p>disease</p></th>
<th class="head"><p>gender</p></th>
<th class="head"><p>species</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>GSE52463Nance2014</p></td>
<td><p>EX08_001</p></td>
<td><p>Lung - Normal</p></td>
<td><p>Lung</p></td>
<td><p></p></td>
<td><p></p></td>
<td><p></p></td>
<td><p>Normal</p></td>
<td><p>male</p></td>
<td><p>Homo sapiens</p></td>
</tr>
<tr class="row-odd"><td><p>GSE52463Nance2014</p></td>
<td><p>EX08_015</p></td>
<td><p>Lung - IPF</p></td>
<td><p>Lung</p></td>
<td><p></p></td>
<td><p></p></td>
<td><p></p></td>
<td><p>Idiopathic Pulmonary Fibrosis</p></td>
<td><p>male</p></td>
<td><p>Homo sapiens</p></td>
</tr>
<tr class="row-even"><td><p>GSE116987Marcher2019</p></td>
<td><p>EX101_1</p></td>
<td><p>HSC CCl4-treated w0</p></td>
<td><p></p></td>
<td><p></p></td>
<td><p>Hepatic Stellate Cell</p></td>
<td><p></p></td>
<td><p>NA</p></td>
<td><p></p></td>
<td><p>Mus musculus</p></td>
</tr>
</tbody>
</table>
<p><em>Table 1: Patient metadata example</em></p>
</div>
<div class="section" id="step-2-select-terms-from-reference-ontologies">
<h3><span class="section-number">8.6.3. </span>Step 2: Select terms from reference ontologies<a class="headerlink" href="#step-2-select-terms-from-reference-ontologies" title="Permalink to this headline">¶</a></h3>
<div class="section" id="step-2-1-select-source-ontologies">
<h4><span class="section-number">8.6.3.1. </span>Step 2.1 Select source ontologies<a class="headerlink" href="#step-2-1-select-source-ontologies" title="Permalink to this headline">¶</a></h4>
<p>To build an <code class="docutils literal notranslate"><span class="pre">application</span> <span class="pre">ontology</span></code> that supports communication between different data resources, it is recommended to reuse existing terms from existing reference ontologies instead of creating new terms.</p>
<blockquote>
<div><p>⚡ <em>Reference ontology:</em> a semantic artifact with a canonical and comprehensive representation of the entities in a specific domain.</p>
<p>⚡ <em>Source ontology:</em> An ontology to be integrated into the application ontology, usually a reference ontology.</p>
</div></blockquote>
<p>When selecting reference ontologies as source ontologies, the reusability and sustainability of the source ontology need to be considered. Many ontologies in <a class="reference external" href="http://www.obofoundry.org/">the OBO foundry</a> use the <a class="reference external" href="https://creativecommons.org/licenses/by/2.0/">CC-BY</a> licence, allowing sharing and adaptation. Such ontologies can be directly used as source ontology. Reference ontologies with similar umbrella structures can be conveniently combined in the application ontology. The format and maintenance of the reference ontology shall also be considered.</p>
<p>Specific use cases and requirements from the target dataset also affect the choice of source ontologies. For use cases focusing on extracting data from heterogeneous datasets with complicated data types and data relationships, reference ontologies with rich term relationships are preferred. The interpretation of each term also depends on the context and requirements of the target dataset. For example, <em>“hypertension”</em> can be either interpreted as a <em>phenotype</em> and mapped to phenotype ontologies, or a <em>disease</em> mapped to disease ontologies.</p>
<p>The example dataset includes metadata related to disease, species, cell lines, tissues and biological sex. <em>Table 2</em> lists some reference ontologies available in corresponding domains. In this recipe, we selected <em>MONDO</em> for disease domain, <em>UBERON</em> for anatomy, <em>NCBItaxon</em> for species taxonomy and <em>PATO</em> for biological sex.</p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Domain</p></th>
<th class="head"><p>Reference Ontologies</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Disease</p></td>
<td><p><a class="reference external" href="http://www.obofoundry.org/ontology/mondo.html">Mondo Disease Ontology, MONDO</a><br> <a class="reference external" href="https://disease-ontology.org/">Disease Ontology, DOID</a><br></p></td>
</tr>
<tr class="row-odd"><td><p>Species</p></td>
<td><p><a class="reference external" href="http://www.obofoundry.org/ontology/ncbitaxon.html">NCBI organismal classification, NCBItaxon</a></p></td>
</tr>
<tr class="row-even"><td><p>Cell line</p></td>
<td><p><a class="reference external" href="http://www.obofoundry.org/ontology/cl.html">Cell Ontology, CL</a><br> <a class="reference external" href="http://www.obofoundry.org/ontology/clo.html">Cell Line Ontology, CLO</a></p></td>
</tr>
<tr class="row-odd"><td><p>Tissue</p></td>
<td><p><a class="reference external" href="http://www.obofoundry.org/ontology/ncit.html">NCI Thesaurus OBO Edition, NCIT</a><br> <a class="reference external" href="http://www.obofoundry.org/ontology/omit.html">Ontology for MIRNA Target, OMIT</a><br> <a class="reference external" href="http://www.obofoundry.org/ontology/uberon.html">Uberon multi-species anatomy ontology, UBERON</a></p></td>
</tr>
<tr class="row-even"><td><p>Biological Sex</p></td>
<td><p><a class="reference external" href="http://www.obofoundry.org/ontology/pato.html">Phenotype And Trait Ontology, PATO</a></p></td>
</tr>
</tbody>
</table>
<p><em>Table 2: Available reference ontologies in selected domains</em></p>
</div>
<div class="section" id="step-2-2-select-seed-ontology-terms">
<h4><span class="section-number">8.6.3.2. </span>Step 2.2 Select seed ontology terms<a class="headerlink" href="#step-2-2-select-seed-ontology-terms" title="Permalink to this headline">¶</a></h4>
<blockquote>
<div><p>⚡ <em>Seed ontology terms</em>: A set of entities extracted from reference ontologies for the application ontology.</p>
</div></blockquote>
<p>This step identifies the seeds needed to perform the knowledge extraction from external sources, i.e., the set of entities to extract in order to be integrated on the application ontology. Ontology Developer can provide the tools to ease and to scale the identification of the seeds. Domain experts can identify the right seeds for a given application ontology.</p>
<p>Besides, the fact that is always possible to manually identify the set of seeds needed for the performing of the concept extraction, to have a helper tool allows to run the task at scale. Following, an automatable approach based on using widely known life sciences annotators  - <a class="reference external" href="https://www.ebi.ac.uk/spot/zooma/">Zooma</a> and <a class="reference external" href="https://bioportal.bioontology.org/annotator">NCBO Annotator</a> - are depicted.</p>
<details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="ontology-robot-recipe-figure2">
<img alt="Ontology seed term selection approaches." src="../../../_images/ontology-robot-recipe.md-figure2.mmd.png" />
<p class="caption"><span class="caption-number">Fig. 8.3 </span><span class="caption-text">Ontology seed term selection approaches.</span><a class="headerlink" href="#ontology-robot-recipe-figure2" title="Permalink to this image">¶</a></p>
</div>
</div>
</details><p><a class="reference external" href="https://www.ebi.ac.uk/spot/zooma/">ZOOMA</a> is a web service for discovering optimal ontology mappings, developed by the Samples, Phenotypes and Ontologies Team at the <a class="reference external" href="https://www.ebi.ac.uk">European Bioinformatics Institute (EMBL-EBI)</a>
. It can be used to automatically annotate plain text about biological entities with ontology classes. Zooma allows to limit the sources used to perform the annotations. These sources are either curated data sources, or ontologies from the <a class="reference external" href="https://www.ebi.ac.uk/ols/index">Ontology Lookup Service (OLS)</a>. Zooma contains a linked data repository of annotation knowledge and highly annotated data that has been fed with manually curated annotation derived from publicly available databases. Because the source data has been curated by hand, it is a rich source of knowledge that is optimised for this task of semantic markup of keywords, in contrast with text-mining approaches.</p>
<p>The <a class="reference external" href="https://bioportal.bioontology.org/annotator">NCBO Annotator</a> is an ontology-based web service that annotates public datasets with biomedical ontology concepts based on their textual metadata. It can be used to automatically tag data  with ontology concepts. These concepts come from the Unified Medical Language System (UMLS) Metathesaurus and the National Center for Biomedical Ontology (NCBO) BioPortal ontologies.</p>
<p>Both the <code class="docutils literal notranslate"><span class="pre">Zooma</span></code> and <code class="docutils literal notranslate"><span class="pre">NCBO</span> <span class="pre">Annotator</span> <span class="pre">service</span></code> provides a <code class="docutils literal notranslate"><span class="pre">web</span> <span class="pre">interface</span></code> and a <code class="docutils literal notranslate"><span class="pre">REST</span> <span class="pre">API</span></code> to identify the seed terms by annotation of free text. Two scripts able to automate the annotation of a set of free text terms are shown following.</p>
</div>
<div class="section" id="seed-term-extraction-with-zooma">
<h4><span class="section-number">8.6.3.3. </span>Seed term extraction with ZOOMA<a class="headerlink" href="#seed-term-extraction-with-zooma" title="Permalink to this headline">¶</a></h4>
<p>The following sample script uses the <code class="docutils literal notranslate"><span class="pre">Zooma</span> <span class="pre">web</span> <span class="pre">service</span></code> to get a list of seed terms - i.e., the URIs of ontology classes -. The service also states the level of confidence of every seed proposed.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1">#Python3</span>
<span class="c1">#zooma-annotator-script.py file</span>

<span class="k">def</span> <span class="nf">get_annotations</span><span class="p">(</span><span class="n">propertyType</span><span class="p">,</span> <span class="n">propertyValues</span><span class="p">,</span> <span class="n">filters</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Get Zooma annotations for the values of a given property of a given type.</span>
<span class="sd">    &quot;&quot;&quot;</span>
    
    <span class="kn">import</span> <span class="nn">requests</span>
    
    <span class="n">annotations</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="n">no_annotations</span> <span class="o">=</span> <span class="p">[]</span>

    <span class="k">for</span> <span class="n">value</span> <span class="ow">in</span> <span class="n">propertyValues</span><span class="p">:</span>
        <span class="n">ploads</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;propertyValue&#39;</span><span class="p">:</span> <span class="n">value</span><span class="p">,</span>
                  <span class="s1">&#39;propertyType&#39;</span><span class="p">:</span> <span class="n">propertyType</span><span class="p">,</span>
                  <span class="s1">&#39;filter&#39;</span><span class="p">:</span> <span class="n">filters</span><span class="p">}</span>
        <span class="n">r</span> <span class="o">=</span> <span class="n">requests</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s1">&#39;http://www.ebi.ac.uk/spot/zooma/v2/api/services/annotate&#39;</span><span class="p">,</span>
                         <span class="n">params</span><span class="o">=</span><span class="n">ploads</span><span class="p">)</span>
            
        <span class="kn">import</span> <span class="nn">json</span>
        <span class="n">data</span> <span class="o">=</span> <span class="n">json</span><span class="o">.</span><span class="n">loads</span><span class="p">(</span><span class="n">r</span><span class="o">.</span><span class="n">text</span><span class="p">)</span>
        
        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">data</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
            <span class="n">no_annotations</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">value</span><span class="p">)</span>
        
        <span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="n">data</span><span class="p">:</span>
            <span class="n">annotations</span><span class="o">.</span><span class="n">append</span><span class="p">((</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">item</span><span class="p">[</span><span class="s1">&#39;semanticTags&#39;</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s2"> &quot;</span>
                                <span class="sa">f</span><span class="s2">&quot;# </span><span class="si">{</span><span class="n">value</span><span class="si">}</span><span class="s2">&quot;</span>
                                <span class="sa">f</span><span class="s2">&quot;-Confidence:</span><span class="si">{</span><span class="n">item</span><span class="p">[</span><span class="s1">&#39;confidence&#39;</span><span class="p">]</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">))</span>

    <span class="k">return</span> <span class="n">annotations</span><span class="p">,</span> <span class="n">no_annotations</span>

<span class="k">if</span> <span class="vm">__name__</span> <span class="o">==</span> <span class="s2">&quot;__main__&quot;</span><span class="p">:</span>
    <span class="n">propertyType</span> <span class="o">=</span> <span class="s1">&#39;gender&#39;</span>
    <span class="n">propertyValues</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;male&#39;</span><span class="p">,</span> <span class="s1">&#39;female&#39;</span><span class="p">,</span> <span class="s1">&#39;unknown&#39;</span><span class="p">]</span>

    <span class="n">annotations</span><span class="p">,</span> <span class="n">no_annotations</span> <span class="o">=</span> <span class="n">get_annotations</span><span class="p">(</span><span class="n">propertyType</span><span class="p">,</span> <span class="n">propertyValues</span><span class="p">)</span>

    <span class="kn">from</span> <span class="nn">pprint</span> <span class="kn">import</span> <span class="n">pprint</span>
    <span class="n">pprint</span><span class="p">(</span><span class="n">annotations</span><span class="p">)</span>
    <span class="n">pprint</span><span class="p">(</span><span class="n">no_annotations</span><span class="p">)</span>
</pre></div>
</div>
<p>Running the above script to get the seeds for the terms <code class="docutils literal notranslate"><span class="pre">male</span></code>, <code class="docutils literal notranslate"><span class="pre">female</span></code>, and <code class="docutils literal notranslate"><span class="pre">unknown</span></code> generates the following results:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="p">[</span><span class="s1">&#39;https://purl.obolibrary.org/obo/PATO_0000384 # male-Confidence:HIGH&#39;</span><span class="p">,</span>
 <span class="s1">&#39;https://purl.obolibrary.org/obo/PATO_0000383 # female-Confidence:HIGH&#39;</span><span class="p">,</span>
 <span class="s1">&#39;https://www.orpha.net/ORDO/Orphanet_288 # unknown-Confidence:MEDIUM&#39;</span><span class="p">,</span>
 <span class="s1">&#39;https://www.ebi.ac.uk/efo/EFO_0003850 # unknown-Confidence:MEDIUM&#39;</span><span class="p">,</span>
 <span class="s1">&#39;https://www.ebi.ac.uk/efo/EFO_0003952 # unknown-Confidence:MEDIUM&#39;</span><span class="p">,</span>
 <span class="s1">&#39;https://www.ebi.ac.uk/efo/EFO_0009471 # unknown-Confidence:MEDIUM&#39;</span><span class="p">,</span>
 <span class="s1">&#39;https://www.ebi.ac.uk/efo/EFO_0000203 # unknown-Confidence:MEDIUM&#39;</span><span class="p">,</span>
 <span class="s1">&#39;https://www.ebi.ac.uk/efo/EFO_0003863 # unknown-Confidence:MEDIUM&#39;</span><span class="p">,</span>
 <span class="s1">&#39;https://www.ebi.ac.uk/efo/EFO_0000616 # unknown-Confidence:MEDIUM&#39;</span><span class="p">,</span>
 <span class="s1">&#39;https://purl.obolibrary.org/obo/HP_0000952 # unknown-Confidence:MEDIUM&#39;</span><span class="p">,</span>
 <span class="s1">&#39;https://www.ebi.ac.uk/efo/EFO_0003853 # unknown-Confidence:MEDIUM&#39;</span><span class="p">,</span>
 <span class="s1">&#39;https://www.ebi.ac.uk/efo/EFO_1001331 # unknown-Confidence:MEDIUM&#39;</span><span class="p">,</span>
 <span class="s1">&#39;https://www.ebi.ac.uk/efo/EFO_0003769 # unknown-Confidence:MEDIUM&#39;</span><span class="p">,</span>
 <span class="s1">&#39;https://purl.obolibrary.org/obo/HP_0000132 # unknown-Confidence:MEDIUM&#39;</span><span class="p">,</span>
 <span class="s1">&#39;https://www.ebi.ac.uk/efo/EFO_0000408 # unknown-Confidence:MEDIUM&#39;</span><span class="p">,</span>
 <span class="s1">&#39;https://www.ebi.ac.uk/efo/EFO_0008549 # unknown-Confidence:MEDIUM&#39;</span><span class="p">,</span>
 <span class="s1">&#39;https://www.ebi.ac.uk/efo/EFO_0001642 # unknown-Confidence:MEDIUM&#39;</span><span class="p">]</span>
<span class="p">[]</span>
</pre></div>
</div>
</div>
<div class="section" id="seed-term-extraction-with-ncbo-annotator">
<h4><span class="section-number">8.6.3.4. </span>Seed term extraction with NCBO Annotator<a class="headerlink" href="#seed-term-extraction-with-ncbo-annotator" title="Permalink to this headline">¶</a></h4>
<p>The following sample script uses the NCBO Annotator web service to get a list of seed terms - i.e., the URIs of ontology classes -.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1">#Python3</span>
<span class="c1">#ncbo-annotator-script.py file</span>

<span class="k">def</span> <span class="nf">get_annotations</span><span class="p">(</span><span class="n">propertyValues</span><span class="p">,</span> <span class="n">ontologies</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Get NCBO Annotations for the values of a given property.</span>
<span class="sd">    &quot;&quot;&quot;</span>
        
    <span class="kn">import</span> <span class="nn">requests</span>
    
    <span class="n">annotations</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="n">no_annotations</span> <span class="o">=</span> <span class="p">[]</span>
    
    <span class="k">for</span> <span class="n">value</span> <span class="ow">in</span> <span class="n">propertyValues</span><span class="p">:</span>
        <span class="n">ploads</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;apikey&#39;</span><span class="p">:</span> <span class="s1">&#39;8b5b7825-538d-40e0-9e9e-5ab9274a9aeb&#39;</span><span class="p">,</span>
                  <span class="s1">&#39;text&#39;</span><span class="p">:</span> <span class="n">value</span><span class="p">,</span>
                  <span class="s1">&#39;ontologies&#39;</span><span class="p">:</span> <span class="n">ontologies</span><span class="p">,</span>
                  <span class="s1">&#39;longest_only&#39;</span><span class="p">:</span> <span class="s1">&#39;true&#39;</span><span class="p">,</span>
                  <span class="s1">&#39;exclude_numbers&#39;</span><span class="p">:</span> <span class="s1">&#39;false&#39;</span><span class="p">,</span>
                  <span class="s1">&#39;whole_word_only&#39;</span><span class="p">:</span> <span class="s1">&#39;true&#39;</span><span class="p">,</span>
                  <span class="s1">&#39;exclude_synonyms&#39;</span><span class="p">:</span> <span class="s1">&#39;false&#39;</span><span class="p">}</span>
        <span class="n">r</span> <span class="o">=</span> <span class="n">requests</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s1">&#39;https://data.bioontology.org/annotator&#39;</span><span class="p">,</span> <span class="n">params</span><span class="o">=</span><span class="n">ploads</span><span class="p">)</span>
        
        <span class="kn">import</span> <span class="nn">json</span>
        <span class="n">data</span> <span class="o">=</span> <span class="n">json</span><span class="o">.</span><span class="n">loads</span><span class="p">(</span><span class="n">r</span><span class="o">.</span><span class="n">text</span><span class="p">)</span>
        
        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">data</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
            <span class="n">no_annotations</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">value</span><span class="p">)</span>
        
        <span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="n">data</span><span class="p">:</span>
            <span class="n">annotations</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">item</span><span class="p">[</span><span class="s1">&#39;annotatedClass&#39;</span><span class="p">][</span><span class="s1">&#39;@id&#39;</span><span class="p">]</span><span class="si">}</span><span class="s2"> # </span><span class="si">{</span><span class="n">value</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">annotations</span><span class="p">,</span> <span class="n">no_annotations</span>

<span class="k">if</span> <span class="vm">__name__</span> <span class="o">==</span> <span class="s2">&quot;__main__&quot;</span><span class="p">:</span>
    <span class="n">propertyType</span> <span class="o">=</span> <span class="s1">&#39;gender&#39;</span>
    <span class="n">propertyValues</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;male&#39;</span><span class="p">,</span> <span class="s1">&#39;female&#39;</span><span class="p">,</span> <span class="s1">&#39;unknown&#39;</span><span class="p">]</span>

    <span class="n">annotations</span><span class="p">,</span> <span class="n">no_annotations</span> <span class="o">=</span> <span class="n">get_annotations</span><span class="p">(</span><span class="n">propertyType</span><span class="p">,</span> <span class="n">propertyValues</span><span class="p">)</span>

    <span class="kn">from</span> <span class="nn">pprint</span> <span class="kn">import</span> <span class="n">pprint</span>
    <span class="n">pprint</span><span class="p">(</span><span class="n">annotations</span><span class="p">)</span>
    <span class="n">pprint</span><span class="p">(</span><span class="n">no_annotations</span><span class="p">)</span>
</pre></div>
</div>
<p>Running the above script to get the seeds for the terms <code class="docutils literal notranslate"><span class="pre">male</span></code>, <code class="docutils literal notranslate"><span class="pre">female</span></code>, and <code class="docutils literal notranslate"><span class="pre">unknown</span></code> generates the following results:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="p">[</span><span class="s1">&#39;http://purl.obolibrary.org/obo/UBERON_0003101 # male&#39;</span><span class="p">,</span>
 <span class="s1">&#39;http://purl.obolibrary.org/obo/UBERON_0003100 # female&#39;</span><span class="p">]</span>
<span class="p">[</span><span class="s1">&#39;unknown&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="section" id="step-2-2-3-seed-term-extraction-with-sparql">
<h4><span class="section-number">8.6.3.5. </span>Step 2.2.3 Seed term extraction with SPARQL<a class="headerlink" href="#step-2-2-3-seed-term-extraction-with-sparql" title="Permalink to this headline">¶</a></h4>
<p>Instead of manually maintaining a list of seed terms to generate a module, a term list can be generated on the fly using a <code class="docutils literal notranslate"><span class="pre">SPARQL</span> <span class="pre">query</span></code>. Here, we generate a subset of <code class="docutils literal notranslate"><span class="pre">UBERON</span></code> terms which have a cross-reference to either <code class="docutils literal notranslate"><span class="pre">FMA</span> <span class="pre">(for</span> <span class="pre">human</span> <span class="pre">anatomy)</span></code> or <code class="docutils literal notranslate"><span class="pre">MA</span> <span class="pre">(for</span> <span class="pre">mouse</span> <span class="pre">anatomy)</span></code> terms, since our example datasets includes human, mouse and rat data.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>PREFIX scdo: &lt;http://scdontology.h3abionet.org/ontology/&gt;
PREFIX obo: &lt;http://purl.obolibrary.org/obo/&gt;
PREFIX rdfs: &lt;http://www.w3.org/2000/01/rdf-schema#&gt;
PREFIX oboInOwl: &lt;http://www.geneontology.org/formats/oboInOwl#&gt;
PREFIX owl: &lt;http://www.w3.org/2002/07/owl#&gt;

SELECT ?uri WHERE <span class="o">{</span>

<span class="o">{</span>
 <span class="o">{</span>
  ?uri oboInOwl:hasDbXref ?xref .
 <span class="o">}</span>
UNION
 <span class="o">{</span>
  ?axiom a owl:Axiom<span class="p">;</span>
  	owl:annotatedSource ?uri<span class="p">;</span>
        oboInOwl:hasDbXref ?xref .
 <span class="o">}</span>
<span class="o">}</span>

?uri a owl:Class

FILTER isLiteral<span class="o">(</span>?xref<span class="o">)</span>
FILTER regex<span class="o">(</span> ?xref, <span class="s2">&quot;^FMA|^MA:&quot;</span>, <span class="s2">&quot;i&quot;</span><span class="o">)</span> 

<span class="o">}</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="step-3-extract-ontology-modules-from-source-ontologies">
<h3><span class="section-number">8.6.4. </span>Step 3: Extract ontology modules from source ontologies<a class="headerlink" href="#step-3-extract-ontology-modules-from-source-ontologies" title="Permalink to this headline">¶</a></h3>
<p>Module extractions from ontologies can be run manually and in an ad hoc fashion. We would however recommend collecting all steps together into a script or <code class="docutils literal notranslate"><span class="pre">Makefile</span></code> to avoid missing steps. <code class="docutils literal notranslate"><span class="pre">ROBOT</span></code> steps can in theory be chained together into single large commands. Practical experience however teaches that this can have unexpected consequences as well as making debugging difficult in the event of an issue. It is therefore advisable to split extractions and merges out into individual steps with intermediate artefacts which can be deleted at the end of the process chain.</p>
<div class="section" id="step-3-1-get-source-ontology-files">
<h4><span class="section-number">8.6.4.1. </span>Step 3.1 Get source ontology files<a class="headerlink" href="#step-3-1-get-source-ontology-files" title="Permalink to this headline">¶</a></h4>
<p>We recommend starting each (re)build of the application ontology with the latest versions of the source ontologies unless there is a good reason not to update to the latest version. Ideally, this should be done automatically, for example through a shell script that CURLs all ontologies from their source URIs, e.g.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>curl -L http://purl.obolibrary.org/obo/uberon.owl &gt; uberon.owl
</pre></div>
</div>
</div>
<div class="section" id="step-3-2-choose-ontology-extraction-algorithms">
<h4><span class="section-number">8.6.4.2. </span>Step 3.2 Choose ontology extraction algorithms<a class="headerlink" href="#step-3-2-choose-ontology-extraction-algorithms" title="Permalink to this headline">¶</a></h4>
<p><code class="docutils literal notranslate"><span class="pre">ROBOT</span></code> supports two types of ontology module extraction, <code class="docutils literal notranslate"><span class="pre">Syntactic</span> <span class="pre">Locality</span> <span class="pre">Module</span> <span class="pre">Extractor</span></code> (SLME) and <code class="docutils literal notranslate"><span class="pre">Minimum</span> <span class="pre">Information</span> <span class="pre">to</span> <span class="pre">Reference</span> <span class="pre">an</span> <span class="pre">External</span> <span class="pre">Ontology</span> <span class="pre">Term</span></code> (MIREOT). <code class="docutils literal notranslate"><span class="pre">SLME</span></code> extractions can be further subdivided into TOP (top module), <code class="docutils literal notranslate"><span class="pre">BOT</span></code> (bottom module) and <code class="docutils literal notranslate"><span class="pre">STAR</span></code> (fixpoint-nested module). For full details of what these different options entail, see <a class="reference external" href="http://robot.obolibrary.org/extract.html">http://robot.obolibrary.org/extract.html</a>. We recommend the use of BOT for comprehensive modules that preserve the links between ontology classes and the use of <code class="docutils literal notranslate"><span class="pre">MIREOT</span></code> if relationships apart from parent-child ones are less important.</p>
</div>
<div class="section" id="step-3-3-extract-modules-using-seed-terms">
<h4><span class="section-number">8.6.4.3. </span>Step 3.3 Extract modules using seed terms<a class="headerlink" href="#step-3-3-extract-modules-using-seed-terms" title="Permalink to this headline">¶</a></h4>
<p>Using <code class="docutils literal notranslate"><span class="pre">robot</span></code> tool provided by the <code class="docutils literal notranslate"><span class="pre">OBO</span> <span class="pre">foundry</span></code>, the creation of a module can be done with one command:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>robot extract --method &lt;some_selection&gt; <span class="se">\</span>
    --input &lt;some_input.owl&gt; <span class="se">\</span>
    --term-file &lt;list_of_classes_of_interest_in_ontology.txt&gt; <span class="se">\</span>
    --intermediates &lt;choose_from_option&gt; <span class="se">\</span>
    --output ./ontology_modules/extracted_module.owl
</pre></div>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">robot</span></code> extract command takes several arguments:</p>
<ul class="simple">
<li><p><em>method</em>: <code class="docutils literal notranslate"><span class="pre">ROBOT</span></code> uses 4 different algorithms to generate a module.  <code class="docutils literal notranslate"><span class="pre">TOP</span></code>, <code class="docutils literal notranslate"><span class="pre">BOT</span></code>, <code class="docutils literal notranslate"><span class="pre">STAR</span></code> (all from the SLME method), and <code class="docutils literal notranslate"><span class="pre">MIREOT</span></code>. The first two will create a module  below or above the seed classes (the classes of interest in the target ontology) respectively. The <code class="docutils literal notranslate"><span class="pre">STAR</span></code> method creates a module by pulling all the properties and axioms of the seed classes but nothing else. <code class="docutils literal notranslate"><span class="pre">MIREOT</span></code> uses a different methods and offers some more options, in particular when it comes to how many levels up or down (parent and children) are needed.</p></li>
<li><p><em>input</em>: this argument is to specify the target ontology you want to extract a module from. It can be the original artefact or a filtered version of it.</p></li>
<li><p><em>imports</em>: this argument allows specifying whether or not to include imported ontologies. Note that the default is to do so using the value <code class="docutils literal notranslate"><span class="pre">include</span></code>. Choose <code class="docutils literal notranslate"><span class="pre">exclude</span></code> otherwise.</p></li>
<li><p><em>term-file</em>: the text file holding the list of classes of interested identified by their iri (e.g. <a class="reference external" href="http://purl.obolibrary.org/obo/UBERON_0001235">http://purl.obolibrary.org/obo/UBERON_0001235</a> # adrenal cortex).</p></li>
<li><p><em>intermediates</em>: specific to the <code class="docutils literal notranslate"><span class="pre">MIREOT</span></code> method, it allows to let the algorithm know how much or how little to extract. It has 3 levels (<code class="docutils literal notranslate"><span class="pre">all</span></code>,<code class="docutils literal notranslate"><span class="pre">minimal</span></code>, <code class="docutils literal notranslate"><span class="pre">none</span></code>).</p></li>
<li><p><em>output</em>: the path to the owl file holding the results of the module extraction.</p></li>
<li><p><em>copy-ontology-annotations</em>: a boolean value true/false to pull or not any class annotation from the parent ontology. default is <code class="docutils literal notranslate"><span class="pre">false</span></code></p></li>
<li><p><em>sources</em>: optional, a pointer to a file mapping .</p></li>
<li><p><em>annotate-with-source</em>: a boolean value true/false. Default is <code class="docutils literal notranslate"><span class="pre">false</span></code>.</p></li>
</ul>
<p>The above query, saved under <code class="docutils literal notranslate"><span class="pre">select_anatomy_subset.sparql</span></code> can be used to generate a dynamic seed list, then do a <code class="docutils literal notranslate"><span class="pre">BOT</span></code> extraction:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>robot query --input uberon.owl --query select_anatomy_subset.sparql uberon_seed_list.txt

robot extract --method BOT --input uberon.owl --term-file uberon_seed_list.txt -o uberon_subset.owl
</pre></div>
</div>
</div>
<div class="section" id="step-3-4-assess-extracted-modules">
<h4><span class="section-number">8.6.4.4. </span>Step 3.4 Assess extracted modules<a class="headerlink" href="#step-3-4-assess-extracted-modules" title="Permalink to this headline">¶</a></h4>
<p>The extracted ontology module should include all seed terms and represent the term relationships correctly. It should also preserve the correct hierarchical structure of the source ontology and have consistent granularity.</p>
</div>
</div>
<div class="section" id="step-4-build-the-upper-level-umbrella-ontology">
<h3><span class="section-number">8.6.5. </span>Step 4: Build the upper level umbrella ontology<a class="headerlink" href="#step-4-build-the-upper-level-umbrella-ontology" title="Permalink to this headline">¶</a></h3>
<p>The umbrella ontology is the root structure of the ontology. Building the umbrella ontology aims to model the main entity of the use case and its relationships with the ontology classes extracted on the previous step. The main identity of the ontology and relationships with extracted modules can be identified by domain experts, or specified by the use case owner.</p>
<p><code class="docutils literal notranslate"><span class="pre">ROBOT</span></code> provides a template-driven ontology term generation system. A <code class="docutils literal notranslate"><span class="pre">ROBOT</span> <span class="pre">template</span></code> is a <code class="docutils literal notranslate"><span class="pre">tab-separated</span> <span class="pre">values</span></code> (TSV) or <code class="docutils literal notranslate"><span class="pre">comma-separated</span> <span class="pre">values</span></code> (CSV) file that depicts a set of patterns to build ontology entities - i.e., classes, properties, and individuals -. These templates can be used for modular ontology development.
After the <em>Domain Expert</em> and the <em>Use Case/Scenario Owner</em> specify the main entity of the use case and its relationships with remaining ontology entities, the <em>Ontology Developer</em> generates the <code class="docutils literal notranslate"><span class="pre">ROBOT</span></code> templates depicting the set of patterns aimed to build the <em>umbrella</em> ontology.</p>
<p>A <code class="docutils literal notranslate"><span class="pre">ROBOT</span></code> command using a template to build an ontology is shown below:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>robot template --template template.csv <span class="se">\</span>
    --prefix <span class="s2">&quot;r4: https://fairplus-project.eu/ontologies/R4/&quot;</span> <span class="se">\</span>
    --ontology-iri <span class="s2">&quot;https://fairplus-project.eu/ontologies/R4/&quot;</span> <span class="se">\</span>
    --output ./templates/umbrella.owl
</pre></div>
</div>
<p>And a template sample is presented following:</p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>ID</p></th>
<th class="head"><p>Label</p></th>
<th class="head"><p>Entity Type</p></th>
<th class="head"><p>Equivalent Axioms</p></th>
<th class="head"><p>Characteristic</p></th>
<th class="head"><p>Domain</p></th>
<th class="head"><p>Range</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>ID</p></td>
<td><p>LABEL</p></td>
<td><p>TYPE</p></td>
<td><p>EC %</p></td>
<td><p>CHARACTERISTIC</p></td>
<td><p>DOMAIN</p></td>
<td><p>RANGE</p></td>
</tr>
<tr class="row-odd"><td><p>ex:cl_2</p></td>
<td><p>Class_2</p></td>
<td><p>class</p></td>
<td><p>-</p></td>
<td><p>-</p></td>
<td><p>-</p></td>
<td><p>-</p></td>
</tr>
<tr class="row-even"><td><p>ex:cl_3</p></td>
<td><p>Class_3</p></td>
<td><p>class</p></td>
<td><p>ex:cl_2</p></td>
<td><p>-</p></td>
<td><p>-</p></td>
<td><p>-</p></td>
</tr>
<tr class="row-odd"><td><p>ex:op_1</p></td>
<td><p>Prop_1</p></td>
<td><p>object property</p></td>
<td><p>-</p></td>
<td><p>Class_2</p></td>
<td><p>Class_3</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p>ex:dp_1</p></td>
<td><p>Prop_2</p></td>
<td><p>data property</p></td>
<td><p>-</p></td>
<td><p>functional</p></td>
<td><p>Class_2</p></td>
<td><p>xsd:string</p></td>
</tr>
</tbody>
</table>
<div class="tip admonition">
<p class="admonition-title">Tip</p>
<p>⚡ The generated ontology can be visualized by using the <a class="reference external" href="https://protege.stanford.edu/">Protégé tool</a> or local deployment of OLS.
The <a class="reference internal" href="../infrastructure/vocabularies/UC3_R13_local_ontology_services.html#fcb-infra-localols"><span class="std std-ref">Deploying the EBI OLS</span></a> option is recommended by this recipe, given that <code class="docutils literal notranslate"><span class="pre">Protégé</span></code> may crash when loading medium or large size ontologies.</p>
</div>
</div>
<div class="section" id="step-5-merge-ontology-modules-and-umbrella-ontology">
<h3><span class="section-number">8.6.6. </span>Step 5: Merge ontology modules and umbrella ontology<a class="headerlink" href="#step-5-merge-ontology-modules-and-umbrella-ontology" title="Permalink to this headline">¶</a></h3>
<p>Merging the ontology modules previously extracted and the umbrella ontology locally built produces a first draft of the application ontology.</p>
<p>The <code class="docutils literal notranslate"><span class="pre">merge</span></code> ROBOT command allows combining two or more separate input ontology modules into a single ontology. Following, the <code class="docutils literal notranslate"><span class="pre">ROBOT</span></code> command merging the umbrella ontology and the modules is shown:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>java -jar robot.jar merge <span class="se">\</span>
    --input ./ontology_modules/iao_mireot_robot_module_1.owl <span class="se">\</span>
    --input ./ontology_modules/obi_mireot_robot_module_2.owl <span class="se">\</span>
    --input ./ontology_modules/duo_mireot_robot_module_3.owl <span class="se">\</span>
    --input ./templates/umbrella.owl <span class="se">\</span>
    --output ./results/r4_app_ontology.owl
</pre></div>
</div>
</div>
<div class="section" id="step-6-post-merge-modifications">
<h3><span class="section-number">8.6.7. </span>Step 6: Post-merge modifications<a class="headerlink" href="#step-6-post-merge-modifications" title="Permalink to this headline">¶</a></h3>
<div class="section" id="step-6-1-materialize-and-reasoning">
<h4><span class="section-number">8.6.7.1. </span>Step 6.1: Materialize and Reasoning<a class="headerlink" href="#step-6-1-materialize-and-reasoning" title="Permalink to this headline">¶</a></h4>
<p>Commands below infer superclasses and superclasses and reduce duplicated axioms merged term:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>robot materialize <span class="se">\</span>
--reasoner ELK  <span class="se">\</span>
--input merged_owl  <span class="se">\</span>
reduce <span class="se">\</span>
--output materialized.owl 
</pre></div>
</div>
<p>The ontology materialization uses <code class="docutils literal notranslate"><span class="pre">OWL</span> <span class="pre">reasoners</span></code>. <code class="docutils literal notranslate"><span class="pre">ROBOT</span></code> provides <a class="reference external" href="http://robot.obolibrary.org/reason">several ontology reasoners</a>.</p>
<p>It is also possible to identify issues in the ontology with some <code class="docutils literal notranslate"><span class="pre">quality</span> <span class="pre">control</span> <span class="pre">SPARQL</span> <span class="pre">queries</span></code>.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>robot report --input edit.owl --output report.tsv
</pre></div>
</div>
</div>
<div class="section" id="step-6-2-annotate">
<h4><span class="section-number">8.6.7.2. </span>Step 6.2: Annotate<a class="headerlink" href="#step-6-2-annotate" title="Permalink to this headline">¶</a></h4>
<p>Ontology annotation adds metadata to the owl file. It is recommended to provide <code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">IRIs</span></code>, <code class="docutils literal notranslate"><span class="pre">version</span> <span class="pre">IRIs</span></code>, <code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">title</span></code>, <code class="docutils literal notranslate"><span class="pre">descriptions</span></code> and <code class="docutils literal notranslate"><span class="pre">license</span></code> to support future usage and management.</p>
<p>The annotation can be added either line-by-line or provided in a turtle (.ttl) file.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1">#ANNOTATE</span>
robot annotate --input materialized.owl <span class="se">\</span>
  --ontology-iri <span class="s2">&quot;https://github.com/ontodev/robot/examples/annotated.owl&quot;</span> <span class="se">\</span>
  --version-iri <span class="s2">&quot;https://github.com/ontodev/robot/examples/annotated-1.owl&quot;</span> <span class="se">\</span>
  --annotation rdfs:comment <span class="s2">&quot;Comment&quot;</span> <span class="se">\</span>
  --annotation rdfs:label <span class="s2">&quot;Label&quot;</span> <span class="se">\</span>
  --annotation-file annotations.ttl <span class="se">\</span>
  --output results/annotated.owl
</pre></div>
</div>
<p>Below is an example annotation file.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>@prefix rdf:     &lt;http://www.w3.org/1999/02/22-rdf-syntax-ns#&gt; .
@prefix rdfs:    &lt;http://www.w3.org/2000/01/rdf-schema#&gt; .
@prefix owl:     &lt;http://www.w3.org/2002/07/owl#&gt; .
@prefix example: &lt;https://github.com/ontodev/robot/examples/&gt; .
@prefix dcterms: &lt;http://purl.org/dc/terms/&gt; .
example:annotated.owl
  rdf:type owl:Ontology <span class="p">;</span>
  rdfs:comment <span class="s2">&quot;Comment from annotations.ttl file.&quot;</span> .
  dcterms:title <span class="s2">&quot;Example title&quot;</span>
  dcterms:license &lt;http://creativecommons.org/publicdomain/zero/1.0/&gt;
  owl:versionIRI &lt;https://github.com/ontodev/robot/examples/annotated-1.owl&gt;
</pre></div>
</div>
</div>
<div class="section" id="step-6-3-convert">
<h4><span class="section-number">8.6.7.3. </span>Step 6.3: Convert<a class="headerlink" href="#step-6-3-convert" title="Permalink to this headline">¶</a></h4>
<p>Besides <code class="docutils literal notranslate"><span class="pre">OWL</span> <span class="pre">format</span></code>, the <code class="docutils literal notranslate"><span class="pre">Open</span> <span class="pre">Biomedical</span> <span class="pre">Ontology</span> <span class="pre">(OBO)</span> <span class="pre">format</span></code> is also widely used in life science related ontologies. It is possible to convert an <code class="docutils literal notranslate"><span class="pre">.owl</span></code> file to <code class="docutils literal notranslate"><span class="pre">.obo</span></code> file using:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1">#CONVERT:</span>
robot convert <span class="se">\</span>
--input  annotated.owl<span class="se">\</span>
--format obo <span class="se">\</span>
--output results/annotated.obo
</pre></div>
</div>
</div>
</div>
<div class="section" id="step-7-assess-coverage-of-the-ontology-scope">
<h3><span class="section-number">8.6.8. </span>Step 7: Assess coverage of the ontology scope<a class="headerlink" href="#step-7-assess-coverage-of-the-ontology-scope" title="Permalink to this headline">¶</a></h3>
<p>The final step of the ontology construction is to assess coverage of the ontology scope by verifying if it is able to answer the competency questions predefined. The competency questions can be implemented as a set of <code class="docutils literal notranslate"><span class="pre">SPARQL</span> <span class="pre">queries</span></code> and run against the developed ontology to check if the answers/results are aligned with the scope of the ontology. The <code class="docutils literal notranslate"><span class="pre">use</span> <span class="pre">case</span> <span class="pre">owner</span></code> and <code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">developer</span></code> can also collaborate on the assessment of the competency questions answers/results.</p>
<p><code class="docutils literal notranslate"><span class="pre">ROBOT</span></code> provides the <code class="docutils literal notranslate"><span class="pre">query</span></code> command to perform <code class="docutils literal notranslate"><span class="pre">SPARQL</span> <span class="pre">queries</span></code>  against an ontology to verify and validate the ontology.</p>
<p>The query command runs SPARQL <code class="docutils literal notranslate"><span class="pre">ASK</span></code>, <code class="docutils literal notranslate"><span class="pre">SELECT</span></code>, and <code class="docutils literal notranslate"><span class="pre">CONSTRUCT</span></code> queries by using the <code class="docutils literal notranslate"><span class="pre">--query</span></code> option with two arguments: <code class="docutils literal notranslate"><span class="pre">a</span> <span class="pre">query</span> <span class="pre">file</span></code> and <code class="docutils literal notranslate"><span class="pre">an</span> <span class="pre">output</span> <span class="pre">file</span></code>. Instead of specifying one or more pairs (query file, output file), it is also possible to specify a single <code class="docutils literal notranslate"><span class="pre">--output-dir</span></code> and use the <code class="docutils literal notranslate"><span class="pre">--queries</span></code> option to provide one or more queries of any type. Each output file will be written to the output directory with the same base name as the query file that produced it. A pattern example of this command is shown as follows.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>robot query --input &lt;input_ontology_file&gt; <span class="se">\</span>
    --queries &lt;query_file&gt; <span class="se">\</span>
    --output-dir &lt;path_to_results&gt; results/
</pre></div>
</div>
</div>
</div>
<div class="section" id="conclusions">
<h2><span class="section-number">8.7. </span>Conclusions<a class="headerlink" href="#conclusions" title="Permalink to this headline">¶</a></h2>
<p>Creation an application ontology and semantic model to support knowledge discovery is an important process in the data management life cycle.
This more advanced recipe has identified and described all the different steps that one needs to consider building such a resource.
While this is important, one should bear in mind the costs associated with maintaining those artefacts and keeping them up to date.
It is therefore also critical to understand the benefits of contributing to existing open efforts.</p>
<div class="section" id="what-to-read-next">
<h3><span class="section-number">8.7.1. </span>What to read next?<a class="headerlink" href="#what-to-read-next" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p>OBO Foundry Semantic Engineering training and the <a class="reference external" href="https://oboacademy.github.io/obook/tutorial/robot-tutorial-1/">Robot tutorial</a></p></li>
<li><p><a class="reference internal" href="creating-minimal-metadata-profiles.html#fcb-interop-metadataprofile"><span class="std std-ref">Creating a metadata profile</span></a>  <!-- How to establish a minimal metadata profile? --> <!-- TODO add a link to corresponding document --></p></li>
<li><p><a class="reference internal" href="ontology-new-term-request-recipe.html#fcb-interop-ontorequest"><span class="std std-ref">Requesting new terms</span></a> <!-- How to submit/request terms to an ontology ?--> <!-- TODO add a link to corresponding document --></p></li>
</ul>
<div class="card w-100 border-2 col-md-4 p-0 docutils">
<div class="card-header bg-primary pa_dark docutils text-center">
<img alt="FAIRsharing logo" height="40px" src="/_static/images/logo/fairsharing-logo.png">
</div>
<div class="card-body docutils">
<p class="mb-0">FAIRsharing records appearing in this recipe:</p>
<ul><li><a href="https://doi.org/10.25504/FAIRsharing.4m97ah" target="_blank">BioPortal</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.4dvtcz" target="_blank">Cell Line Ontology (CLO)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.1943d4" target="_blank">Comma-separated Values (CSV)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.8b6wfq" target="_blank">Disease Ontology (DOID)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.x56jsy" target="_blank">Foundational Model of Anatomy (FMA)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.b2979t" target="_blank">Monarch Disease Ontology (MONDO)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.pdwqcr" target="_blank">Mouse Adult Gross Anatomy Ontology (MA)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.fj07xj" target="_blank">NCBI Taxonomy (NCBITAXON)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.847069" target="_blank">OBO Foundry (OBO)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.Mkl9RR" target="_blank">Ontology Lookup Service (OLS)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.mf91p5" target="_blank">Ontology for MicroRNA Target (OMIT)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.ezwdhz" target="_blank">Phenotypic QualiTy Ontology (PATO)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.87ccfd" target="_blank">Simple Protocol and RDF Query Language Overview (SPARQL)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.a978c9" target="_blank">Tab-separated values (TSV)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.4c0b6b" target="_blank">UBER anatomy ONtology (UBERON)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.atygwy" target="_blank">Web Ontology Language (OWL)</a></li></ul>
</div>
</div>
</div>
</div>
<div class="section" id="references">
<h2><span class="section-number">8.8. </span>References<a class="headerlink" href="#references" title="Permalink to this headline">¶</a></h2>
<details class="sphinx-bs dropdown card mb-3">
<summary class="summary-title card-header">
<strong>References</strong><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<p class="card-text" id="bibtex-footbibliography-content/recipes/interoperability/ontology-robot-recipe-0"><dl class="footnote brackets">
<dt class="label" id="arp2015"><span class="brackets"><a class="fn-backref" href="#id2">1</a></span></dt>
<dd><p class="card-text">Robert Arp, Barry Smith, and Andrew D. Spear. <em>Building Ontologies with Basic Formal Ontology</em>. The MIT Press, 2021/05/14/ 2015. ISBN 9780262527811. URL: <a class="reference external" href="http://www.jstor.org/stable/j.ctt17kk7vw">http://www.jstor.org/stable/j.ctt17kk7vw</a>.</p>
</dd>
<dt class="label" id="pmid31357927"><span class="brackets"><a class="fn-backref" href="#id1">2</a></span></dt>
<dd><p class="card-text">R. C. Jackson, J. P. Balhoff, E. Douglass, N. L. Harris, C. J. Mungall, and J. A. Overton. ROBOT: A Tool for Automating Ontology Workflows. <em>BMC Bioinformatics</em>, 20(1):407, Jul 2019.</p>
</dd>
</dl>
</p>
</div>
</details><!-- ```{bibliography} ../../../_bibliography/bibliography-identifier-mapping.bib
:filter: docname in docnames
``` -->
<!-- - R.C. Jackson, J.P. Balhoff, E. Douglass, N.L. Harris, C.J. Mungall, and J.A. Overton. [_ROBOT: A tool for automating ontology workflows_](https://link.springer.com/epdf/10.1186/s12859-019-3002-3?author_access_token=bB8BLjFWrdh42vR6DjT-nG_BpE1tBhCbnbw3BuzI2RPCZ2BK7EeexaCNYfT-cCz8Q_mrZomT2_svoQf12CW661Sagzw6JGF9DhJq3Q3fTPdMGFMtais7MRgx8-kDhp6uC9g2qcVh5FumTsveV22XVQ%3D%3D). BMC Bioinformatics, vol. 20, July 2019
- Arp, Robert, Barry Smith, and Andrew D. Spear. _Building ontologies with basic formal ontology_. Mit Press, 2015. -->
</div>
<div class="section" id="supplementary-material">
<h2><span class="section-number">8.9. </span>Supplementary material<a class="headerlink" href="#supplementary-material" title="Permalink to this headline">¶</a></h2>
<details class="sphinx-bs dropdown card mb-3">
<summary class="summary-title card-header">
<strong>Supplementary material</strong><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<ul class="simple">
<li><p class="card-text"><a class="reference download internal" download="" href="../../../_downloads/5591660813f83203f617deec6d135104/ExternalStudiesKQ.xlsx"><code class="xref download docutils literal notranslate"><span class="pre">Example</span> <span class="pre">Dataset</span></code></a></p></li>
<li><p class="card-text"><a class="reference internal" href="ontology-robot-recipe/competency-questions.html"><span class="doc std std-doc">Competency questions</span></a></p></li>
<li><p class="card-text"><a class="reference internal" href="ontology-robot-recipe/MSIO-robot-build-process.html"><span class="doc std std-doc">IPython Notebook for the MSIO Build Process</span></a></p></li>
</ul>
</div>
</details></div>
<div class="section" id="authors">
<h2><span class="section-number">8.10. </span>Authors<a class="headerlink" href="#authors" title="Permalink to this headline">¶</a></h2>
<details class="sphinx-bs dropdown card mb-3">
<summary class="summary-title card-header">
<strong>Authors</strong><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p class="card-text">Name</p></th>
<th class="head"><p class="card-text">ORCID</p></th>
<th class="head"><p class="card-text">Affiliation</p></th>
<th class="head"><p class="card-text">Type</p></th>
<th class="head"><p class="card-text">ELIXIR Node</p></th>
<th class="head"><p class="card-text">Contribution</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p class="card-text"><div class="firstCol"><a target="_blank" href="https://github.com/{github_handle}"><img class="avatar-style" src="https://avatars.githubusercontent.com/daniwelter"></img><div class="d-block">Danielle Welter</div></a></div></p></td>
<td><p class="card-text"><a target="_blank" href="https://orcid.org/0000-0003-1058-2668"><i class="fab fa-orcid fa-2x text--orange"></i></a></p></td>
<td><p class="card-text">University of Luxembourg</p></td>
<td><p class="card-text"><i class="fas fa-graduation-cap fa-1x text--orange" alt="Academic"></i></p></td>
<td><p class="card-text"><img class="elixir-style" src="../../../_static/images/logo/Elixir/ELIXIR-LU.svg" ></img></p></td>
<td><p class="card-text">Writing - Original Draft, Code</p></td>
</tr>
<tr class="row-odd"><td><p class="card-text"><div class="firstCol"><img class="avatar-style" src="https://avatars.githubusercontent.com/NONE"></img><div class="d-block">Karsten Quast</div></div></p></td>
<td><p class="card-text"></p></td>
<td><p class="card-text">Boehringer-Ingelheim AG</p></td>
<td><p class="card-text"><i class="fas fa-industry fa-1x text--purple-light" alt="EFPIA"></i></p></td>
<td><p class="card-text"></p></td>
<td><p class="card-text">Writing - Original Draft</p></td>
</tr>
<tr class="row-even"><td><p class="card-text"><div class="firstCol"><a target="_blank" href="https://github.com/{github_handle}"><img class="avatar-style" src="https://avatars.githubusercontent.com/proccaserra"></img><div class="d-block">Philippe Rocca-Serra</div></a></div></p></td>
<td><p class="card-text"><a target="_blank" href="https://orcid.org/0000-0001-9853-5668"><i class="fab fa-orcid fa-2x text--orange"></i></a></p></td>
<td><p class="card-text">University of Oxford</p></td>
<td><p class="card-text"><i class="fas fa-graduation-cap fa-1x text--orange" alt="Academic"></i></p></td>
<td><p class="card-text"><img class="elixir-style" src="../../../_static/images/logo/Elixir/ELIXIR-UK.svg" ></img></p></td>
<td><p class="card-text">Writing - Review &amp; Editing, Conceptualization</p></td>
</tr>
<tr class="row-odd"><td><p class="card-text"><div class="firstCol"><a target="_blank" href="https://github.com/{github_handle}"><img class="avatar-style" src="https://avatars.githubusercontent.com/fuqix"></img><div class="d-block">Fuqi Xu</div></a></div></p></td>
<td><p class="card-text"><a target="_blank" href="https://orcid.org/0000-0002-5923-3859"><i class="fab fa-orcid fa-2x text--orange"></i></a></p></td>
<td><p class="card-text">EMBL-EBI</p></td>
<td><p class="card-text"><i class="fas fa-graduation-cap fa-1x text--orange" alt="Academic"></i></p></td>
<td><p class="card-text"><img class="elixir-style" src="../../../_static/images/logo/Elixir/embl.png" ></img></p></td>
<td><p class="card-text">Writing - Original Draft, Writing - Review &amp; Editing</p></td>
</tr>
<tr class="row-even"><td><p class="card-text"><div class="firstCol"><a target="_blank" href="https://github.com/{github_handle}"><img class="avatar-style" src="https://avatars.githubusercontent.com/ereynrs"></img><div class="d-block">Emiliano Reynares</div></a></div></p></td>
<td><p class="card-text"><a target="_blank" href="https://orcid.org/0000-0002-5109-3716"><i class="fab fa-orcid fa-2x text--orange"></i></a></p></td>
<td><p class="card-text">Boehringer-Ingelheim AG</p></td>
<td><p class="card-text"><i class="fas fa-industry fa-1x text--purple-light" alt="EFPIA"></i></p></td>
<td><p class="card-text"></p></td>
<td><p class="card-text">Writing - Review &amp; Editing</p></td>
</tr>
</tbody>
</table>
</div>
</details></div>
<div class="section" id="license">
<h2><span class="section-number">8.11. </span>License<a class="headerlink" href="#license" title="Permalink to this headline">¶</a></h2>
<details class="sphinx-bs dropdown card mb-3">
<summary class="summary-title card-header">
<strong>License</strong><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<a href="https://creativecommons.org/licenses/by/4.0/">
    <img src="../../../_static/images/logo/cc.svg"  height="25"/>
    <img src="../../../_static/images/logo/by.svg"  height="25"/></a>
    <a href="https://creativecommons.org/licenses/by/4.0/"> The Creative Commons 4.0 BY license</a></br>
    <br/>
</div>
</details><div class="toctree-wrapper compound">
</div>
<hr class="footnotes docutils" />
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./content/recipes/interoperability"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                                            </v-col>
                                            <div class="d-none d-md-block col-md-2 bd-toc show collapse custom-bdtoc pt-5">
                                                <div style="position: fixed">
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i>
            Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#main-objectives">
   8.1. Main Objectives
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#graphical-overview">
   8.2. Graphical Overview
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#fairification-objectives-inputs-and-outputs">
   8.3. FAIRification Objectives, Inputs and Outputs
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#table-of-data-standards">
   8.4. Table of Data Standards
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#ingredients">
   8.5. Ingredients
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#step-by-step-process">
   8.6. Step-by-step process
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#preliminary-requirements">
     8.6.1. Preliminary requirements
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#step-1-define-the-goal-of-the-application-ontology">
     8.6.2. Step 1: Define the goal of the application ontology
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#step-2-select-terms-from-reference-ontologies">
     8.6.3. Step 2: Select terms from reference ontologies
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#step-2-1-select-source-ontologies">
       8.6.3.1. Step 2.1 Select source ontologies
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#step-2-2-select-seed-ontology-terms">
       8.6.3.2. Step 2.2 Select seed ontology terms
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#seed-term-extraction-with-zooma">
       8.6.3.3. Seed term extraction with ZOOMA
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#seed-term-extraction-with-ncbo-annotator">
       8.6.3.4. Seed term extraction with NCBO Annotator
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#step-2-2-3-seed-term-extraction-with-sparql">
       8.6.3.5. Step 2.2.3 Seed term extraction with SPARQL
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#step-3-extract-ontology-modules-from-source-ontologies">
     8.6.4. Step 3: Extract ontology modules from source ontologies
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#step-3-1-get-source-ontology-files">
       8.6.4.1. Step 3.1 Get source ontology files
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#step-3-2-choose-ontology-extraction-algorithms">
       8.6.4.2. Step 3.2 Choose ontology extraction algorithms
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#step-3-3-extract-modules-using-seed-terms">
       8.6.4.3. Step 3.3 Extract modules using seed terms
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#step-3-4-assess-extracted-modules">
       8.6.4.4. Step 3.4 Assess extracted modules
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#step-4-build-the-upper-level-umbrella-ontology">
     8.6.5. Step 4: Build the upper level umbrella ontology
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#step-5-merge-ontology-modules-and-umbrella-ontology">
     8.6.6. Step 5: Merge ontology modules and umbrella ontology
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#step-6-post-merge-modifications">
     8.6.7. Step 6: Post-merge modifications
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#step-6-1-materialize-and-reasoning">
       8.6.7.1. Step 6.1: Materialize and Reasoning
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#step-6-2-annotate">
       8.6.7.2. Step 6.2: Annotate
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#step-6-3-convert">
       8.6.7.3. Step 6.3: Convert
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#step-7-assess-coverage-of-the-ontology-scope">
     8.6.8. Step 7: Assess coverage of the ontology scope
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#conclusions">
   8.7. Conclusions
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#what-to-read-next">
     8.7.1. What to read next?
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#references">
   8.8. References
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#supplementary-material">
   8.9. Supplementary material
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#authors">
   8.10. Authors
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#license">
   8.11. License
  </a>
 </li>
</ul>

        </nav>
        </div>
                                            </div>
                                        
                                    </v-row>

                                    
                                        <v-row class="mb-5">
                                            <div class='pl-10 prev-next-bottom mr-10' style="width:82%">
                                                
<a class='left-prev' id="prev-link" href="ontology-operations-tools.html" title="previous page"><span class="section-number">7. </span>Ontology-related tools and services</a>
<a class='right-next' id="next-link" href="ontology-robot-recipe/MSIO-robot-build-process.html" title="next page"><span class="section-number">8.11.1. </span>Application ontology for metabolomics</a>

                                            </div>
                                        </v-row>
                                    
                                </v-container>
                            </main>
                        
                    
                </v-row>
                
            </v-container>
        </v-row>
    </v-container>
</v-app>


  <script src="../../../_static/js/index.d3f166471bb80abb5163.js"></script>



<!-- Google Analytics -->
<script>
    window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
    ga('create', 'UA-171697285-1', 'auto');
    ga('set', 'anonymizeIp', true);
    ga('send', 'pageview');
</script>
<script async src='https://www.google-analytics.com/analytics.js'></script>
<!-- End Google Analytics -->

<v-container fluid>
    <v-row no-gutter>
        <v-col cols="12"><footer class="footer" style=" margin-top: 12px;  ">
    <div class="container-md mt-4">
      <div class="row" style="display:flex; justify-content: center">
          <div class="col-12 col-sm-12 col-md-4 col-lg-4 col-xl-3">
            <div class="footerImages">
              <a href="https://europa.eu/"><img src="/_static/images/logo/eu-flag.png" height="40" alt="EU flag"></a>
              <a href="https://www.imi.europa.eu/"><img src="/_static/images/logo/imi-logo.png" height="40" alt="IMI logo"></a>
              <a href="https://efpia.eu"><img src="/_static/images/logo/efpia-logo.png" height="40" alt="EFPIA logo"></a>
            </div>
            <div class="mb-3"> Grant agreement 802750 </div>
          </div>
          <div class="col-5 col-sm-2 col-xs-5 col-md-2 col-lg-2 col-xl-2">
              <ul>
                <li><a href="mailto:faircookbook-ed@elixir-europe.org"> Contact us </a></li>
                <li><a href="https://fairplus-project.eu" target="_blank"> FAIRplus </a> </li>
              </ul>
          </div>
          <div class="col-5 col-sm-2 col-xs-5 col-md-2 col-lg-2 col-xl-2">
              <ul>
                <li> <a href="https://faircookbook.elixir-europe.org/content/recipes/afterword/people.html  "> Contributors </a> </li> </li>
                <li> <a href="https://faircookbook.elixir-europe.org/content/recipes/help.html"> How to contribute </a> </li>
              </ul>
          </div>
          <div class="col-12 col-sm-4 col-xs-12 col-md-3 col-lg-3 col-xl-3 pb-5">
            <a class="mr-3" href="https://elixir-europe.org/" target="_blank"><img src="/_static/images/logo/Elixir/ELIXIR.png" height="50" alt="Elixir logo"></a>
            <a href="https://fairplus-project.eu/"><img src="https://fairplus-project.eu/images/fairplus-logo.png" alt="FAIRplus logo" style="background:white;padding:5px;height:33px;"></a>
          </div>
          <div class="col-12 pb-4">
              Recommended in the
              <a
                href="https://www.imi.europa.eu/sites/default/files/uploads/documents/resources-for-projects/IMI2_OpenAcesGuidelines_Updated2021.pdf"
                target="_blank"
                class="primary--text"
              >
                 IMI/IHI Project Guidelines
              </a> and the
              <a
                href="link to https://ec.europa.eu/info/funding-tenders/opportunities/docs/2021-2027/horizon/wp-call/2023-2024/wp-4-health_horizon-2023-2024_en.pdf"
                target="_blank"
                class="primary--text"
              >
                  Horizon Europe Work Programme for Health
              </a> as the resource for FAIR Data Management guidelines and good practices for the Life Sciences.
          </div>
      </div>
    </div>

    <div class="container-fluid" style="padding:0">
        <div class="row" style="padding:0; margin:0">
            <div class="col-12 purple-dark">
                The FAIR Cookbook is licensed under the
                <a href="https://creativecommons.org/licenses/by/4.0/" target="_blank">
                    Creative Commons Attribution 4.0 International
                </a> (CC BY 4.0) license.
            </div>
        </div>
        <div class="row extra_footer d-none">
            <script> mermaid.init() </script>
        </div>
    </div>

</footer> </v-col>
    </v-row>
</v-container>
<script type="application/javascript">
    const vuetify = new Vuetify({
        theme: {
            themes: {
                light: {
                    primary: '#421C6E',
                    secondary: '#FC7A4A',
                    accent: '#8038D1',
                    error: '#FF5252',
                    info: '#2196F3',
                    success: '#7E0038',
                    warning: '#FFC107'
                }
            }
        },
        icons: { iconfont: 'fa4' }
    })
    const my_vue = new Vue({ el: '#app', vuetify: vuetify })

</script>
  </body>
</html>