
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>11.5.1. Metadata profile for transcriptomics</title>

  <link rel="stylesheet" href="../../../_static/css/index.f658d18f9b420779cfdf24aa0a7e2d77.css">


  <link rel="stylesheet"
    href="../../../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">



  <link rel="stylesheet"
    href="../../../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../../../_static/vendor/lato_latin-ext/1.44.1/index.css">


    <link rel="stylesheet" href="../../../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../../../_static/sphinx-book-theme.e7340bb3dbd8dde6db86f25597f54a1b.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/custom.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/vuetify.min.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/animate.min.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/search-wizard.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/search.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/materialdesignicons.min.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />

  <link rel="preload" as="script" href="../../../_static/js/index.d3f166471bb80abb5163.js">

    <script id="documentation_options" data-url_root="../../../" src="../../../_static/documentation_options.js"></script>
    <script src="../../../_static/jquery.js"></script>
    <script src="../../../_static/underscore.js"></script>
    <script src="../../../_static/doctools.js"></script>
    <script src="../../../_static/togglebutton.js"></script>
    <script src="../../../_static/clipboard.min.js"></script>
    <script src="../../../_static/copybutton.js"></script>
    <script src="../../../_static/js/mermaid.js"></script>
    <script src="../../../_static/js/searchNodal.js"></script>
    <script src="../../../_static/js/vue.min.js"></script>
    <script src="../../../_static/js/mermaid.core.js"></script>
    <script src="../../../_static/js/mermaid.min.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../../../_static/sphinx-book-theme.7d483ff0a819d6edff12ce0b1ead3928.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../../../_static/sphinx-thebe.js"></script>
    <link rel="canonical" href="https://fairplus.github.io/the-fair-cookbook/content/recipes/interoperability/transcriptomics-metadata.html" />
    <link rel="shortcut icon" href="../../../_static/favicon.ico"/>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" />
    <link rel="next" title="11.5.2. Metadata profile validation in RDF" href="covid19-sample-metadata-profile-shex-use-case.html" />
    <link rel="prev" title="11. Creating a metadata profile" href="creating-minimal-metadata-profiles.html" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<meta name="docsearch:language" content="en" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/vuetify/2.6.9/vuetify.js"
        integrity="sha512-cgrSlRKhr/ovJePDoGIdcfZ3wiGcc0fT4Q7hAo0x3P1lyLwUmv2NgJtCo2gzFbxqFjKvNnCfVUjZ1XADm0/ZdQ=="
        crossorigin="anonymous" referrerpolicy="no-referrer">
</script>

<!-- Opengraph tags -->
<meta property="og:url"         content="https://fairplus.github.io/the-fair-cookbook/content/recipes/interoperability/transcriptomics-metadata.html" />
<meta property="og:type"        content="article" />
<meta property="og:title"       content="Metadata profile for transcriptomics" />
<meta property="og:description" content="An online, open and live resource for the Life Sciences with recipes that help you to make and keep data Findable, Accessible, Interoperable and Reusable; in one word FAIR." />
 <meta property="og:image"       content="/_static/images/homepage.png" /> 
<meta name="twitter:card" content="summary" />
  </head>
<body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">



<v-app id="app" class="pa-0">
    <v-container fluid class="pa-0">
        <v-row class="pa-0" no-gutters>
            <v-container fluid class="pa-0">
                <v-row>
                    <div class="topbar container-xl fixed-top pl-0 primary lighten-1 py-2" style="height:65px !important;">
    <div class="topbar-contents row" style="margin: initial !important;">
        <div class="col topbar-main d-flex align-center">
            
                <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                        data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                        aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                        title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                    <i class="fas fa-bars"></i>
                    <i class="fas fa-arrow-left"></i>
                    <i class="fas fa-arrow-up"></i>
                </button>
            

            
                <a href="../../../index.html" style="width: 300px">
                    <img
                        src="../../../_static/cookbook-logo-small.png"
                        class="logo"
                        alt="The FAIR Cookbook"
                        style="width: 100%"
                    >
                </a>
            

            <v-spacer> </v-spacer>
            <div class="mr-4 extraButtons">
                <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger d-flex justify-center align-center" style="height:100%">

    <v-menu offset-y content-class="elevation-0" open-on-hover>
        <template v-slot:activator="{ on, attrs }">
            <v-btn
              id="dropdown-buttons-trigger"
              color="white"
              outlined
              v-bind="attrs"
              v-on="on"
              flat
              solo
              rounded
              :fab="$vuetify.breakpoint.mdAndDown"
              :small="$vuetify.breakpoint.mdAndDown"
              :class="$vuetify.breakpoint.mdAndDown ? 'pa-0' : ''"
            >
                <v-icon :class="$vuetify.breakpoint.lgAndUp ? 'mr-3' : 'mx-1'" left>fab fa-github</v-icon>
                <span v-if="$vuetify.breakpoint.lgAndUp"> GitHub </span>
            </v-btn>
        </template>
        <div class="homepageMenu d-flex flex-column">
            
            <v-btn
              color="grey darken-1"
              dark
              flat
              solo
              rounded
              href="https://github.com/FAIRplus/the-fair-cookbook"
              class="mt-1"
              target="_blank"
            >
                <v-icon class="mr-3" left>fab fa-github</v-icon>
                repository
            </v-btn>
            

            
            <v-btn
              color="grey darken-1"
              dark
              flat
              solo
              rounded
              href="https://github.com/FAIRplus/the-fair-cookbook/issues/new?title=Issue%20on%20page%20%2Fcontent/recipes/interoperability/transcriptomics-metadata.html&body=Your%20issue%20content%20here."
              class="mt-1"
              target="_blank"
            >
                <v-icon class="mr-3" left>fas fa-lightbulb</v-icon>
                open issue
            </v-btn>
            

            
            <v-btn
              color="grey darken-1"
              dark
              flat
              solo
              rounded
              href="https://github.com/FAIRplus/the-fair-cookbook/edit/main/content/recipes/interoperability/transcriptomics-metadata.md"
              class="mt-1"
              target="_blank"
            >
                <v-icon class="mx-3" left>fas fa-pencil-alt</v-icon>
                suggest edit
            </v-btn>
            
        </div>
    </v-menu>
</div>

                <!-- Launch buttons -->

            </div>
                
                    <script>
    Vue.component("Searchfield", {
      data () { return {
        query: null
      } },
      methods: {
        search() {
            if (this.query) window.location.href = `../../search-wizard.html?q=` + this.query
            else window.location.href = `../../search-wizard.html`
        }
      },
      template: `
      <div style="width:250px">
            <v-text-field
              v-model="query"
              append-icon="fa-search"
              label="Search Wizard..."
              dense
              rounded
              single-line
              solo
              hide-details
              class="white primary--text"
              @click:append="search"
              @keyup.enter="search"
            />
      </div>
      `
    })
</script>

<Searchfield> </Searchfield>






                
                
                
        </div>
    </div>
</div>
                </v-row>
                
                    <v-row class="d-flex">
                    
                        <div
                            class=" bd-sidebar site-navigation show"
                            id="site-navigation"
                            style="width:350px"
                        >
                            
                                <div class="nav-drawer d-flex flex-column grey lighten-3" style="height:100%;padding-top: 40px;">

    <div class="testTOC">
        <script>
    Vue.component("Navigation", {
        data() { return { recipes: {} }},
        methods: {
            async getData(){
                const recipes = await fetch('/_static/recipes.json')
                this.recipes = await recipes.json();
                this.parseDom()
            },
            parseDom() {
                const links = Array.prototype.slice.call(document.getElementsByClassName("reference internal"))
                const icon = '<i class="fas fa-play-circle grey lighten-3 secondary--text ml-2"></i>'
                links.forEach(link => {
                    const recipeSplit = link.innerText.split(". ")
                    const recipeName = recipeSplit[recipeSplit.length - 1].trim()
                    recipeSplit.pop()
                    recipeSplit.forEach(pref => { pref = pref.trim() })
                    const prefix = recipeSplit.length > 0 ? recipeSplit.join(".") + "." : null
                    const content = prefix ? `${prefix} ${recipeName}` : recipeName
                    if (Object.keys(this.recipes).includes(recipeName)) {
                        if (this.recipes[recipeName].executable === 'Yes') link.innerHTML = content + icon
                    }
                })
            }
        },
        mounted() { this.getData() }
    })
</script>


<Navigation> </Navigation>

    </div>

    <div class="nav-tree grey px-5 lighten-3 pb-5" style="flex-grow: 1; border-bottom: 1px solid #ccc !important">
        
        
            <nav class="bd-links pt-10" id="bd-docs-nav" aria-label="Main navigation">
    <p class="caption collapsible-parent">
 <span class="caption-text">
  FOREWORD
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/FAIR-cookbook-audience.html">
   Introduction
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/brief-FAIR-principles.html">
   Introducing the FAIR Principles
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/FAIRplus-values.html">
   Reflecting on the ethical values of FAIR
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/fairification-process.html">
   Introducing our FAIRification framework
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/priorization.html">
   Prioritizing projects for FAIRification
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/metadata-fair.html">
   Framing FAIR and the notion of metadata
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/FAIR-and-knowledge-graphs.html">
   Understanding the relation between FAIR and Knowledge Graphs
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/training-datasets.html">
   Training for FAIRification with open or synthetic biomedical datasets
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/public-kg-resource-integration.html">
   Raising Awareness in Public Knowledge Graphs for Life Sciences
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/fair-cro.html">
   Reflecting on Practical Considerations for CROs to play FAIR
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/dpia.html">
   Data Protection Impact Assessment and Data Privacy
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/glossary-fair-terms.html">
   Glossary
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Recipes at a Glance
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../../search-wizard.html">
   All Recipes In a Table
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  FAIR RECIPES
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../findability.html">
   Findability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/identifiers.html">
     1. Introducing unique, persistent identifiers
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/chemical-identities.html">
     2. Creating InChI &amp; SMILES identifiers for chemical structures
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/iupac-names.html">
     3. Creating InChIKeys for IUPAC names
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/identifiers-minids.html">
     4. Minting identifiers with Globus Minid client
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/zenodo-deposition.html">
     5. Depositing to generic repositories - Zenodo use case
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/registeringDatasets.html">
     6. Registering datasets with Wikidata
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/checksum-create.html">
     7. Creating file checksums
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../findability/checksum-validate.html">
     8. Validating checksums to verify file integrity
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../findability/seo.html">
     9. Introducing Search Engine Optimization (SEO)
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../findability/seo/bioschemas-data-page.html">
       9.9.1. Marking up Data pages with Schema.org &amp; Bioschemas for SEO
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../findability/seo/bioschemas-dataset.html">
       9.9.2. Marking up Dataset page with Schema.org &amp; Bioschemas for SEO
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../findability/seo/bioschemas-datacatalog.html">
       9.9.3. Marking up Data Catalogue page with Schema.org &amp; Bioschemas for SEO
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../accessibility.html">
   Accessibility
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../accessibility/sftp.html">
     1. Transferring data with SFTP protocol
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../accessibility/aspera.html">
     2. Downloading data with Aspera protocol
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="../interoperability.html">
   Interoperability
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="registering-swisslipids-identifiers.html">
     1. Registering SwissLipids identifiers in Wikidata
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="identifier-mapping.html">
     2. Interlinking data from different sources
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="bridgedb-recipe.html">
     3. Mapping identifiers with BridgeDb
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="bridgedb/bridgedb-example.html">
       3.11.1. Mapping identifiers using BridgeDb web services
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="introduction-terminologies-ontologies.html">
     4. Introducing terminologies and ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="selecting-ontologies.html">
     5. Selecting terminologies and ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ontology-new-term-request-recipe.html">
     6. Requesting new terms from terminologies and ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ontology-operations-tools.html">
     7. Introducing ontology-related tools and services
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="ontology-robot-recipe.html">
     8. Building an application ontology with ROBOT
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="ontology-robot-recipe/MSIO-robot-build-process.html">
       8.11.1. Building an application ontology for metabolomics - MSIO
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="ontology-robot-recipe/competency-questions.html">
       8.11.2. Defining competency questions for the Ontology ROBOT use case
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ontology-alignment-oxo.html">
     9. Mapping Ontologies with OxO, EBI Ontology Xref Service
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="creating-data-dictionary.html">
     10. Creating a data/variable dictionary
    </a>
   </li>
   <li class="toctree-l2 current active collapsible-parent">
    <a class="reference internal" href="creating-minimal-metadata-profiles.html">
     11. Creating a metadata profile
    </a>
    <ul class="current collapse-ul">
     <li class="toctree-l3 current active">
      <a class="current reference internal" href="#">
       11.5.1. Outlining a metadata profile for transcriptomics
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="covid19-sample-metadata-profile-shex-use-case.html">
       11.5.2. Building a community compliant metadata profile - The Covid19 sample profile use case
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="bioactivity-profile.html">
       11.5.3. Outlining a metadata profile for Bioactivity data
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="from-proprietary-to-open-standard-mzml-exemplar.html">
     12. Converting from proprietary to open format
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="fastq-file-format-validators.html">
     13. Validating file format - FASTQ example
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="rdf-conversion.html">
     14. Inventorying tools for converting data to RDF
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ETL-tools.html">
     15. Surveying extraction, transformation, load (ETL) tools
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="VCF-to-FHIR-conversion.html">
     16. Expressing Clinical Genetic Information as FHIR JSON
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="vcf2fhir/FAIRcookbook-vcf2fhir.html">
       16.9.1. Converting VCF file to FHIR JSON
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="creating-knowledge-graph-from-unstructured-text.html">
     17. Creating knowledge graphs from unstructured text
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="nlp2kg/creating-knowledge-graph-from-text.html">
       17.12.1. unstructured text to graph as executable notebook
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../reusability.html">
   Reusability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/ATI-licensing.html">
     1. Licensing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/ATI_licensing_software.html">
     2. Licensing Software
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/ATI_licensing_data.html">
     3. Licensing Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/expressing-data-use.html">
     4. Declaring data permitted uses
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/provenance.html">
     5. Introducing Provenance Information
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../infrastructure.html">
   Infrastructure
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/id-resolution.html">
     1. Introducing identifier resolution services
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/gupri.html">
     2. Creating resolvable identifiers
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/data-catalog.html">
     3. Building a catalogue of datasets
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/dats-model.html">
     4. Introducing the DATS model
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/data-catalog-deployment.html">
     5. Deploying a data catalogue - The IMI data catalogue example
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../infrastructure/vocabularies.html">
     6. Vocabulary management
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../infrastructure/vocabularies/technical-and-architectural-selection-criteria-of-ontology-lookup-services.html">
       6.1. Introducing vocabulary portals and lookup services
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../infrastructure/vocabularies/Selecting_and_using_ontology_lookup_services.html">
       6.2. Selecting an ontology lookup service
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../infrastructure/vocabularies/UC3_R13_local_ontology_services.html">
       6.3. Deploying EBI Ontology Loopkup Service
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/openrefine-nvt.html">
     7. Using OpenRefine and Karma for FAIRification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../infrastructure/fair-api.html">
     8. Developing FAIR API for the Web
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../assessing-fairness.html">
   Assessment
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../assessing-fairness/fair-assessment-fairevaluator.html">
     1. Assessing with FAIR Evaluator
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../assessing-fairness/fair-assessment-fairshake.html">
     2. Assessing with FAIRshake
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  FAIR Maturity
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../maturity.html">
   Changing culture with the Dataset Maturity Model
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../reusability/miappe.html">
     Improving dataset maturity - the MIAPPE use case
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../maturity/isa-maturity-progression.html">
     Moving through maturity levels with ISA
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../maturity/create-a-simple-ISA-descriptor.html">
       Minimal Data Maturity with ISA - ISA-Tab and free text
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../maturity/isa-api-programmatic-rebuild-of-BII-S-3.html">
       Enhanced Data Maturity with ISA - ISA-JSON and ontology markup
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../maturity/isa-json-conversion-to-rdf-linked-data.html">
       Moving to a semantically typed version - ISA-JSON-LD
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../maturity/isa-as-RO.html">
       Dissemination - Packaging ISA as a Research Object (RO)
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../applied-examples/fair-data-matrix-recipe.html">
     Making an omics data matrix FAIR
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/0-rose-metabolites-Python-data-handling.html">
       FAIRifying Data Matrices - Step1 - Starting material
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/1-rose-metabolites-Python-analysis.html">
       FAIRifying Data Matrices - Step2 - Structuring data
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/2-rose-metabolites-Python-RDF-querying-analysis.html">
       FAIRifying Data Matrices - Step3 - Exploring data with SPARQL
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../applied-examples/fair-data-matrix/3-rose-metabolites-R-analysis.html">
       FAIRifying Data Matrices - Step4 - Integrating data
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/fair-workflows.html">
     Making Computational Workflows FAIR
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  FAIRified Datasets
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../applied-examples.html">
   Applied examples
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/etox.html">
     1. IMI eTox - toxicity datasets
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/nd4bb.html">
     2. IMI nd4bb - chemical activities datasets
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/oncotrack_raw/rawRecipe-oncotrack.html">
     3. Readying IMI Oncotrack - clinical cohort datasets for deposition to EBI Biosamples
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/resolute_raw/resolute.html">
     4. Depositing IMI ReSOLUTE transcriptomics datasets to EBI repositories
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/ehden-ohdsi.html">
     5. Enhancing discoverability of EHDEN OHDSI data with Schema.org markup
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/eubopen-hcs-bioimage.html">
     6. Depositing IMI EUBOPEN High-Content Screening data to EBI BioImage Archive
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../applied-examples/approach-cdisc.html">
     7. Mapping IMI APPROACH datasets to CDISC-SDTM standard
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  AFTERWORD
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../afterword/fair-reflections.html">
   Lessons learned from the FAIR journey and project outlook
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  GLOSSARY
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../introduction/glossary-fair-terms.html">
   Glossary of terms and abbreviations
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  JOIN US
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../community.html">
   Community
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../afterword/people.html">
     Boards and contributors
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../afterword/code_of_conduct.html">
     Code of conduct
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../afterword.html">
     Platform
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../afterword/the-turing-way.html">
       Leveraging the Turing Way Book
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../help.html">
   Contribute
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../afterword/contributing.html">
     1. How to contribute
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/how-to-create-recipe-with-gdoc.html">
     2. Add via GoogleDoc
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/how-to-create-recipe-with-hackmd.html">
     3. Add via HackMD
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/how-to-create-recipe-with-git.html">
     4. Add via Git
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/recipe-template.html">
     5. Git recipe template
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/tips-tricks.html">
     6. Tips and tricks
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../help/myst.html">
     7. Markdown cheatsheet
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>

</nav>
        
        
             <!-- To handle the deprecated key -->

        
    </div>

    <div class="text-center px-5 lighten-3 grey py-2">
        Powered by <a href="https://jupyterbook.org" class="secondary--text">Jupyter Book</a>
    </div>
</div>
                        </div>
                        
                            <main class="bd-content" role="main" style="width:1%; flex-grow:1;">
                                <v-container fluid id="main-content" class="pa-0">
                                    <v-row class="d-flex">
                                        
                                            <v-col class="col-xl-10 col-lg-10 col-md-9 col-12 px-10 pt-10 mt-3">
                                                
  <div class="section" id="metadata-profile-for-transcriptomics">
<span id="fcb-interop-txmetadata"></span><h1><span class="section-number">11.5.1. </span>Metadata profile for transcriptomics<a class="headerlink" href="#metadata-profile-for-transcriptomics" title="Permalink to this headline">¶</a></h1>
<script type="application/ld+json"> {"@context": {"bs": "https://bioschema.org/", "dc": "http://purl.org/dc/terms/", "owl": "http://www.w3.org/2002/07/owl#", "sdo": "https://schema.org/"}, "@type": ["sdo:HowTo", "sdo:LearningResource"], "@id": "https://w3id.org/faircookbook/FCB027", "sdo:name": "Outlining a metadata profile for transcriptomics", "sdo:description": "Outlining a metadata profile for transcriptomics", "sdo:keywords": [], "dc:conformsTo": {"@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE", "@type": "CreativeWork"}, "sdo:inLanguage": ["en-GB"], "sdo:audience": ["Principal Investigator|T4FS_0000441", "Data Manager|T4FS_0000520", "Terminology Manager|T4FS_0000568", "Data Scientist|T4FS_0000566", "Ontologist|T4FS_0000567"], "sdo:isPartOf": [{"@type": "CreativeWork", "@id": "https://fairplus.github.io/the-fair-cookbook/", "sdo:url": "https://fairplus.github.io/the-fair-cookbook/", "sdo:name": "The FAIR Cookbook"}], "sdo:learningResourceType": "Guidance", "sdo:identifier": {"@type": "PropertyValue", "@id": "", "sdo:name": "FCB027", "dso:url": "https://w3id.org/faircookbook/FCB027", "owl:sameAs": "https://identifiers.org/fcb:FCB027"}, "sdo:license": {"@type": "CreativeWork", "@id": "https://creativecommons.org/licenses/by/4.0/", "sdo:url": "https://creativecommons.org/licenses/by/4.0/", "sdo:name": "CC-BY-4.0"}, "sdo:timeRequired": "PT0H30M0S", "sdo:totalTime": "PT0H30M0S", "sdo:step": [{"@type": "sdo:HowToSection", "sdo:position": "1", "sdo:name": "Overview", "sdo:url": "https://w3id.org/faircookbook/FCB027#main-objectives", "sdo:itemListElement": [{"@type": "sdo:HowToStep", "sdo:position": "1", "sdo:name": "", "sdo:itemListElement": [{"@type": "sdo:HowToDirection", "sdo:position": "1", "sdo:name": "step-by-step section", "sdo:text": "Follow the step-by-step section.", "sdo:url": "https://w3id.org/faircookbook/FCB027#main-objectives"}, {"@type": "sdo:HowToTip", "sdo:position": "2", "sdo:name": "Reporting an issue", "sdo:text": "Use our GitHub issue tracker to report any error. Thanks!", "sdo:url": "https://github.com/FAIRplus/the-fair-cookbook/issues/new/choose"}]}]}, {"@type": "sdo:HowToSection", "sdo:position": "2", "sdo:name": "Conclusion", "sdo:url": "https://w3id.org/faircookbook/FCB027#conclusion", "sdo:itemListElement": [{"@type": "sdo:HowToStep", "sdo:position": "1", "sdo:name": "", "sdo:itemListElement": [{"@type": "sdo:HowToDirection", "sdo:position": "1", "sdo:name": "conclusion section", "sdo:text": "What to read next?", "sdo:url": "https://w3id.org/faircookbook/FCB027#conclusion"}, {"@type": "sdo:HowToTip", "sdo:position": "2", "sdo:name": "Reporting an issue", "sdo:text": "Use our GitHub issue tracker to report any error. Thanks!", "sdo:url": "https://github.com/FAIRplus/the-fair-cookbook/issues/new/choose"}]}]}]} </script>
<div class="card my-10 recipe d-flex">
<div class="card-header purple-dark text--white">
<div class="overview">
<div class="overviewLabel">Recipe Overview</div>
<div class="overviewContent">
<div class="item"><div class="itemContainer">
<i class="itemIcon fa fa-clock fa-2x text--white"></i>
<div class="itemContent">
<div class="label"> Reading Time </div>
30 minutes
</div>
</div></div>
<div class="item"><div class="itemContainer">
<i class="itemIcon fa fa-play-circle fa-2x text--white"></i>
<div class="itemContent">
<div class="label"> Executable Code </div>
No
</div>
</div></div>
<div class="item" style="margin-bottom:10px"><div class="itemContainer">
<i class="itemIcon fa fa-fire fa-2x text--white" style="margin-left:4px;margin-right: 5px;"></i>
<div class="itemContent">
<div class="label" style="position:relative; left:-6px">Difficulty</div>
<i class="fa fa-fire"></i>
<i class="fa fa-fire"></i>
<i class="fa fa-fire"></i>
<i class="fa fa-fire" style="color:grey"></i>
<i class="fa fa-fire" style="color:grey"></i>
</div>
</div></div>
</div></div>
</div>
<div class="card-body text--purple-dark">
<div class="title">
<div style="flex-grow:1; margin-right:5px">Outlining a metadata profile for transcriptomics</div>
<img src="https://fairplus.github.io/the-fair-cookbook/_static/images/fairplus-mini.png" alt="FAIRPlus logo"> </img>
</div>
<div class="section">
<i class="sectionIcon fa fa-laptop fa-2x"></i><div class="sectionContent">
<div class="label">Recipe Type</div>
<div class="sectionValue">Guidance</div>
</div>
</div>
<div class="section" style="flex-grow: 1;">
<i class="sectionIcon fa fa-users fa-2x"></i><div class="sectionContent">
<div class="label">Audience</div>
<div class="sectionValue"><a href="http://purl.obolibrary.org/obo/T4FS_0000441">Principal Investigator</a>, <a href="http://purl.obolibrary.org/obo/T4FS_0000520">Data Manager</a>, <a href="http://purl.obolibrary.org/obo/T4FS_0000568">Terminology Manager</a>, <a href="http://purl.obolibrary.org/obo/T4FS_0000566">Data Scientist</a>, <a href="http://purl.obolibrary.org/obo/T4FS_0000567">Ontologist</a></div>
</div>
</div>
<div class="section" style="flex-grow: 1;">
<svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="signal" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 600" class="bars level-3">
<path class="bar5" d="M216 288h-48c-8.84 0-16 7.16-16 16v192c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V304c0-8.84-7.16-16-16-16zM88 384H40c-8.84 0-16 7.16-16 16v96c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16v-96c0-8.84-7.16-16-16-16zm256-192h-48c-8.84 0-16 7.16-16 16v288c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V208c0-8.84-7.16-16-16-16zm128-96h-48c-8.84 0-16 7.16-16 16v384c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V112c0-8.84-7.16-16-16-16zM600 0h-48c-8.84 0-16 7.16-16 16v480c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V16c0-8.84-7.16-16-16-16z"/>
<path class="bar4" d="M216 288h-48c-8.84 0-16 7.16-16 16v192c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V304c0-8.84-7.16-16-16-16zM88 384H40c-8.84 0-16 7.16-16 16v96c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16v-96c0-8.84-7.16-16-16-16zm256-192h-48c-8.84 0-16 7.16-16 16v288c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V208c0-8.84-7.16-16-16-16zm128-96h-48c-8.84 0-16 7.16-16 16v384c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V112c0-8.84-7.16-16-16-16zM600" />
<path class="bar3" d="M216 288h-48c-8.84 0-16 7.16-16 16v192c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V304c0-8.84-7.16-16-16-16zM88 384H40c-8.84 0-16 7.16-16 16v96c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16v-96c0-8.84-7.16-16-16-16zm256-192h-48c-8.84 0-16 7.16-16 16v288c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V208c0-8.84-7.16-16-16-16zm128-96h-48c-8.84" />
<path class="bar1" d="M216 288h-48c-8.84 0-16 7.16-16 16v192c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V304c0-8.84-7.16-16-16-16zM88 384H40c-8.84 0-16 7.16-16 16v96c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16v-96c0-8.84-7.16-16-16-16zm256-192h-48c-8.84"/>
<path class="bar2" d="M216 288h-48c-8.84 0-16 7.16-16 16v192c0 8.84 7.16 16 16 16h48c8.84 0 16-7.16 16-16V304c0-8.84-7.16-16-16-16zM88 384H40c-8.84"/>
</svg>
<div class="sectionContent">
<div class="label">Maturity Level & Indicator</div>
<div class="sectionValue"><v-chip-group><v-chip href="https://w3id.org/Data-Maturity/DSM-3-C1" class="secondary white--text" target="_blank" label>DSM-3-C1</v-chip></v-chip-group></div>
<div class="tooltip">hover me
<span class="tooltiptext">Tooltip text</span></div>
</div>
</div><div class="card-footer text--orange sphinx-bs.badge.badge-primary"> Cite me with 
<a href="https://w3id.org/faircookbook/FCB027" class="text--purple-dark">FCB027
</a></div>
</div>
</div>
<div class="section" id="main-objectives">
<h2><span class="section-number">11.5.1.1. </span>Main Objectives:<a class="headerlink" href="#main-objectives" title="Permalink to this headline">¶</a></h2>
<p>The main purpose of this recipe is:</p>
<blockquote>
<div><p>To provide guidance on the minimum set of metadata and semantics required to describe any transcriptomics experiments,
from standard case-control to dosage-response designs, and from microarrays to single cell RNA sequencing.</p>
</div></blockquote>
<div class="section" id="who-is-this-recipe-aimed-at">
<h3><span class="section-number">11.5.1.1.1. </span>Who is this recipe aimed at?<a class="headerlink" href="#who-is-this-recipe-aimed-at" title="Permalink to this headline">¶</a></h3>
<p>This document is aimed at anyone interested in the metadata, and specifically the ontology annotations, required to capture variables and experimental factors related to a transcriptomics experiment. General knowledge of transcriptomics experiments would be beneficial. No specific technical knowledge is required.</p>
</div>
</div>
<hr class="docutils" />
<div class="section" id="transcriptomics-data-model">
<h2><span class="section-number">11.5.1.2. </span>Transcriptomics Data model<a class="headerlink" href="#transcriptomics-data-model" title="Permalink to this headline">¶</a></h2>
<p>Large sections of any transcriptomics data model are not unique to transcriptomics experiments but are common to a wide range of experiments in the biomedical domain. This includes general project- and sample-level information, which will be briefly covered in this recipe but will also be covered in more depth elsewhere.</p>
<p>Where possible, metadata should be mapped to ontologies to maximise the FAIRness of any dataset. This recipe will suggest possible ontologies for metadata fields where these are available. These lists may however not be exhaustive as new ontologies emerge regularly.</p>
<div class="section" id="existing-standards-and-checklists">
<h3><span class="section-number">11.5.1.2.1. </span>Existing standards and checklists<a class="headerlink" href="#existing-standards-and-checklists" title="Permalink to this headline">¶</a></h3>
<p>A set of well-established standards and minimum metadata checklists exist for various aspects of transcriptomics. They include:</p>
<p>Minimum Information About a Microarray Experiment (MIAME) - MIAME is intended to specify all the information necessary for an unambiguous interpretation of a microarray experiment, and potentially to reproduce it. MIAME defines the content but not the format for this information. The MIAME standard has been in existence for over 20 years and has been widely adopted across the scientific community. The data models of the major transcriptomics repositories such as ArrayExpress, the Expression Atlas and the Gene Expression Omnibus (GEO) are MIAME-compliant.</p>
<p>Minimum Information about a high-throughput nucleotide SEQuencing Experiment (MINSEQE) - MINSEQE describes the minimum metadata that is needed to enable the unambiguous interpretation and facilitate reproduction of the results of the experiment. By analogy to the MIAME guidelines for microarray experiments, adherence to the MINSEQE guidelines will improve integration of multiple experiments across different modalities, thereby maximising the value of high-throughput research. MINSEQE has been integrated into a number of transcriptomics and sequencing archives.</p>
<p>Functional Annotation of Animal Genomes (FAANG) - FAANG provides a set of orthogonal standards for the capture of well-structured metadata for experiments, samples and analyses in the animal genomics domain. The FAANG standards support the MIAME and MINSEQE guidelines, and aim to convert them to a concrete specification.</p>
<p>Human Cell Atlas Metadata (HCA-Metadata) - the HCA metadata model provides a concrete specification for capturing the metadata for single cell sequencing experiments. It is based on the three standards above, while focusing on the specific requirements of the single cell space.</p>
</div>
<div class="section" id="minimum-metadata-vs-desirable-metadata">
<h3><span class="section-number">11.5.1.2.2. </span>Minimum metadata vs desirable metadata<a class="headerlink" href="#minimum-metadata-vs-desirable-metadata" title="Permalink to this headline">¶</a></h3>
<p>A minimum metadata set constitutes metadata items that always need to be supplied with any experimental data. For validation purposes, these should not be omittable. Desirable or recommended metadata items on the other hand should be supplied if available but will not cause a validation failure if absent.</p>
</div>
<div class="section" id="common-metadata">
<h3><span class="section-number">11.5.1.2.3. </span>Common metadata<a class="headerlink" href="#common-metadata" title="Permalink to this headline">¶</a></h3>
<p>Common metadata include any information that is not specific to transcriptomics experiments but applies to most experiments in the biomedical domain. They include:</p>
<ul class="simple">
<li><p>Project level metadata</p></li>
<li><p>Common sample level metadata such as species, tissue, cell type etc.</p></li>
</ul>
<div class="section" id="suggested-metadata-fields">
<h4><span class="section-number">11.5.1.2.3.1. </span>Suggested metadata fields<a class="headerlink" href="#suggested-metadata-fields" title="Permalink to this headline">¶</a></h4>
<p>The following table contains a non-exhaustive list of suggested minimum metadata fields for biological samples. The collection is based on a range of existing metadata standards, including MIAME, MINSEQE, FAANG and HCA. Fields were included if they occurred in at least two of the standards.</p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Metadata field</p></th>
<th class="head"><p>Required?</p></th>
<th class="head"><p>Definition</p></th>
<th class="head"><p>Comment</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>unique ID</p></td>
<td><p>required</p></td>
<td><p>Identifier for a sample that is at least unique within the project</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>sample type</p></td>
<td><p>required</p></td>
<td><p>The type of the collected specimen, eg tissue biopsy, blood draw or throat swab</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. OBI or EFO</p></td>
</tr>
<tr class="row-even"><td><p>species</p></td>
<td><p>required</p></td>
<td><p>The primary species of the specimen, preferably the taxonomic identifier</p></td>
<td><p>This may not be the same as the “host” organism, eg in the case of a PDX tissue sample, the host may be a mouse but the tissue may be human. Ontology field - NCBITaxonomy</p></td>
</tr>
<tr class="row-odd"><td><p>tissue/organism part</p></td>
<td><p>required</p></td>
<td><p>The tissue from which the sample was taken</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. Uberon</p></td>
</tr>
<tr class="row-even"><td><p>disease</p></td>
<td><p>required</p></td>
<td><p>Any diseases that may affect the sample</p></td>
<td><p>This may not necessarily be the same as the host’s disease, eg healthy brain tissue might be collected from a host with type II diabetes while cirrhotic liver tissue might be collected from an otherwise healthy individual. Ontology field - e.g. MONDO or DO</p></td>
</tr>
<tr class="row-odd"><td><p>sex</p></td>
<td><p>required</p></td>
<td><p>The biological/genetic sex of the sample</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. PATO</p></td>
</tr>
<tr class="row-even"><td><p>development stage</p></td>
<td><p>required</p></td>
<td><p>The developmental stage of the sample</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. Uberon or Hsadpdv; species dependent</p></td>
</tr>
<tr class="row-odd"><td><p>collection date</p></td>
<td><p>required</p></td>
<td><p>The date on which the sample was collected, in a standardised format</p></td>
<td><p>Collection date in combination with other fields such as location and disease may be sufficient to de-anonymise a sample</p></td>
</tr>
<tr class="row-even"><td><p>external accessions</p></td>
<td><p>recommended</p></td>
<td><p>Accession numbers from any external resources to which the sample was submitted</p></td>
<td><p>eg Biosamples, Biostudies</p></td>
</tr>
<tr class="row-odd"><td><p>strain</p></td>
<td><p>recommended</p></td>
<td><p>Strain of the species from which the sample was collected, if applicable</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. NCBITaxonomy</p></td>
</tr>
<tr class="row-even"><td><p>ancestry/ethnicity</p></td>
<td><p>recommended</p></td>
<td><p>Ancestry or ethnic group of the individual from which the sample was collected</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. HANCESTRO</p></td>
</tr>
<tr class="row-odd"><td><p>age</p></td>
<td><p>recommended</p></td>
<td><p>Age of the organism from which the sample was collected</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p>age unit</p></td>
<td><p>recommended</p></td>
<td><p>Unit of the value of the age field</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. UO</p></td>
</tr>
<tr class="row-odd"><td><p>BMI</p></td>
<td><p>recommended</p></td>
<td><p>Body mass index of the individual from which the sample was collected</p></td>
<td><p>Only applies to human samples</p></td>
</tr>
<tr class="row-even"><td><p>treatment category</p></td>
<td><p>recommended</p></td>
<td><p>Treatments that the sample might have undergone after collection</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. OBI, NCIt or OGMS</p></td>
</tr>
<tr class="row-odd"><td><p>cell type</p></td>
<td><p>recommended</p></td>
<td><p>The cell type(s) known or selected to be present in the sample</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. CL</p></td>
</tr>
<tr class="row-even"><td><p>growth conditions</p></td>
<td><p>recommended</p></td>
<td><p>Features relating to the growth and/or maintenance of the sample</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>genetic variation</p></td>
<td><p>recommended</p></td>
<td><p>Any relevant genetic differences from the specimen or sample to the expected genomic information for this species, eg abnormal chromosome counts, major translocations or indels</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p>sample collection technique</p></td>
<td><p>recommended</p></td>
<td><p>The technique used to collect the specimen, eg blood draw or surgical resection</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. EFO or OBI</p></td>
</tr>
<tr class="row-odd"><td><p>phenotype</p></td>
<td><p>recommended</p></td>
<td><p>Any relevant (usually abnormal) phenotypes of the specimen or sample</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. HP or MP; species dependent</p></td>
</tr>
<tr class="row-even"><td><p>cell cycle</p></td>
<td><p>recommended</p></td>
<td><p>The cell cycle phase of the sample (for synchronized growing cells or a single-cell sample), if known</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. GO</p></td>
</tr>
<tr class="row-odd"><td><p>cell location</p></td>
<td><p>recommended</p></td>
<td><p>The cell location from which genetic material was collected (usually either nucleus or mitochondria)</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. GO</p></td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section" id="assay-metadata">
<h3><span class="section-number">11.5.1.2.4. </span>Assay metadata<a class="headerlink" href="#assay-metadata" title="Permalink to this headline">¶</a></h3>
<p>Assay-level metadata covers any metadata directly related to the preparation of the biomaterial undergoing the assay and the process of performing the assay. Most, though not all, of this metadata is specific to transcriptomics. Examples include:</p>
<ul class="simple">
<li><p>Library information, including what sample the library was originally derived from and whether any replicates are biological or technical</p></li>
<li><p>Process (wet experiment) metadata</p></li>
<li><p>Technology type (e.g. bulk RNA-Seq, scRNA-Seq, CITE-Seq)</p></li>
<li><p>Instrument metadata</p></li>
<li><p>Other assay-specific metadata</p></li>
<li><p>QC information</p></li>
<li><p>Workflow metadata</p></li>
</ul>
<p>The figure below shows an excerpt of the Human Cell Atlas json schema for describing the process of sequencing.</p>
<details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="human-cell-atlas-sequencing-json-schema">
<img alt="human cell atlas sequencing json schema." src="../../../_images/hca_sequencing_json_schema.png" />
<p class="caption"><span class="caption-number">Fig. 11.2 </span><span class="caption-text">human cell atlas sequencing json schema.</span><a class="headerlink" href="#human-cell-atlas-sequencing-json-schema" title="Permalink to this image">¶</a></p>
</div>
</div>
</details><div class="admonition note">
<p class="admonition-title">Note</p>
<p>Note the key attributes for which controlled vocabulary may be directly specified as a:</p>
<p><code class="docutils literal notranslate"><span class="pre">json</span> <span class="pre">schema</span> <span class="pre">ENUM</span></code></p>
<p>as found at line 79, or may be specified via a reference to a dedicated ontology schema as in:</p>
<p><code class="docutils literal notranslate"> <span class="pre">&quot;$ref&quot;:</span> <span class="pre">&quot;module/ontology/biological_macromolecule_ontology.json&quot;</span></code></p>
<p>as found at line 82.</p>
</div>
<details class="sphinx-bs dropdown card mb-3" open="open">
<summary class="summary-title card-header">
<svg viewBox="0 0 36 24" width="36" height="16" xmlns="http://www.w3.org/2000/svg"
    class="octicon no-title" aria-hidden="true">
  <g xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
    <circle cx="0" cy="12" r="6"></circle>
    <circle cx="18" cy="12" r="6"></circle>
    <circle cx="36" cy="12" r="6"></circle>
  </g>
</svg><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="figure align-default" id="hca-library-amplification-ontology-json-schema">
<img alt="hca_library_amplification_ontology_json_schema." src="../../../_images/hca_library_amplification_ontology_json_schema.png" />
<p class="caption"><span class="caption-number">Fig. 11.3 </span><span class="caption-text">human cell atlas sequencing json schema.</span><a class="headerlink" href="#hca-library-amplification-ontology-json-schema" title="Permalink to this image">¶</a></p>
</div>
</div>
</details><div class="admonition note">
<p class="admonition-title">Note</p>
<p>Note how the <code class="docutils literal notranslate"><span class="pre">ontology</span></code> element defines a <code class="docutils literal notranslate"><span class="pre">graph</span> <span class="pre">restriction</span></code> pointing to a branch in a semantic resource, (EFO in this case).</p>
</div>
<div class="section" id="id1">
<h4><span class="section-number">11.5.1.2.4.1. </span>Suggested metadata fields<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h4>
<p>The following table contains a non-exhaustive list of suggested minimum metadata fields for assays. The collection is based on a range of existing metadata standards, including MIAME, MINSEQE and HCA. This list can and should be further broken down based on specific technologies used, such as microarrays or whole genome sequencing.</p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Metadata field</p></th>
<th class="head"><p>Required?</p></th>
<th class="head"><p>Definition</p></th>
<th class="head"><p>Comment</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>unique ID</p></td>
<td><p>required</p></td>
<td><p>Identifier for the assay that is at least unique within the project</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>experiment type</p></td>
<td><p>required</p></td>
<td><p>The type of experiment performed, eg ATAC-seq or seqFISH</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. EFO or OBI</p></td>
</tr>
<tr class="row-even"><td><p>extracted nucleic acid/material type</p></td>
<td><p>required</p></td>
<td><p>The type of material that was extracted from the sample, eg polyA RNA</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. ChEBI or EFO</p></td>
</tr>
<tr class="row-odd"><td><p>platform</p></td>
<td><p>required</p></td>
<td><p>The type of instrument used to perform the assay, eg Illumina HiSeq 4000 or Fluidigm C1 microfluidics platform</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. EFO or OBI</p></td>
</tr>
<tr class="row-even"><td><p>nucleic acid extraction method</p></td>
<td><p>required</p></td>
<td><p>Technique used to extract the nucleic acid from the cell</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. EFO or OBI</p></td>
</tr>
<tr class="row-odd"><td><p>cDNA library amplication method</p></td>
<td><p>required</p></td>
<td><p>Technique used to amplify a cDNA library</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. EFO or OBI</p></td>
</tr>
<tr class="row-even"><td><p>array or sequencing method</p></td>
<td><p>required</p></td>
<td><p>The array or sequencing technology used - may be the same as <code class="docutils literal notranslate"><span class="pre">experiment</span> <span class="pre">type</span></code> or can be a more specific term</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. EFO or OBI</p></td>
</tr>
<tr class="row-odd"><td><p>biological or technical replicate</p></td>
<td><p>required</p></td>
<td><p>Information whether the sample on which the assay was performed was biological or technical replicate.</p></td>
<td><p>boolean or CV</p></td>
</tr>
<tr class="row-even"><td><p>end bias</p></td>
<td><p>required</p></td>
<td><p>The type of tag or end bias the library has, eg 3 prime tag or 5 prime end bias</p></td>
<td><p>standardised field or ontology</p></td>
</tr>
<tr class="row-odd"><td><p>external accessions</p></td>
<td><p>recommended</p></td>
<td><p>Accession numbers from external resources to which assay or protocol information was submitted</p></td>
<td><p>eg <a class="reference external" href="http://protocols.io">protocols.io</a>, AE</p></td>
</tr>
<tr class="row-even"><td><p>instrument model</p></td>
<td><p>required</p></td>
<td><p>The specific instrument on which the assay was performed. Essential for QC purposes.</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. EFO or OBI</p></td>
</tr>
<tr class="row-odd"><td><p>assay start time</p></td>
<td><p>recommended</p></td>
<td><p>The exact time at which the assay was started</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p>assay end time</p></td>
<td><p>recommended</p></td>
<td><p>The exact time at which the assay was completed</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>assay duration</p></td>
<td><p>recommended</p></td>
<td><p>The duration, in a relevant time unit (eg minutes or hours), of the assay from start to finish</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p>array quality</p></td>
<td><p>recommended</p></td>
<td><p>The overall quality of the array</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>chemical compound</p></td>
<td><p>recommended</p></td>
<td><p>Any relevant chemical compounds used in the assay</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. ChEBI</p></td>
</tr>
<tr class="row-even"><td><p>labeling molecule used</p></td>
<td><p>recommended</p></td>
<td><p>The type of labeling molecule used in an array-based experiment</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. ChEBI</p></td>
</tr>
<tr class="row-odd"><td><p>spike-in kit used</p></td>
<td><p>recommended</p></td>
<td><p>Information about the spike-in kit used during sequencing library preparation</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p>cDNA primer</p></td>
<td><p>recommended</p></td>
<td><p>Type of primer used for cDNA synthesis from RNA, eg polyA or random</p></td>
<td><p>standardised field or ontology</p></td>
</tr>
<tr class="row-odd"><td><p>library strandedness</p></td>
<td><p>recommended</p></td>
<td><p>The strandedness of the cDNA library</p></td>
<td><p>standardised field or ontology</p></td>
</tr>
<tr class="row-even"><td><p>cell quality</p></td>
<td><p>recommended</p></td>
<td><p>Information about the quality of a single cell such as morphology or percent viability</p></td>
<td><p>standardised field or ontology</p></td>
</tr>
<tr class="row-odd"><td><p>cell barcode</p></td>
<td><p>recommended</p></td>
<td><p>Information about the cell identifier barcode used to tag individual cells in single cell sequencing</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p>UMI barcode</p></td>
<td><p>recommended</p></td>
<td><p>Information about the Unique Molecular Identifier barcodes used to tag DNA fragments</p></td>
<td><p></p></td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section" id="analysis-metadata">
<h3><span class="section-number">11.5.1.2.5. </span>Analysis metadata<a class="headerlink" href="#analysis-metadata" title="Permalink to this headline">¶</a></h3>
<p>Analysis-level metadata includes any metadata related to the files that come out of the experiment, from the sequencing or imaging files generated directly by the machine to files generated during the various stages of processing and analysis, as well as details of any analyses performed. It is very important to always capture versions of software and reference genomes used in order to allow accurate replication of results. Analysis metadata includes</p>
<ul class="simple">
<li><p>Type of analysis</p></li>
<li><p>File formats, e.g. BAM, fastq or gene count</p></li>
<li><p>File location e.g. URL</p></li>
<li><p>Summarisation of data, e.g. enrichment analysis</p></li>
</ul>
<div class="section" id="id2">
<h4><span class="section-number">11.5.1.2.5.1. </span>Suggested metadata fields<a class="headerlink" href="#id2" title="Permalink to this headline">¶</a></h4>
<p>The following table contains a non-exhaustive list of suggested minimum metadata fields for analyses. The collection is based on a range of existing metadata standards, including MIAME, MINSEQE and HCA. This list can and should be further broken down based on the specific analysis type (primary, secondary or teriatry analysis, meta-analysis etc)</p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Metadata field</p></th>
<th class="head"><p>Required?</p></th>
<th class="head"><p>Definition</p></th>
<th class="head"><p>Comment</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>analysis type</p></td>
<td><p>required</p></td>
<td><p>The type of analysis performed, eg genome assembly or variant calling</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. EFO, OBI or EDAM</p></td>
</tr>
<tr class="row-odd"><td><p>computational method</p></td>
<td><p>required</p></td>
<td><p>The specific computational method or algorithm used as part of the analysis</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. EFO or EDAM</p></td>
</tr>
<tr class="row-even"><td><p>normalisation strategy</p></td>
<td><p>required</p></td>
<td><p>The approach used to normalise the data</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. EFO or EDAM</p></td>
</tr>
<tr class="row-odd"><td><p>file format</p></td>
<td><p>required</p></td>
<td><p>The file format in which the analysis is provided</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">ontology</span> <span class="pre">field</span></code> - e.g. EDAM</p></td>
</tr>
<tr class="row-even"><td><p>file storage location</p></td>
<td><p>required</p></td>
<td><p>The location in which the data files are stored</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>software package</p></td>
<td><p>recommended</p></td>
<td><p>The software package used for data analysis</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p>software version</p></td>
<td><p>recommended</p></td>
<td><p>The exact version number of the software package</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>analysis date</p></td>
<td><p>recommended</p></td>
<td><p>The date on which the analysis was performed</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p>read index</p></td>
<td><p>recommended</p></td>
<td><p>The sequencing read a specific file represents, eg read1 or index1</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>read length</p></td>
<td><p>recommended</p></td>
<td><p>The length of a sequenced read in this file, in nucleotides.</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p>assembly type</p></td>
<td><p>recommended</p></td>
<td><p>The assembly type of the genome reference file, eg primary, complete or patch assembly.</p></td>
<td><p>standardised field or ontology</p></td>
</tr>
<tr class="row-odd"><td><p>reference genome version</p></td>
<td><p>recommended</p></td>
<td><p>The genome version of the reference file.</p></td>
<td><p></p></td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="section" id="ontologies-for-transcriptomics-data">
<h2><span class="section-number">11.5.1.3. </span>Ontologies for transcriptomics data<a class="headerlink" href="#ontologies-for-transcriptomics-data" title="Permalink to this headline">¶</a></h2>
<p>While it is essential that any transcriptomics metadata be annotated with ontology terms wherever possible, there is no absolute set of ontologies that must be used above all others. There is however a consensus set of ontologies and other standardised resources that are commonly used in transcriptomics metadata, including in the main data archives. The most commonly used ontologies and fields they apply to are listed below. This table represents an absolute minimum of ontology annotations that should be included in a transcriptomics metadata set for it to be considered as FAIR. Not all fields suggested for ontology annotation in the previous section are repeated here for this reason.</p>
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p>Data type</p></th>
<th class="head"><p>Ontology/Entity sources</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Notes</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Species</p></td>
<td><p>NCBI taxonomy Scientific name + ID</p></td>
<td><p>Ontology</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>Tissue</p></td>
<td><p>Uberon term and Id</p></td>
<td><p>Ontology</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p>Cell type</p></td>
<td><p>CL term and Id</p></td>
<td><p>Ontology</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>Disease</p></td>
<td><p>MONDO, DO or MeSH</p></td>
<td><p>Ontology</p></td>
<td><p>no single solution - options vary depending on resource and individual requirements</p></td>
</tr>
<tr class="row-even"><td><p>Phenotype/Trait</p></td>
<td><p>HPO (human),  MP(other mammals), various others for model organisms (yeast, zebrafish, Xenopus, C. elegans)</p></td>
<td><p>Ontology</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>Experiment Type</p></td>
<td><p>EFO, OBI</p></td>
<td><p>Ontology</p></td>
<td><p>e.g. RNASeq, CITESeq etc. -</p></td>
</tr>
<tr class="row-even"><td><p>Cell location/cycle</p></td>
<td><p>GO</p></td>
<td><p>Ontology</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>Developmental stage</p></td>
<td><p>HSAPDV/Uberon</p></td>
<td><p>Ontology</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p>Chemical compound</p></td>
<td><p>ChEBI</p></td>
<td><p>Ontology</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>Chemical compound</p></td>
<td><p>UniChem, SMILE, InChi</p></td>
<td><p>Entity</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p>Gene/protein</p></td>
<td><p>ENSEMBL, ENTREZ_GENE, UNIPROT, HGNC ID, INSDC</p></td>
<td><p>Entity</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>Metabolites</p></td>
<td><p>MetaboLights compound accession, ChEBI</p></td>
<td><p>Entity</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p>Nucleotide reference sequence</p></td>
<td><p>ReqSeq</p></td>
<td><p>Entity</p></td>
<td><p></p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="conclusion">
<h2><span class="section-number">11.5.1.4. </span>Conclusion<a class="headerlink" href="#conclusion" title="Permalink to this headline">¶</a></h2>
<p>Using common transcriptomics metadata standards, in particular the fields listed above as guidance, it is possible to
easily define a comprehensive metadata template to capture all the experimental variables to describe any
transcriptomics experiment in a FAIR-compliant way. A generic step-by-step guide for designing a metadata
template is provided <a class="reference internal" href="creating-minimal-metadata-profiles.html"><span class="doc std std-doc">here</span></a></p>
<details class="sphinx-bs dropdown card mb-3">
<summary class="summary-title card-header">
<strong>References</strong><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
</div>
</details><div class="section" id="what-to-read-next">
<h3><span class="section-number">11.5.1.4.1. </span>What to read next?<a class="headerlink" href="#what-to-read-next" title="Permalink to this headline">¶</a></h3>
<div class="card w-100 border-2 col-md-4 p-0 docutils">
<div class="card-header bg-primary pa_dark docutils text-center">
<img alt="FAIRsharing logo" height="40px" src="/_static/images/logo/fairsharing-logo.png">
</div>
<div class="card-body docutils">
<p class="mb-0">FAIRsharing records appearing in this recipe:</p>
<ul><li><a href="https://doi.org/10.25504/FAIRsharing.6k0kwd" target="_blank">ArrayExpress</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.hza1ec" target="_blank">Binary Alignment Map Format (BAM)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.mtjvme" target="_blank">BioStudies</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.j9y503" target="_blank">Cell Ontology (CL)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.62qk8w" target="_blank">Chemical Entities of Biological Interest (ChEBI)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.a6r7zs" target="_blank">EMBRACE Data and Methods Ontology (EDAM)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.fx0mw7" target="_blank">Ensembl</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.1gr4tz" target="_blank">Experimental Factor Ontology (EFO)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.f5zx00" target="_blank">Expression Atlas</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.5hc8vt" target="_blank">Gene Expression Omnibus (GEO)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.6xq0ee" target="_blank">Gene Ontology (GO)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.29we0s" target="_blank">HUGO Gene Nomenclature Committee (HGNC)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.rja8qp" target="_blank">Human Ancestry Ontology (HANCESTRO)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.c6vhm3" target="_blank">Human Developmental Stages Ontology (HSAPDV)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.kg1x4z" target="_blank">Mammalian Phenotype Ontology (MP)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.kkdpxe" target="_blank">MetaboLights (MTBLS)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.a55z32" target="_blank">Minimal Information about a high throughput SEQuencing Experiment (MINSEQE)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.32b10v" target="_blank">Minimum Information About a Microarray Experiment (MIAME)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.b2979t" target="_blank">Monarch Disease Ontology (MONDO)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.fj07xj" target="_blank">NCBI Taxonomy (NCBITAXON)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.4cvwxa" target="_blank">NCI Thesaurus (NCIt)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.284e1z" target="_blank">Ontology for Biomedical Investigations (OBI)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.rvz0m9" target="_blank">Ontology for General Medical Science (OGMS)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.WWI10U" target="_blank">The FAIR Principles (FAIR)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.mjnypw" target="_blank">Units Ontology (UO)</a></li><li><a href="https://doi.org/10.25504/FAIRsharing.132b10" target="_blank">protocols.io</a></li></ul>
</div>
</div>
</div>
</div>
<div class="section" id="authors">
<h2><span class="section-number">11.5.1.5. </span>Authors<a class="headerlink" href="#authors" title="Permalink to this headline">¶</a></h2>
<!-- TODO clarify the contribution of Karsten and Peter -->
<details class="sphinx-bs dropdown card mb-3">
<summary class="summary-title card-header">
<strong>Authors</strong><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<table class="colwidths-auto table">
<thead>
<tr class="row-odd"><th class="head"><p class="card-text">Name</p></th>
<th class="head"><p class="card-text">ORCID</p></th>
<th class="head"><p class="card-text">Affiliation</p></th>
<th class="head"><p class="card-text">Type</p></th>
<th class="head"><p class="card-text">ELIXIR Node</p></th>
<th class="head"><p class="card-text">Contribution</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p class="card-text"><div class="firstCol"><a target="_blank" href="https://github.com/{github_handle}"><img class="avatar-style" src="https://avatars.githubusercontent.com/daniwelter"></img><div class="d-block">Danielle Welter</div></a></div></p></td>
<td><p class="card-text"><a target="_blank" href="https://orcid.org/0000-0003-1058-2668"><i class="fab fa-orcid fa-2x text--orange"></i></a></p></td>
<td><p class="card-text">University of Luxembourg</p></td>
<td><p class="card-text"><i class="fas fa-graduation-cap fa-1x text--orange" alt="Academic"></i></p></td>
<td><p class="card-text"><img class="elixir-style" src="../../../_static/images/logo/Elixir/ELIXIR-LU.svg" ></img></p></td>
<td><p class="card-text">Writing - Review &amp; Editing</p></td>
</tr>
<tr class="row-odd"><td><p class="card-text"><div class="firstCol"><img class="avatar-style" src="https://avatars.githubusercontent.com/NONE"></img><div class="d-block">Karsten Quast</div></div></p></td>
<td><p class="card-text"></p></td>
<td><p class="card-text">Boehringer-Ingelheim AG</p></td>
<td><p class="card-text"><i class="fas fa-industry fa-1x text--purple-light" alt="EFPIA"></i></p></td>
<td><p class="card-text"></p></td>
<td><p class="card-text"><!-- TODO --></p></td>
</tr>
<tr class="row-even"><td><p class="card-text"><div class="firstCol"><a target="_blank" href="https://github.com/{github_handle}"><img class="avatar-style" src="https://avatars.githubusercontent.com/PeterWoollard"></img><div class="d-block">Peter Woollard</div></a></div></p></td>
<td><p class="card-text"><a target="_blank" href="https://orcid.org/0000-0002-7654-6902"><i class="fab fa-orcid fa-2x text--orange"></i></a></p></td>
<td><p class="card-text">GSK</p></td>
<td><p class="card-text"><i class="fas fa-industry fa-1x text--purple-light" alt="EFPIA"></i></p></td>
<td><p class="card-text"></p></td>
<td><p class="card-text"><!-- TODO --></p></td>
</tr>
</tbody>
</table>
</div>
</details></div>
<div class="section" id="license">
<h2><span class="section-number">11.5.1.6. </span>License<a class="headerlink" href="#license" title="Permalink to this headline">¶</a></h2>
<details class="sphinx-bs dropdown card mb-3">
<summary class="summary-title card-header">
<strong>License</strong><div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<a href="https://creativecommons.org/licenses/by/4.0/">
    <img src="../../../_static/images/logo/cc.svg"  height="25"/>
    <img src="../../../_static/images/logo/by.svg"  height="25"/></a>
    <a href="https://creativecommons.org/licenses/by/4.0/"> The Creative Commons 4.0 BY license</a></br>
    <br/>
</div>
</details></div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./content/recipes/interoperability"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                                            </v-col>
                                            <div class="d-none d-md-block col-md-2 bd-toc show collapse custom-bdtoc pt-5">
                                                <div style="position: fixed">
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i>
            Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#main-objectives">
   11.5.1.1. Main Objectives:
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#who-is-this-recipe-aimed-at">
     11.5.1.1.1. Who is this recipe aimed at?
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#transcriptomics-data-model">
   11.5.1.2. Transcriptomics Data model
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#existing-standards-and-checklists">
     11.5.1.2.1. Existing standards and checklists
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#minimum-metadata-vs-desirable-metadata">
     11.5.1.2.2. Minimum metadata vs desirable metadata
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#common-metadata">
     11.5.1.2.3. Common metadata
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#suggested-metadata-fields">
       11.5.1.2.3.1. Suggested metadata fields
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#assay-metadata">
     11.5.1.2.4. Assay metadata
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#id1">
       11.5.1.2.4.1. Suggested metadata fields
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#analysis-metadata">
     11.5.1.2.5. Analysis metadata
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#id2">
       11.5.1.2.5.1. Suggested metadata fields
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#ontologies-for-transcriptomics-data">
   11.5.1.3. Ontologies for transcriptomics data
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#conclusion">
   11.5.1.4. Conclusion
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#what-to-read-next">
     11.5.1.4.1. What to read next?
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#authors">
   11.5.1.5. Authors
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#license">
   11.5.1.6. License
  </a>
 </li>
</ul>

        </nav>
        </div>
                                            </div>
                                        
                                    </v-row>

                                    
                                        <v-row class="mb-5">
                                            <div class='pl-10 prev-next-bottom mr-10' style="width:82%">
                                                
<a class='left-prev' id="prev-link" href="creating-minimal-metadata-profiles.html" title="previous page"><span class="section-number">11. </span>Creating a metadata profile</a>
<a class='right-next' id="next-link" href="covid19-sample-metadata-profile-shex-use-case.html" title="next page"><span class="section-number">11.5.2. </span>Metadata profile validation in RDF</a>

                                            </div>
                                        </v-row>
                                    
                                </v-container>
                            </main>
                        
                    
                </v-row>
                
            </v-container>
        </v-row>
    </v-container>
</v-app>


  <script src="../../../_static/js/index.d3f166471bb80abb5163.js"></script>



<!-- Google Analytics -->
<script>
    window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
    ga('create', 'UA-171697285-1', 'auto');
    ga('set', 'anonymizeIp', true);
    ga('send', 'pageview');
</script>
<script async src='https://www.google-analytics.com/analytics.js'></script>
<!-- End Google Analytics -->

<v-container fluid>
    <v-row no-gutter>
        <v-col cols="12"><footer class="footer" style=" margin-top: 12px;  ">
    <div class="container-md mt-4">
      <div class="row" style="display:flex; justify-content: center">
          <div class="col-12 col-sm-12 col-md-4 col-lg-4 col-xl-3">
            <div class="footerImages">
              <a href="https://europa.eu/"><img src="/_static/images/logo/eu-flag.png" height="40" alt="EU flag"></a>
              <a href="https://www.imi.europa.eu/"><img src="/_static/images/logo/imi-logo.png" height="40" alt="IMI logo"></a>
              <a href="https://efpia.eu"><img src="/_static/images/logo/efpia-logo.png" height="40" alt="EFPIA logo"></a>
            </div>
            <div class="mb-3"> Grant agreement 802750 </div>
          </div>
          <div class="col-5 col-sm-2 col-xs-5 col-md-2 col-lg-2 col-xl-2">
              <ul>
                <li><a href="mailto:faircookbook-ed@elixir-europe.org"> Contact us </a></li>
                <li><a href="https://fairplus-project.eu" target="_blank"> FAIRplus </a> </li>
              </ul>
          </div>
          <div class="col-5 col-sm-2 col-xs-5 col-md-2 col-lg-2 col-xl-2">
              <ul>
                <li> <a href="https://faircookbook.elixir-europe.org/content/recipes/afterword/people.html  "> Contributors </a> </li> </li>
                <li> <a href="https://faircookbook.elixir-europe.org/content/recipes/help.html"> How to contribute </a> </li>
              </ul>
          </div>
          <div class="col-12 col-sm-4 col-xs-12 col-md-3 col-lg-3 col-xl-3 pb-5">
            <a class="mr-3" href="https://elixir-europe.org/" target="_blank"><img src="/_static/images/logo/Elixir/ELIXIR.png" height="50" alt="Elixir logo"></a>
            <a href="https://fairplus-project.eu/"><img src="https://fairplus-project.eu/images/fairplus-logo.png" alt="FAIRplus logo" style="background:white;padding:5px;height:33px;"></a>
          </div>
          <div class="col-12 pb-4">
              Recommended in the
              <a
                href="https://www.imi.europa.eu/sites/default/files/uploads/documents/resources-for-projects/IMI2_OpenAcesGuidelines_Updated2021.pdf"
                target="_blank"
                class="primary--text"
              >
                 IMI/IHI Project Guidelines
              </a> and the
              <a
                href="link to https://ec.europa.eu/info/funding-tenders/opportunities/docs/2021-2027/horizon/wp-call/2023-2024/wp-4-health_horizon-2023-2024_en.pdf"
                target="_blank"
                class="primary--text"
              >
                  Horizon Europe Work Programme for Health
              </a> as the resource for FAIR Data Management guidelines and good practices for the Life Sciences.
          </div>
      </div>
    </div>

    <div class="container-fluid" style="padding:0">
        <div class="row" style="padding:0; margin:0">
            <div class="col-12 purple-dark">
                The FAIR Cookbook is licensed under the
                <a href="https://creativecommons.org/licenses/by/4.0/" target="_blank">
                    Creative Commons Attribution 4.0 International
                </a> (CC BY 4.0) license.
            </div>
        </div>
        <div class="row extra_footer d-none">
            <script> mermaid.init() </script>
        </div>
    </div>

</footer> </v-col>
    </v-row>
</v-container>
<script type="application/javascript">
    const vuetify = new Vuetify({
        theme: {
            themes: {
                light: {
                    primary: '#421C6E',
                    secondary: '#FC7A4A',
                    accent: '#8038D1',
                    error: '#FF5252',
                    info: '#2196F3',
                    success: '#7E0038',
                    warning: '#FFC107'
                }
            }
        },
        icons: { iconfont: 'fa4' }
    })
    const my_vue = new Vue({ el: '#app', vuetify: vuetify })

</script>
  </body>
</html>