
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Exploring and comparing Rose Scent profiles stored as tabular data packages with plotnine, a python port of R ggplot2</title>
    
  <link rel="stylesheet" href="../../../../_static/css/index.f658d18f9b420779cfdf24aa0a7e2d77.css">

    
  <link rel="stylesheet"
    href="../../../../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../../../../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../../../../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../../../../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../../../../_static/sphinx-book-theme.e7340bb3dbd8dde6db86f25597f54a1b.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/custom.css" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../../../../_static/js/index.d3f166471bb80abb5163.js">

    <script id="documentation_options" data-url_root="../../../../" src="../../../../_static/documentation_options.js"></script>
    <script src="../../../../_static/jquery.js"></script>
    <script src="../../../../_static/underscore.js"></script>
    <script src="../../../../_static/doctools.js"></script>
    <script src="../../../../_static/togglebutton.js"></script>
    <script src="../../../../_static/clipboard.min.js"></script>
    <script src="../../../../_static/copybutton.js"></script>
    <script src="../../../../_static/mermaid.js"></script>
    <script src="../../../../_static/mermaid.min.js"></script>
    <script src="../../../../_static/mermaid.core.js"></script>
    <script async="async" kind="hypothesis" src="https://hypothes.is/embed.js"></script>
    <script kind="utterances">

    var commentsRunWhenDOMLoaded = cb => {
    if (document.readyState != 'loading') {
        cb()
    } else if (document.addEventListener) {
        document.addEventListener('DOMContentLoaded', cb)
    } else {
        document.attachEvent('onreadystatechange', function() {
        if (document.readyState == 'complete') cb()
        })
    }
}

var addUtterances = () => {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src = "https://utteranc.es/client.js";
    script.async = "async";

    script.setAttribute("repo", "FAIRplus/the-fair-cookbook");
    script.setAttribute("issue-term", "pathname");
    script.setAttribute("theme", "github-light");
    script.setAttribute("label", "üí¨ comment");
    script.setAttribute("crossorigin", "anonymous");

    sections = document.querySelectorAll("div.section");
    if (sections !== null) {
        section = sections[sections.length-1];
        section.appendChild(script);
    }
}
commentsRunWhenDOMLoaded(addUtterances);
</script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../../../../_static/sphinx-book-theme.7d483ff0a819d6edff12ce0b1ead3928.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../../../../_static/sphinx-thebe.js"></script>
    <link rel="canonical" href="https://fairplus.github.io/the-fair-cookbook/content/recipes/applied-examples/fair-data-matrix/1-rose-metabolites-Python-analysis.html" />
    <link rel="shortcut icon" href="../../../../_static/favicon.ico"/>
    <link rel="index" title="Index" href="../../../../genindex.html" />
    <link rel="search" title="Search" href="../../../../search.html" />
    <link rel="next" title="Exploring a rose garden with SPARQL" href="2-rose-metabolites-Python-RDF-querying-analysis.html" />
    <link rel="prev" title="Bibliographic Reference:" href="0-rose-metabolites-Python-data-handling.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />


<!-- Opengraph tags -->
<meta property="og:url"         content="https://fairplus.github.io/the-fair-cookbook/content/recipes/applied-examples/fair-data-matrix/1-rose-metabolites-Python-analysis.html" />
<meta property="og:type"        content="article" />
<meta property="og:title"       content="Exploring and comparing Rose Scent profiles stored as tabular data packages with plotnine, a python port of R ggplot2" />
<meta property="og:description" content="Exploring and comparing Rose Scent profiles stored as tabular data packages with plotnine, a python port of R ggplot2  &lt;i class=&#39;fa fa-qrcode fa-2x&#39; style=&#39;colo" />
<meta property="og:image"       content="https://fairplus.github.io/the-fair-cookbook/_static/cookbook-logo-small.png" />

<meta name="twitter:card" content="summary" />


  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../../../../index.html">
  
  <img src="../../../../_static/cookbook-logo-small.png" class="logo" alt="logo">
  
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../../../../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../../../home.html">
   FAIR Cookbook
  </a>
 </li>
</ul>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../introduction.html">
   Introduction
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../introduction/the-turing-way.html">
     1. The Turing Way Book of Data Science
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../introduction/FAIR-cookbook-audience.html">
     2. Who should read the FAIR Cookbook?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../introduction/brief-FAIR-principles.html">
     3. What are the FAIR Principles?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../introduction/what-is-IMI.html">
     4. What is IMI?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../introduction/FAIR-in-IMI-context.html">
     5. FAIR in the context of IMI
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../introduction/IMI-Data-Access-Request-Process.html">
     6. Engaging with IMI Projects
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../introduction/FAIRplus-values.html">
     7. The Ethical Values behind FAIR and the FAIRPlus project
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../introduction/data-sharing-models.html">
     8. High level data sharing models
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../assessing-fairness.html">
   Assessing FAIR
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../assessing-fairness/cmmi/Data-Management-Maturity-Model.html">
     1. Data Management Maturity Model
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../assessing-fairness/cmmi/Datasets-Maturity-Model.html">
     2. FAIRplus Datasets Maturity Model
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../assessing-fairness/cmmi/Data-Usage-Areas.html">
     3. Data Usage Areas
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../../assessing-fairness/fair-indicators.html">
     4. FAIR Indicators
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../../assessing-fairness/cmmi/RDA-Indicators.html">
       4.4.1. RDA Indicators
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../../assessing-fairness/cmmi/FAIRPlus-Indicators.html">
       4.4.2. IMI FAIRPlus Indicators
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../../assessing-fairness/manual-fair-assessment.html">
     5. FAIR assessment by hand
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../../assessing-fairness/cmmi/evaluation-method.html">
       5.4.1. Evaluation Method
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../../assessing-fairness/cmmi/rda-evaluation.html">
       5.4.2. RDA Evaluation Methods
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../../assessing-fairness/automatic-fair-assessment.html">
     6. FAIR assessment by software
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../../assessing-fairness/fair-assessment-fairevaluator.html">
       6.4.1. Assessing FAIRness with FAIREvaluator
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../../assessing-fairness/fair-assessment-fairshake.html">
       6.4.2. FAIR assessment with FAIRShake
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../infrastructure.html">
   Infrastructure for FAIR
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../infrastructure/id-resolution.html">
     1. Identifier resolution services
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../infrastructure/data-catalog.html">
     2. Building a Dataset Catalogue
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../infrastructure/data-catalog-deployment.html">
     3. Deploying the IMI data catalogue
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../../infrastructure/vocabularies.html">
     4. Vocabularies and Terminology Infrastructure
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../../infrastructure/vocabularies/technical-and-architectural-selection-criteria-of-ontology-lookup-services.html">
       4.1. Technical &amp; architectural selection criteria of ontology lookup services
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../../infrastructure/vocabularies/Selecting_and_using_ontology_lookup_services.html">
       4.2. Selection and exploitation of Ontology Lookup Services
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../../infrastructure/vocabularies/UC3_R13_local_ontology_services.html">
       4.3. How to deploy a terminology service - EBI Ontology Lookup Service
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../findability.html">
   Improving Findability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../findability/identifiers.html">
     1. How to generate globally unique, resolvable and persistent identifiers
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../infrastructure/chemical-identities.html">
     2. Chemical Identity - InChi and SMILES
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../findability/identifiers-minids.html">
     3. Identifier Minting Service with Minid Client
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../findability/zenodo-deposition.html">
     4. Deposition to Zenodo Archive
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../../findability/seo.html">
     5. Findability: Search Engine Optimization
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../../findability/seo/bioschemas-data-page.html">
       5.8.1. Findability: Data Page Markup with Bioschemas
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../../findability/seo/bioschemas-dataset.html">
       5.8.2. Findability: Dataset Markup
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../../findability/seo/bioschemas-datacatalog.html">
       5.8.3. Findability: Data Catalog Markup
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../accessibility.html">
   Improving Accessibility
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../accessibility/anonymizing-patient-data.html">
     1. Anonymizing Patient Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../accessibility/how-to-use-sftp-for-transfer.html">
     2. How to use SFTP to transfer data files between institutions.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../accessibility/aspera.html">
     3. How to download files with Aspera
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../interoperability.html">
   Improving Interoperability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../interoperability/identifier-mapping.html">
     How to interlink data from different sources?
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../../interoperability/bridgedb-recipe.html">
     Identifier mapping with BridgeDB
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../../interoperability/bridgedb/bridgedb-example.html">
       Identifier mapping using BridgeDb WS
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../interoperability/selecting-ontologies.html">
     Which vocabulary to use?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../interoperability/ontology-new-term-request-recipe.html">
     Requesting terms addition to terminology artefacts
    </a>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../../interoperability/ontology-robot-recipe.html">
     Building an application ontology with Robot
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../../interoperability/ontology-robot/MSIO-robot-build-process.html">
       MSIO, an application ontology for Metabolomics
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2 collapsible-parent">
    <a class="reference internal" href="../../interoperability/creating-minimal-metadata-profiles.html">
     Creating a Metadata Profile
    </a>
    <ul class="collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="../../interoperability/transcriptomics-metadata.html">
       Establishing a Metadata Profile for Transcriptomics
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="../../interoperability/covid19-sample-metadata-profile-shex-use-case.html">
       Metadata Profile Validation in RDF with Shape Expression: The Covid-19 sample metadata use case
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../interoperability/from-proprietary-to-open-standard-mzml-exemplar.html">
     From proprietary to open standard data format
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../interoperability/fastq-file-format-validators.html">
     File format validation, an example with FASTQ files
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../reusability.html">
   Improving Reusability
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../reusability/ATI-licensing.html">
     1. Licensing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../reusability/ATI_licensing_software.html">
     2. Software Licenses
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../reusability/ATI_licensing_data.html">
     3. Data Licenses
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../reusability/expressing-data-use.html">
     4. Patterns and practice for expressing allowed data usage
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../reusability/provenance.html">
     5. Provenance Information
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../reusability/community-standards.html">
     6. Compliance with community annotation standards
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="../../applied-examples.html">
   Making IMI data FAIR
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2 current active collapsible-parent">
    <a class="reference internal" href="../fair-data-matrix-recipe.html">
     Making a Data Matrix FAIR
    </a>
    <ul class="current collapse-ul">
     <li class="toctree-l3">
      <a class="reference internal" href="0-rose-metabolites-Python-data-handling.html">
       Bibliographic Reference:
      </a>
     </li>
     <li class="toctree-l3 current active">
      <a class="current reference internal" href="#">
       Exploring and comparing Rose Scent profiles stored as tabular data packages with plotnine, a python port of R ggplot2
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="2-rose-metabolites-Python-RDF-querying-analysis.html">
       Exploring a rose garden with SPARQL
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="3-rose-metabolites-R-analysis.html">
       Exploring and comparing Rose Scent profiles stored as tabular data packages with ggplot2 R library
      </a>
     </li>
    </ul>
    <i class="fas fa-chevron-down">
    </i>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../etox_raw/FAIRplus_Recipe_Ontology_mapping_the_eTox_dataset_scenario.html">
     Data normalization and integration through entity linking
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../nd4bb_raw/FAIRification_CookBook_Recipe1_V02.html">
     ND4BB dataset FAIRification recipe
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../oncotrack_raw/rawRecipe-oncotrack.html">
     Oncotrack FAIRification recipe
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../resolute_raw/rawRecipe_resolute.html">
     RESOLUTE FAIRification Recipe
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../../glossary-fair-terms.html">
   Glossary of FAIR things
  </a>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../help.html">
   Help
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../help/tips-tricks.html">
     1. Recipe Tips and Tricks
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../help/recipe-template.html">
     2. Recipe Template
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../help/how-to-create-recipe-with-gdoc.html">
     3. Using GoogleDocs to contribute a FAIR Cookbook Recipe
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../help/how-to-create-recipe-with-hackmd.html">
     4. How to use HackMD to write a FAIR Cookbook Recipe
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../help/how-to-create-recipe-with-git.html">
     5. How to create a FAIRPlus Coobook Recipe ?
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../boilerplate.html">
   Miscellaneous
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../boilerplate/license.html">
     1. License for this book
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../boilerplate/people.html">
     2. FAIR Cookbook Steering Council and Institutional Partners
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../boilerplate/governance.html">
     3. FAIR Cookbook Governance
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../boilerplate/code_of_conduct.html">
     4. FAIR Cookbook Code of Conduct
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../boilerplate/project_information.html">
     5. FAIR CookBook BoilerPlate
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../boilerplate/roadmap.html">
     6. FAIR cookbook development roadmap
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../boilerplate/contributing.html">
     7. Contributing to the FAIR cookbook
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../boilerplate/disclaimer.html">
     8. Disclaimer
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../../../../_sources/content/recipes/applied-examples/fair-data-matrix/1-rose-metabolites-Python-analysis.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.ipynb</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/FAIRplus/the-fair-cookbook"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/FAIRplus/the-fair-cookbook/issues/new?title=Issue%20on%20page%20%2Fcontent/recipes/applied-examples/fair-data-matrix/1-rose-metabolites-Python-analysis.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        <a class="edit-button" href="https://github.com/FAIRplus/the-fair-cookbook/edit/migrating/content/recipes/applied-examples/fair-data-matrix/1-rose-metabolites-Python-analysis.ipynb"><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Edit this page"><i class="fas fa-pencil-alt"></i>suggest edit</button></a>
    </div>
</div>


            <!-- Full screen (wrap in <a> to have style consistency -->
            <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                    data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                    title="Fullscreen mode"><i
                        class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Launch interactive content"><i class="fas fa-rocket"></i></button>
    <div class="dropdown-buttons">
        
        <a class="binder-button" href="https://mybinder.org/v2/gh/FAIRplus/the-fair-cookbook/migrating?urlpath=lab/tree/content/recipes/applied-examples/fair-data-matrix/1-rose-metabolites-Python-analysis.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Launch Binder" data-toggle="tooltip"
                data-placement="left"><img class="binder-button-logo"
                    src="../../../../_static/images/logo_binder.svg"
                    alt="Interact on binder">Binder</button></a>
        
        
        
        <a class="colab-button" href="https://colab.research.google.com/github/FAIRplus/the-fair-cookbook/blob/migrating/content/recipes/applied-examples/fair-data-matrix/1-rose-metabolites-Python-analysis.ipynb"><button type="button" class="btn btn-secondary topbarbtn"
                title="Launch Colab" data-toggle="tooltip" data-placement="left"><img class="colab-button-logo"
                    src="../../../../_static/images/logo_colab.png"
                    alt="Interact on Colab">Colab</button></a>
        
        
    </div>
</div>

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i>
            Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#bibliographic-reference">
   Bibliographic Reference:
  </a>
 </li>
</ul>

        </nav>
        
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="exploring-and-comparing-rose-scent-profiles-stored-as-tabular-data-packages-with-plotnine-a-python-port-of-r-ggplot2">
<h1>Exploring and comparing Rose Scent profiles stored as tabular data packages with plotnine, a python port of R ggplot2<a class="headerlink" href="#exploring-and-comparing-rose-scent-profiles-stored-as-tabular-data-packages-with-plotnine-a-python-port-of-r-ggplot2" title="Permalink to this headline">¬∂</a></h1>
<div class="sphinx-bs container-lg pb-3&lt;br&gt; docutils">
<div class="row docutils">
<div class="d-flex col-lg-3 col-md-4 col-sm-6 col-xs-12 p-1&lt;br&gt; docutils">
<div class="card w-100 rounded&lt;br&gt; docutils">
<div class="card-header docutils">
<p class="card-text"><i class='fa fa-qrcode fa-2x' style='color:#7e0038;'></i><br></p>
</div>
<div class="card-body docutils">
<h4><b>Recipe metadata</b></h4>
 identifier: <a href="">RX.X</a>
 version: <a href="">v1.0</a>
</div>
</div>
</div>
<div class="d-flex col-lg-3 col-md-4 col-sm-6 col-xs-12 p-1&lt;br&gt; docutils">
<div class="card w-100 rounded&lt;br&gt; docutils">
<div class="card-header docutils">
<p class="card-text"><i class='fa fa-fire fa-2x' style='color:#7e0038;'></i></p>
</div>
<div class="card-body docutils">
<h4><b>Difficulty level</b></h4>
<i class='fa fa-fire fa-lg' style='color:#7e0038;'></i>
<i class='fa fa-fire fa-lg' style='color:#7e0038;'></i>
<i class='fa fa-fire fa-lg' style='color:#7e0038;'></i>
<i class='fa fa-fire fa-lg' style='color:#7e0038;'></i>
<i class='fa fa-fire fa-lg' style='color:lightgrey'></i>
</div>
</div>
</div>
<div class="d-flex col-lg-3 col-md-4 col-sm-6 col-xs-12 p-1&lt;br&gt; docutils">
<div class="card w-100 rounded&lt;br&gt; docutils">
<div class="card-header docutils">
<p class="card-text"><i class='fas fa-clock fa-2x' style='color:#7e0038;'></i></p>
</div>
<div class="card-body docutils">
<h4><b>Reading Time</b></h4>
<i class='fa fa-clock fa-lg' style='color:#7e0038;'></i> 30 minutes
<h4><b>Recipe Type</b></h4>
<i class='fa fa-globe fa-lg' style='color:#7e0038;'></i> Hands-on
<h4><b>Executable Code</b></h4>
<i class='fa fa-play-circle fa-lg' style='color:#7e0038;'></i> Yes
</div>
</div>
</div>
<div class="d-flex col-lg-3 col-md-4 col-sm-6 col-xs-12 p-1&lt;br&gt; docutils">
<div class="card w-100 rounded&lt;br&gt; docutils">
<div class="card-header docutils">
<p class="card-text"><i class='fa fa-users fa-2x' style='color:#7e0038;'></i></p>
</div>
<div class="card-body docutils">
<h4><b>Intended Audience</b></h4>
<!-- <p> <i class='fa fa-user-md fa-lg' style='color:#7e0038;'></i> Principal Investigator </p> -->
<p> <i class='fa fa-database fa-lg' style='color:#7e0038;'></i> Data Manager </p>
<p> <i class='fa fa-wrench fa-lg' style='color:#7e0038;'></i> Data Scientist </p>
</div>
</div>
</div>
</div>
</div>
<hr class="docutils" />
<p><strong>title:</strong> <code class="docutils literal notranslate"> </code> <br>
<strong>author:</strong> <code class="docutils literal notranslate"><span class="pre">philippe.rocca-serra&#64;oerc.ox.ac.uk</span></code> <br>
<strong>date:</strong> <code class="docutils literal notranslate"><span class="pre">2020-05-11</span></code> <br>
<strong>license:</strong> <a class="reference external" href="https://creativecommons.org/licenses/by/4.0/"><code class="docutils literal notranslate"><span class="pre">creative-commons-by-sa-4.0</span></code></a> <br>
<strong>keywords:</strong> <code class="docutils literal notranslate"> </code> <br>
<strong>description:</strong> <code class="docutils literal notranslate"> </code> <br></p>
<p>*Philippe Rocca-Serra (philippe.rocca-serra[at]<a class="reference external" href="http://oerc.ox.ac.uk">oerc.ox.ac.uk</a>), University of Oxford e-Research Centre</p>
<p>Experimental results such as metabolite profiling data published in [1,2] can be straightfowardly reported using <a class="reference external" href="https://frictionlessdata.io/specs/tabular-data-package/">OKFN Data Packages</a>. Such components can be easily parsed as data frames and exploiting for data visualization purpose using libraries implementing graphical grammar concepts. Here, we show how to use python equivalent of ggplot2 , the rich R graphical libraries (<a class="reference external" href="https://ggplot2.tidyverse.org/">https://ggplot2.tidyverse.org/</a>).
A few line of codes allow to query datasets and rapidly explore the information. Most importantly, this rapid exploration is possible because of independent variables and their levels have been clearly and unambiguously declared in the Tabular Data Package itself.</p>
<ol class="simple">
<li><p>Let‚Äôs begin by installing the Python packages allowing easy access and use of data formatted as JSON Data Package</p></li>
</ol>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">from</span> <span class="nn">plotnine</span> <span class="kn">import</span> <span class="o">*</span>
</pre></div>
</div>
</div>
</div>
<ol class="simple">
<li><p>Reading the data:</p></li>
</ol>
<p>We now simply read in the comma-separated-file associated with the tabular data package (a ‚Äúlong‚Äù table)</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">data</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">&quot;../data/processed/rose-data/rose-aroma-naturegenetics2018-treatment-group-mean-sem-report-table-example.csv&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">FileNotFoundError</span><span class="g g-Whitespace">                         </span>Traceback (most recent call last)
<span class="o">&lt;</span><span class="n">ipython</span><span class="o">-</span><span class="nb">input</span><span class="o">-</span><span class="mi">2</span><span class="o">-</span><span class="mi">39</span><span class="n">f76dc87f95</span><span class="o">&gt;</span> <span class="ow">in</span> <span class="o">&lt;</span><span class="n">module</span><span class="o">&gt;</span>
<span class="ne">----&gt; </span><span class="mi">1</span> <span class="n">data</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">&quot;../data/processed/rose-data/rose-aroma-naturegenetics2018-treatment-group-mean-sem-report-table-example.csv&quot;</span><span class="p">)</span>

<span class="nn">~/.pyenv/versions/venv372/lib/python3.7/site-packages/pandas/io/parsers.py</span> in <span class="ni">parser_f</span><span class="nt">(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, dialect, error_bad_lines, warn_bad_lines, delim_whitespace, low_memory, memory_map, float_precision)</span>
<span class="g g-Whitespace">    </span><span class="mi">683</span>         <span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">684</span> 
<span class="ne">--&gt; </span><span class="mi">685</span>         <span class="k">return</span> <span class="n">_read</span><span class="p">(</span><span class="n">filepath_or_buffer</span><span class="p">,</span> <span class="n">kwds</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">686</span> 
<span class="g g-Whitespace">    </span><span class="mi">687</span>     <span class="n">parser_f</span><span class="o">.</span><span class="vm">__name__</span> <span class="o">=</span> <span class="n">name</span>

<span class="nn">~/.pyenv/versions/venv372/lib/python3.7/site-packages/pandas/io/parsers.py</span> in <span class="ni">_read</span><span class="nt">(filepath_or_buffer, kwds)</span>
<span class="g g-Whitespace">    </span><span class="mi">455</span> 
<span class="g g-Whitespace">    </span><span class="mi">456</span>     <span class="c1"># Create the parser.</span>
<span class="ne">--&gt; </span><span class="mi">457</span>     <span class="n">parser</span> <span class="o">=</span> <span class="n">TextFileReader</span><span class="p">(</span><span class="n">fp_or_buf</span><span class="p">,</span> <span class="o">**</span><span class="n">kwds</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">458</span> 
<span class="g g-Whitespace">    </span><span class="mi">459</span>     <span class="k">if</span> <span class="n">chunksize</span> <span class="ow">or</span> <span class="n">iterator</span><span class="p">:</span>

<span class="nn">~/.pyenv/versions/venv372/lib/python3.7/site-packages/pandas/io/parsers.py</span> in <span class="ni">__init__</span><span class="nt">(self, f, engine, **kwds)</span>
<span class="g g-Whitespace">    </span><span class="mi">893</span>             <span class="bp">self</span><span class="o">.</span><span class="n">options</span><span class="p">[</span><span class="s2">&quot;has_index_names&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">kwds</span><span class="p">[</span><span class="s2">&quot;has_index_names&quot;</span><span class="p">]</span>
<span class="g g-Whitespace">    </span><span class="mi">894</span> 
<span class="ne">--&gt; </span><span class="mi">895</span>         <span class="bp">self</span><span class="o">.</span><span class="n">_make_engine</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">engine</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">896</span> 
<span class="g g-Whitespace">    </span><span class="mi">897</span>     <span class="k">def</span> <span class="nf">close</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>

<span class="nn">~/.pyenv/versions/venv372/lib/python3.7/site-packages/pandas/io/parsers.py</span> in <span class="ni">_make_engine</span><span class="nt">(self, engine)</span>
<span class="g g-Whitespace">   </span><span class="mi">1133</span>     <span class="k">def</span> <span class="nf">_make_engine</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">engine</span><span class="o">=</span><span class="s2">&quot;c&quot;</span><span class="p">):</span>
<span class="g g-Whitespace">   </span><span class="mi">1134</span>         <span class="k">if</span> <span class="n">engine</span> <span class="o">==</span> <span class="s2">&quot;c&quot;</span><span class="p">:</span>
<span class="ne">-&gt; </span><span class="mi">1135</span>             <span class="bp">self</span><span class="o">.</span><span class="n">_engine</span> <span class="o">=</span> <span class="n">CParserWrapper</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">f</span><span class="p">,</span> <span class="o">**</span><span class="bp">self</span><span class="o">.</span><span class="n">options</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1136</span>         <span class="k">else</span><span class="p">:</span>
<span class="g g-Whitespace">   </span><span class="mi">1137</span>             <span class="k">if</span> <span class="n">engine</span> <span class="o">==</span> <span class="s2">&quot;python&quot;</span><span class="p">:</span>

<span class="nn">~/.pyenv/versions/venv372/lib/python3.7/site-packages/pandas/io/parsers.py</span> in <span class="ni">__init__</span><span class="nt">(self, src, **kwds)</span>
<span class="g g-Whitespace">   </span><span class="mi">1915</span>         <span class="n">kwds</span><span class="p">[</span><span class="s2">&quot;usecols&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">usecols</span>
<span class="g g-Whitespace">   </span><span class="mi">1916</span> 
<span class="ne">-&gt; </span><span class="mi">1917</span>         <span class="bp">self</span><span class="o">.</span><span class="n">_reader</span> <span class="o">=</span> <span class="n">parsers</span><span class="o">.</span><span class="n">TextReader</span><span class="p">(</span><span class="n">src</span><span class="p">,</span> <span class="o">**</span><span class="n">kwds</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1918</span>         <span class="bp">self</span><span class="o">.</span><span class="n">unnamed_cols</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_reader</span><span class="o">.</span><span class="n">unnamed_cols</span>
<span class="g g-Whitespace">   </span><span class="mi">1919</span> 

<span class="nn">pandas/_libs/parsers.pyx</span> in <span class="ni">pandas._libs.parsers.TextReader.__cinit__</span><span class="nt">()</span>

<span class="nn">pandas/_libs/parsers.pyx</span> in <span class="ni">pandas._libs.parsers.TextReader._setup_parser_source</span><span class="nt">()</span>

<span class="ne">FileNotFoundError</span>: [Errno 2] File b&#39;../data/processed/rose-data/rose-aroma-naturegenetics2018-treatment-group-mean-sem-report-table-example.csv&#39; does not exist: b&#39;../data/processed/rose-data/rose-aroma-naturegenetics2018-treatment-group-mean-sem-report-table-example.csv&#39;
</pre></div>
</div>
</div>
</div>
<p>Alternately, one could read the relevant data file from the corresponding zenodo dataset</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#data = pd.read_csv(&quot;https://zenodo.org/api/files/ba3fbc84-14af-4858-a9ed-e6cfe8d4efd2/rose-aroma-naturegenetics2018-treatment-group-mean-sem-report-table-example.csv&quot;) </span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">data</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">NameError</span><span class="g g-Whitespace">                                 </span>Traceback (most recent call last)
<span class="o">&lt;</span><span class="n">ipython</span><span class="o">-</span><span class="nb">input</span><span class="o">-</span><span class="mi">4</span><span class="o">-</span><span class="mi">304</span><span class="n">fa4ce4ebd</span><span class="o">&gt;</span> <span class="ow">in</span> <span class="o">&lt;</span><span class="n">module</span><span class="o">&gt;</span>
<span class="ne">----&gt; </span><span class="mi">1</span> <span class="n">data</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>

<span class="ne">NameError</span>: name &#39;data&#39; is not defined
</pre></div>
</div>
</div>
</div>
<ol class="simple">
<li><p>Plotting the data: We then generate a barplot using the python plotnine library, which delivers a similar functionality as the R ggplot2 package.</p></li>
</ol>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># width = figure_size[0]</span>
<span class="c1"># height = figure_size[0] * aspect_ratio</span>
<span class="n">gray</span> <span class="o">=</span> <span class="s1">&#39;#666666&#39;</span>
<span class="n">orange</span> <span class="o">=</span> <span class="s1">&#39;#FF8000&#39;</span>
<span class="n">blue</span> <span class="o">=</span> <span class="s1">&#39;#3333FF&#39;</span>

<span class="n">p1</span> <span class="o">=</span> <span class="p">(</span><span class="n">ggplot</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
 <span class="o">+</span> <span class="n">aes</span><span class="p">(</span><span class="s1">&#39;chemical_name&#39;</span><span class="p">,</span> <span class="s1">&#39;sample_mean&#39;</span><span class="p">,</span><span class="n">fill</span><span class="o">=</span><span class="s1">&#39;factor(treatment)&#39;</span><span class="p">)</span>
 <span class="o">+</span> <span class="n">geom_col</span><span class="p">()</span>
 
 <span class="o">+</span> <span class="n">theme</span><span class="p">(</span><span class="n">axis_text_x</span><span class="o">=</span><span class="n">element_text</span><span class="p">(</span><span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span> <span class="n">hjust</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">6</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="n">blue</span><span class="p">))</span>
 <span class="o">+</span> <span class="n">theme</span><span class="p">(</span><span class="n">axis_text_y</span><span class="o">=</span><span class="n">element_text</span><span class="p">(</span><span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span> <span class="n">hjust</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">6</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="n">orange</span><span class="p">))</span>
 <span class="o">+</span> <span class="n">scale_y_continuous</span><span class="p">(</span><span class="n">expand</span> <span class="o">=</span> <span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">))</span>   
 <span class="o">+</span> <span class="n">facet_wrap</span><span class="p">(</span><span class="s1">&#39;~treatment&#39;</span><span class="p">,</span> <span class="nb">dir</span><span class="o">=</span><span class="s1">&#39;v&#39;</span><span class="p">,</span><span class="n">ncol</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
 <span class="o">+</span> <span class="n">theme</span><span class="p">(</span><span class="n">figure_size</span> <span class="o">=</span> <span class="p">(</span><span class="mi">8</span><span class="p">,</span> <span class="mi">16</span><span class="p">))</span>      
<span class="p">)</span>

<span class="n">p1</span> <span class="o">+</span> <span class="n">theme</span><span class="p">(</span><span class="n">panel_background</span><span class="o">=</span><span class="n">element_rect</span><span class="p">(</span><span class="n">fill</span><span class="o">=</span><span class="n">blue</span><span class="p">)</span>
       <span class="p">)</span>

<span class="n">p1</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">NameError</span><span class="g g-Whitespace">                                 </span>Traceback (most recent call last)
<span class="o">&lt;</span><span class="n">ipython</span><span class="o">-</span><span class="nb">input</span><span class="o">-</span><span class="mi">5</span><span class="o">-</span><span class="mi">8</span><span class="n">b6cd2b54fc3</span><span class="o">&gt;</span> <span class="ow">in</span> <span class="o">&lt;</span><span class="n">module</span><span class="o">&gt;</span>
<span class="g g-Whitespace">     </span><span class="mi">13</span>  <span class="o">+</span> <span class="n">scale_y_continuous</span><span class="p">(</span><span class="n">expand</span> <span class="o">=</span> <span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">))</span>
<span class="g g-Whitespace">     </span><span class="mi">14</span>  <span class="o">+</span> <span class="n">facet_wrap</span><span class="p">(</span><span class="s1">&#39;~treatment&#39;</span><span class="p">,</span> <span class="nb">dir</span><span class="o">=</span><span class="s1">&#39;v&#39;</span><span class="p">,</span><span class="n">ncol</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="ne">---&gt; </span><span class="mi">15</span>  <span class="o">+</span> <span class="n">theme</span><span class="p">(</span><span class="n">figure_size</span> <span class="o">=</span> <span class="p">(</span><span class="mi">8</span><span class="p">,</span> <span class="mi">16</span><span class="p">))</span>
<span class="g g-Whitespace">     </span><span class="mi">16</span> <span class="p">)</span>
<span class="g g-Whitespace">     </span><span class="mi">17</span> 

<span class="ne">NameError</span>: name &#39;data&#39; is not defined
</pre></div>
</div>
</div>
</div>
<ol class="simple">
<li><p>Let‚Äôs now compare the dataset generated in 2015 and the dataset generated in 2018.</p></li>
</ol>
<p>Both datasets have been generated by the same team, on the same genotype (Rosa Chinensis ‚ÄòOld Blush‚Äô) and organism part (‚Äòsepals‚Äô). Both datasets are held in a Tabular Data Package with the same structure. To perform the comparison, we have simply created another tabular data package, which retains the exact same structure and that simply holds the measurements for the relevant conditions extracted from each dataset (the function to create such file is omitted).</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">ng2018sc2015</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">&quot;../data/processed/rose-data/rose_aroma_compound_science2015_vs_NG2018_data_integration.csv&quot;</span><span class="p">)</span>
<span class="c1"># ng2018sc2015 = pd.read_csv(&quot;https://zenodo.org/api/files/268f29fc-8ead-4049-bb86-181b72073682/rose_aroma_compound_science2015_vs_NG2018_data_integration.csv&quot;)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">FileNotFoundError</span><span class="g g-Whitespace">                         </span>Traceback (most recent call last)
<span class="o">&lt;</span><span class="n">ipython</span><span class="o">-</span><span class="nb">input</span><span class="o">-</span><span class="mi">6</span><span class="o">-</span><span class="mi">33683</span><span class="n">c5df1e2</span><span class="o">&gt;</span> <span class="ow">in</span> <span class="o">&lt;</span><span class="n">module</span><span class="o">&gt;</span>
<span class="ne">----&gt; </span><span class="mi">1</span> <span class="n">ng2018sc2015</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">&quot;../data/processed/rose-data/rose_aroma_compound_science2015_vs_NG2018_data_integration.csv&quot;</span><span class="p">)</span>
<span class="g g-Whitespace">      </span><span class="mi">2</span> <span class="c1"># ng2018sc2015 = pd.read_csv(&quot;https://zenodo.org/api/files/268f29fc-8ead-4049-bb86-181b72073682/rose_aroma_compound_science2015_vs_NG2018_data_integration.csv&quot;)</span>

<span class="nn">~/.pyenv/versions/venv372/lib/python3.7/site-packages/pandas/io/parsers.py</span> in <span class="ni">parser_f</span><span class="nt">(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, dialect, error_bad_lines, warn_bad_lines, delim_whitespace, low_memory, memory_map, float_precision)</span>
<span class="g g-Whitespace">    </span><span class="mi">683</span>         <span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">684</span> 
<span class="ne">--&gt; </span><span class="mi">685</span>         <span class="k">return</span> <span class="n">_read</span><span class="p">(</span><span class="n">filepath_or_buffer</span><span class="p">,</span> <span class="n">kwds</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">686</span> 
<span class="g g-Whitespace">    </span><span class="mi">687</span>     <span class="n">parser_f</span><span class="o">.</span><span class="vm">__name__</span> <span class="o">=</span> <span class="n">name</span>

<span class="nn">~/.pyenv/versions/venv372/lib/python3.7/site-packages/pandas/io/parsers.py</span> in <span class="ni">_read</span><span class="nt">(filepath_or_buffer, kwds)</span>
<span class="g g-Whitespace">    </span><span class="mi">455</span> 
<span class="g g-Whitespace">    </span><span class="mi">456</span>     <span class="c1"># Create the parser.</span>
<span class="ne">--&gt; </span><span class="mi">457</span>     <span class="n">parser</span> <span class="o">=</span> <span class="n">TextFileReader</span><span class="p">(</span><span class="n">fp_or_buf</span><span class="p">,</span> <span class="o">**</span><span class="n">kwds</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">458</span> 
<span class="g g-Whitespace">    </span><span class="mi">459</span>     <span class="k">if</span> <span class="n">chunksize</span> <span class="ow">or</span> <span class="n">iterator</span><span class="p">:</span>

<span class="nn">~/.pyenv/versions/venv372/lib/python3.7/site-packages/pandas/io/parsers.py</span> in <span class="ni">__init__</span><span class="nt">(self, f, engine, **kwds)</span>
<span class="g g-Whitespace">    </span><span class="mi">893</span>             <span class="bp">self</span><span class="o">.</span><span class="n">options</span><span class="p">[</span><span class="s2">&quot;has_index_names&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">kwds</span><span class="p">[</span><span class="s2">&quot;has_index_names&quot;</span><span class="p">]</span>
<span class="g g-Whitespace">    </span><span class="mi">894</span> 
<span class="ne">--&gt; </span><span class="mi">895</span>         <span class="bp">self</span><span class="o">.</span><span class="n">_make_engine</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">engine</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">896</span> 
<span class="g g-Whitespace">    </span><span class="mi">897</span>     <span class="k">def</span> <span class="nf">close</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>

<span class="nn">~/.pyenv/versions/venv372/lib/python3.7/site-packages/pandas/io/parsers.py</span> in <span class="ni">_make_engine</span><span class="nt">(self, engine)</span>
<span class="g g-Whitespace">   </span><span class="mi">1133</span>     <span class="k">def</span> <span class="nf">_make_engine</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">engine</span><span class="o">=</span><span class="s2">&quot;c&quot;</span><span class="p">):</span>
<span class="g g-Whitespace">   </span><span class="mi">1134</span>         <span class="k">if</span> <span class="n">engine</span> <span class="o">==</span> <span class="s2">&quot;c&quot;</span><span class="p">:</span>
<span class="ne">-&gt; </span><span class="mi">1135</span>             <span class="bp">self</span><span class="o">.</span><span class="n">_engine</span> <span class="o">=</span> <span class="n">CParserWrapper</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">f</span><span class="p">,</span> <span class="o">**</span><span class="bp">self</span><span class="o">.</span><span class="n">options</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1136</span>         <span class="k">else</span><span class="p">:</span>
<span class="g g-Whitespace">   </span><span class="mi">1137</span>             <span class="k">if</span> <span class="n">engine</span> <span class="o">==</span> <span class="s2">&quot;python&quot;</span><span class="p">:</span>

<span class="nn">~/.pyenv/versions/venv372/lib/python3.7/site-packages/pandas/io/parsers.py</span> in <span class="ni">__init__</span><span class="nt">(self, src, **kwds)</span>
<span class="g g-Whitespace">   </span><span class="mi">1915</span>         <span class="n">kwds</span><span class="p">[</span><span class="s2">&quot;usecols&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">usecols</span>
<span class="g g-Whitespace">   </span><span class="mi">1916</span> 
<span class="ne">-&gt; </span><span class="mi">1917</span>         <span class="bp">self</span><span class="o">.</span><span class="n">_reader</span> <span class="o">=</span> <span class="n">parsers</span><span class="o">.</span><span class="n">TextReader</span><span class="p">(</span><span class="n">src</span><span class="p">,</span> <span class="o">**</span><span class="n">kwds</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1918</span>         <span class="bp">self</span><span class="o">.</span><span class="n">unnamed_cols</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_reader</span><span class="o">.</span><span class="n">unnamed_cols</span>
<span class="g g-Whitespace">   </span><span class="mi">1919</span> 

<span class="nn">pandas/_libs/parsers.pyx</span> in <span class="ni">pandas._libs.parsers.TextReader.__cinit__</span><span class="nt">()</span>

<span class="nn">pandas/_libs/parsers.pyx</span> in <span class="ni">pandas._libs.parsers.TextReader._setup_parser_source</span><span class="nt">()</span>

<span class="ne">FileNotFoundError</span>: [Errno 2] File b&#39;../data/processed/rose-data/rose_aroma_compound_science2015_vs_NG2018_data_integration.csv&#39; does not exist: b&#39;../data/processed/rose-data/rose_aroma_compound_science2015_vs_NG2018_data_integration.csv&#39;
</pre></div>
</div>
</div>
</div>
<ol class="simple">
<li><p>We generate another barplot, which shows the concentration of the chemicals targeted by the GC-MS profiling assay.</p></li>
</ol>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="p">(</span><span class="n">ggplot</span><span class="p">(</span><span class="n">ng2018sc2015</span><span class="p">)</span>
 <span class="o">+</span> <span class="n">aes</span><span class="p">(</span><span class="s1">&#39;chemical_name&#39;</span><span class="p">,</span> <span class="s1">&#39;normalized_to_total_sum_concentration&#39;</span><span class="p">,</span><span class="n">fill</span><span class="o">=</span><span class="s1">&#39;factor(publication_year)&#39;</span><span class="p">)</span>
 <span class="o">+</span> <span class="n">geom_col</span><span class="p">()</span>
 <span class="o">+</span> <span class="n">facet_wrap</span><span class="p">(</span><span class="s1">&#39;~publication_year&#39;</span><span class="p">,</span> <span class="nb">dir</span><span class="o">=</span><span class="s1">&#39;h&#39;</span><span class="p">,</span> <span class="n">ncol</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
 <span class="o">+</span> <span class="n">theme</span><span class="p">(</span><span class="n">axis_text_x</span><span class="o">=</span><span class="n">element_text</span><span class="p">(</span><span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span> <span class="n">hjust</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">6</span><span class="p">))</span>

<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">NameError</span><span class="g g-Whitespace">                                 </span>Traceback (most recent call last)
<span class="o">&lt;</span><span class="n">ipython</span><span class="o">-</span><span class="nb">input</span><span class="o">-</span><span class="mi">7</span><span class="o">-</span><span class="mi">86</span><span class="n">bb638daf4f</span><span class="o">&gt;</span> <span class="ow">in</span> <span class="o">&lt;</span><span class="n">module</span><span class="o">&gt;</span>
<span class="g g-Whitespace">      </span><span class="mi">3</span>  <span class="o">+</span> <span class="n">geom_col</span><span class="p">()</span>
<span class="g g-Whitespace">      </span><span class="mi">4</span>  <span class="o">+</span> <span class="n">facet_wrap</span><span class="p">(</span><span class="s1">&#39;~publication_year&#39;</span><span class="p">,</span> <span class="nb">dir</span><span class="o">=</span><span class="s1">&#39;h&#39;</span><span class="p">,</span> <span class="n">ncol</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="ne">----&gt; </span><span class="mi">5</span>  <span class="o">+</span> <span class="n">theme</span><span class="p">(</span><span class="n">axis_text_x</span><span class="o">=</span><span class="n">element_text</span><span class="p">(</span><span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span> <span class="n">hjust</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">6</span><span class="p">))</span>
<span class="g g-Whitespace">      </span><span class="mi">6</span> 
<span class="g g-Whitespace">      </span><span class="mi">7</span> <span class="p">)</span>

<span class="ne">NameError</span>: name &#39;ng2018sc2015&#39; is not defined
</pre></div>
</div>
</div>
</div>
<p>What do we see? The figure shows how consistant the chemical profile of the scent between the 2 studies is, which prevalent compounds such as X, Y, and Z showing roughtly similar relative amount within and across studies.</p>
<div class="section" id="bibliographic-reference">
<h2>Bibliographic Reference:<a class="headerlink" href="#bibliographic-reference" title="Permalink to this headline">¬∂</a></h2>
<p><a class="reference external" href="https://doi.org/10.1126/science.aab0696">1.Magnard JL, Roccia A, Caissard JC, Vergne P, Sun P, Hecquet R, Dubois A, Hibrand-Saint Oyant L, Jullien F, Nicol√® F, Raymond O, Huguet S, Baltenweck R, Meyer S, Claudel P, Jeauffre J, Rohmer M, Foucher F, Hugueney P, Bendahmane M, Baudino S. PLANT VOLATILES. Biosynthesis of monoterpene scent compounds in roses. Science. 2015 Jul 3;349(6243):81-3.</a></p>
<p><a class="reference external" href="https://doi.org/10.1038/s41588-018-0110-3">2.Raymond O, Gouzy J, Just J, Badouin H, Verdenaud M, Lemainque A, Vergne P, Moja S, Choisne N, Pont C, Carr√®re S, Caissard JC, Couloux A, Cottret L, Aury JM, Sz√©csi J, Latrasse D, Madoui MA, Fran√ßois L, Fu X, Yang SH, Dubois A, Piola F, Larrieu A, Perez M, Labadie K, Perrier L, Govetto B, Labrousse Y, Villand P, Bardoux C, Boltz V, Lopez-Roques C, Heitzler P, Vernoux T, Vandenbussche M, Quesneville H, Boualem A, Bendahmane A, Liu C, Le Bris M, Salse J, Baudino S, Benhamed M, Wincker P, Bendahmane M. The Rosa genome provides new insights into the domestication of modern roses. Nat Genet. 2018 Jun;50(6):772-777.</a></p>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "venv372"
        },
        kernelOptions: {
            kernelName: "venv372",
            path: "./content/recipes/applied-examples/fair-data-matrix"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'venv372'</script>

              </div>
              
        
        <div class='prev-next-bottom'>
            
    <a class='left-prev' id="prev-link" href="0-rose-metabolites-Python-data-handling.html" title="previous page">Bibliographic Reference:</a>
    <a class='right-next' id="next-link" href="2-rose-metabolites-Python-RDF-querying-analysis.html" title="next page">Exploring a rose garden with SPARQL</a>

        </div>
        
        </div>
    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By The FAIRplus project members<br/>
        
            &copy; Copyright 2020 (CC-BY 4.0).<br/>
          <div class="extra_footer">
            <script>mermaid.init();</script> <div class="row_footer"> This project has received funding from the <a href="https://www.imi.europa.eu/">Innovative Medicines Initiative</a> 2 Joint Undertaking under grant agreement No 802750. This Joint Undertaking receives support from the European Union‚Äôs Horizon 2020 research and innovation programme and EFPIA Companies. This communication reflects the views of the authors and neither IMI nor the European Union, EFPIA or any Associated Partners are liable for any use that may be made of the information contained herein. </div>
<div class="row_footer"> This page was created by <a href="https://github.com/jupyter/jupyter-book/graphs/contributors">The Jupyter Book Community</a> </div>
<div class="row_footer"> <div class="column" style="center"><a href="https://europa.eu/"><img src="https://raw.githubusercontent.com/FAIRplus/the-fair-cookbook/migrating/_static/images/logo/eu-flag-recolored-fairplus.png" height="20" alt="EU flag" style="float:center""></a></div> <div class="column"><a href="https://www.imi.europa.eu/"><img src="https://raw.githubusercontent.com/FAIRplus/the-fair-cookbook/migrating/_static/images/logo/imi-logo-fairplus-v1.png" height="20" alt="IMI logo" style="float:center""></a></div> <div class="column"><a href="https://efpia.eu"><img src="https://raw.githubusercontent.com/FAIRplus/the-fair-cookbook/migrating/_static/images/logo/efpia-logo-fairplus2.png" height="20" alt="EFPIA logo" style="float:center""></a></div> <br/> </div>
          </div>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../../../../_static/js/index.d3f166471bb80abb5163.js"></script>


    
    <!-- Google Analytics -->
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-171697285-1', 'auto');
      ga('set', 'anonymizeIp', true);
      ga('send', 'pageview');
    </script>
    <script async src='https://www.google-analytics.com/analytics.js'></script>
    <!-- End Google Analytics -->
    
  </body>
</html>